<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki - Boise State University" />

<meta name="date" content="2023-01-16" />

<title>BIOL 497/597 - Genomics &amp; Bioinformatics</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="R_Functions.html">R functions</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BIOL 497/597 - Genomics &amp; Bioinformatics</h1>
<h3 class="subtitle">Chapters</h3>
<h4 class="author">Sven Buerki - Boise State University</h4>
<h4 class="date">2023-01-16</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>This webpage contains information related to the chapters covered in the Genomics &amp; Bioinformatics course at BSU. The instructor will provide links to download the presentations as well as tutorials to complete in-class exercises (see e.g. <a href="#Chap4">Chapter 4</a>).</p>
</div>
<div id="syllabus" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Syllabus</h1>
<p>The presentation slides (in <code>pdf</code> format) of the Syllabus can be downloaded <a href="Lectures/Chapter_00_Syllabus_Spring.pdf" target="_blank">here</a>.</p>
</div>
<div id="chapter-1" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Chapter 1</h1>
<p>This chapter provides the theoretical background information (incl. definitions; see <a href="Lexicon.html">Lexicon</a>) necessary to support material taught in this course. We will review information on DNA and RNA that are important for genomic projects, especially to support genome sequencing, assembly and annotation.</p>
<div id="learning-outcomes" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Learning outcomes</h2>
<p>The key learning outcomes of this chapter are:</p>
<ul>
<li>Understand the differences between “Genetics” and “Genomics”.</li>
<li>Know the basic dogma that DNA is transcribed to RNA, which is translated to protein. This will be especially important for the <a href="Mining_draft_genome.html">Lab. report</a>.</li>
<li>Appreciate the diversity of genome organization in organelles, prokaryotes and eukaryotes.</li>
<li>Appreciate that organelle genomes have forged eukaryotic chromosomes and that this has to be accounted for in designing genome assembly projects.</li>
<li>Appreciate that eukaryotic genomes contain extensive repetitive regions of several different kinds.</li>
<li>Appreciate the diversity of published genomes and their differences in the level of completeness.</li>
<li>Understand the importance of computer science and bioinformatics in producing the raw sequence data and assembly genomes.</li>
<li>Understand the challenges in sequencing a genome.</li>
</ul>
</div>
<div id="download-presentation" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Download presentation</h2>
<p>The presentation slides (in <code>pdf</code> format) of Chapter 1 can be downloaded <a href="Lectures/Chapter_01_Spring.pdf" target="_blank">here</a>.</p>
</div>
</div>
<div id="chapter-2" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Chapter 2</h1>
<p>This chapter focuses on learning about next-generation sequencing and students will have to produce individual mini-reports on a sequencing technology (see <a href="#minrep1">below</a>).</p>
<div id="learning-outcomes-1" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Learning outcomes</h2>
<ul>
<li>Learn the next-generation sequencing (NGS) jargon underpinning genome assembly and annotation.</li>
<li>Understand the general principle of genome sequencing (wet-lab) and assembly (bioinformatics workflow).</li>
<li>Be familiar and know the limitations of the following NGS platforms (<a href="#minrep1">Mini-report 1</a>):
<ul>
<li>Illumina.</li>
<li>PacBio.</li>
<li>Oxford Nanopore.</li>
</ul></li>
<li>Be accustomed to handle NGS data output, especially FASTA and FASTQ files.</li>
<li>Learn protocols to quantify the quality of the nucleobases generated by NGS platforms using Phred quality score.</li>
</ul>
</div>
<div id="resources" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Resources</h2>
<ul>
<li><strong>An Overview of Next-Generation Sequencing</strong>: This <a href="https://www.technologynetworks.com/genomics/articles/an-overview-of-next-generation-sequencing-346532">webpage</a> contains information related to next-generation sequencing.</li>
</ul>
</div>
<div id="download-presentation-1" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Download presentation</h2>
<p>The presentation slides (in <code>pdf</code> format) of Chapter 2 can be downloaded <a href="Lectures/Chapter_02_Spring.pdf" target="_blank">here</a>.</p>
</div>
<div id="minrep1" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Mini-report 1</h2>
<p>As part of this chapter, students are tasked to to produce individual mini-reports on the following sequencing platforms and their associated technologies:</p>
<ul>
<li><strong>Sequencing platform 1:</strong> Illumina.</li>
<li><strong>Sequencing platform 2:</strong> PacBio.</li>
<li><strong>Sequencing platform 3:</strong> Oxford Nanopore.</li>
</ul>
<p>More details on this assignment is available <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini1">here</a>.</p>
</div>
</div>
<div id="chapter-3" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Chapter 3</h1>
<div id="learning-outcomes-2" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Learning outcomes</h2>
<ul>
<li>Understand the importance of computer science and bioinformatics in:
<ul>
<li>Producing raw data.</li>
<li>Creating databases in molecular biology.</li>
<li>Archiving and careful curation of the data.</li>
<li>Distributing them via the Internet.</li>
<li>Creating information-retrieval tools to allow effective mining of the data for research application.</li>
</ul></li>
<li>Gain knowledge on major molecular biology databases, which are key to the assembly and annotation of genomes:
<ul>
<li>Nucleic acid sequences databases.</li>
<li>Protein sequences databases.</li>
<li>Gene ontology databases.</li>
<li>Metabolic pathways databases.</li>
<li>Specialized annotated genome portals.</li>
</ul></li>
<li>Learn protocols to query genomic information and remotely download files from GenBank.</li>
</ul>
</div>
<div id="resources-1" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Resources</h2>
<ul>
<li><strong>NCBI BioProject Help Book</strong>: This <a href="https://www.ncbi.nlm.nih.gov/books/NBK54364/">book</a> provides definitions of terms associated to submitted data to NCBI.</li>
</ul>
</div>
<div id="download-presentation-2" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Download presentation</h2>
<p>The presentation slides (in <code>pdf</code> format) of Chapter 3 can be downloaded <a href="Lectures/Chapter_03_Spring2022.pdf" target="_blank">here</a>.</p>
</div>
<div id="mini2" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Mini-report 2 – Molecular biology databases</h2>
<p>To have a full overview of available molecular biology databases and assess their contributions to genome annotation, students are tasked to produce individual mini-reports on the following molecular databases:</p>
<ul>
<li><strong>Molecular database 1:</strong> Protein sequences databases.</li>
<li><strong>Molecular database 2:</strong> Gene ontology databases.</li>
<li><strong>Molecular database 3:</strong> Metabolic pathways databases.</li>
</ul>
<p>More details on this assignment is available <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini2">here</a>.</p>
</div>
</div>
<div id="Chap4" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Chapter 4</h1>
<div id="objectives" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Objectives</h2>
<p>In this chapter, we aim at training students in producing a draft nuclear genome for a non-model organism using Illumina data.</p>
<p>The whole-genome shotgun (WGS) sequencing dataset published by <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> on the orchid species <em>Apostasia shenzhenica</em> is used as case-study. This dataset provides an opportunity to gain firsthand experience in analyzing NGS data. <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> have also produced RNA-Seq data, which were used in their study to support genome annotation. <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> is available in our shared <a href="https://drive.google.com/file/d/1S1FWnEmxBC3_Ghl40S5GhcTVecQ2X4qd/view?usp=sharing">Google Drive</a>.</p>
<p>The chapter is subdivided into three parts:</p>
<ul>
<li><a href="#PART1"><strong>PART 1:</strong></a> Preparing/cleaning Illumina reads for <em>de novo</em> nuclear genome assembly and inferring genome size and complexity.</li>
<li><a href="#PART2"><strong>PART 2:</strong></a> <em>De novo</em> genome assembly.</li>
<li><a href="#PART3"><strong>PART 3:</strong></a> Validation of the draft genome.</li>
</ul>
<p>Before you start working on Chapter 4, please make sure that you have completed the tutorials available <a href="Tutorials.html">here</a>.</p>
</div>
<div id="download-presentations" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Download presentations</h2>
<p>The two presentations (in <code>pdf</code> format) associated to material presented in Chapter 4 can be downloaded here:</p>
<ul>
<li><a href="Lectures/Chapter_04_Spring2022.pdf" target="_blank">Overview of <em>de novo</em> genome assembly procedure</a>.</li>
<li><a href="Lectures/Chapter_04_Orchid_Genome_presentation.pdf" target="_blank">Presentation of orchid dataset, sequencing approach and results</a>.</li>
</ul>
</div>
<div id="presenting-the-ngs-data" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Presenting the NGS data</h2>
<p>Details on the WGS PE Illumina library studied here are provided in Figure <a href="#fig:SRA">6.1</a>. The SRA accession number is <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR5759389">SRR5759389</a> and the most important information to know about this data are that the fragments insert-size of the library is 180 bp and that the length of each PE read is 90 bp. Note that the authors have performed a round of PCR amplification during their library preparation. This step is know to potentially introduce errors in reads, which could be identified by performing k-mer analyses.</p>
<p>The number of bases shown in Figure <a href="#fig:SRA">6.1</a> was inferred as follows: <span class="math inline">\(\text{N bases} = \text{N. spots * reads length (bp)}\)</span>. In this example, <span class="math inline">\(\text{N bases} = 84.1e6 * 180 (90 + 90) = 15.1e9bp = 15.1Gbp\)</span>.</p>
<p>You can also obtain a rough estimate of haploid genome coverage (x) by using the following equation: <span class="math inline">\(\text{Raw haploid genome coverage (x)} = \text{N bases / Genome size (haploid)}\)</span>. In this example, <span class="math inline">\(\text{Raw haploid genome coverage (x)} = 15.1e9 / 471.0e6 = 32x\)</span>. The estimation of haploid genome size was taken from <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:SRA"></span>
<img src="Images/SRR5759389_details.png" alt="Details on the WGS library (SRA accession number: SRR5759389) used in this tutorial. " width="100%" />
<p class="caption">
Figure 6.1: Details on the WGS library (SRA accession number: SRR5759389) used in this tutorial.
</p>
</div>
</div>
<div id="write-your-own-scripts" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Write your own scripts</h2>
<p>As you go along this document and perform analyses, please COPY ALL COMMAND LINES into an <code>.Rmd</code> document saved in a folder entitled <code>Report/</code> (create this folder in your working directory; see below). Remember to comment your script (using #s). This will greatly help you in repeating your analyses or using parts of your code to create new scripts. Enjoy scripting!</p>
</div>
<div id="files-location" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Files location</h2>
<p>Files for this chapter are deposited on each group Linux computer under the following path, which is shared between all the users on the computer: <code>/home/Genomics_shared/Chapter_04/</code>.</p>
<p>There are three folders in this project:</p>
<ul>
<li><code>01_SRA/</code>: This folder contains the <code>SRA</code> file (<code>SRR5759389_pe12.fastq</code>) downloaded from the <code>SRA</code> database. This file is very large, &gt;35GB!</li>
<li><code>02_Kmers_analyses/</code>: This folder contains the structure that will be used for this tutorial. Each student will have to copy this folder onto the account prior to starting the analyses.</li>
<li><code>03_Output_files/</code>: This folder contains all the outputs files. Students can look at these files to help solve potential coding issues.</li>
</ul>
</div>
<div id="PART1" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> PART 1: QCs, read cleaning and genome size</h2>
<div id="analytical-workflow" class="section level3" number="6.6.1">
<h3><span class="header-section-number">6.6.1</span> Analytical workflow</h3>
<p>The overarching objective of this tutorial is to gain theoretical and bioinformatics knowledge on the steps required to prepare PE Illumina reads for <em>de novo</em> nuclear genome assembly as well as infer genome size and complexity (see Figure <a href="#fig:part1">6.2</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:part1"></span>
<img src="Images/Chapter_4_Part1_steps_covered.png" alt="Workflow applied to assemble and annotate a genome for non-model organisms. Steps associated to PART 1 are highlighted in grey." width="80%" />
<p class="caption">
Figure 6.2: Workflow applied to assemble and annotate a genome for non-model organisms. Steps associated to PART 1 are highlighted in grey.
</p>
</div>
<p>The five main steps associated to the objectives of PART 1 are as follows (see Figure <a href="#fig:part1workflow">6.3</a>):</p>
<ul>
<li><strong>Step 1:</strong> Download SRA file containing raw WGS data and simultaneously convert it into an interleaved PE <code>fasqt</code> format file. This latter format (where both reads are paired and combined into one file) is the input for most bioinformatic programs.</li>
<li><strong>Step 2:</strong> Infer reads Quality Checks (QCs) using standard statistics implemented in <code>FastQC</code> <span class="citation">(<a href="#ref-FastQC" role="doc-biblioref">Andrews, 2012</a>)</span>.</li>
<li><strong>Step 3:</strong> Perform reads cleaning based on Phread scores and k-mer frequencies.</li>
<li><strong>Step 4:</strong> What’s “in” the reads? This will be done in two phases as follows:
<ul>
<li>Infer and plot clean reads GC contents to assess potential contamination patterns.</li>
<li>Map clean reads against reference genomes to assess the proportions of reads belonging to either nuclear or chloroplast genomes.</li>
</ul></li>
<li><strong>Step 5:</strong> Estimate genome size and complexity (especially looking at repetitive elements and heterozygosity rate) using k-mer frequencies.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:part1workflow"></span>
<img src="Images/Cleaning_reads.png" alt="Overview of the analytical workflow applied here to prepare/clean Illumina reads for genome assembly and inferring genome size and complexity. Details on associated bioinformatic tools (here software) and file formats are also provided." width="100%" />
<p class="caption">
Figure 6.3: Overview of the analytical workflow applied here to prepare/clean Illumina reads for genome assembly and inferring genome size and complexity. Details on associated bioinformatic tools (here software) and file formats are also provided.
</p>
</div>
</div>
<div id="getting-ready" class="section level3" number="6.6.2">
<h3><span class="header-section-number">6.6.2</span> Getting ready!</h3>
<p>Before starting coding, each student has to copy the <code>02_Kmers_analyses/</code> folder located in <code>/home/Genomics_shared/Chapter_04</code> in their <code>~/Documents/</code> folder.</p>
<p>To do that, please execute the following commands in a <code>Terminal</code> window:</p>
<pre class="bash"><code>#Navigate to /home/Genomics_shared/Chapter_04
cd /home/Genomics_shared/Chapter_04

#Copy all files in 02_Kmers_analyses/ in ~/Documents/
cp -r 02_Kmers_analyses/ ~/Documents/

#Navigate to ~/Documents/02_Kmers_analyses/ to start the analyses
cd ~/Documents/02_Kmers_analyses/</code></pre>
<p>The instructor encourages you to explore the content of <code>02_Kmers_analyses/</code> using the UNIX command <code>ls -al</code>. In the tutorial, the instructor will only refer to the name of the folder containing information about the analyses and not provide their full paths.</p>
</div>
<div id="bioinformatic-tools" class="section level3" number="6.6.3">
<h3><span class="header-section-number">6.6.3</span> Bioinformatic tools</h3>
<p>Although all the bioinformatic tools (software) necessary to complete this tutorial are already installed on your Linux computers, URLs to their repositories (incl. documentations) are provided below together with details on their publications (when available). Software are sorted by steps in our analytical workflow (see Figure <a href="#fig:part1workflow">6.3</a>):</p>
<ul>
<li><strong>Step 1:</strong> <code>fastq-dump</code>: <a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/" class="uri">https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/</a>.</li>
<li><strong>Step 2:</strong> <code>FastQC</code> <span class="citation">(<a href="#ref-FastQC" role="doc-biblioref">Andrews, 2012</a>)</span>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>.</li>
<li><strong>Step 3:</strong>
<ul>
<li><code>seqtk</code>: <a href="https://github.com/lh3/seqtk" class="uri">https://github.com/lh3/seqtk</a>.</li>
<li><code>khmer</code> <span class="citation">(<a href="#ref-khmer2015" role="doc-biblioref">Crusoe et al., 2015</a>)</span>: <a href="https://github.com/dib-lab/khmer" class="uri">https://github.com/dib-lab/khmer</a>.</li>
</ul></li>
<li><strong>Step 4:</strong>
<ul>
<li><code>bwa</code> <span class="citation">(<a href="#ref-bwa2009" role="doc-biblioref">Li and Durbin, 2009</a>)</span>: <a href="http://bio-bwa.sourceforge.net" class="uri">http://bio-bwa.sourceforge.net</a>.</li>
<li><code>samtools</code><span class="citation">(<a href="#ref-samtools2009" role="doc-biblioref">Li, Handsaker, et al., 2009</a>)</span>: <a href="http://samtools.sourceforge.net" class="uri">http://samtools.sourceforge.net</a>.</li>
</ul></li>
<li><strong>Step 5:</strong>
<ul>
<li><code>jellyfish</code> <span class="citation">(<a href="#ref-Jellyfish2011" role="doc-biblioref">Marçais and Kingsford, 2011</a>)</span>: <a href="http://www.genome.umd.edu/jellyfish.html" class="uri">http://www.genome.umd.edu/jellyfish.html</a>.</li>
<li><code>GenomeScope</code> <span class="citation">(<a href="#ref-genomescope" role="doc-biblioref">Vurture et al., 2017</a>)</span>: <a href="http://qb.cshl.edu/genomescope/" class="uri">http://qb.cshl.edu/genomescope/</a>.</li>
</ul></li>
</ul>
<p>If you are remotely connecting to the Linux computers please <strong>install <a href="https://filezilla-project.org/">FileZilla</a> to support file transfer</strong>. Please see section <a href="#sftp">6.6.6.2</a> for more details on how to set up remote connection for transferring files. Finally, the <code>R</code> software is also used in steps 4 and 5. You can find more details on this software in the <a href="index.html">Syllabus</a> and <a href="Mini_reports.html">Mini-Report</a> webpages.</p>
</div>
<div id="step-1-download-sra-file" class="section level3" number="6.6.4">
<h3><span class="header-section-number">6.6.4</span> Step 1: Download SRA file</h3>
<p>The SRA format (containing raw NGS data) can be quite difficult to play with. Here, we aim at downloading the WGS data, split PE reads, but store both reads (R1 and R2) in the same file using the interleaved format (Figure <a href="#fig:part1workflow">6.3</a>). This format is very convenient and is the entry point for most reads cleaning programs. This task can be done by using <code>fastq-dump</code> from the <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">SRA Toolkit</a>.</p>
<p><strong>!!! DON’T EXECUTE THIS CODE !!!</strong></p>
<blockquote>
<p>The command to download the SRA file is available below, but please don’t execute it since the file weighs &gt;35GB and it takes a LONG time to download.</p>
</blockquote>
<p>Before starting, open a Terminal and navigate to the <code>/home/Genomics_shared/Chapter_04/01_SRA</code> folder using the <code>cd</code> command. This folder contains the <code>SRA</code> file (<code>SRR5759389_pe12.fastq</code>) that we will be using in this tutorial. You could download it by executing the following command:</p>
<pre class="bash"><code>#Download PE reads (properly edited) in interleave format using SRA Toolkit
fastq-dump --split-files --defline-seq &#39;@$ac.$si.$sg/$ri&#39; --defline-qual &#39;+&#39; -Z SRR5759389  &gt; SRR5759389_pe12.fastq</code></pre>
</div>
<div id="Step2" class="section level3" number="6.6.5">
<h3><span class="header-section-number">6.6.5</span> Step 2: Reads QCs</h3>
<p>The program <code>FastQC</code> is used to obtain preliminary information about the Illumina library (see Figure <a href="#fig:SRA">6.1</a>). To run this program navigate to the <code>FastQC/</code> folder using <code>cd</code> and execute the following command:</p>
<pre class="bash"><code>#Reads QCs with FastQC 
./FastQC_program/fastqc -o ~/Documents/02_Kmers_analyses/FastQC/ /home/Genomics_shared/Chapter_04/01_SRA/SRR5759389_pe12.fastq </code></pre>
<p>This analysis generates the <code>SRR5759389_pe12_fastqc.html</code> output file. We will have a group discussion to inspect this file and draw general conclusions on the quality of the raw data. This step will have to be repeated after completion of the step3 (reads cleaning) to confirm that our reads trimming and cleaning procedure was successful. To inspect this file click <a href="Data/SRR5759389_pe12_fastqc.html">here</a>.</p>
</div>
<div id="file-transferring" class="section level3" number="6.6.6">
<h3><span class="header-section-number">6.6.6</span> File transferring</h3>
<p>At this stage, you will need to transfer <code>SRR5759389_pe12_fastqc.html</code> on your local computer to be able to inspect the results. This can be done using i) the <code>scp</code> protocol (but it only works on Unix-based operating systems) or ii) the <code>sftp</code> protocol as implemented in FileZilla (this works on all operating systems).</p>
<div id="scp-protocol" class="section level4" number="6.6.6.1">
<h4><span class="header-section-number">6.6.6.1</span> <code>scp</code> protocol</h4>
<p><strong>WARNING:</strong> This protocol works only on UNIX OS (Mac, Linux).</p>
<p>You can use the <code>scp</code> command to copy <code>SRR5759389_pe12_fastqc.html</code> or any other file to your local computer. This can be done as follows:</p>
<ol style="list-style-type: decimal">
<li><p>Open a <code>Terminal</code> window on your computer and navigate (using <code>cd</code>) where you want to store the target file.</p></li>
<li><p>Type this command line:</p></li>
</ol>
<pre class="bash"><code>#General syntax
scp USER@IP:PATH_ON_REMOTE_COMPUTER PATH_ON_YOUR_COMPUTER

#Example:
# To copy SRR5759389_pe12_fastqc.html on my computer from bio_11 user do:
scp bio_11@132.178.142.214:~/Documents/02_Kmers_analyses/FastQC/SRR5759389_pe12_fastqc.html .</code></pre>
</div>
<div id="sftp" class="section level4" number="6.6.6.2">
<h4><span class="header-section-number">6.6.6.2</span> <code>sftp</code> protocol</h4>
<p>Open FileZilla on your personal computer and establish remote connection with Linux computer as shown in Figure <a href="#fig:FileZilla">6.4</a>. Once the connection is established:</p>
<ol style="list-style-type: decimal">
<li>Navigate where the target file (<code>SRR5759389_pe12_fastqc.html</code>) is located on the remote computer and select it.</li>
<li>Identify where the target file needs to be saved on your personal computer.</li>
<li>Copy the file on your personal computer by dragging the target file from the remote to the local location.</li>
</ol>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:FileZilla"></span>
<img src="Images/FileZilla_snapshot.png" alt="Snapshot of FileZilla program used to transfer files between computers." width="80%" />
<p class="caption">
Figure 6.4: Snapshot of FileZilla program used to transfer files between computers.
</p>
</div>
<p>This protocol will have to be replicated at different stages of this tutorial.</p>
</div>
</div>
<div id="Step3" class="section level3" number="6.6.7">
<h3><span class="header-section-number">6.6.7</span> Step 3: Reads cleaning</h3>
<p>This step is at the core of our analytical workflow. It is paramount to conduct thorough reads cleaning when assembling a nuclear genome due to the very nature of this genome (i.e. repetitive elements, heterozygosity, recombination, etc…). In this case, aim at minimizing the effect of PCR errors, which could jeopardize our assembly by creating false polymorphism. In addition, <em>de novo</em> genome assembly is computing intensive and by properly cleaning reads we will dramatically decrease RAM requirements.</p>
<div id="approach-applied-here-to-clean-reads" class="section level4" number="6.6.7.1">
<h4><span class="header-section-number">6.6.7.1</span> Approach applied here to clean reads</h4>
<p>Reads will be cleaned/trimmed based on (see Figure <a href="#fig:part1workflow">6.3</a>):</p>
<ul>
<li>Phred quality scores (33) to conduct a first round of trimming.</li>
<li>K-mer frequencies (k=21) to:
<ul>
<li>Normalize high coverage reads (higher than 100x) based on median reads coverage.</li>
<li>Filter low abundance reads (where PCR errors will most likely take place).</li>
</ul></li>
<li>A final round of cleaning by removing low quality bases, short sequences, and non-paired reads.</li>
</ul>
<p>Finally, we will format the clean data for <em>de novo</em> genome assembly, which will be conducted using <code>SOAPdenovo2</code> <span class="citation">(<a href="#ref-Luo2012" role="doc-biblioref">Luo et al., 2012</a>)</span>.</p>
</div>
<div id="what-is-a-k-mer" class="section level4" number="6.6.7.2">
<h4><span class="header-section-number">6.6.7.2</span> What is a k-mer?</h4>
<p>A <code>k-mer</code> is a substring of length <span class="math inline">\(k\)</span> in a string of DNA bases. For a given sequence of length <span class="math inline">\(L\)</span>, and a k-mer size of <span class="math inline">\(k\)</span>, the total k-mer’s possible (<span class="math inline">\(n\)</span>) will be given by <span class="math inline">\((L-k) + 1\)</span>. For instance, in a sequence of length of 9 (<span class="math inline">\(L\)</span>), and a k-mer length of 2 (<span class="math inline">\(k\)</span>), the number of k-mer’s generated will be: <span class="math inline">\(n = (9-2) + 1 = 8\)</span>.</p>
<blockquote>
<p>All eight 2-mers of the sequence “AATTGGCCG” are:
AA, AT, TT, TG, GG, GC, CC, CG</p>
</blockquote>
<p>In most studies, the authors provide an estimate of sequencing coverage prior to assembly <span class="citation">(e.g. 73 fold in the case of the giant panda genome, <a href="#ref-Li2009" role="doc-biblioref">Li, Fan, et al., 2009</a>)</span>, but the real coverage distribution will be influenced by factors including DNA quality, library preparation and local GC content. On average, you might expect most of the genome (especially the single/low copy genes) to be covered between 20 and 70x. One way of rapidly examining the coverage distribution before assembling a reference genome is to chop your cleaned sequence reads into short “k-mers” of 21 nucleotides, and count how often you get each possible k-mer. By doing so, you will find out that:</p>
<ul>
<li>Many sequences are extremely rare (e.g., once). These are likely to be PCR errors that appeared during library preparation or sequencing, or could be rare somatic mutations). Such sequences can confuse assembly software; eliminating them can decrease subsequent memory and CPU requirements.</li>
<li>Other sequences may exist at 10,000x coverage. These could be pathogens or repetitive elements. Often, there is no benefit to retaining all copies of such sequences because the assembly software will be confused by them; while retaining a small proportion such reads could significantly reduce CPU, memory and space requirements.</li>
</ul>
<p>Please find below the plot of k-mer frequencies inferred from the trimmed library studied here (<code>SRR5759389</code>), which suggests that the haploid genome of <em>Apostasia shenzhenica</em> is sequenced at ca. 25x (Figure <a href="#fig:kmerstrimmed">6.5</a>). The methodology used to infer this graph is explained in Step 4.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:kmerstrimmed"></span>
<img src="Images/GenomeSize_trimmed.png" alt="Distribution of 21-mer frequencies based on the trimmed library with insert fragment size of 180 bp. Given only one peak in the k-mer distribution, we predict that the *Apostasia shenzhenica* genome has limited heterozygosity or in other words, that this genome is inbred." width="100%" />
<p class="caption">
Figure 6.5: Distribution of 21-mer frequencies based on the trimmed library with insert fragment size of 180 bp. Given only one peak in the k-mer distribution, we predict that the <em>Apostasia shenzhenica</em> genome has limited heterozygosity or in other words, that this genome is inbred.
</p>
</div>
</div>
<div id="kmers" class="section level4" number="6.6.7.3">
<h4><span class="header-section-number">6.6.7.3</span> K-mer graph and PCR effect</h4>
<p>The peak around 25 in Figure <a href="#fig:kmerstrimmed">6.5</a> is the coverage with the highest number of different 21-mers. This means that there are <span class="math inline">\(1.4e^7\)</span> unique 21-mers (frequency) that have been observed 25 times (coverage). The normal-like distribution is due to the fact that the sequencing did not provide a perfect coverage of the whole genome. Some regions have less coverage, whereas others have a little more coverage, but the average coverage depth is around 25.</p>
<p>The large number of unique k-mers (<span class="math inline">\(1.7e^7\)</span>) that have a frequency of 1 (right left side of the graph; Figure <a href="#fig:kmerstrimmed">6.5</a>) is most likely due to PCR errors. Please find below an example explaining how PCR errors impact on the k-mer procedure.</p>
<p>All 3-mers of the sequence “AATTGGCCG” are:</p>
<ul>
<li>AAT, ATT, TTG, TGG, GGC, GCC, CCG</li>
</ul>
<p>Now, let’s consider that the 4th letter (<code>T</code>) in the sequence above is replaced with a <code>C</code> to simulate a PCR error. In this context, all 3-mers of this sequence “AAT<strong>C</strong>GGCCG” are:</p>
<ul>
<li>AAT, <strong>ATC</strong>, <strong>TCG</strong>, <strong>CGG</strong>, GGC, GCC, CCG.</li>
</ul>
<p>The k-mers in bold are the incorrect 3-mers that are now unique and end up at the beginning of the graph in Figure <a href="#fig:kmerstrimmed">6.5</a>.</p>
<p>Overall, the <strong>general rule</strong> is that for a given sequence, <strong>a single PCR error will result in <span class="math inline">\(k\)</span> unique and incorrect k-mers</strong>.</p>
</div>
<div id="trim-reads-based-on-phred-quality-scores" class="section level4" number="6.6.7.4">
<h4><span class="header-section-number">6.6.7.4</span> Trim reads based on Phred quality scores</h4>
<p>Let’s start the cleaning of our interleaved PE reads by trimming reads using Phred scores (33) as implemented in <code>seqtk</code>. Before executing the commands below, <code>cd</code> to the <code>khmer/</code> folder:</p>
<pre class="bash"><code>#Trim reads based on Phred scores (default of 33)
seqtk trimfq /home/Genomics_shared/Chapter_04/01_SRA/SRR5759389_pe12.fastq &gt; SRR5759389_pe12.trimmed.fastq</code></pre>
<p>Using Phred scores to clean our data is not going to deal with PCR errors and potential effects of high repeated DNA sequences on the <em>de novo</em> assembly. Below, we will use information on k-mer distributions to address these issues and clean our dataset accordingly.</p>
</div>
<div id="normalize-high-coverage-reads-remove-pcr-errors-and-low-quality-reads" class="section level4" number="6.6.7.5">
<h4><span class="header-section-number">6.6.7.5</span> Normalize high coverage reads, remove PCR errors and low quality reads</h4>
<p>In the next commands, <code>khmer</code> <span class="citation">(<a href="#ref-khmer2015" role="doc-biblioref">Crusoe et al., 2015</a>)</span> will be used to:</p>
<ol style="list-style-type: decimal">
<li>Normalize high coverage reads (&gt;100x) based on median coverage (to optimize RAM requirements for <em>de novo</em> genome assembly).</li>
<li>Filter low abundance reads (most likely PCR errors).</li>
<li>Final clean-up of reads to remove low quality bases, short sequences, and non-paired reads</li>
</ol>
<p>To conduct these analyses, run the following commands in the Terminal:</p>
<pre class="bash"><code>#1. Normalize high coverage reads (&gt;100x) based on median reads coverage using k-mer frequencies
khmer normalize-by-median.py -p --ksize 21 -C 100 -M 1e9 -s kmer.counts -o SRR5759389_pe12.max100.trimmed.fastq SRR5759389_pe12.trimmed.fastq</code></pre>
<pre class="bash"><code>#2. Filter low abundance reads based on k-mer frequencies to minimize PCR errors
khmer filter-abund.py -V kmer.counts -o SRR5759389_pe12.max100.trimmed.norare.fastq SRR5759389_pe12.max100.trimmed.fastq</code></pre>
<pre class="bash"><code>#3. Final clean-up of reads (remove low quality bases, short sequences, and non-paired reads)
seqtk seq -q 10 -N -L 80 SRR5759389_pe12.max100.trimmed.norare.fastq | seqtk dropse &gt; SRR5759389_pe12.max100.trimmed.norare.noshort.fastq</code></pre>
</div>
<div id="prepreads" class="section level4" number="6.6.7.6">
<h4><span class="header-section-number">6.6.7.6</span> Prepare data for <em>de novo</em> genome assembly</h4>
<p>In this last section, our cleaned reads are prepared for the <em>de novo</em> genome assembly analysis conducted in <code>SOAPdenovo2</code> <span class="citation">(<a href="#ref-Luo2012" role="doc-biblioref">Luo et al., 2012</a>)</span> (done in <a href="#PART2">PART 2</a>). This is achieved by using <code>khmer</code> and renaming files using <code>mv</code> as follows:</p>
<pre class="bash"><code>#De-interleave filtered reads (to be ready for de novo assembly in SOAPdenovo2)
khmer split-paired-reads.py SRR5759389_pe12.max100.trimmed.norare.noshort.fastq

#Rename output reads to something more human-friendly
mv SRR5759389_pe12.max100.trimmed.norare.noshort.fastq.1 SRR5759389.pe1.clean.fastq
mv SRR5759389_pe12.max100.trimmed.norare.noshort.fastq.2 SRR5759389.pe2.clean.fastq</code></pre>
</div>
<div id="reads-qcs-on-cleaned-reads" class="section level4" number="6.6.7.7">
<h4><span class="header-section-number">6.6.7.7</span> Reads QCs on cleaned reads</h4>
<p>Use <code>FastQC</code> to validate the reads cleaning conducted above. Please see <a href="#Step2">Step 2</a> for more details on the methodology.</p>
</div>
</div>
<div id="step-4-whats-in-the-reads" class="section level3" number="6.6.8">
<h3><span class="header-section-number">6.6.8</span> Step 4: What’s “in” the reads?</h3>
<p>Before conducting the <em>de novo</em> genome assembly (using the cleaned reads produced in <a href="#Step3">Step 3</a>, it is paramount to assess whether the data are contaminated and also to provide a first estimate of proportions of reads belonging to either nuclear or plastid genomes. To investigate these topics, the instructor proposes that we assess i) potential contamination by inferring reads GC contents and ii) map the clean reads against nuclear and chloroplast genome assemblies to assess proportions of reads belonging to either genomes in our library.</p>
<div id="reads-gc-contents" class="section level4" number="6.6.8.1">
<h4><span class="header-section-number">6.6.8.1</span> Reads GC contents</h4>
<p>The GC content of a sequence library can provide evidence of contamination or the presence of sequence from multiple organisms (since most organisms have specific GC profiles). A GC plot inferred from a non-contaminated library would display a smooth, unimodal distribution. The existence of shoulders, or in more extreme cases a bimodal distribution, could be indicative of the presence of sequence reads from an organism with a different GC content, which is most likely a contaminant (see Figure <a href="#fig:part1workflow">6.3</a>).</p>
<p>Here we infer GC contents for all clean reads using the perl script <code>GC_content.pl</code>. This script requires reads to be in <code>fasta</code> format. We will then start by converting our <code>fastq</code> file into <code>fasta</code> by using <code>seqtk</code>. Data are available in the <code>GC_content/</code> folder. Please see below for the commands to be executed:</p>
<pre class="bash"><code>#Convert fastq to fasta
seqtk seq -a ~/Documents/02_Kmers_analyses/khmer/    SRR5759389_pe12.max100.trimmed.norare.noshort.fastq &gt; SRR5759389_pe12.max100.trimmed.norare.noshort.fasta</code></pre>
<p><strong>PLEASE DON’T EXECUTE THE COMMAND BELOW!</strong> It takes few hours to calculate reads GC contents. Use the output of this command <code>gc_out.txt</code> (weighing ca. 5GB), which is available in the <code>GC_content/</code> folder to carry on our analyses.</p>
<pre class="bash"><code>#Infer GC content using fasta file as input (this can take a long time)
./GC_content.pl SRR5759389_pe12.max100.trimmed.norare.noshort.fasta &gt; gc_out.txt</code></pre>
<p>The next command extracts the column containing the GC content per read from <code>gc_out.txt</code> (located in <code>GC_content/</code>). Since the file is very big, we use the BASH command <code>awk</code> to extract this information instead of <code>R</code>.</p>
<pre class="bash"><code>#Extract reads GC contents (the file is big and can&#39;t be processed easily)
awk &#39;{print$2}&#39; gc_out.txt &gt; gc_simple.txt</code></pre>
<p>The last part of the analysis will be executed in <code>R</code> where we will load the data and create a histogram to look at the distribution of reads GC contents.</p>
<pre class="r"><code>###~~~
#Infer GC content per reads
###~~~
#Load the data in R
gc_frac &lt;- read.table(&#39;gc_simple.txt&#39;, header=T)

#Create pdf to save output of hist
pdf(&quot;GC_content_hist.pdf&quot;)
#Do the histogram of GC content 
hist(as.numeric(gc_frac[,1]), main=&quot;Histogram of GC content per reads&quot;, xlab=&quot;GC fraction&quot;)
#Close pdf file
dev.off()</code></pre>
<p>The histogram displaying the distribution of GC values based on cleaned reads is provided in Figure <a href="#fig:GC">6.6</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:GC"></span>
<img src="Images/GC_content_hist.png" alt="Histogram of GC values inferred from the cleaned library of reads (SRR5759389). See text for more details." width="80%" />
<p class="caption">
Figure 6.6: Histogram of GC values inferred from the cleaned library of reads (SRR5759389). See text for more details.
</p>
</div>
<div id="question" class="section level5" number="6.6.8.1.1">
<h5><span class="header-section-number">6.6.8.1.1</span> Question</h5>
<blockquote>
<p>To your knowledge, based on data presented in Figure <a href="#fig:GC">6.6</a>, is the SRR5759389 library contaminated with alien DNA?</p>
</blockquote>
</div>
</div>
<div id="mapping-reads-against-reference-genomes" class="section level4" number="6.6.8.2">
<h4><span class="header-section-number">6.6.8.2</span> Mapping reads against reference genomes</h4>
<p>Here we aim at assessing the proportions of reads in the clean library belonging to either nuclear or chloroplast genomes by mapping our clean reads against two reference genomes using <code>bwa</code> <span class="citation">(<a href="#ref-bwa2009" role="doc-biblioref">Li and Durbin, 2009</a>)</span> and <code>samtools</code> <span class="citation">(<a href="#ref-samtools2009" role="doc-biblioref">Li, Handsaker, et al., 2009</a>)</span>. Data are available in <code>Map_reads/</code>.</p>
<div id="mapping-reads-against-the-nuclear-genome" class="section level5" number="6.6.8.2.1">
<h5><span class="header-section-number">6.6.8.2.1</span> Mapping reads against the nuclear genome</h5>
<p>The nuclear genome of <em>Apostasia shenzhenica</em> published by <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> is not listed in the NCBI database as “Genome”, but rather as “Assembly”. Please see Figure <a href="#fig:NuclearNCBI">6.7</a> for more details on the genome. This file (a compressed FASTA) is already available in your folder, but you could download it as follows:</p>
<pre class="bash"><code>#Get Fasta files for genomes of reference
#PEFY01 nuclear genome of Apostasia 
wget ftp://ftp.ncbi.nlm.nih.gov/sra/wgs_aux/PE/FY/PEFY01/PEFY01.1.fsa_nt.gz</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:NuclearNCBI"></span>
<img src="Images/Nuclear_genome_assembly.png" alt="Screen shot of the NCBI website showing details about the accession containing the nuclear genome assembly of *Apostasia shenzhenica* under the accession PEFY01." width="100%" />
<p class="caption">
Figure 6.7: Screen shot of the NCBI website showing details about the accession containing the nuclear genome assembly of <em>Apostasia shenzhenica</em> under the accession PEFY01.
</p>
</div>
<p>Please go through the commands below to map your clean reads against the reference nuclear genome assembly:</p>
<pre class="bash"><code>#Mapping clean reads against reference genome using bwa and converting into human readable format using samtools

#Before starting, you have to create an index for your reference genome to allow efficient mapping
bwa index -a bwtsw PEFY01.1.fsa_nt.gz
bwa mem -t 20 PEFY01.1.fsa_nt.gz ~/Documents/02_Kmers_analyses/khmer/SRR5759389_pe12.max100.trimmed.norare.noshort.fastq | samtools view -buS - | samtools sort - -o PEFY01_map_pe.sorted.bam 
samtools index PEFY01_map_pe.sorted.bam</code></pre>
<p>We will be devoting time to decipher these commands during our Lab sessions. For now, here is a short description of <code>bwa</code> from the <a href="http://bio-bwa.sourceforge.net/bwa.shtml">manual</a>:</p>
<blockquote>
<p>BWA is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome. It consists of three algorithms: BWA-backtrack, BWA-SW and BWA-MEM. The first algorithm is designed for Illumina sequence reads up to 100bp, while the rest two for longer sequences ranged from 70bp to 1Mbp. BWA-MEM and BWA-SW share similar features such as long-read support and split alignment, but BWA-MEM, which is the latest, is generally recommended for high-quality queries as it is faster and more accurate. BWA-MEM also has better performance than BWA-backtrack for 70-100bp Illumina reads.</p>
</blockquote>
</div>
<div id="mapping-reads-against-the-chloroplast-genome" class="section level5" number="6.6.8.2.2">
<h5><span class="header-section-number">6.6.8.2.2</span> Mapping reads against the chloroplast genome</h5>
<p>The chloroplast genome of <em>Apostasia shenzhenica</em> was not published. To assess the proportion of chloroplastic reads in our Illumina cleaned reads, we will use the chloroplast genome of a sister species, *A. wallichii <span class="citation">(<a href="#ref-Niu2017" role="doc-biblioref">Niu et al., 2017</a>)</span>, which is available on the NCBI website under the accession number <code>LC199394</code>. This file (in FASTA) is already available in your folder and you do not have to download it.</p>
<p>Please execute the commands below to map the clean reads against the reference chloroplast genome of the sister species:</p>
<pre class="bash"><code>#Mapping reads against the chloropolast genome of a sister species
# following the same approach as with the nuclear genome
bwa index -a bwtsw LC199394.fasta
bwa mem -t 20 LC199394.fasta ~/Documents/02_Kmers_analyses/khmer/SRR5759389_pe12.max100.trimmed.norare.noshort.fastq | samtools view -buS - | 
    samtools sort - -o LC199394_map_pe.sorted.bam
samtools index LC199394_map_pe.sorted.bam</code></pre>
</div>
<div id="counting-reads-belonging-to-either-nuclear-or-plastid-genomes" class="section level5" number="6.6.8.2.3">
<h5><span class="header-section-number">6.6.8.2.3</span> Counting reads belonging to either nuclear or plastid genomes</h5>
<p>We have now all the information at hand to count how many reads are matching with either nuclear or chloroplast genomes (and infer proportions). Remember that it is just a very preliminary assessment, which aims at assessing the quality of the library and its suitability for <em>de novo</em> genome assembly.</p>
<p>To reach this goal, we will start by counting the total number of cleaned reads using <code>count_fastq.sh</code> as follows:</p>
<pre class="bash"><code>#Count the tot. number of clean reads
./count_fastq.sh ~/Documents/02_Kmers_analyses/khmer/SRR5759389_pe12.max100.trimmed.norare.noshort.fastq</code></pre>
<p>Now, we will use <code>samtools</code> to count how many unique reads mapped the reference genomes:</p>
<pre class="bash"><code>#How many unique reads mapped the nuclear genome
samtools view -F 0x904 -c PEFY01_map_pe.sorted.bam

#How many unique reads mapped the chloroplast genome
samtools view -F 0x904 -c LC199394_map_pe.sorted.bam</code></pre>
</div>
<div id="question-1" class="section level5" number="6.6.8.2.4">
<h5><span class="header-section-number">6.6.8.2.4</span> Question</h5>
<blockquote>
<p>What are the proportions of nuclear and chloroplast reads in your library as well as potential contaminants?</p>
</blockquote>
<blockquote>
<p>Based on your previous answer, is this library suitable to assemble the nuclear genome of <em>Apostasia shenzhenica</em>?</p>
</blockquote>
</div>
</div>
</div>
<div id="step-5-estimate-genome-size-and-other-genomic-features-using-k-mer-frequencies" class="section level3" number="6.6.9">
<h3><span class="header-section-number">6.6.9</span> Step 5: Estimate genome size and other genomic features using k-mer frequencies</h3>
<p>In this section, data on <a href="#kmers">k-mer</a> frequencies are used to estimate nuclear genome size and tease apart the proportion of the genome associated with low copy genes (also referred to as unique sequences) and repetitive DNA. The <code>JELLYFISH</code> <span class="citation">(<a href="#ref-Jellyfish2011" role="doc-biblioref">Marçais and Kingsford, 2011</a>)</span> program is a tool for fast, memory-efficient counting of k-mers in DNA. <code>JELLYFISH</code> is a command-line program that reads FASTA and multi-FASTA files containing DNA sequences and it outputs k-mer counts in an binary format (which can easily be translated into a human-readable text). We will be using the apporach implemented in <code>JELLYFISH</code> to estimate the genome size and complexity of the <em>Apostasia shenzhenica</em> nuclear genome based on the Illumina data.</p>
<div id="estimate-genome-size" class="section level4" number="6.6.9.1">
<h4><span class="header-section-number">6.6.9.1</span> Estimate genome size</h4>
<p>Here, we are using <code>JELLYFISH</code> to generate countings of 21-mers based on the cleaned Illumina reads. We will subsequently input the output of this program into <code>R</code> to estimate the genome size based on k-mer frequencies. The data are available in <code>Jellyfish/</code>.</p>
<pre class="bash"><code> #Run jellyfish to obtain k-mer counts and frequencies (based on trimmed, normalized and filtered reads)
jellyfish count -t 30 -C -m 21 -s 5G -o     Orchid_21mer_SRR5759389_pe12_max100.trimmed.norare ~/Documents/02_Kmers_analyses/khmer/SRR5759389_pe12.max100.trimmed.norare.noshort.fastq

jellyfish histo -o Orchid_21mer_SRR5759389_pe12_max100.trimmed.norare.histo Orchid_21mer_SRR5759389_pe12_max100.trimmed.norare</code></pre>
<p>Once the analyses are completed, please open an <code>R</code> session (in e.g. <code>RStudio</code>) and execute the following command to estimate the haploid (1x) genome size. Here, genome size is estimate as follows:</p>
<pre class="r"><code>#Genome size (N) is equal to:
N = Total numbers of k-mers / Peak of coverage (here ca. 22x)
N = Area under the curve / Peak of coverage</code></pre>
<p>The <code>R</code> script to infer the genome size of <em>Apostasia shenzhenica</em> based on the data generated by <code>JELLYFISH</code> is presented below:</p>
<!--- 
Print the R code in Chapter_4_QC_Jellyfish.R
by calling a specially tag in the file
See https://bookdown.org/yihui/rmarkdown-cookbook/read-chunk.html
-->
<pre class="r"><code>#1. Load the output of Jellyfish in R
Clean &lt;- read.table(&#39;Orchid_21mer_SRR5759389_pe12_max100.trimmed.norare.histo&#39;)

#2. Execute this user defined function to infer genome size and plot k-mer distribution
#A function to infer genome size (1C or haploid) on jellyfish output
#input - hist = table containing output of Jellyfish (hist file)
#         - plotKmers if &quot;TRUE&quot; outputs a graph
GenomeSize &lt;- function(hist, plotKmers){
  #Infer Genome size
  x &lt;- grep(min(hist[1:20,2]),hist[,2])
  tmp &lt;- hist[x:1000,1]*hist[x:1000,2]
  #Infer the total number of kmers in the distribution
  Totkmers &lt;- sum(as.numeric(as.vector(tmp)))
  #Infer Peak of coverage (when enough data is equal to mean) of the genome (13x)
  PeakCov &lt;- grep(max(hist[x:1000,2]), hist[,2])
  #Genome size (1C or haploid) is the area under the curve, which is the Total numner of K-mers/Mean coverage
  GenomeSizeOut &lt;- Totkmers/PeakCov
  GenomeSizeOut &lt;- round(GenomeSizeOut/1000000,2)
  print(GenomeSizeOut)
  if(plotKmers == &quot;TRUE&quot;){
    #Plot 
    plot(hist[2:200,], type=&#39;n&#39;, ylim=c(0,20000000), xlab=&#39;Coverage&#39;, ylab=&#39;Frequency&#39;, main=&quot;K-mer (k=21) distribution&quot;)
    #Plot trimmed data
    lines(hist[2:200,])
    #These are low coverage K-mers (K=1:8), which have very high frequencies
    rect(xleft=0, ybottom=0, xright=x, ytop=20000000, col=&#39;red&#39;)
    text(x=25, y=hist[PeakCov,2]+3000000, paste(&quot;Error k-mers&quot;, sep=&#39;\n&#39;), cex=.7, col=&#39;red&#39;)
    #Add the distribution line
    lines(hist[2:200,], type=&#39;l&#39;)
    #Add a line showing the peak of coverage (here at 26x)
    segments(x0 = PeakCov, y0 = 0, x1=PeakCov, y1=hist[PeakCov,2], lty=2)
    text(x=26, y=hist[PeakCov,2]+900000, paste(&quot;Peak cov.: &quot;, PeakCov, &quot;x&quot;, sep=&#39;&#39;), cex=.6)
    text(x=50, y=8000000, paste(&quot;True k-mers&quot;, sep=&#39;\n&#39;), cex=.8)
    text(x=170, y=18500000, paste(&quot;Estimated genome size: &quot;, GenomeSizeOut, &quot;Mb&quot;, sep=&#39;&#39;), cex=.6)
  }
}

#3. Estimate genome size on cleaned reads using the function GenomeSize
pdf(&quot;GenomeSize.pdf&quot;)
GenomeSize(Clean, plotKmers = &quot;TRUE&quot;)
dev.off()</code></pre>
<div id="questions" class="section level5" number="6.6.9.1.1">
<h5><span class="header-section-number">6.6.9.1.1</span> Questions</h5>
<blockquote>
<p>What is the haploid genome size of <em>Apostasia shenzhenica</em>?</p>
</blockquote>
<blockquote>
<p>How does this genome size estimation compare with the one conducted based on reads that were only trimmed provided in Figure <a href="#fig:kmerstrimmed">6.5</a>?</p>
</blockquote>
<blockquote>
<p>Finally, are the latter two estimations close to the 349 Mb provided in <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span>?</p>
</blockquote>
</div>
</div>
<div id="use-genomescope-to-estimate-heterozygosity-unique-genes-repetitive-dna-and-pcr-errors" class="section level4" number="6.6.9.2">
<h4><span class="header-section-number">6.6.9.2</span> Use GenomeScope to estimate heterozygosity, unique genes, repetitive DNA and PCR errors</h4>
<p>Finally, the last step of our preliminary analyses prior to <em>de novo</em> genome assembly is aimed at estimating heterozygosity, unique genes, repetitive DNA and PCR errors. These analyses are conducted by using the approach implemented in the online tool <code>GenomeScope</code> <span class="citation">(<a href="#ref-genomescope" role="doc-biblioref">Vurture et al., 2017</a>)</span>. The authors of this program developed an equation to model the shape and size of the k-mer graph by using four negative binomial peaks which shape and size are determined by % heterozygosity, % PCR duplication, and % PCR Error.</p>
<div id="lets-run-the-analysis" class="section level5" number="6.6.9.2.1">
<h5><span class="header-section-number">6.6.9.2.1</span> Let’s run the analysis</h5>
<p>Open a web browser and navigate to the GenomeScope website (<a href="http://qb.cshl.edu/genomescope/" class="uri">http://qb.cshl.edu/genomescope/</a>). Once on the website, upload <code>Orchid_21mer_SRR5759389_pe12_max100.trimmed.norare.histo</code> by dragging your file onto the small window and set the analysis as provided in Figure <a href="#fig:GenScope1">6.8</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:GenScope1"></span>
<img src="Images/Screen.png" alt="Screen shot of the GenomeScope portal to help you setting your analysis." width="100%" />
<p class="caption">
Figure 6.8: Screen shot of the GenomeScope portal to help you setting your analysis.
</p>
</div>
</div>
<div id="interpreting-results-of-genomescope" class="section level5" number="6.6.9.2.2">
<h5><span class="header-section-number">6.6.9.2.2</span> Interpreting results of GenomeScope</h5>
<p>The output of <code>GenomeScope</code> is displayed in Figure <a href="#fig:GenScope2">6.9</a>. The program outputs two plots and the only difference between those is that the the second plot is using a log transformation for the axes. This transformation allows better teasing apart the different elements studied here. The big peak at 25 in the graph above is in fact the homozygous portions of the genome that account for the identical 21-mers from both strands of the DNA (haploid genome mostly composed of single-copy genes). The dotted line corresponds to the predicted center of that peak. The absence of a shoulder to the left of the peak suggests that this genome has a very low level of heterozygosity. The red line on the left corresponds to PCR errors. Finally, the proportion of unique sequences, which would correspond to low-copy genes is depicted by the yellow line and the amount of repetitive DNA is estimated by comparing the “space” between the black and yellow lines (on the right side of the graph).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:GenScope2"></span>
<img src="Images/genomescope1.png" alt="Output of the GenomeScope analysis." width="80%" />
<p class="caption">
Figure 6.9: Output of the GenomeScope analysis.
</p>
</div>
</div>
</div>
</div>
</div>
<div id="PART2" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> PART 2: <em>De novo</em> genome assembly</h2>
<div id="objective" class="section level3" number="6.7.1">
<h3><span class="header-section-number">6.7.1</span> Objective</h3>
<p>The overarching objective of PART 2 is to gain theoretical and bioinformatics knowledge on the steps required to <strong>perform a <em>de novo</em> genome assembly based on cleaned Illumina reads</strong>. The procedure applied to assess and validate the quality of the genome assembly will also be covered in PART 3.</p>
</div>
<div id="dataPART2" class="section level3" number="6.7.2">
<h3><span class="header-section-number">6.7.2</span> Presenting the data and their location</h3>
<p>PART 2 will be using the output files of the PART 1 <a href="#Step3">reads cleaning step</a>, which are deposited in <code>~/Documents/02_Kmers_analyses/khmer/</code> and named as follows:</p>
<ul>
<li><code>SRR5759389.pe1.clean.fastq</code></li>
<li><code>SRR5759389.pe2.clean.fastq</code></li>
</ul>
<p><strong>WARNING:</strong> You have to complete section <a href="#prepreads">6.6.7.6</a> before starting this section.</p>
</div>
<div id="analytical-workflow-1" class="section level3" number="6.7.3">
<h3><span class="header-section-number">6.7.3</span> Analytical workflow</h3>
<p>To achieve our overarching objective, our next classes will be divided into two steps:</p>
<ul>
<li><strong>Step 1:</strong> Set-up and perform a <em>de novo</em> genome assembly based on cleaned paired-end (PE) Illumina reads using <code>SOAPdenovo2</code> <span class="citation">(<a href="#ref-Luo2012" role="doc-biblioref">Luo et al., 2012</a>)</span>.</li>
<li><strong>Step 2:</strong> Provide theoretical knowledge on <em>de novo</em> genome assembly methods by focusing on de Bruijn graphs.</li>
</ul>
</div>
<div id="step-1-perform-de-novo-genome-assembly-analysis" class="section level3" number="6.7.4">
<h3><span class="header-section-number">6.7.4</span> Step 1: Perform <em>de novo</em> genome assembly analysis</h3>
<div id="specific-objective" class="section level4" number="6.7.4.1">
<h4><span class="header-section-number">6.7.4.1</span> Specific objective</h4>
<p>Here, students will be learning the procedure to set-up and perform a <em>de novo</em> genome assembly using cleaned PE Illumina reads as implemented in <code>SOAPdenovo2</code> <span class="citation">(<a href="#ref-Luo2012" role="doc-biblioref">Luo et al., 2012</a>)</span>. The analysis will be conducted on the output files obtained after cleaning reads based on Phred scores and k-mer distributions (see section <a href="#dataPART2">6.7.2</a> for more details on the input files).</p>
<p><code>SOAPdenovo2</code> is the updated version of the program that was used by <span class="citation">Li, Fan, et al. (<a href="#ref-Li2009" role="doc-biblioref">2009</a>)</span> to assemble the nuclear genome of the giant panda. The draft nuclear genome of <em>Apostasia shenzhenica</em> was not inferred with this latter program, but with <code>ALLPATHS-LG</code> <span class="citation">(<a href="#ref-Gnerre1513" role="doc-biblioref">Gnerre et al., 2011</a>)</span>. The draft assembly was obtained based on multiple Illumina short-reads libraries exhibiting different insert-sizes <span class="citation">(see <a href="#ref-Zhang2017" role="doc-biblioref">Zhang et al., 2017</a>)</span>. The authors have conducted a final round of gap closures to polish their assembly based on PacBio long-reads. It will therefore be interesting to compare our assembly based on one Illumina library with the one obtained by <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> inferred uisng a combination of short and long read data. Further information justifying using <code>SOAPdenovo2</code> instead of <code>ALLPATHS-LG</code> are provided below.</p>
</div>
<div id="background-on-soapdenovo2" class="section level4" number="6.7.4.2">
<h4><span class="header-section-number">6.7.4.2</span> Background on <code>SOAPdenovo2</code></h4>
<p><code>SOAPdenovo2</code> <span class="citation">(<a href="#ref-Luo2012" role="doc-biblioref">Luo et al., 2012</a>)</span> is a short-read genome assembler, which was especially developed to handle Illumina reads and to build <code>de novo</code> draft assemblies for human-sized genomes. This program creates new opportunities for building reference sequences and carrying out accurate analyses of unexplored genomes in a cost effective way. As stated above, <code>SOAPdenovo2</code> aims for large eukaryote genomes, but it also works well on bacteria and fungi genomes. It runs on 64-bit Linux system with a minimum of 5GB of physical memory. For big genomes like human (3.2Gbp), about 150GB of memory are required.</p>
<p>In our case, we have estimated the haploid genome size of <em>Apostasia shenzhenica</em> to be around 340Mbp in our previous tutorial. This latter evidence therefore suggests that is adapted to assemble a draft genome for this species of orchid.</p>
</div>
<div id="structure-of-soapdenovo2" class="section level4" number="6.7.4.3">
<h4><span class="header-section-number">6.7.4.3</span> Structure of <code>SOAPdenovo2</code></h4>
<p>This program is made up of six modules handling:</p>
<ol style="list-style-type: decimal">
<li>Read error correction.</li>
<li>de Bruijn graph construction.</li>
<li>Contig assembly.</li>
<li>Paired-end reads mapping.</li>
<li>Scaffold construction.</li>
<li>Gap closure.</li>
</ol>
<p>Please read <span class="citation">Luo et al. (<a href="#ref-Luo2012" role="doc-biblioref">2012</a>)</span> for more details on the approach implemented in <code>SOAPdenovo2</code>.</p>
</div>
<div id="why-are-we-using-soapdenovo2-to-assemble-this-genome" class="section level4" number="6.7.4.4">
<h4><span class="header-section-number">6.7.4.4</span> Why are we using SOAPdenovo2 to assemble this genome?</h4>
<p>Once reads are fully cleaned and ready for <strong>de novo</strong> assembly, one of the main questions that you will ask yourself is: What <strong>de novo</strong> program should I use to assemble my genome? Or in other words, which program best fit my data?</p>
<p>Most researchers working on Illumina short-reads are using either <code>SOAPdenovo2</code> or <code>ALLPATHS-LG</code> to assemble nuclear genomes of eukaryote species. Please find below a short comparison of both programs:</p>
<ul>
<li>Unlike <code>SOAPdenovo2</code>, <code>ALLPATHS‐-LG</code> requires high sequence coverage of the genome in order to compensate for the shortness of the Illumina reads. The precise coverage required depends on the length and quality of the paired reads, but typically is of the order 100x or above. This is raw read coverage, before any error correction or filtering.</li>
<li>Unlike <code>SOAPdenovo2</code>, <code>ALLPATHS‐-LG</code> requires a minimum of 2 paired-end libraries – one short and one long. The short library average separation size must be slightly less than twice the read size, such that the reads from a pair will likely overlap – for example, for 100 base reads the insert size should be 180 bases. The distribution of sizes should be as small as possible, with a standard deviation of less than 20%. The long library insert size should be approximately 3000 bases long and can have a larger size distribution. Additional optional longer insert libraries can be used to help disambiguate larger repeat structures and may be generated at lower coverage.</li>
</ul>
<p>Overall, because we estimated in the previous tutorial that this library provided a coverage depth of ca. 25x for the haploid genome and that we only have one paired-end library, <code>SOAPdenovo2</code> therefore best fits our data.</p>
</div>
<div id="running-soapdenovo2-analysis" class="section level4" number="6.7.4.5">
<h4><span class="header-section-number">6.7.4.5</span> Running <code>SOAPdenovo2</code> analysis</h4>
<p>Running a <code>SOAPdenovo2</code> analysis is a three steps process:</p>
<ul>
<li><strong>Step 1:</strong> Create a folder and copy the de-interleaved cleaned paired-end ‘fastq’ files.</li>
<li><strong>Step 2:</strong> Create a <code>SOAPdenovo2</code> configuration file providing settings for the analysis.</li>
<li><strong>Step 3:</strong> Run the <em>de novo</em> genome assembly analysis.</li>
</ul>
<div id="step-1" class="section level5" number="6.7.4.5.1">
<h5><span class="header-section-number">6.7.4.5.1</span> Step 1</h5>
<p>Use the <code>mkdir</code> command to create a new folder entitled <code>SOAPdenovo/</code> in <code>~/Documents/02_Kmers_analyses/</code> as follows:</p>
<pre class="bash"><code>mkdir SOAPdenovo2</code></pre>
<p>Copy the paired-end cleaned reads files (i.e. input files for the <em>de novo</em> analysis) in <code>SOAPdenovo2/</code> using the <code>cp</code> command as follows:</p>
<pre class="bash"><code>#Copy the split pe files in SOAPdenovo2
cp ~/Documents/02_Kmers_analyses/khmer/SRR5759389.pe* ~/Documents/02_Kmers_analyses/SOAPdenovo2/</code></pre>
</div>
<div id="step-2" class="section level5" number="6.7.4.5.2">
<h5><span class="header-section-number">6.7.4.5.2</span> Step 2</h5>
<p><code>SOAPdenovo2</code> requires a configuration file to run the analysis. Please create a file entitled <code>soap_config.txt</code> deposited in <code>SOAPdenovo2/</code> using <code>vim</code>. Fill the latter file with the information provided below to set-up the analysis:</p>
<pre class="bash"><code>max_rd_len=90          # maximal read length
[LIB]            # One [LIB] section per library
avg_ins=180             # average insert size
reverse_seq=0           # if sequence needs to be reversed
asm_flags=3             # use for contig building and subsequent scaffolding
rank=1                  # in which order the reads are used while scaffolding
q1=SRR5759389.pe1.clean.fastq
q2=SRR5759389.pe2.clean.fastq</code></pre>
<p>To better understand the settings contained in this file, comments have been provided to explain each parameter. The last 2 lines contain the names of the input files. This is a very simple analysis based only on one paired-end Illumina library (<code>[LIB]</code>).</p>
</div>
<div id="step-3" class="section level5" number="6.7.4.5.3">
<h5><span class="header-section-number">6.7.4.5.3</span> Step 3</h5>
<p>Once the input and configuration files are sorted, just type the following command in your terminal to start the analysis:</p>
<pre class="bash"><code>#Run the de novo analysis
soapdenovo2-63mer all -s soap_config.txt -K 63 -R -p 16 -o SRR5759239_assembly</code></pre>
<p>Here, <code>SOAPdenovo2</code> works with a k-mer size of 63 (<code>-K</code>). The analysis will be performed on 16 threads (or CPUs) (<code>-p</code>) and the program will save the output files under <code>SRR5759239_assembly</code> as shown by the <code>-o</code> argument.</p>
<p>The analysis will run for SEVERAL HOURS. Please log out from your sessions, but DON’T SWITCH OFF COMPUTERS.</p>
<p>The output file of the <code>SOAPdenovo2</code> analysis are presented in section <a href="#outputdenovo">6.8.2</a>.</p>
</div>
</div>
</div>
</div>
<div id="PART3" class="section level2" number="6.8">
<h2><span class="header-section-number">6.8</span> PART 3: Validation of the draft genome</h2>
<div id="objective-1" class="section level3" number="6.8.1">
<h3><span class="header-section-number">6.8.1</span> Objective</h3>
<p>In this part, students are learning how to use <code>QUAST</code> to assess the quality of the <em>de novo</em> genome assembly inferred in PART 2 using <code>SOAPdenovo2</code>. <code>QUAST</code> is used to validate the assembly (based on one PE Illumina library; <code>SRR5759389</code>) by comparing it with the assembly published by <span class="citation">Zhang et al. (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> (available under the GenBank accession number: <code>PEFY01</code>). Such approach is especially useful to assess the level of misassemblies (see Figure <a href="#fig:assemblyworkflow">6.10</a>). Please keep in mind that this assembly has only been inferred from one PE Illumina library.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:assemblyworkflow"></span>
<img src="Images/Assembly_workflow.png" alt="Overview of the assembly workflow. Here are few key definitions: Contig: A contiguous sequence of bases. Unitig: A type of contig for which there are no competing choices in terms of internal overlaps (they usually stop before a repeat sequence). Scaffold: A sequence of contigs separated by gaps (Ns). See PART 2 for more details." width="80%" />
<p class="caption">
Figure 6.10: Overview of the assembly workflow. Here are few key definitions: Contig: A contiguous sequence of bases. Unitig: A type of contig for which there are no competing choices in terms of internal overlaps (they usually stop before a repeat sequence). Scaffold: A sequence of contigs separated by gaps (Ns). See PART 2 for more details.
</p>
</div>
</div>
<div id="outputdenovo" class="section level3" number="6.8.2">
<h3><span class="header-section-number">6.8.2</span> <code>SOAPdenovo2</code> assembly files</h3>
<p>The input of the <code>QUAST</code> analysis is the <code>fasta</code> assembly file produced by <code>SOAPdenovo2</code> containing scaffolds (see Figure <a href="#fig:assemblyworkflow">6.10</a>). This file is entitled <code>SRR5759239_assembly.scafSeq</code> and it is located in the <code>SOAPdenovo2/</code> folder (see PART 2 for more details).</p>
<p>Navigate to the file location and inspect it by using the <code>head</code> command. It should be similar to the example provided below:</p>
<pre class="bash"><code>&gt;scaffold2  6.9
TATATATATTAGCATTAAATTAACTTTATTTTCATTCTTTCACAAAGCTCGAAACAAAGAT
ATCACCAAACAACTATGGCTTAGCAAACACATCATTTATACGAGAAATATTAACGATTTG
AGTGGTGTATGTAGGTTTATGTTCTGAAGTTAACTCGCCCTTTTTTTTTTTTTTCTTAAGT
ACTGATTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTCTTCTTCT
TCTAATCATAATATATGTGGTGCTTGCATTTGTGTGCCATGGTGTCACCCTCCCTTCTTG
CATTGGCCATTGACTCTACTCTTCCCCTCATCAGTGTCTTCCTGTTTAAATTGTAAACCA</code></pre>
<p>You can also inspect some general statistics on your assembly (both at scaffold and contig levels) by consulting <code>SRR5759239_assembly.scafStatistics</code>. Please find below a print out of the content of this latter file:</p>
<pre class="bash"><code>&lt;-- Information for assembly Scaffold &#39;SRR5759239_assembly.scafSeq&#39;.(cut_off_length &lt; 100bp) --&gt;

Size_includeN   319316154
Size_withoutN   316496824
Scaffold_Num    206180
Mean_Size       1548
Median_Size     339
Longest_Seq     58464
Shortest_Seq    100
Singleton_Num   148213
Average_length_of_break(N)_in_scaffold  13

Known_genome_size       NaN
Total_scaffold_length_as_percentage_of_known_genome_size        NaN

scaffolds&gt;100   205022  99.44%
scaffolds&gt;500   84690   41.08%
scaffolds&gt;1K    60675   29.43%
scaffolds&gt;10K   6062    2.94%
scaffolds&gt;100K  0       0.00%
scaffolds&gt;1M    0       0.00%

Nucleotide_A    105171893       32.94%
Nucleotide_C    54211428        16.98%
Nucleotide_G    53194169        16.66%
Nucleotide_T    103919334       32.54%
GapContent_N    2819330 0.88%
Non_ACGTN       0       0.00%
GC_Content      33.94%          (G+C)/(A+C+G+T)

N10     17086   1447
N20     12498   3661
N30     9573    6593
N40     7459    10380
N50     5728    15270
N60     4208    21769
N70     2902    30880
N80     1712    45089
N90     649     74738

NG50    NaN     NaN
N50_scaffold-NG50_scaffold_length_difference    NaN

&lt;-- Information for assembly Contig &#39;SRR5759239_assembly.contig&#39;.(cut_off_length &lt; 100bp) --&gt;

Size_includeN   323703442
Size_withoutN   323703442
Contig_Num      326278
Mean_Size       992
Median_Size     328
Longest_Seq     32360
Shortest_Seq    100

Contig&gt;100      324883  99.57%
Contig&gt;500      125875  38.58%
Contig&gt;1K       81379   24.94%
Contig&gt;10K      1818    0.56%
Contig&gt;100K     0       0.00%
Contig&gt;1M       0       0.00%

Nucleotide_A    108192955       33.42%
Nucleotide_C    55873838        17.26%
Nucleotide_G    54056600        16.70%
Nucleotide_T    105580049       32.62%
GapContent_N    0       0.00%
Non_ACGTN       0       0.00%
GC_Content      33.96%          (G+C)/(A+C+G+T)

N10     8835    2797
N20     6338    7177
N30     4798    13093
N40     3648    20857
N50     2743    31105
N60     1995    44959
N70     1350    64640
N80     786     95910
N90     342     158749

NG50    NaN     NaN
N50_contig-NG50_contig_length_difference        NaN

Number_of_contigs_in_scaffolds  178065
Number_of_contigs_not_in_scaffolds(Singleton)   148213
Average_number_of_contigs_per_scaffold  3.1</code></pre>
<div id="questions-1" class="section level4" number="6.8.2.1">
<h4><span class="header-section-number">6.8.2.1</span> Questions</h4>
<blockquote>
<p>Why does the scaffold sequence contain Ns?</p>
</blockquote>
<p>Use the workflow presented in Figure <a href="#fig:assemblyworkflow">6.10</a> to answer this question.</p>
<blockquote>
<p>What is the assembly length and its overall quality? What kind of analyses could be conducted on this assembly?</p>
</blockquote>
</div>
</div>
<div id="introduction-to-quast" class="section level3" number="6.8.3">
<h3><span class="header-section-number">6.8.3</span> Introduction to QUAST</h3>
<p><code>QUAST</code> stands for <strong>QU</strong>ality <strong>AS</strong>sessment <strong>T</strong>ool for genome assemblies <span class="citation">(<a href="#ref-QUAST" role="doc-biblioref">Gurevich et al., 2013</a>)</span>. The manual of this program is available here: <a href="http://quast.sourceforge.net/docs/manual.html" class="uri">http://quast.sourceforge.net/docs/manual.html</a></p>
<p><code>QUAST</code> is using <code>MUMmer v3.23</code> <span class="citation">(<a href="#ref-Kurtz2004" role="doc-biblioref">Kurtz et al., 2004</a>)</span> to align assemblies to a reference genome and to provide metrics depending on alignments. Unlike other similar software, <code>QUAST</code> can also compute metrics that are useful for assessing assemblies of previously un-sequenced species (i.e. non model organisms).</p>
<div id="quast-metrics" class="section level4" number="6.8.3.1">
<h4><span class="header-section-number">6.8.3.1</span> <code>QUAST</code> metrics</h4>
<p>The metrics evaluated by <code>QUAST</code> can be subidvided into several groups as follows:</p>
<ul>
<li><strong>Contig sizes:</strong> The following metrics (except for NGx) can be evaluated with or without a reference genome.
<ul>
<li>No. of contigs: The total number of contigs in the assembly.</li>
<li>Largest contig: The length of the largest contig in the assembly.</li>
<li>Total length: The total number of bases in the assembly.</li>
<li>Nx (where <span class="math inline">\(0\leq x \leq 100\)</span>): The largest contig length, <span class="math inline">\(L\)</span>, such that using contigs of length <span class="math inline">\(\geq L\)</span> accounts for at least <span class="math inline">\(x\)</span>% of the bases of the assembly.</li>
<li>NGx, Genome Nx: The contig length such that using equal or longer length contigs produces <span class="math inline">\(x\)</span>% of the length of the reference genome, rather than <span class="math inline">\(x\)</span>% of the assembly length.</li>
<li>N50: It is the length for which the collection of all contigs of that length or longer covers at least half an assembly.</li>
<li>NG50: It is the length for which the collection of all contigs of that length or longer covers at least half the reference genome. This metric is computed only if the reference genome is provided.</li>
</ul></li>
<li><strong>Misassemblies and structural variations:</strong> The following metrics describe structural errors in the contigs (see Figure <a href="#fig:assemblyworkflow">6.10</a>. <code>QUAST</code> can evaluate them only with respect to a known reference genome. If the reference genome exactly matches the dataset being assembled, differences may be attributed to misassemblies by the software or to sequencing errors, such as chimeric reads. Sometimes one uses a reference genome that is related to but different than the dataset being sequenced. In this case, the differences may still be misassemblies, but they may also be true structural variations, such as rearrangements, large indels, different repeat copy numbers and so forth.
<ul>
<li>No. of misassemblies: The number of misassemblies. A misassembly breakpoint is defined as a position in the assembled contigs where the left flanking sequence aligns over 1 kb away from the right flanking sequence on the reference, or they overlap by <span class="math inline">\(&gt; 1\)</span> kb, or the flanking sequences align on opposite strands or different chromosomes.</li>
<li>No. of misassembled contigs: The number of contigs that contain misassembly breakpoints.</li>
<li>Misassembled contigs length: The total number of bases contained in all contigs that have one or more misassemblies.</li>
<li>No. of unaligned contigs: The number of contigs that have no alignment to the reference sequence.</li>
<li>No. of ambiguously mapped contigs: The number of contigs that have high-scoring reference alignments of equal quality in multiple locations on the reference genome.</li>
<li>In addition to these summary statistics, <code>QUAST</code> also generates reports with detailed information about each contig, including whether the contig is unaligned, ambiguously mapped, misassembled or correct.</li>
</ul></li>
<li><strong>Genome representation:</strong> This section lists metrics evaluating genome representation in contigs. Most of these latter metrics require a reference genome.
<ul>
<li>Genome fraction (%): The total number of aligned bases in the reference, divided by the genome size. A base in the reference genome is counted as aligned if at least one contig has at least one alignment to this base. Contigs from repeat regions may map to multiple places, and thus may be counted multiple times in this quantity.</li>
<li>Duplication ratio: The total number of aligned bases in the assembly (i.e. total length minus unaligned contigs length), divided by the total number of aligned bases in the reference (see genome fraction metric above). If the assembly contains many contigs that cover the same regions of the reference, its duplication ratio may be much <span class="math inline">\(&gt;1\)</span>. This may occur due to overestimating repeat multiplicities and due to small overlaps between contigs, among other reasons.</li>
<li>GC (%): The total number of G and C nucleotides in the assembly, divided by the total length of the assembly. This metric can be computed without a reference genome.</li>
<li>No. of mismatches per 100 kb: The average number of mismatches per 100,000 aligned bases. <code>QUAST</code> also generates a more detailed report with the coordinates of mismatches. This metric does not distinguish between single-nucleotide polymorphisms, which are true differences in the assembled genome versus the reference genome, and single-nucleotide errors, which are due to errors in reads or errors in the assembly algorithm.</li>
<li>No. of indels per 100 kb: The average number of single nucleotide insertions or deletions per 100,000 aligned bases. A detailed report with coordinates of indels for all assemblies is also available.</li>
</ul></li>
<li><strong>Visualization:</strong> <code>QUAST</code> presents a number of statistics in graphical form and supports <code>svg</code>, <code>png</code> and <code>pdf</code> formats. These plots are divided into several groups as follows:
<ul>
<li>Nx-like plots: These show the trends of Nx and NGx, metrics as <span class="math inline">\(x\)</span> varies. This is more informative than just using N50.</li>
<li>Cumulative plots: Contigs are ordered from largest to smallest (in number of bases) for all the types of cumulative plots considered. The cumulative length plot shows the number of bases in the first <span class="math inline">\(x\)</span> contigs, as <span class="math inline">\(x\)</span> varies from zero to the number of contigs.</li>
<li>GC content plots: These show the distribution of GC content in the contigs. The <span class="math inline">\(x\)</span> value shows the per cent of GC (from 0 to 100). The <span class="math inline">\(y\)</span> value shows the number of non-overlapping 100 bp windows whose GC content is <span class="math inline">\(x\)</span>. This distribution is often Gaussian; however, if there are contaminants with a different GC content, there will often be a superposition of multiple Gaussians.</li>
<li>Contig alignment plots: These show alignment of contigs to the reference genome and the positions of misassemblies in these contigs. Colour coding indicates when block boundaries are shared by multiple assemblies, and to show misassembled blocks.</li>
</ul></li>
</ul>
</div>
</div>
<div id="running-quast" class="section level3" number="6.8.4">
<h3><span class="header-section-number">6.8.4</span> Running QUAST</h3>
<p><code>QUAST</code> is written in <code>python</code> and can be run on Linux or MacOS. The assembly and reference genome files have to be in <code>fasta</code> formats (they can be compressed). The output files are in <code>html</code> and <code>pdf</code> formats.</p>
<div id="fitting-the-quast-analysis-to-our-data-and-resources" class="section level4" number="6.8.4.1">
<h4><span class="header-section-number">6.8.4.1</span> Fitting the QUAST analysis to our data and resources</h4>
<p>In this section, we are reviewing our data and computational resources and learning the procedure to best-fit those to the options provided in the <code>QUAST</code> program:</p>
<ul>
<li>The <code>SOAPdenovo2</code> assembly is made of scaffolds and not contigs. This will have to be stated by using the <code>--scaffolds</code> option.</li>
<li>The reference nuclear genome (<code>PEFY01.1.fsa_nt.gz</code>) is compressed. <code>QUAST</code> can handle compressed files, but not all software can.</li>
<li>The reference genome is fragmented (i.e. there are still gaps). This will have to be stated by using the <code>--fragmented</code> option.</li>
<li><code>QUAST</code> can be run in parallel (i.e. simultaneously using multiple threads/CPUs to speed up the analysis). We will take advantage of the 32 cores of our computers and set the <code>--threads</code> option to 20.</li>
</ul>
</div>
<div id="setting-up-the-analysis" class="section level4" number="6.8.4.2">
<h4><span class="header-section-number">6.8.4.2</span> Setting-up the analysis</h4>
<p><code>QUAST</code> runs from a command line as follows:</p>
<pre class="bash"><code>#General syntax
quast.py [options] &lt;contig_file(s)&gt;</code></pre>
<p><strong>Options relevant to our data are as follows:</strong></p>
<ul>
<li><code>-R &lt;path&gt;</code>: Reference genome file. Optional. Many metrics can’t be evaluated without a reference. If this is omitted, <code>QUAST</code> will only report the metrics that can be evaluated without a reference.</li>
<li><code>--threads (or -t) &lt;int&gt;</code>: Maximum number of threads. The default value is 25% of all available CPUs but not less than 1. If <code>QUAST</code> fails to determine the number of CPUs, maximum threads number is set to 4.</li>
<li><code>--scaffolds (or -s)</code>: The assemblies are scaffolds (rather than contigs). <code>QUAST</code> will add split versions of assemblies to the comparison (named <code>&lt;assembly_name&gt;_broken</code>). Assemblies are split by continuous fragments of N’s of length <span class="math inline">\(\geq 10\)</span>. <strong>This is especially important to assess the level of misassemblies constructed by the algorithm.</strong></li>
<li><code>--fragmented</code>: Reference genome is fragmented (e.g. a scaffold reference). <code>QUAST</code> will try to detect misassemblies caused by the fragmentation and mark them fake (will be excluded from <code># misassemblies</code>).</li>
</ul>
<p>If you want to get more information about options available in <code>QUAST</code>, please consult the <a href="http://quast.sourceforge.net/docs/manual.html">manual</a>.</p>
<p>The command provided below will run a <code>QUAST</code> analysis on the <code>SOAPdenovo2</code> <em>de novo</em> assembly file using <code>PEFY01.1.fsa_nt.gz</code> as reference genome sequence. The output of the analysis will also contain statistics solely based on the <code>SOAPdenovo2</code> <em>de novo</em> assembly.</p>
<p>Before starting the <code>QUAST</code> analysis execute the following tasks:</p>
<ol style="list-style-type: decimal">
<li>Open a terminal window and go to <code>02_Kmers_analyses/</code>.</li>
<li>Create a new folder entitled <code>QUAST/</code> using the <code>mkdir</code> command.</li>
<li>Use the <code>cd</code> command to go to <code>QUAST/</code>.</li>
<li>Check that <code>PEFY01.1.fsa_nt.gz</code> is in this path: <code>~/Documents/02_Kmers_analyses/Map_reads/</code>.</li>
<li>Check that <code>SRR5759239_assembly.scafSeq</code> is in this path: <code>~/Documents/02_Kmers_analyses/SOAPdenovo2/</code>.</li>
</ol>
<p>Once you have completed all the tasks, execute the following command:</p>
<pre class="bash"><code>#Run QUAST on the SOAPdenovo2 assembly using the reference nuclear genome
quast.py -R ~/Documents/02_Kmers_analyses/Map_reads/PEFY01.1.fsa_nt.gz --threads 20 --scaffolds --fragmented ~/Documents/02_Kmers_analyses/SOAPdenovo2/SRR5759239_assembly.scafSeq  </code></pre>
</div>
</div>
<div id="quast-output-files" class="section level3" number="6.8.5">
<h3><span class="header-section-number">6.8.5</span> <code>QUAST</code> output files</h3>
<p>By default, the output files of the analysis will be saved in a folder named as follows:</p>
<pre class="bash"><code>quast_results/results_&lt;date_time&gt;</code></pre>
<p>In each folder, there will be a suite of output files, please find below the ones that are more relevant to our goals:</p>
<ul>
<li><code>report.txt</code>: Assessment summary in plain text format.</li>
<li><code>report.html</code>: HTML version of the report with interactive plots inside.</li>
<li><code>report.pdf</code>: All plots combined with all tables (see Figure <a href="#fig:QUAST">6.11</a>).</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:QUAST"></span>
<img src="Images/QUAST_output.png" alt="Screenshot of the report.html file showing the statistics and plots inferred by QUAST on the results of the SOAPdenovo2 assembly." width="80%" />
<p class="caption">
Figure 6.11: Screenshot of the report.html file showing the statistics and plots inferred by QUAST on the results of the SOAPdenovo2 assembly.
</p>
</div>
<ul>
<li><code>icarus.html</code>: Icarus main menu with links to interactive viewers. Icarus generates contig size viewer and one or more contig alignment viewers (if reference genome is provided). All Icarus viewers contain a legend with color scheme description (see Figure <a href="#fig:ICARUS">6.12</a>).</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ICARUS"></span>
<img src="Images/ICARUS_output.png" alt="Screenshot of the icarus.html file showing the alignment of the SOAPdenovo2 de novo assembly against the reference genome." width="80%" />
<p class="caption">
Figure 6.12: Screenshot of the icarus.html file showing the alignment of the SOAPdenovo2 de novo assembly against the reference genome.
</p>
</div>
<div id="task" class="section level4" number="6.8.5.1">
<h4><span class="header-section-number">6.8.5.1</span> Task</h4>
<p>Open <a href="Data/PART3_QUAST_results/report.html">report.html</a> and look at the statistics, plots and alignments (available by clicking on the <code>Icarus</code> tab; see Figure <a href="#fig:QUAST">6.11</a>).</p>
</div>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-FastQC" class="csl-entry">
<span class="smallcaps">Andrews, S.</span> 2012. <span class="nocase">FastQC A Quality Control tool for High Throughput Sequence Data</span>. Available at: <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>.
</div>
<div id="ref-khmer2015" class="csl-entry">
<span class="smallcaps">Crusoe, M.</span>, <span class="smallcaps">H. Alameldin</span>, <span class="smallcaps">S. Awad</span>, <span class="smallcaps">E. Boucher</span>, <span class="smallcaps">A. Caldwell</span>, <span class="smallcaps">R. Cartwright</span>, <span class="smallcaps">A. Charbonneau</span>, et al. 2015. <a href="https://doi.org/10.12688/f1000research.6924.1">The khmer software package: Enabling efficient nucleotide sequence analysis [version 1; referees: 2 approved, 1 approved with reservations]</a>. <em>F1000Research</em> 4:
</div>
<div id="ref-Gnerre1513" class="csl-entry">
<span class="smallcaps">Gnerre, S.</span>, <span class="smallcaps">I. MacCallum</span>, <span class="smallcaps">D. Przybylski</span>, <span class="smallcaps">F.J. Ribeiro</span>, <span class="smallcaps">J.N. Burton</span>, <span class="smallcaps">B.J. Walker</span>, <span class="smallcaps">T. Sharpe</span>, et al. 2011. High-quality draft assemblies of mammalian genomes from massively parallel sequence data. <em>Proceedings of the National Academy of Sciences</em> 108: 1513–1518. Available at: <a href="http://www.pnas.org/content/108/4/1513">http://www.pnas.org/content/108/4/1513</a>.
</div>
<div id="ref-QUAST" class="csl-entry">
<span class="smallcaps">Gurevich, A.</span>, <span class="smallcaps">V. Saveliev</span>, <span class="smallcaps">N. Vyahhi</span>, and <span class="smallcaps">G. Tesler</span>. 2013. QUAST: Quality assessment tool for genome assemblies. <em>Bioinformatics</em> 29: 1072–1075. Available at:<a href=" + http://dx.doi.org/10.1093/bioinformatics/btt086"> + http://dx.doi.org/10.1093/bioinformatics/btt086</a>.
</div>
<div id="ref-Kurtz2004" class="csl-entry">
<span class="smallcaps">Kurtz, S.</span>, <span class="smallcaps">A. Phillippy</span>, <span class="smallcaps">A.L. Delcher</span>, <span class="smallcaps">M. Smoot</span>, <span class="smallcaps">M. Shumway</span>, <span class="smallcaps">C. Antonescu</span>, and <span class="smallcaps">S.L. Salzberg</span>. 2004. Versatile and open software for comparing large genomes. <em>Genome Biology</em> 5: R12. Available at: <a href="https://doi.org/10.1186/gb-2004-5-2-r12">https://doi.org/10.1186/gb-2004-5-2-r12</a>.
</div>
<div id="ref-bwa2009" class="csl-entry">
<span class="smallcaps">Li, H.</span>, and <span class="smallcaps">R. Durbin</span>. 2009. Fast and accurate short read alignment with burrows–wheeler transform. <em>Bioinformatics</em> 25: 1754–1760. Available at:<a href=" + http://dx.doi.org/10.1093/bioinformatics/btp324"> + http://dx.doi.org/10.1093/bioinformatics/btp324</a>.
</div>
<div id="ref-samtools2009" class="csl-entry">
<span class="smallcaps">Li, H.</span>, <span class="smallcaps">B. Handsaker</span>, <span class="smallcaps">A. Wysoker</span>, <span class="smallcaps">T. Fennell</span>, <span class="smallcaps">J. Ruan</span>, <span class="smallcaps">N. Homer</span>, <span class="smallcaps">G. Marth</span>, et al. 2009. The sequence alignment/map format and SAMtools. <em>Bioinformatics</em> 25: 2078–2079. Available at:<a href=" + http://dx.doi.org/10.1093/bioinformatics/btp352"> + http://dx.doi.org/10.1093/bioinformatics/btp352</a>.
</div>
<div id="ref-Li2009" class="csl-entry">
<span class="smallcaps">Li, R.</span>, <span class="smallcaps">W. Fan</span>, <span class="smallcaps">G. Tian</span>, <span class="smallcaps">H. Zhu</span>, <span class="smallcaps">L. He</span>, <span class="smallcaps">J. Cai</span>, <span class="smallcaps">Q. Huang</span>, et al. 2009. The sequence and de novo assembly of the giant panda genome. <em>Nature</em> 463: 311 EP–. Available at: <a href="http://dx.doi.org/10.1038/nature08696">http://dx.doi.org/10.1038/nature08696</a>.
</div>
<div id="ref-Luo2012" class="csl-entry">
<span class="smallcaps">Luo, R.</span>, <span class="smallcaps">B. Liu</span>, <span class="smallcaps">Y. Xie</span>, <span class="smallcaps">Z. Li</span>, <span class="smallcaps">W. Huang</span>, <span class="smallcaps">J. Yuan</span>, <span class="smallcaps">G. He</span>, et al. 2012. SOAPdenovo2: An empirically improved memory-efficient short-read de novo assembler. <em>GigaScience</em> 1: 18. Available at: <a href="https://doi.org/10.1186/2047-217X-1-18">https://doi.org/10.1186/2047-217X-1-18</a>.
</div>
<div id="ref-Jellyfish2011" class="csl-entry">
<span class="smallcaps">Marçais, G.</span>, and <span class="smallcaps">C. Kingsford</span>. 2011. A fast, lock-free approach for efficient parallel counting of occurrences of k-mers. <em>Bioinformatics</em> 27: 764–770. Available at:<a href=" + http://dx.doi.org/10.1093/bioinformatics/btr011"> + http://dx.doi.org/10.1093/bioinformatics/btr011</a>.
</div>
<div id="ref-Niu2017" class="csl-entry">
<span class="smallcaps">Niu, Z.</span>, <span class="smallcaps">J. Pan</span>, <span class="smallcaps">S. Zhu</span>, <span class="smallcaps">L. Li</span>, <span class="smallcaps">Q. Xue</span>, <span class="smallcaps">W. Liu</span>, and <span class="smallcaps">X. Ding</span>. 2017. Comparative analysis of the complete plastomes of apostasia wallichii and neuwiedia singapureana (apostasioideae) reveals different evolutionary dynamics of IR/SSC boundary among photosynthetic orchids. <em>Frontiers in Plant Science</em> 8: 1713. Available at: <a href="https://www.frontiersin.org/article/10.3389/fpls.2017.01713">https://www.frontiersin.org/article/10.3389/fpls.2017.01713</a>.
</div>
<div id="ref-genomescope" class="csl-entry">
<span class="smallcaps">Vurture, G.W.</span>, <span class="smallcaps">F.J. Sedlazeck</span>, <span class="smallcaps">M. Nattestad</span>, <span class="smallcaps">C.J. Underwood</span>, <span class="smallcaps">H. Fang</span>, <span class="smallcaps">J. Gurtowski</span>, and <span class="smallcaps">M.C. Schatz</span>. 2017. GenomeScope: Fast reference-free genome profiling from short reads. <em>Bioinformatics</em> 33: 2202–2204. Available at:<a href=" + http://dx.doi.org/10.1093/bioinformatics/btx153"> + http://dx.doi.org/10.1093/bioinformatics/btx153</a>.
</div>
<div id="ref-Zhang2017" class="csl-entry">
<span class="smallcaps">Zhang, G.-Q.</span>, <span class="smallcaps">K.-W. Liu</span>, <span class="smallcaps">Z. Li</span>, <span class="smallcaps">R. Lohaus</span>, <span class="smallcaps">Y.-Y. Hsiao</span>, <span class="smallcaps">S.-C. Niu</span>, <span class="smallcaps">J.-Y. Wang</span>, et al. 2017. The apostasia genome and the evolution of orchids. <em>Nature</em> 549: 379 EP–. Available at: <a href="http://dx.doi.org/10.1038/nature23897">http://dx.doi.org/10.1038/nature23897</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
