<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki - Boise State University" />

<meta name="date" content="2021-03-22" />

<title>BIOL 497/597 - Genomics &amp; Bioinformatics</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">BIOL 497/597 - Genomics &amp; Bioinformatics</h1>
<h3 class="subtitle">Chapters</h3>
<h4 class="author">Sven Buerki - Boise State University</h4>
<h4 class="date">2021-03-22</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div id="chapter-4" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Chapter 4</h1>
<div id="objectives" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Objectives</h2>
<p>In this chapter, we aim at training students in producing a draft nuclear genome for a non-model organism using Illumina data.</p>
<p>The whole-genome shotgun (WGS) sequencing dataset published by <span class="citation"><a href="#ref-Zhang2017" role="doc-biblioref">Zhang et al.</a> (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> on the orchid species <em>Apostasia shenzhenica</em> is used as case-study. This dataset provides an opportunity to gain firsthand experience in analyzing NGS data. <span class="citation"><a href="#ref-Zhang2017" role="doc-biblioref">Zhang et al.</a> (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span> have also produced RNA-Seq data, which were used in their study to support genome annotation.</p>
<p>The chapter is subdivided into three parts:</p>
<ul>
<li><strong>PART 1:</strong> Preparing/cleaning Illumina reads for <em>de novo</em> nuclear genome assembly and inferring genome size and complexity.</li>
<li><strong>PART 2:</strong> <em>De novo</em> genome assembly.</li>
<li><strong>PART 3:</strong> Validation of draft genome.</li>
</ul>
<p>Before you start working on Chapter 4, please make sure that you have completed the tutorials available <a href="Tutorials.html">here</a>.</p>
</div>
<div id="presenting-the-ngs-data" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Presenting the NGS data</h2>
<p>Details on the WGS PE Illumina library studied here are provided in Figure <a href="#fig:SRA">1.1</a>. The SRA accession number is <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR5759389">SRR5759389</a> and the most important information to know about this data are that the fragments insert-size of the library is 180 bp and that the length of each PE read is 90 bp. Note that the authors have performed a round of PCR amplification during their library preparation. This step is know to potentially introduce errors in reads, which could be identified by performing k-mer analyses.</p>
<p>The number of bases shown in Figure <a href="#fig:SRA">1.1</a> was inferred as follows: <span class="math inline">\(\text{N bases} = \text{N. spots * reads length (bp)}\)</span>. In this example, <span class="math inline">\(\text{N bases} = 84.1e6 * 180 (90 + 90) = 15.1e9bp = 15.1Gbp\)</span>.</p>
<p>You can also obtain a rough estimate of haploid genome coverage (x) by using the following equation: <span class="math inline">\(\text{Raw haploid genome coverage (x)} = \text{N bases / Genome size (haploid)}\)</span>. In this example, <span class="math inline">\(\text{Raw haploid genome coverage (x)} = 15.1e9 / 471.0e6 = 32x\)</span>. The estimation of haploid genome size was taken from <span class="citation"><a href="#ref-Zhang2017" role="doc-biblioref">Zhang et al.</a> (<a href="#ref-Zhang2017" role="doc-biblioref">2017</a>)</span>.</p>
<div class="figure" style="text-align: center"><span id="fig:SRA"></span>
<img src="Images/SRR5759389_details.png" alt="Details on the WGS library (SRA accession number: SRR5759389) used in this tutorial. " width="100%" />
<p class="caption">
Figure 1.1: Details on the WGS library (SRA accession number: SRR5759389) used in this tutorial.
</p>
</div>
</div>
<div id="write-your-own-scripts" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Write your own scripts</h2>
<p>As you go along this document and perform analyses, please COPY ALL COMMAND LINES into an <code>.Rmd</code> document saved in a folder entitled <code>Report/</code> (create this folder in your working directory; see below). Remember to comment your script (using #s). This will greatly help you in repeating your analyses or using parts of your code to create new scripts. Enjoy scripting!</p>
</div>
<div id="part-1" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> PART 1</h2>
<div id="analytical-workflow" class="section level3" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Analytical workflow</h3>
<p>The overarching objective of this tutorial is to gain theoretical and bioinformatics knowledge on the steps required to prepare PE Illumina reads for <em>de novo</em> nuclear genome assembly as well as infer genome size and complexity (see Figure <a href="#fig:part1">1.2</a>).</p>
<div class="figure" style="text-align: center"><span id="fig:part1"></span>
<img src="Images/Chapter_4_Part1_steps_covered.png" alt="Workflow applied to assemble and annotate a genome for non-model organisms. Steps associated to PART 1 are highlighted in grey." width="80%" />
<p class="caption">
Figure 1.2: Workflow applied to assemble and annotate a genome for non-model organisms. Steps associated to PART 1 are highlighted in grey.
</p>
</div>
<p>The five main steps associated to the objectives of PART 1 are as follows (see Figure <a href="#fig:part1workflow">1.3</a>):</p>
<ul>
<li><strong>Step 1:</strong> Download SRA file containing raw WGS data and simultaneously convert it into an interleaved PE <code>fasqt</code> format file. This latter format (where both reads are paired and combined into one file) is the input for most bioinformatic programs.</li>
<li><strong>Step 2:</strong> Infer reads Quality Checks (QCs) using standard statistics implemented in <code>FastQC</code> <span class="citation">(<a href="#ref-FastQC" role="doc-biblioref">Andrews, 2012</a>)</span>.</li>
<li><strong>Step 3:</strong> Perform reads cleaning based on Phread scores and k-mer frequencies.</li>
<li><strong>Step 4:</strong> What’s “in” the reads? This will be done in two phases as follows:
<ul>
<li>Infer and plot clean reads GC contents to assess potential contamination patterns.</li>
<li>Map clean reads against reference genomes to assess the proportions of reads belonging to either nuclear or chloroplast genomes.</li>
</ul></li>
<li><strong>Step 5:</strong> Estimate genome size and complexity (especially looking at repetitive elements and heterozygosity rate) using k-mer frequencies.</li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:part1workflow"></span>
<img src="Images/Cleaning_reads.png" alt="Overview of the analytical workflow applied here to prepare/clean Illumina reads for genome assembly and inferring genome size and complexity. Details on associated bioinformatic tools (here software) and file formats are also provided." width="100%" />
<p class="caption">
Figure 1.3: Overview of the analytical workflow applied here to prepare/clean Illumina reads for genome assembly and inferring genome size and complexity. Details on associated bioinformatic tools (here software) and file formats are also provided.
</p>
</div>
</div>
<div id="location-of-the-data-on-linux-computers" class="section level3" number="1.4.2">
<h3><span class="header-section-number">1.4.2</span> Location of the data on Linux computers</h3>
<p>A folder entitled <code>Kmers_analyses/</code> (deposited onto your <code>bioinformatics</code> sessions) is located in <code>Documents/</code> and contains all the required files and programs to conduct the analyses. The instructor encourages you to explore the content of this folder using the UNIX command <code>ls -al</code>. Within each folder associated with each major step of this tutorial, there is a folder entitled <code>output/</code> containing the output files of the analyses. This is a back-up plan in case you have some troubles with the analyses or are running out of time. Finally, in the tutorial, the instructor will only refer to the name of the folder containing information about the analyses and not provide their full paths.</p>
</div>
<div id="bioinformatic-tools" class="section level3" number="1.4.3">
<h3><span class="header-section-number">1.4.3</span> Bioinformatic tools</h3>
<p>Although all the bioinformatic tools (software) necessary to complete this tutorial are already installed on your Linux computers, URLs to their repositories (incl. documentations) are provided below together with details on their publications (when available). Software are sorted by steps in our analytical workflow (see Figure <a href="#fig:part1workflow">1.3</a>):</p>
<ul>
<li><strong>Step 1:</strong> <code>fastq-dump</code>: <a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/" class="uri">https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/</a>.</li>
<li><strong>Step 2:</strong> <code>FastQC</code> <span class="citation">(<a href="#ref-FastQC" role="doc-biblioref">Andrews, 2012</a>)</span>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>.</li>
<li><strong>Step 3:</strong>
<ul>
<li><code>seqtk</code>: <a href="https://github.com/lh3/seqtk" class="uri">https://github.com/lh3/seqtk</a>.</li>
<li><code>khmer</code> <span class="citation">(<a href="#ref-khmer2015" role="doc-biblioref">Crusoe et al., 2015</a>)</span>: <a href="https://github.com/dib-lab/khmer" class="uri">https://github.com/dib-lab/khmer</a>.</li>
</ul></li>
<li><strong>Step 4:</strong>
<ul>
<li><code>bwa</code> <span class="citation">(<a href="#ref-bwa2009" role="doc-biblioref">Li and Durbin, 2009</a>)</span>: <a href="http://bio-bwa.sourceforge.net" class="uri">http://bio-bwa.sourceforge.net</a>.</li>
<li><code>samtools</code><span class="citation">(<a href="#ref-samtools2009" role="doc-biblioref">Li, Handsaker, et al., 2009</a>)</span>: <a href="http://samtools.sourceforge.net" class="uri">http://samtools.sourceforge.net</a>.</li>
</ul></li>
<li><strong>Step 5:</strong>
<ul>
<li><code>jellyfish</code> <span class="citation">(<a href="#ref-Jellyfish2011" role="doc-biblioref">Marçais and Kingsford, 2011</a>)</span>: <a href="http://www.genome.umd.edu/jellyfish.html" class="uri">http://www.genome.umd.edu/jellyfish.html</a>.</li>
<li><code>GenomeScope</code> <span class="citation">(<a href="#ref-genomescope" role="doc-biblioref">Vurture et al., 2017</a>)</span>: <a href="http://qb.cshl.edu/genomescope/" class="uri">http://qb.cshl.edu/genomescope/</a>.</li>
</ul></li>
</ul>
<p>The <code>R</code> software is also used in steps 4 and 5. You can find more details on this software in the <a href="index.html">Syllabus</a> and <a href="Mini_reports.html">Mini-Report</a> webpages.</p>
</div>
<div id="step-1-download-sra-file" class="section level3" number="1.4.4">
<h3><span class="header-section-number">1.4.4</span> Step 1: Download SRA file</h3>
<p>The SRA format (containing raw NGS data) can be quite difficult to play with. Here, we aim at downloading the WGS data, split PE reads, but store both reads (R1 and R2) in the same file using the interleaved format (Figure <a href="#fig:part1workflow">1.3</a>). This format is very convenient and is the entry point for most reads cleaning programs. This task can be done by using <code>fastq-dump</code> from the <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">SRA Toolkit</a>.</p>
<p>Before starting, open a Terminal and navigate to the <code>SRA/</code> folder using the <code>cd</code> command. Once there, execute the following command:</p>
<pre class="bash"><code>#Download PE reads (properly edited) in interleave format using SRA Toolkit
fastq-dump --split-files --defline-seq &#39;@$ac.$si.$sg/$ri&#39; --defline-qual &#39;+&#39; -Z SRR5759389  &gt; SRR5759389_pe12.fastq</code></pre>
<p>Downloading the file can take a while!</p>
</div>
<div id="Step2" class="section level3" number="1.4.5">
<h3><span class="header-section-number">1.4.5</span> Step 2: Reads QCs</h3>
<p>The program <code>FastQC</code> is used to obtain preliminary information about the Illumina library (see Figure <a href="#fig:SRA">1.1</a>). To run this program navigate to the <code>FastQC/</code> folder uisng <code>cd</code> and execute the following command:</p>
<pre class="bash"><code>#Reads QCs with FastQC 
./fastqc -o ~/Documents/Kmers_analyses/FastQC/ ~/Documents/Kmers_analyses/SRA/SRR5759389_pe12.fastq</code></pre>
<p>Please inspect the <code>SRR5759389_pe12_fastqc.html</code> output file. We will have a group discussion to inspect this file and draw general conclusions on the quality of the raw data. This step will have to be repeated after completion of the step3 (reads cleaning) to confirm that our reads trimming and cleaning procedure was successful.</p>
</div>
<div id="Step3" class="section level3" number="1.4.6">
<h3><span class="header-section-number">1.4.6</span> Step 3: Reads cleaning</h3>
<p>This step is at the core of our analytical workflow. It is paramount to conduct thorough reads cleaning when assembling a nuclear genome due to the very nature of this genome (i.e. repetitive elements, heterozygosity, recombination, etc…). In this case, aim at minimizing the effect of PCR errors, which could jeopardize our assembly by creating false polymorphism. In addition, <em>de novo</em> genome assembly is computing intensive and by properly cleaning reads we will dramatically decrease RAM requirements.</p>
<div id="approach-applied-here-to-clean-reads" class="section level4" number="1.4.6.1">
<h4><span class="header-section-number">1.4.6.1</span> Approach applied here to clean reads</h4>
<p>Reads will be cleaned/trimmed based on (see Figure <a href="#fig:part1workflow">1.3</a>):</p>
<ul>
<li>Phred quality scores (33) to conduct a first round of trimming.</li>
<li>K-mer frequencies (k=21) to:
<ul>
<li>Normalize high coverage reads (higher than 100x) based on median reads coverage.</li>
<li>Filter low abundance reads (where PCR errors will most likely take place).</li>
</ul></li>
<li>A final round of cleaning by removing low quality bases, short sequences, and non-paired reads.</li>
</ul>
<p>Finally, we will format the clean data for <em>de novo</em> genome assembly, which will be conducted using <code>SOAPdenovo2</code> <span class="citation">(<a href="#ref-Luo2012" role="doc-biblioref">Luo et al., 2012</a>)</span>.</p>
</div>
<div id="what-is-a-k-mer" class="section level4" number="1.4.6.2">
<h4><span class="header-section-number">1.4.6.2</span> What is a k-mer?</h4>
<p>A <code>k-mer</code> is a substring of length <span class="math inline">\(k\)</span> in a string of DNA bases. For a given sequence of length <span class="math inline">\(L\)</span>, and a k-mer size of <span class="math inline">\(k\)</span>, the total k-mer’s possible (<span class="math inline">\(n\)</span>) will be given by <span class="math inline">\((L-k) + 1\)</span>. For instance, in a sequence of length of 9 (<span class="math inline">\(L\)</span>), and a k-mer length of 2 (<span class="math inline">\(k\)</span>), the number of k-mer’s generated will be: <span class="math inline">\(n = (9-2) + 1 = 8\)</span>.</p>
<blockquote>
<p>All eight 2-mers of the sequence “AATTGGCCG” are:
AA, AT, TT, TG, GG, GC, CC, CG</p>
</blockquote>
<p>In most studies, the authors provide an estimate of sequencing coverage prior to assembly <span class="citation">(e.g. 73 fold in the case of the giant panda genome, <a href="#ref-Li2009" role="doc-biblioref">Li, Fan, et al., 2009</a>)</span>, but the real coverage distribution will be influenced by factors including DNA quality, library preparation and local GC content. On average, you might expect most of the genome (especially the single/low copy genes) to be covered between 20 and 70x. One way of rapidly examining the coverage distribution before assembling a reference genome is to chop your cleaned sequence reads into short “k-mers” of 21 nucleotides, and count how often you get each possible k-mer. By doing so, you will find out that:</p>
<ul>
<li>Many sequences are extremely rare (e.g., once). These are likely to be PCR errors that appeared during library preparation or sequencing, or could be rare somatic mutations). Such sequences can confuse assembly software; eliminating them can decrease subsequent memory and CPU requirements.</li>
<li>Other sequences may exist at 10,000x coverage. These could be pathogens or repetitive elements. Often, there is no benefit to retaining all copies of such sequences because the assembly software will be confused by them; while retaining a small proportion such reads could significantly reduce CPU, memory and space requirements.</li>
</ul>
<p>Please find below the plot of k-mer frequencies inferred from the trimmed library studied here (<code>SRR5759389</code>), which suggests that the haploid genome of <em>Apostasia shenzhenica</em> is sequenced at ca. 25x (Figure <a href="#fig:kmerstrimmed">1.4</a>). The methodology used to infer this graph is explained in Step 4.</p>
<div class="figure" style="text-align: center"><span id="fig:kmerstrimmed"></span>
<img src="Images/GenomeSize_trimmed.png" alt="Distribution of 21-mer frequencies based on the trimmed library with insert fragment size of 180 bp. Given only one peak in the k-mer distribution, we predict that the *Apostasia shenzhenica* genome has limited heterozygosity or in other words, that this genome is inbred." width="100%" />
<p class="caption">
Figure 1.4: Distribution of 21-mer frequencies based on the trimmed library with insert fragment size of 180 bp. Given only one peak in the k-mer distribution, we predict that the <em>Apostasia shenzhenica</em> genome has limited heterozygosity or in other words, that this genome is inbred.
</p>
</div>
</div>
<div id="k-mer-graph-and-pcr-effect" class="section level4" number="1.4.6.3">
<h4><span class="header-section-number">1.4.6.3</span> K-mer graph and PCR effect</h4>
<p>The peak around 25 in Figure <a href="#fig:kmerstrimmed">1.4</a> is the coverage with the highest number of different 21-mers. This means that there are <span class="math inline">\(1.4e^7\)</span> unique 21-mers (frequency) that have been observed 25 times (coverage). The normal-like distribution is due to the fact that the sequencing did not provide a perfect coverage of the whole genome. Some regions have less coverage, whereas others have a little more coverage, but the average coverage depth is around 25.</p>
<p>The large number of unique k-mers (<span class="math inline">\(1.7e^7\)</span>) that have a frequency of 1 (right left side of the graph; Figure <a href="#fig:kmerstrimmed">1.4</a>) is most likely due to PCR errors. Please find below an example explaining how PCR errors impact on the k-mer procedure.</p>
<p>All 3-mers of the sequence “AATTGGCCG” are:</p>
<ul>
<li>AAT, ATT, TTG, TGG, GGC, GCC, CCG</li>
</ul>
<p>Now, let’s consider that the 4th letter (<code>T</code>) in the sequence above is replaced with a <code>C</code> to simulate a PCR error. In this context, all 3-mers of this sequence “AAT<strong>C</strong>GGCCG” are:</p>
<ul>
<li>AAT, <strong>ATC</strong>, <strong>TCG</strong>, <strong>CGG</strong>, GGC, GCC, CCG.</li>
</ul>
<p>The k-mers in bold are the incorrect 3-mers that are now unique and end up at the beginning of the graph in Figure <a href="#fig:kmerstrimmed">1.4</a>.</p>
<p>Overall, the <strong>general rule</strong> is that for a given sequence, <strong>a single PCR error will result in <span class="math inline">\(k\)</span> unique and incorrect k-mers</strong>.</p>
</div>
<div id="trim-reads-based-on-phred-quality-scores" class="section level4" number="1.4.6.4">
<h4><span class="header-section-number">1.4.6.4</span> Trim reads based on Phred quality scores</h4>
<p>Let’s start the cleaning of our interleaved PE reads by trimming reads using Phred scores (33) as implemented in <code>seqtk</code>. Before executing the commands below, <code>cd</code> to the <code>khmer/</code> folder:</p>
<pre class="bash"><code>#Trim reads based on Phred scores (default of 33)
seqtk trimfq ~/Documents/Kmers_analyses/SRA/SRR5759389_pe12.fastq &gt; SRR5759389_pe12.trimmed.fastq</code></pre>
<p>Using Phred scores to clean our data is not going to deal with PCR errors and potential effects of high repeated DNA sequences on the <em>de novo</em> assembly. Below, we will use information on k-mer distributions to address these issues and clean our dataset accordingly.</p>
</div>
<div id="normalize-high-coverage-reads-remove-pcr-errors-and-low-quality-reads" class="section level4" number="1.4.6.5">
<h4><span class="header-section-number">1.4.6.5</span> Normalize high coverage reads, remove PCR errors and low quality reads</h4>
<p>In the next commands, <code>khmer</code> <span class="citation">(<a href="#ref-khmer2015" role="doc-biblioref">Crusoe et al., 2015</a>)</span> will be used to:</p>
<ol style="list-style-type: decimal">
<li>Normalize high coverage reads (&gt;100x) based on median coverage (to optimize RAM requirements for <em>de novo</em> genome assembly).</li>
<li>Filter low abundance reads (most likely PCR errors).</li>
<li>Final clean-up of reads to remove low quality bases, short sequences, and non-paired reads</li>
</ol>
<p>To conduct these analyses, run the following commands in the Terminal:</p>
<pre class="bash"><code>#1. Normalize high coverage reads (&gt;100x) based on median reads coverage using k-mer frequencies
khmer normalize-by-median.py -p --ksize 21 -C 100 -M 1e9 -s kmer.counts -o SRR5759389_pe12.max100.trimmed.fastq SRR5759389_pe12.trimmed.fastq</code></pre>
<pre class="bash"><code>#2. Filter low abundance reads based on k-mer frequencies to minimize PCR errors
khmer filter-abund.py -V kmer.counts -o SRR5759389_pe12.max100.trimmed.norare.fastq SRR5759389_pe12.max100.trimmed.fastq</code></pre>
<pre class="bash"><code>#3. Final clean-up of reads (remove low quality bases, short sequences, and non-paired reads)
seqtk seq -q 10 -N -L 80 SRR5759389_pe12.max100.trimmed.norare.fastq | seqtk dropse &gt; SRR5759389_pe12.max100.trimmed.norare.noshort.fastq</code></pre>
</div>
<div id="prepare-data-for-de-novo-genome-assembly" class="section level4" number="1.4.6.6">
<h4><span class="header-section-number">1.4.6.6</span> Prepare data for <em>de novo</em> genome assembly</h4>
<p>In this last section, our cleaned reads are prepared for the <em>de novo</em> genome assembly analysis conducted in <code>SOAPdenovo2</code> <span class="citation">(<a href="#ref-Luo2012" role="doc-biblioref">Luo et al., 2012</a>)</span> (done in PART 2). This is achieved by using <code>khmer</code> and renaming files using <code>mv</code> as follows:</p>
<pre class="bash"><code>#De-interleave filtered reads (to be ready for de novo assembly in SOAPdenovo2)
khmer split-paired-reads.py SRR5759389_pe12.max100.trimmed.norare.noshort.fastq

#Rename output reads to something more human-friendly
mv SRR5759389_pe12.max100.trimmed.norare.noshort.fastq.1 SRR5759389.pe1.clean.fastq
mv SRR5759389_pe12.max100.trimmed.norare.noshort.fastq.2 SRR5759389.pe2.clean.fastq</code></pre>
</div>
<div id="reads-qcs-on-cleaned-reads" class="section level4" number="1.4.6.7">
<h4><span class="header-section-number">1.4.6.7</span> Reads QCs on cleaned reads</h4>
<p>Use <code>FastQC</code> to validate the reads cleaning conducted above. Please see <a href="#Step2">Step 2</a> for more details on the methodology.</p>
</div>
</div>
<div id="step-4-whats-in-the-reads" class="section level3" number="1.4.7">
<h3><span class="header-section-number">1.4.7</span> Step 4: What’s “in” the reads?</h3>
<p>Before conducting the <em>de novo</em> genome assembly (using the cleaned reads produced in <a href="#Step3">Step 3</a>, it is paramount to assess whether the data are contaminated and also to provide a first estimate of proportions of reads belonging to either nuclear or plastid genomes. To investigate these topics, the instructor proposes that we assess i) potential contamination by inferring reads GC contents and ii) map the clean reads against nuclear and chloroplast genome assemblies to assess proportions of reads belonging to either genomes in our library.</p>
<div id="reads-gc-contents" class="section level4" number="1.4.7.1">
<h4><span class="header-section-number">1.4.7.1</span> Reads GC contents</h4>
<p>The GC content of a sequence library can provide evidence of contamination or the presence of sequence from multiple organisms (since most organisms have specific GC profiles). A GC plot inferred from a non-contaminated library would display a smooth, unimodal distribution. The existence of shoulders, or in more extreme cases a bimodal distribution, could be indicative of the presence of sequence reads from an organism with a different GC content, which is most likely a contaminant (see Figure <a href="#fig:part1workflow">1.3</a>).</p>
<p>Here we infer GC contents for all clean reads using the perl script <code>GC_content.pl</code>. This script requires reads to be in <code>fasta</code> format. We will then start by converting our <code>fastq</code> file into <code>fasta</code> by using <code>seqtk</code>. Data are available in the <code>GC_content/</code> folder. Please see below for the commands to be executed:</p>
<pre class="bash"><code>#Convert fastq to fasta
seqtk seq -a ~/Documents/Kmers_analyses/khmer/   SRR5759389_pe12.max100.trimmed.norare.noshort.fastq &gt; SRR5759389_pe12.max100.trimmed.norare.noshort.fasta</code></pre>
<p><strong>PLEASE DON’T EXECUTE THE COMMAND BELOW!</strong> It takes few hours to calculate reads GC contents. Use the output of this command <code>gc_out.txt</code>, which is available in the <code>GC_content/</code> folder to carry on our analyses.</p>
<pre class="bash"><code>#Infer GC content using fasta file as input (this can take a long time)
./GC_content.pl SRR5759389_pe12.max100.trimmed.norare.noshort.fasta &gt; gc_out.txt</code></pre>
<p>The next command extracts the column containing the GC content per read. Since the file is very big, we use the BASH command <code>awk</code> to extract this information instead of <code>R</code>.</p>
<pre class="bash"><code>#Extract reads GC contents (the file is big and can&#39;t be processed easily)
awk &#39;{print$2}&#39; gc_out.txt &gt; gc_simple.txt</code></pre>
<p>The last part of the analysis will be executed in <code>R</code> where we will load the data and create a histogram to look at the distribution of reads GC contents.</p>
<pre class="r"><code>###~~~
#Infer GC content per reads
###~~~
#Load the data in R
gc_frac &lt;- read.table(&#39;gc_simple.txt&#39;, header=T)

#Create pdf to save output of hist
pdf(&quot;GC_content_hist.pdf&quot;)
#Do the histogram of GC content 
hist(as.numeric(gc_frac[,1]), main=&quot;Histogram of GC content per reads&quot;, xlab=&quot;GC fraction&quot;)
#Close pdf file
dev.off()</code></pre>
<p>The histogram displaying the distribution of GC values based on cleaned reads is provided in Figure <a href="#fig:GC">1.5</a>.</p>
<div class="figure" style="text-align: center"><span id="fig:GC"></span>
<img src="Images/GC_content_hist.png" alt="Histogram of GC values inferred from the cleaned library of reads (SRR5759389). See text for more details." width="80%" />
<p class="caption">
Figure 1.5: Histogram of GC values inferred from the cleaned library of reads (SRR5759389). See text for more details.
</p>
</div>
<div id="question" class="section level5" number="1.4.7.1.1">
<h5><span class="header-section-number">1.4.7.1.1</span> Question</h5>
<blockquote>
<p>To your knowledge, based on data presented in Figure <a href="#fig:GC">1.5</a>, is the SRR5759389 library contaminated with alien DNA?</p>
</blockquote>
</div>
</div>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-FastQC" class="csl-entry">
<span class="smallcaps">Andrews, S.</span> 2012. <span class="nocase">FastQC A Quality Control tool for High Throughput Sequence Data</span>. Available at: <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>.
</div>
<div id="ref-khmer2015" class="csl-entry">
<span class="smallcaps">Crusoe, M.</span>, <span class="smallcaps">H. Alameldin</span>, <span class="smallcaps">S. Awad</span>, <span class="smallcaps">E. Boucher</span>, <span class="smallcaps">A. Caldwell</span>, <span class="smallcaps">R. Cartwright</span>, <span class="smallcaps">A. Charbonneau</span>, et al. 2015. The khmer software package: Enabling efficient nucleotide sequence analysis [version 1; referees: 2 approved, 1 approved with reservations]. <em>F1000Research</em> 4:
</div>
<div id="ref-bwa2009" class="csl-entry">
<span class="smallcaps">Li, H.</span>, and <span class="smallcaps">R. Durbin</span>. 2009. Fast and accurate short read alignment with burrows–wheeler transform. <em>Bioinformatics</em> 25: 1754–1760. Available at: <a href="+ http://dx.doi.org/10.1093/bioinformatics/btp324">+ http://dx.doi.org/10.1093/bioinformatics/btp324</a>.
</div>
<div id="ref-samtools2009" class="csl-entry">
<span class="smallcaps">Li, H.</span>, <span class="smallcaps">B. Handsaker</span>, <span class="smallcaps">A. Wysoker</span>, <span class="smallcaps">T. Fennell</span>, <span class="smallcaps">J. Ruan</span>, <span class="smallcaps">N. Homer</span>, <span class="smallcaps">G. Marth</span>, et al. 2009. The sequence alignment/map format and SAMtools. <em>Bioinformatics</em> 25: 2078–2079. Available at: <a href="+ http://dx.doi.org/10.1093/bioinformatics/btp352">+ http://dx.doi.org/10.1093/bioinformatics/btp352</a>.
</div>
<div id="ref-Li2009" class="csl-entry">
<span class="smallcaps">Li, R.</span>, <span class="smallcaps">W. Fan</span>, <span class="smallcaps">G. Tian</span>, <span class="smallcaps">H. Zhu</span>, <span class="smallcaps">L. He</span>, <span class="smallcaps">J. Cai</span>, <span class="smallcaps">Q. Huang</span>, et al. 2009. The sequence and de novo assembly of the giant panda genome. <em>Nature</em> 463: 311 EP–. Available at: <a href="http://dx.doi.org/10.1038/nature08696">http://dx.doi.org/10.1038/nature08696</a>.
</div>
<div id="ref-Luo2012" class="csl-entry">
<span class="smallcaps">Luo, R.</span>, <span class="smallcaps">B. Liu</span>, <span class="smallcaps">Y. Xie</span>, <span class="smallcaps">Z. Li</span>, <span class="smallcaps">W. Huang</span>, <span class="smallcaps">J. Yuan</span>, <span class="smallcaps">G. He</span>, et al. 2012. SOAPdenovo2: An empirically improved memory-efficient short-read de novo assembler. <em>GigaScience</em> 1: 18. Available at: <a href="https://doi.org/10.1186/2047-217X-1-18">https://doi.org/10.1186/2047-217X-1-18</a>.
</div>
<div id="ref-Jellyfish2011" class="csl-entry">
<span class="smallcaps">Marçais, G.</span>, and <span class="smallcaps">C. Kingsford</span>. 2011. A fast, lock-free approach for efficient parallel counting of occurrences of k-mers. <em>Bioinformatics</em> 27: 764–770. Available at: <a href="+ http://dx.doi.org/10.1093/bioinformatics/btr011">+ http://dx.doi.org/10.1093/bioinformatics/btr011</a>.
</div>
<div id="ref-genomescope" class="csl-entry">
<span class="smallcaps">Vurture, G.W.</span>, <span class="smallcaps">F.J. Sedlazeck</span>, <span class="smallcaps">M. Nattestad</span>, <span class="smallcaps">C.J. Underwood</span>, <span class="smallcaps">H. Fang</span>, <span class="smallcaps">J. Gurtowski</span>, and <span class="smallcaps">M.C. Schatz</span>. 2017. GenomeScope: Fast reference-free genome profiling from short reads. <em>Bioinformatics</em> 33: 2202–2204. Available at: <a href="+ http://dx.doi.org/10.1093/bioinformatics/btx153">+ http://dx.doi.org/10.1093/bioinformatics/btx153</a>.
</div>
<div id="ref-Zhang2017" class="csl-entry">
<span class="smallcaps">Zhang, G.-Q.</span>, <span class="smallcaps">K.-W. Liu</span>, <span class="smallcaps">Z. Li</span>, <span class="smallcaps">R. Lohaus</span>, <span class="smallcaps">Y.-Y. Hsiao</span>, <span class="smallcaps">S.-C. Niu</span>, <span class="smallcaps">J.-Y. Wang</span>, et al. 2017. The apostasia genome and the evolution of orchids. <em>Nature</em> 549: 379 EP–. Available at: <a href="http://dx.doi.org/10.1038/nature23897">http://dx.doi.org/10.1038/nature23897</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
