<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki - Boise State University" />

<meta name="date" content="2026-01-05" />

<title>BIOL 497/597 - Genomics &amp; Bioinformatics</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Timetable.html">Schedule</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-Reports</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="R_Functions.html">R Functions</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BIOL 497/597 - Genomics &amp; Bioinformatics</h1>
<h3 class="subtitle">Syllabus</h3>
<h4 class="author">Sven Buerki - Boise State University</h4>
<h4 class="date">2026-01-05</h4>

</div>


<script>
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div style="text-align: right">
<a href="BIOL497_597_Genomics_Bioinformatics_Syllabus_BUERKI.pdf">Download pdf version</a>
</div>
<div style="text-align: right">
<a href="https://github.com/svenbuerki/Genomics-Bioinformatics">Raw data on GitHub</a>
</div>
<div id="meeting-time-and-place" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Meeting Time and Place</h1>
<ul>
<li><strong>Day:</strong> Wednesday</li>
<li><strong>Time:</strong> 9:00 AM to 11:45 AM</li>
<li><strong>Location:</strong> <a href="https://maps.boisestate.edu/?id=715#!m/115308?share">Raptor Research Center Building, Rm 119</a></li>
<li><strong>Important Information:</strong> Students must bring their <strong>Bronco Card</strong> to access the building using key card readers located at the east, west, or north entrances. Because the Raptor Research Building is on the edge of campus, it is kept secure and requires card access for entry. <strong>Please carry your Bronco Card to every class session held in this building, as you will not be able to enter without it.</strong></li>
</ul>
<hr />
</div>
<div id="instructor-information" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Instructor Information</h1>
<ul>
<li>Name: Sven Buerki (he/his)</li>
<li>Office location: Science building, office 228 (first floor)</li>
<li>Office hours: By appointment</li>
<li>Email address: <a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a></li>
<li>Preferred way to contact me: By email or in class</li>
</ul>
<hr />
</div>
<div id="pre-requisite" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Pre-requisite</h1>
<p><strong>BIOL 310: Genetics (3-0-3; F, S, SU).</strong><br />
A study of the principles of genetics as they relate to living organisms.</p>
<p><strong>Prerequisites:</strong> BIOL 191–192 <em>or</em> BIOL 191 and BIOL 320.<br />
<strong>Pre-/Corequisite:</strong> CHEM 301 or CHEM 307.</p>
<hr />
</div>
<div id="welcome" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Welcome</h1>
<p>Welcome to the course! I am looking forward to getting to know you this semester. To get started, please familiarize yourself with this syllabus and our <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/index.html">website</a>. I developed this course to provide a welcoming environment and effective learning experience for all students. If you encounter barriers in this course, please bring them to my attention so that I may work to address them, and reach out to me at any time if you have questions about course content or assignments.</p>
<hr />
</div>
<div id="course-format" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Course Format</h1>
<p>This is a <strong>face-to-face course</strong> that meets <strong>once a week</strong>. Our class sessions include important information and opportunities to apply what you are learning with your peers.</p>
<p>To achieve the learning outcomes for the course, it is important that you <strong>attend class</strong> and actively engage with the content, with me, and with each other.</p>
<p>You should expect to spend approximately <strong>9 hours per week</strong> on work for this course, including both <strong>in-class</strong> and <strong>out-of-class</strong> time.</p>
<hr />
</div>
<div id="course-description-goal-and-objectives" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Course Description, Goal, and Objectives</h1>
<div id="goal" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Goal</h2>
<p>The primary <strong><em>goal</em></strong> of this course is to provide students with both theoretical and applied knowledge in genomics and bioinformatics required to sequence, assemble, and annotate eukaryotic genomes <span class="citation">(<a href="#ref-Saitou2013">Saitou, 2013</a>; <a href="#ref-Satam2023">Satam et al., 2023</a>)</span>, with particular emphasis on non-model organisms <span class="citation">(<a href="#ref-Russell2017">Russell et al., 2017</a>)</span>. The overall structure of the course and progression of topics are illustrated in Figure <a href="#fig:flowchart">6.1</a>.</p>
<div class="figure"><span style="display:block;" id="fig:flowchart"></span>
<img src="Images/Flowchart_course_crop.png" alt="Overview of the structure of the course. See text for more details." width="100%" />
<p class="caption">
Figure 6.1: Overview of the structure of the course. See text for more details.
</p>
</div>
</div>
<div id="objectives" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Objectives</h2>
<p>To achieve this goal, the course is designed around three core <strong><em>objectives</em></strong>:</p>
<ul>
<li>Building strong <strong>genomics knowledge</strong></li>
<li>Developing practical <strong>bioinformatics skills</strong></li>
<li>Enhancing <strong>scientific dissemination</strong> abilities</li>
</ul>
<p>Together, these objectives prepare students to engage with modern genomic research, from data generation and analysis to effective communication of results.</p>
<p>The first objective is to develop <strong>genomics knowledge</strong> through lectures that introduce foundational concepts in genome biology, sequencing technologies, and genome analysis, reinforced by applied examples that reflect real research workflows (Figure <a href="#fig:flowchart">6.1</a>). To provide an authentic research experience, the course draws on examples and case studies from the instructor’s research program <span class="citation">(<a href="#ref-Melton2021">Melton et al., 2021</a>; e.g., <a href="#ref-Ellestad2022AJB">Ellestad et al., 2022</a>; <a href="#ref-Melton2022">Melton et al., 2022</a>)</span>.</p>
<p>The second objective is to develop <strong>bioinformatics proficiency</strong> through hands-on use of a suite of <a href="https://opensource.com/article/18/5/differences-between-linux-and-unix">unix-based open source software</a> to perform genomic analyses, from raw sequence data to assembled and annotated genomes. Students will work through complete bioinformatics workflows commonly used to generate draft genome assemblies and annotations (Figure <a href="#fig:BioinfoWorkflow">6.2</a>). In addition to learning how to execute bioinformatics programs, students will be taught how to install and manage software, gaining critical and transferable technical expertise. Reports and supporting documents provided by the instructor will give students a structured framework for analyzing and interpreting their own genomic datasets. Computational work will be conducted on computers running the <a href="https://ubuntu.com/">Ubuntu Linux operating system</a>, which is widely used in genomics research. This environment will allow students to become familiar with <code>bash/shell</code>, <code>Python</code>, and <code>R</code>, and will provide an on-ramp to using research computing clusters (e.g., the <a href="https://www.boisestate.edu/rcs/2020/10/12/boise-state-university-launches-borah-a-new-peak-computing-system-for-idaho/">BSU Borah research cluster</a>).</p>
<div class="figure"><span style="display:block;" id="fig:BioinfoWorkflow"></span>
<img src="Images/Workflow_Bioinformatics.png" alt="Overview of an example of an approach applied to produce a draft genome assembly. In this course, students will become accustomed with such approach and master some specific key steps." width="100%" />
<p class="caption">
Figure 6.2: Overview of an example of an approach applied to produce a draft genome assembly. In this course, students will become accustomed with such approach and master some specific key steps.
</p>
</div>
<p>The third objective is to foster <strong>scientific dissemination and critical evaluation</strong> by engaging students in writing, presenting, and discussing scientific papers and their own genomic analyses. Through these activities, students will develop the ability to clearly communicate results and to critically assess the methodological quality, rigor, and reproducibility of published genomic studies.</p>
<hr />
</div>
</div>
<div id="course-content" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Course Content</h1>
<p>Throughout the course, <strong>bioinformatics will be used as a central conduit for studying genomics</strong>, allowing students to explore genomic concepts through hands-on data analysis and computational workflows.</p>
<p>The course is subdivided into nine chapters distributed between lecture and laboratory sessions as follows (see Figure <a href="#fig:flowchart">6.1</a>):</p>
<p><strong>Lecture sessions:</strong></p>
<ul>
<li><strong>Chapter 1</strong>: DNA and RNA.</li>
<li><strong>Chapter 2</strong>: Next-generation sequencing.</li>
<li><strong>Chapter 3</strong>: Genome databases.</li>
<li><strong>Chapter 4</strong>: Genome assembly.</li>
<li><strong>Chapter 5</strong>: Transcriptome assembly.</li>
<li><strong>Chapter 6</strong>: Structural annotation.</li>
<li><strong>Chapter 7</strong>: Functional annotation.</li>
<li><strong>Chapter 8</strong>: Comparative genomics.</li>
</ul>
<p><strong>Laboratory sessions:</strong></p>
<ul>
<li><strong>Chapter 9</strong>: Applied bioinformatics.</li>
</ul>
<hr />
</div>
<div id="course-learning-outcomes" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Course Learning Outcomes</h1>
<p>Below is a non-exhaustive list of learning outcomes associated with each chapter of the course (see Figure <a href="#fig:flowchart">6.1</a>). Collectively, these outcomes align with three core learning outcomes that span the entire course and reflect its emphasis on <strong>genomics</strong>, <strong>bioinformatics</strong>, and <strong>scientific dissemination</strong>.</p>
<div id="core-learning-outcomes" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Core Learning Outcomes</h2>
<p>By the end of this course, students will be able to:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Genomics</strong>: Explain key concepts in genome biology and critically evaluate genome sequencing, assembly, annotation, and comparative genomics strategies, with particular attention to eukaryotic and non-model organisms.</p></li>
<li><p><strong>Bioinformatics</strong>: Apply bioinformatics tools and computational workflows to analyze genomic and transcriptomic data, including data quality assessment, assembly, annotation, and comparative analyses, using unix-based open-source software.</p></li>
<li><p><strong>Scientific Dissemination</strong>: Effectively communicate genomic research through written reports and oral presentations, and critically evaluate published genomic studies for methodological rigor, data quality, and reproducibility.</p></li>
</ol>
</div>
<div id="chapter-specific-learning-outcomes" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Chapter-Specific Learning Outcomes</h2>
<blockquote>
<p><strong>Note:</strong> Material associated with Chapters 5–8 (Transcriptome Assembly, Structural Annotation, Functional Annotation, and Comparative Genomics) will be studied primarily through <strong>Mini-Report 3</strong> and the <strong>Group Lab Assignment</strong> (corresponding to Chapter 9). This approach is designed to help students engage with the material through authentic, evidence-based research, reinforcing learning by applying theoretical concepts to real genomic datasets.</p>
</blockquote>
<div id="chapter-1-dna-rna" class="section level3" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Chapter 1: DNA &amp; RNA</h3>
<ul>
<li>Understand the differences between <em>genetics</em> and <em>genomics</em>.</li>
<li>Understand the central dogma: DNA is transcribed into RNA, which is translated into protein.</li>
<li>Appreciate the diversity of genome organization in organelles, prokaryotes, and eukaryotes.</li>
<li>Understand how organelle genomes have contributed to the evolution of eukaryotic chromosomes and how this impacts genome assembly project design.</li>
<li>Appreciate that eukaryotic genomes contain extensive and diverse repetitive regions.</li>
<li>Appreciate the diversity of published genomes and variation in their levels of completeness.</li>
<li>Understand the importance of computer science and bioinformatics in generating raw sequencing data and assembling genomes.</li>
<li>Understand the major challenges associated with sequencing and assembling genomes.</li>
</ul>
</div>
<div id="chapter-2-next-generation-sequencing" class="section level3" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Chapter 2: Next-Generation Sequencing</h3>
<ul>
<li>Learn the terminology associated with next-generation sequencing (NGS) that underpins genome assembly and annotation.</li>
<li>Understand the general principles of genome sequencing (wet-lab) and genome assembly (bioinformatics workflows) (see Figure <a href="#fig:BioinfoWorkflow">6.2</a>).</li>
<li>Be familiar with the major NGS platforms and their limitations <span class="citation">(<a href="#ref-Satam2023">Satam et al., 2023</a>)</span>:
<ul>
<li>Illumina<br />
</li>
<li>PacBio<br />
</li>
<li>Oxford Nanopore</li>
</ul></li>
<li>Become proficient in handling NGS data outputs, particularly FASTA and FASTQ files.</li>
<li>Learn how to assess nucleotide quality using Phred quality scores.</li>
</ul>
</div>
<div id="chapter-3-genome-databases" class="section level3" number="8.2.3">
<h3><span class="header-section-number">8.2.3</span> Chapter 3: Genome Databases</h3>
<ul>
<li>Understand the role of computer science and bioinformatics in:
<ul>
<li>Producing raw sequencing data,</li>
<li>Creating molecular biology databases,</li>
<li>Archiving and curating genomic data,</li>
<li>Distributing data via the Internet,</li>
<li>Developing tools for efficient data retrieval and mining.</li>
</ul></li>
<li>Gain knowledge of major molecular biology databases relevant to genome assembly and annotation, including:
<ul>
<li>Nucleic acid sequence databases,</li>
<li>Protein sequence databases,</li>
<li>Gene ontology databases,</li>
<li>Metabolic pathway databases,</li>
<li>Specialized annotated genome portals.</li>
</ul></li>
<li>Learn protocols to query genomic information and remotely download data from GenBank.</li>
</ul>
</div>
<div id="chapter-4-genome-assembly" class="section level3" number="8.2.4">
<h3><span class="header-section-number">8.2.4</span> Chapter 4: Genome Assembly</h3>
<ul>
<li>Understand the key steps involved in producing a genome assembly and strategies to optimize assembly quality.</li>
<li>Perform read-quality assessments, including:
<ul>
<li>Read trimming,</li>
<li>Contaminant screening,</li>
<li><em>k</em>-mer counting,</li>
<li>Genome size estimation,</li>
<li>Read error correction.</li>
</ul></li>
<li>Understand the use of de Bruijn graphs for <em>de novo</em> genome assembly.</li>
<li>Gain an overview of commonly used genome assemblers and their specific applications.</li>
<li>Understand the benefits of long-read sequencing for closing gaps and improving draft genome assemblies.</li>
</ul>
</div>
<div id="chapter-5-transcriptome-assembly" class="section level3" number="8.2.5">
<h3><span class="header-section-number">8.2.5</span> Chapter 5: Transcriptome Assembly</h3>
<ul>
<li>Learn about the different types of RNA molecules in the cell.</li>
<li>Understand the relationship between the transcriptome and proteome through the genetic code.</li>
<li>Understand key steps involved in transcriptome assembly and strategies to optimize results, including comparisons between reference-based, <em>de novo</em>, and combined approaches.</li>
<li>Gain an overview of commonly used transcriptome assemblers and their specific applications.</li>
</ul>
</div>
<div id="chapter-6-structural-annotation" class="section level3" number="8.2.6">
<h3><span class="header-section-number">8.2.6</span> Chapter 6: Structural Annotation</h3>
<ul>
<li>Understand what is meant by the term <em>genome annotation</em>.</li>
<li>Understand key steps involved in structural genome annotation and best practices.</li>
<li>Become familiar with major stages of the structural annotation process:
<ul>
<li>Repeat identification,</li>
<li>Evidence alignment,</li>
<li><em>Ab initio</em> gene prediction,</li>
<li>Evidence-driven gene prediction,</li>
<li>Overview of commonly used structural annotation software.</li>
</ul></li>
</ul>
</div>
<div id="chapter-7-functional-annotation" class="section level3" number="8.2.7">
<h3><span class="header-section-number">8.2.7</span> Chapter 7: Functional Annotation</h3>
<ul>
<li>Understand the challenges associated with accurately assigning gene functions.</li>
<li>Study and compare available pipelines for automated functional genome annotation.</li>
<li>Review approaches used to assess annotation quality.</li>
<li>Become familiar with commonly used functional annotation software.</li>
<li>Gain an overview of procedures for submitting annotated genomes to public databases.</li>
</ul>
</div>
<div id="chapter-8-comparative-genomics" class="section level3" number="8.2.8">
<h3><span class="header-section-number">8.2.8</span> Chapter 8: Comparative Genomics</h3>
<ul>
<li>Learn how to read, interpret, and present scientific papers reporting new genomic or transcriptomic data.</li>
<li>Develop expertise through structured reading and discussion of scientific publications <span class="citation">(e.g., <a href="#ref-Li2009">Li et al., 2009</a>; <a href="#ref-Ellestad2022AJB">Ellestad et al., 2022</a>)</span>.</li>
</ul>
</div>
<div id="chapter-9-applied-bioinformatics" class="section level3" number="8.2.9">
<h3><span class="header-section-number">8.2.9</span> Chapter 9: Applied Bioinformatics</h3>
<p>In this chapter, students will use data produced by the instructor’s laboratory to learn practical bioinformatics protocols to:</p>
<ul>
<li>Analyze Sanger sequencing data,</li>
<li>Conduct phylogenetic analyses,</li>
<li>Perform quality control and cleaning of Illumina reads,</li>
<li>Assemble draft genomes,</li>
<li>Mine draft genome scaffolds for target genes,</li>
<li>Validate gene mining by identifying open reading frames (ORFs), reconstructing protein sequences, and inferring gene phylogenetic trees,</li>
<li>Annotate validated genes.</li>
</ul>
<hr />
</div>
</div>
</div>
<div id="course-schedule" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Course Schedule</h1>
<p>The tentative schedule for this course is available <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Timetable.html">here</a>. The instructor wants to warn students that he might adjust the schedule to accommodate any needs. However, in case of changes, the instructor will make sure to contact enrolled students to keep them posted.</p>
<hr />
</div>
<div id="course-resources" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Course Resources</h1>
<p>Teaching materials for this course will be shared with students via the dedicated <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/index.html">course website</a>, <a href="https://github.com/svenbuerki/Genomics-Bioinformatics">GitHub repository</a>, and Google Drive.</p>
<div id="shared-google-drive" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Shared Google Drive</h2>
<p>Google Drive will primarily be used to share:</p>
<ul>
<li>Sensitive material (e.g., IP addresses and passwords for lab computers).<br />
</li>
<li>Scientific publications.<br />
</li>
<li>Documents related to class assignments.</li>
</ul>
<p>Students will also upload their assignments to Google Drive.</p>
<p>The URL to the shared Google Drive folder is provided below. The instructor will grant access to this folder prior to the start of the semester:</p>
<ul>
<li><a href="https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing">Shared Google Drive Folder</a></li>
</ul>
</div>
<div id="publications-textbooks-and-online-resources" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Publications, Textbooks, and Online Resources</h2>
<p>The reading materials for this course consist of a combination of <strong>scientific publications</strong> and <strong>textbook chapters</strong> <span class="citation">(<a href="#ref-Dale2012">Dale et al., 2012</a>; <a href="#ref-deSalle2013">deSalle and Rosenfeld, 2013</a>; <a href="#ref-Brown2017">Brown, 2017</a>; <a href="#ref-Lesk2017">Lesk, 2017</a>)</span>. The instructor has copies of these textbooks, and students are welcome to consult them at any time.</p>
<p>A comprehensive list of references used in this course is provided <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/References.html">here</a>. In addition, there are numerous online resources dedicated to genomic and transcriptomic data. A selection is provided below:</p>
<ul>
<li><a href="https://www.ebi.ac.uk/training/online/course/ebi-next-generation-sequencing-practical-course">Next-Generation Sequencing Online Course</a><br />
</li>
<li><a href="https://www.integratedbreeding.net/courses/genomics-and-comparative-genomics/www.generationcp.org/genomics/index390e.html?page=1131">Genomics and Comparative Genomics Resources</a></li>
</ul>
<p>Finally, the instructor has assembled a <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Lexicon.html">Lexicon</a> containing definitions of key concepts used in this course. If you encounter a term that is <strong>not included in the Lexicon</strong>, please contact the instructor to have it added.</p>
<hr />
</div>
</div>
<div id="bioinformatics-tools" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Bioinformatics Tools</h1>
<p>Although we will be using software installed on the Linux computers, the instructor is advising students to install the following programs on their personal computers:</p>
<ul>
<li>The <strong>R</strong> statistical language that will allow you to gather data and analyze it.</li>
<li>The <strong>Markdown</strong> and <strong>LaTeX</strong> markup languages that you can use to create documents (slideshows, articles, books, webpages) for presenting your findings.</li>
<li>The <em>knitr</em> and <em>rmarkdown</em> <strong>packages</strong> for R and other tools, including <strong>command-line shell programs</strong> like GNU Make and Git version control, for dynamically tiding your data gathering, analysis, and presentation documents together so that they can be easily reproduced.</li>
<li><strong>RStudio</strong>, a program that brings all of these tools together in one place.</li>
</ul>
<div id="installing-software" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Installing Software</h2>
<p>As shown above, <strong>R</strong> and <strong>RStudio</strong> are at the core of this course and will have to be installed on your computers. This can be easily done by downloading the software from the following websites:</p>
<ul>
<li><strong>R</strong>: <a href="https://www.r-project.org" class="uri">https://www.r-project.org</a></li>
<li><strong>RStudio</strong>: <a href="https://www.rstudio.com/products/rstudio/download/" class="uri">https://www.rstudio.com/products/rstudio/download/</a></li>
</ul>
<p>The download pages for these software tools include detailed installation instructions; please refer to them for more information.</p>
</div>
<div id="installing-markup-software" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Installing Markup Software</h2>
<p>If you are planning to create LaTeX documents, you will need to install a Tex distribution. Please refer to this website for more details: <a href="https://www.latex-project.org/get/" class="uri">https://www.latex-project.org/get/</a></p>
<p>If you want to create Markdown documents you can separately install the <em>rmarkdown</em> package in R (see <a href="#Rpack">below</a> for more details).</p>
</div>
<div id="Rpack" class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> <a name="Rpack"></a>Installing R Packages</h2>
<p>We will be using several R packages specifically designed to support reproducible research. Many of these packages are not included in the default R installation and must be installed separately.</p>
<p>To install the core packages used in this course, copy the following code and paste it into your R console:</p>
<pre class="r"><code>install.packages(c(&quot;brew&quot;, &quot;countrycode&quot;, &quot;devtools&quot;, &quot;dplyr&quot;, &quot;ggplot2&quot;, &quot;googleVis&quot;,
    &quot;knitr&quot;, &quot;rmarkdown&quot;, &quot;tidyr&quot;, &quot;xtable&quot;))</code></pre>
<p>Once you run this code, you may be prompted to select a CRAN “mirror” to download the packages from. Simply choose the mirror closest to your location.</p>
<p>It is also likely that we will need to install additional packages throughout the course. When that happens, you can install them using the same R function <code>install.packages()</code>, or through RStudio by selecting “Tools” → “Install Packages…”, entering the package name in the dialog box, and ensuring the “Install dependencies” option is checked.</p>
</div>
<div id="Cheat" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> <a name="Cheat"></a>RStudio Cheat Sheets</h2>
<p>RStudio offers a collection of cheat sheets accessible from the “Help” menu by selecting “Cheatsheets.”</p>
<p>Five cheat sheets are especially relevant to chapters taught in this course:</p>
<ul>
<li><em>RStudio IDE : Cheat sheet</em></li>
<li><em>Data Manipulation with dplyr, tidyr</em></li>
<li><em>Data Visualization with ggplot2</em></li>
<li><em>R Markdown Cheat Sheet</em></li>
<li><em>R Markdown Reference Guide</em></li>
</ul>
<p>These documents together with the material presented in <a href="#Publis">course resources</a> will provide the basis to design your <a href="#BioTut">bioinformatics tutorials</a>.</p>
</div>
<div id="r-tutorials" class="section level2" number="11.5">
<h2><span class="header-section-number">11.5</span> R Tutorials</h2>
<p>Please find below two documents providing a comprehensive introduction to R:</p>
<ul>
<li>R for beginners (a tutorial by Emmanuel Paradis): <a href="https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf" class="uri">https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf</a></li>
<li>An introduction to R: <a href="https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf" class="uri">https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf</a></li>
</ul>
<hr />
</div>
</div>
<div id="assessments-and-grading" class="section level1" number="12">
<h1><span class="header-section-number">12</span> Assessments and Grading</h1>
<p>Students will develop and demonstrate mastery of the course learning outcomes by producing written reports and delivering oral presentations, either individually or collaboratively in groups. Rather than relying on traditional exams, this course emphasizes hands-on application: you will build both theoretical genomics understanding and practical bioinformatics skills, and apply them directly to authentic research projects.</p>
<p><strong>Graduate students:</strong> Graduate students enrolled in this course are held to a higher standard and are expected to take on <strong>leadership roles</strong> during group assignments. This includes guiding discussions, coordinating group efforts, and ensuring that analyses and reports meet the advanced expectations appropriate for graduate-level work.</p>
<div id="assessments" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Assessments</h2>
<p>Students will be evaluated based on the following five mandatory assessments:</p>
<ul>
<li>Three individual mini-reports (2 × 25 points and 1 × 50 points; <strong>total: 100 points</strong>).<br />
</li>
<li>One group lab report (<strong>150 points</strong>).<br />
</li>
<li>One group lab presentation (<strong>50 points</strong>).</li>
</ul>
<p>These assessments sum to a total of <strong>300 points</strong>. Table <a href="#tab:mygrade">12.1</a> shows the grading scale used in this course.</p>
</div>
<div id="grading-policy" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Grading Policy</h2>
<p>The grading scale for this course is in Table <a href="#tab:mygrade">12.1</a>.</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:mygrade">Table 12.1: </span><span id="tab:mygrade">Table 12.2: </span>Grading scale applied in this course.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Percentage
</th>
<th style="text-align:left;">
Grade
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
100-98
</td>
<td style="text-align:left;">
A+
</td>
</tr>
<tr>
<td style="text-align:left;">
97.9-93
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
92.9-90
</td>
<td style="text-align:left;">
A-
</td>
</tr>
<tr>
<td style="text-align:left;">
89.9-88
</td>
<td style="text-align:left;">
B+
</td>
</tr>
<tr>
<td style="text-align:left;">
87.9-83
</td>
<td style="text-align:left;">
B
</td>
</tr>
<tr>
<td style="text-align:left;">
82.9-80
</td>
<td style="text-align:left;">
B-
</td>
</tr>
<tr>
<td style="text-align:left;">
79.9-78
</td>
<td style="text-align:left;">
C+
</td>
</tr>
<tr>
<td style="text-align:left;">
77.9-73
</td>
<td style="text-align:left;">
C
</td>
</tr>
<tr>
<td style="text-align:left;">
72.9-70
</td>
<td style="text-align:left;">
C-
</td>
</tr>
<tr>
<td style="text-align:left;">
69.9-68
</td>
<td style="text-align:left;">
D+
</td>
</tr>
<tr>
<td style="text-align:left;">
67.9-60
</td>
<td style="text-align:left;">
D
</td>
</tr>
<tr>
<td style="text-align:left;">
59.9-0
</td>
<td style="text-align:left;">
F
</td>
</tr>
</tbody>
</table>
<p>You can look at all of your scores by accessing Grades in the Canvas course menu.</p>
</div>
<div id="minireport" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> Mini-Reports (TOTAL: 100 points)</h2>
<p>To further develop expertise in genomics and bioinformatics, students will produce three <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html">mini-reports</a> that align with the course’s core learning outcomes:</p>
<ul>
<li><strong>Mini-Report 1: Sequencing technologies</strong> (<strong>25 points</strong>; related to Chapter 2)
<ul>
<li><strong>Genomics:</strong> Demonstrate understanding of genome sequencing principles and technologies.<br />
</li>
<li><strong>Bioinformatics:</strong> Explore how sequencing data are generated and processed.<br />
</li>
<li><strong>Scientific dissemination:</strong> Communicate technical information clearly in written form.</li>
</ul></li>
<li><strong>Mini-Report 2: Molecular biology databases</strong> (<strong>25 points</strong>; related to Chapter 3)
<ul>
<li><strong>Genomics:</strong> Explain the role of databases in genome assembly and annotation.<br />
</li>
<li><strong>Bioinformatics:</strong> Apply protocols to query and retrieve genomic data.<br />
</li>
<li><strong>Scientific dissemination:</strong> Summarize and present information from multiple sources accurately.</li>
</ul></li>
<li><strong>Mini-Report 3: DNA barcoding and phylogenetic analysis</strong> (<strong>50 points</strong>; related to Chapter 8)
<ul>
<li><strong>Genomics:</strong> Apply comparative genomics concepts to identify species and infer evolutionary relationships.<br />
</li>
<li><strong>Bioinformatics:</strong> Perform basic analyses, including sequence alignment and phylogenetic reconstruction.<br />
</li>
<li><strong>Scientific dissemination:</strong> Integrate methods and results into a coherent written report and justify the choice of comparative articles.</li>
</ul></li>
</ul>
<p>Time will be allocated during class to work on these mini-reports; however, students are expected to complete them independently outside of class.</p>
</div>
<div id="group-lab-report-total-150-points" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> Group Lab Report (TOTAL: 150 points)</h2>
<p>Students will work in groups to produce a report addressing the following research question:</p>
<blockquote>
<p>What Aquaporin genes are encoded in the sagebrush genome (<em>Artemisia tridentata</em>) and what are their functions?</p>
</blockquote>
<p>To gain further insights into Aquaporins, sagebrush, and the associated bioinformatic analyses, students will read the companion publication <span class="citation">(<a href="#ref-Melton2021">Melton et al., 2021</a>)</span>.</p>
<div id="associated-scripts" class="section level3" number="12.4.1">
<h3><span class="header-section-number">12.4.1</span> Associated Scripts</h3>
<p>Students can access <code>bash</code> and <code>R</code> scripts associated with this lab report <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mining_draft_genome.html">here</a>.</p>
</div>
<div id="details-on-reporting" class="section level3" number="12.4.2">
<h3><span class="header-section-number">12.4.2</span> Details on Reporting</h3>
<p>Lab reports should be formal scientific documents, structured like manuscripts with the following sections: Introduction, Materials &amp; Methods, Results, Discussion, and References.</p>
<ul>
<li>The instructor does not set a minimum word or page count but expects reports to be <strong>concise, clear, and free of redundancy</strong>.<br />
</li>
<li>All evidence and documentation must be provided to allow an <strong>external reviewer to replicate analyses</strong>. Students are expected to include associated data and code to support reproducible science; guidance on these protocols will be provided during lab sessions.<br />
</li>
<li>For group projects, each member must contribute equally. If participation is deemed insufficient, the instructor may assign individual grades.</li>
</ul>
</div>
<div id="alignment-with-course-learning-outcomes" class="section level3" number="12.4.3">
<h3><span class="header-section-number">12.4.3</span> Alignment with Course Learning Outcomes</h3>
<ul>
<li><strong>Genomics:</strong> Apply knowledge of genome structure and gene function to identify and interpret Aquaporin genes in a non-model eukaryotic genome.<br />
</li>
<li><strong>Bioinformatics:</strong> Use computational tools and scripts (<code>bash</code> and <code>R</code>) to analyze sequence data, perform gene mining, and validate results.<br />
</li>
<li><strong>Scientific Dissemination:</strong> Communicate findings effectively through a structured written report, ensuring clarity, reproducibility, and proper scientific documentation.</li>
</ul>
</div>
</div>
<div id="group-lab-presentation-total-50-points" class="section level2" number="12.5">
<h2><span class="header-section-number">12.5</span> Group Lab Presentation (TOTAL: 50 points)</h2>
<p>Students will use their group lab reports to prepare <strong>10-minute presentations</strong>, followed by <strong>5 minutes for questions</strong>, scheduled during the final weeks of the semester.</p>
<ul>
<li>Each student is expected to actively participate in the presentation and contribute to answering questions from the audience.<br />
</li>
<li>Presentations must be submitted in advance, either uploaded to the shared Google Drive folder or sent directly to the instructor.</li>
</ul>
<div id="alignment-with-course-learning-outcomes-1" class="section level3" number="12.5.1">
<h3><span class="header-section-number">12.5.1</span> Alignment with Course Learning Outcomes</h3>
<ul>
<li><strong>Genomics:</strong> Clearly explain the biological context, genome organization, and functional interpretation of Aquaporin genes in <em>Artemisia tridentata</em>.<br />
</li>
<li><strong>Bioinformatics:</strong> Summarize the computational analyses performed, including the use of <code>bash</code> and <code>R</code> scripts, and demonstrate understanding of the bioinformatic workflow.<br />
</li>
<li><strong>Scientific Dissemination:</strong> Communicate research findings effectively in an oral format, respond to questions with clarity, and demonstrate the ability to convey complex genomic and bioinformatic concepts to peers.</li>
</ul>
</div>
</div>
<div id="extra-credits" class="section level2" number="12.6">
<h2><span class="header-section-number">12.6</span> Extra Credits</h2>
<p>For students interested in earning extra credits, the instructor will provide <strong>two opportunities</strong> to earn a total of <strong>30 extra credits</strong>. Students may allocate these extra credits to any of their assessments by sending an email to the instructor specifying the allocation.</p>
<p>These two non-mandatory assignments have <strong>strict deadlines</strong> (see <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Timetable.html">Schedule</a>). <strong>No extensions will be granted</strong>, and any work submitted after the deadline <strong>will not be considered</strong>. Students who do not submit by the deadline will be deemed not interested in earning the extra credits.</p>
<p>To support time management, the number of credits assigned to each question will be provided in advance. These opportunities allow students to <strong>improve their overall grade</strong> while gaining additional experience in genome assembly and annotation. As with other assignments, students may work in groups, but <strong>each student must submit their own individual assignment</strong>.</p>
<hr />
</div>
</div>
<div id="expectation-for-student-success" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Expectation for Student Success</h1>
<p>My goal is that every student is successful in this course, but I need your help to achieve that. In order to do your part to ensure your success in this course, please:</p>
<ul>
<li><strong>Attend class.</strong> We will be using class time to practice and apply what we are learning, so it is important for your and your fellow students’ learning that you are present and participatory. You can miss two class periods without it affecting your grade. If you have to miss beyond two classes, please email me as soon as possible to discuss ways to help you participate in classroom activities asynchronously as appropriate.</li>
<li><strong>Ask questions.</strong> Learning is all about asking questions, so always feel free to do so in this class.</li>
<li><strong>Be respectful of your fellow students.</strong> While working together to build this community, we ask all members to:
<ul>
<li>Share their unique experiences, values, and beliefs, if comfortable doing so.</li>
<li>Listen deeply to one another.</li>
<li>Honor the uniqueness of their peers.</li>
<li>Create a respectful environment in this course and across the campus community.</li>
</ul></li>
<li><strong>Check announcements or e-mails regularly.</strong> Announcements will serve as courtesy reminders and also point you to any new materials or changes.</li>
<li><strong>Do the pre-class assignments.</strong> In order to make the most of our class time, it is important that all students complete the pre-class assignments so we can jump into our application activities.</li>
</ul>
<p>If you are unable to attend class, please contact the instructor as soon as possible via email at <a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a>.</p>
<hr />
</div>
<div id="expectations-for-me" class="section level1" number="14">
<h1><span class="header-section-number">14</span> Expectations for Me</h1>
<p>In support of my goal that every student be successful in this course, you can expect that I:</p>
<ul>
<li><strong>Will be available to answer questions throughout the course.</strong> I encourage you to visit me during Student Hours to ask any questions about the course material or to further your curiosity in the subject matter. You may also email me with questions or concerns or ask to meet via Zoom or on campus outside of the Student Hours, and I will do my best to accommodate your needs.</li>
<li><strong>Will provide regular feedback on your work in a timely fashion.</strong> I will return completed work to you, with feedback, within one week of the due date.</li>
<li><strong>Will continuously work on improving your learning experience through my own class observations and based on your feedback.</strong> You will have the opportunity to provide feedback to me at the midpoint of the semester, and via the end of course evaluations.</li>
</ul>
<hr />
</div>
<div id="this-class-welcomes-everyone" class="section level1" number="15">
<h1><span class="header-section-number">15</span> This Class Welcomes Everyone</h1>
<p>Students in this class represent a rich variety of backgrounds and perspectives. The (program/dept) is committed to providing an environment where similarities and differences are respected, supported, and valued. While working together to build this community, we ask all members to:</p>
<ul>
<li>share their unique experiences, values, and beliefs, if comfortable doing so.</li>
<li>listen deeply to one another.</li>
<li>honor the uniqueness of their peers.</li>
<li>appreciate the opportunity we have to learn from each other in this community.</li>
<li>use this opportunity together to discuss ways in which we can create a welcoming and respectful environment in this course and across the campus community.</li>
<li>recognize opportunities to invite a community member to exhibit more respectful speech or behavior—and then also invite them into further conversation. We also expect community members to respond with gratitude and to take a moment of reflection when they receive such an invitation, rather than react immediately from defensiveness.</li>
<li>keep confidential any discussions that the community has of a personal (or professional) nature, unless the speaker has given explicit permission to share what they have said.</li>
</ul>
<p>As your instructor, my goal is to make sure that our learning environment is effective for everyone. This means, in part, that each student is encouraged to share perspectives relevant to the course material and that our class activities and discussions are conducted in a way that supports everyone’s learning.</p>
<hr />
</div>
<div id="student-well-being" class="section level1" number="16">
<h1><span class="header-section-number">16</span> Student Well-being</h1>
<p>If you are struggling for any reason (e.g., family emergency, financial/basic needs, mental/physical health concerns, caregiving responsibilities, etc.) and believe these struggles may impact your performance in the course, I encourage you to reach out to me if you are comfortable doing so, and I will refer you to an appropriate university resource. You may also reach out directly to the outreach team in the Office of the Dean of Students at (208) 426-1527 or email <a href="mailto:studentoutreach@boisestate.edu" class="email">studentoutreach@boisestate.edu</a> for support. The <a href="https://www.boisestate.edu/student-life/">Student Life Essentials page</a> is also a great place to find helpful resources. If you notice a significant change in your mood, sleep, feelings of hopelessness or a lack of self worth, consider connecting immediately with <a href="https://www.boisestate.edu/healthservices/counseling/">Counseling Services</a> (1529 Belmont Street, Norco Building) at (208) 426-1459 or email <a href="mailto:healthservices@boisestate.edu" class="email">healthservices@boisestate.edu</a>.</p>
<hr />
</div>
<div id="what-do-you-need" class="section level1" number="17">
<h1><span class="header-section-number">17</span> What Do you Need?</h1>
<p>The university has many resources designed to support you as a learner and human being. Among these are:</p>
<ul>
<li><a href="https://library.boisestate.edu/">Albertsons Library</a> provides a treasure trove of physical and electronic resources.
<ul>
<li>As you enter the library, straight ahead you’ll find the Reference Desk, where librarians can help you find the information and resources you need.</li>
<li>The Circulation Desk lets students borrow various technologies.</li>
<li>The MakerLab on the second floor offers tools for student use, and there are friendly staff in the MakerLab to help you learn how.</li>
</ul></li>
<li>The <a href="https://www.boisestate.edu/writingcenter/">Writing Center</a> offers individual consultations tailored to your needs, including making sense of writing assignment instructions, brainstorming, crafting a thesis, organizing an essay, revisions, citations, and more.</li>
<li><a href="https://www.boisestate.edu/healthservices/counseling">Counseling Services</a> helps you tap into your strengths and find resources to deal more effectively with concerns that impact your pursuit of personal and academic goals. It emphasizes prevention and early detection and provides a broad spectrum of short-term counseling, consultative, evaluative, teaching, and training functions. Counseling staff consists of licensed counselors, psychologists, and closely supervised trainees/post-graduate interns.</li>
<li>Food assistance: If you are hungry and cannot afford to purchase food, the campus has some resources to help you. You can visit the <a href="https://www.boisestate.edu/campusfoodpantry/">campus food pantry</a> or <a href="https://www.boisestate.edu/deanofstudents/student-support/food-help/meal-assistance/">get free meals in the campus dining hall</a>.</li>
</ul>
<hr />
</div>
<div id="course-policies" class="section level1" number="18">
<h1><span class="header-section-number">18</span> Course Policies</h1>
<div id="academic-integrity" class="section level2" number="18.1">
<h2><span class="header-section-number">18.1</span> Academic Integrity</h2>
<p>Academic Excellence is a <a href="https://www.boisestate.edu/president/values/statement-of-shared-values/">Shared Value</a> at Boise State, and part of your responsibility in pursuing academic excellence includes avoiding cheating, plagiarism, and any other kind of academic misconduct. If I find a student responsible for academic misconduct in our class, the outcome of their choice to not fully engage in their learning might range from a ‘revise &amp; resubmit’ up to an ‘F (failure) for the course.’ For more info, please read <a href="https://www.boisestate.edu/policy/student-affairs/code-of-conduct/#academic_misconduct">The Student Code of Conduct (Policy 2020), Section 7: Academic Misconduct Complaints, Violations, Processes and Sanctions</a>.</p>
</div>
<div id="artifical-intelligence-ai-use-in-the-course" class="section level2" number="18.2">
<h2><span class="header-section-number">18.2</span> Artifical Intelligence (AI) Use in The Course</h2>
<p>In this course, I want to see your thoughts, understand your reasoning, and hear your voice. However, there are moments in this course where you might find it useful to use generative AI tools in support of your learning.</p>
<p>You may use generative AI tools for specified activities and assignments if their use supports, rather than undermines, your learning. While generative AI can help to advance your learning, its usefulness depends on the purpose of each activity or assignment. You will find guidelines for generative AI use in the instructions for each assignment; please read them very carefully, as these guidelines differ by assignment.</p>
<p>If you use ChatGPT, Gemini, Grammarly, Midjourney, or other AI tools in support of your work in this course, cite any ideas, text, images, or other media generated by the tool using the instructions and format of the <a href="https://style.mla.org/citing-generative-ai/">Modern Language Association</a> (MLA), <a href="https://apastyle.apa.org/blog/how-to-cite-chatgpt">American Psychological Association</a> (APA), <a href="https://www.chicagomanualofstyle.org/qanda/data/faq/topics/Documentation/faq0422.html">Chicago Manual of Style</a>, or other citation style as appropriate. When you use a tool in an assignment, include a brief, clear description of how you used it. If you use generative AI, you must not let this tool replace your thinking and work. In fact, it is your responsibility to ensure you are fully engaging in learning and submitting authentic work. To learn more about how to learn successfully and avoid academic misconduct behaviors, please review the <a href="https://www.boisestate.edu/policy/student-affairs/code-of-conduct/">Student Code of Conduct</a> with special attention to <a href="https://www.boisestate.edu/policy/student-affairs/code-of-conduct/#procedures">Section 8: Procedures for Academic Misconduct</a>.</p>
<p>If you are unsure of whether or when to use generative AI tools in this course, please reach out to me. I am eager to learn about how we might use them in new ways to meaningfully advance your learning and prepare you for your future beyond Boise State.</p>
</div>
<div id="communicable-disease-policy" class="section level2" number="18.3">
<h2><span class="header-section-number">18.3</span> Communicable Disease Policy</h2>
<p>Boise State has a <a href="https://www.boisestate.edu/policy/facilities-planning-campus-safety/communicable-disease/#:~:text=Students%2C%20faculty%2C%20and%20other%20employees,to%20the%20Public%20Health%20Office.">Communicable Disease Policy (Policy 9270)</a> that guides everyone working and learning in our community. The policy has two key implications:</p>
<ol style="list-style-type: decimal">
<li>Any illness covered by the policy must be reported to Boise State Public Health by the impacted individual,</li>
<li>Faculty are required to accommodate any student impacted by any illness covered by the policy as directed by Boise State Public Health.</li>
</ol>
</div>
<div id="late-work-policy" class="section level2" number="18.4">
<h2><span class="header-section-number">18.4</span> Late Work Policy</h2>
<p>Due dates for all assignments are provided on the course <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Timetable.html">schedule</a>. Unless otherwise stated, assignments are <strong>due on those dates</strong>.</p>
<p>The instructor recognizes that unforeseen circumstances may arise. If this happens, please contact the instructor <strong>ahead of the due date</strong>, or <strong>as soon as possible afterwards</strong>, so that we can establish a plan for submitting the assignment as close to the original due date as possible.</p>
<p><strong>Automatic Late Penalty:</strong></p>
<ul>
<li>If a student fails to submit an assignment by the due date <strong>without a valid excuse</strong>, a <strong>10% penalty</strong> will be applied to the assignment grade.<br />
</li>
<li>In such cases, the instructor will reach out to the student to establish a <strong>new submission deadline</strong> and clarify expectations for completing the assignment.</li>
</ul>
<p>Assignments that are not submitted within <strong>two weeks of the due date</strong> and for which the student has not communicated with the instructor will be recorded as <strong>Incomplete</strong> in Canvas until a plan is agreed upon.</p>
<hr />
</div>
</div>
<div id="note-on-course-content-and-idaho-law" class="section level1" number="19">
<h1><span class="header-section-number">19</span> Note on Course Content and Idaho Law</h1>
<p>Under Idaho law (Section § 67-5909D), some university courses with content related to diversity, equity, inclusion, or critical theory may be subject to certain restrictions. However, the law affirms and does not limit free discussion in the learning environment. Like all Boise State courses, this course supports open inquiry, intellectual honesty, and respectful engagement with a range of perspectives, all of which are consistent with student rights and responsibilities described in the <a href="https://www.boisestate.edu/policy/student-affairs/code-of-conduct/">Student Code of Conduct (Policy 2020)</a>.</p>
<p>This course may include content that touches on concepts related to diversity, equity, inclusion (DEI), or critical theory—such as systemic inequality, cultural identity, or gender and race in society. If these topics are included, it is because they are relevant to the learning outcomes for this course and are explored to support critical thinking, deeper understanding, and respectful engagement with different perspectives. As part of the course, you may be asked to apply or explain ideas that come from a particular perspective. However, you are not required to adopt such perspectives as your own.</p>
<p>Our learning environment is a space for open dialogue and thoughtful discussion, including complex or challenging topics. Everyone is expected to engage with curiosity, listen respectfully, and contribute in ways that support a productive and welcoming learning environment. Boise State and the Idaho State Board of Education affirm the importance of free expression and academic inquiry. As outlined in SBOE Policy III.B:</p>
<blockquote>
<p>“Membership in the academic community imposes on administrators, faculty members, other institutional employees, and students an obligation to respect the dignity of others, to acknowledge the right of others to express differing opinions, and to foster and defend intellectual honesty, freedom of inquiry and instruction, and free expression on and off the campus of an institution.”</p>
</blockquote>
<p>Disruptive behavior that interferes with the learning environment will not be tolerated and may result in removal from this course, in line with university policy (See <a href="https://www.boisestate.edu/policy/academic-affairs-student/maintaining-effective-learning-environments/">Policy 3240 Maintaining Effective Learning Environments</a>).</p>
<p>In this course, I will foster critical discussion and analysis, and a respectful consideration of a wide range of ideas, in accordance with the <a href="https://www.boisestate.edu/policy/academic-affairs-faculty-administration/faculty-code-of-rights-responsibilities-and-conduct/">Faculty Code of Rights, Responsibilities, and Conduct (Policy 4000)</a>. You are encouraged to think critically, question ideas, and form your own conclusions. As always, you have the freedom to choose courses that align with your academic goals—if you have concerns about course content, please talk with your instructor or advisor. Refer to the academic calendar for important deadlines related to course withdrawal.</p>
<p>To learn more about the law and its impact at Boise State, visit the <a href="https://www.boisestate.edu/academics/information-regarding-section-67-5909d-course-and-curriculum/">Provost Office’s Information Regarding Section 67-5909D page</a>.</p>
<hr />
</div>
<div id="references" class="section level1" number="20">
<h1><span class="header-section-number">20</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Brown2017" class="csl-entry">
<span class="smallcaps">Brown, T.A.</span> 2017. Genomes 4. Garland Science.
</div>
<div id="ref-Dale2012" class="csl-entry">
<span class="smallcaps">Dale, J.W.</span>, <span class="smallcaps">M. von Schantz</span>, and <span class="smallcaps">N. Plant</span>. 2012. From genes to genomes: Concepts and applications of DNA technology, 3rd edition. Wiley.
</div>
<div id="ref-deSalle2013" class="csl-entry">
<span class="smallcaps">deSalle, R.</span>, and <span class="smallcaps">J. Rosenfeld</span>. 2013. Phylogenomics: A primer 1st edition. Garland Science.
</div>
<div id="ref-Ellestad2022AJB" class="csl-entry">
<span class="smallcaps">Ellestad, P.</span>, <span class="smallcaps">M.A.P. Farrera</span>, <span class="smallcaps">F. Forest</span>, and <span class="smallcaps">S. Buerki</span>. 2022. Uncovering haplotype diversity in cultivated mexican vanilla species. <em>American Journal of Botany</em> 109: 1120–1138. Available at: <a href="https://bsapubs.onlinelibrary.wiley.com/doi/abs/10.1002/ajb2.16024">https://bsapubs.onlinelibrary.wiley.com/doi/abs/10.1002/ajb2.16024</a>.
</div>
<div id="ref-Lesk2017" class="csl-entry">
<span class="smallcaps">Lesk, A.</span> 2017. Introduction to genomics, third edition. Oxford University Press.
</div>
<div id="ref-Li2009" class="csl-entry">
<span class="smallcaps">Li, R.</span>, <span class="smallcaps">W. Fan</span>, <span class="smallcaps">G. Tian</span>, <span class="smallcaps">H. Zhu</span>, <span class="smallcaps">L. He</span>, <span class="smallcaps">J. Cai</span>, <span class="smallcaps">Q. Huang</span>, et al. 2009. The sequence and de novo assembly of the giant panda genome. <em>Nature</em> 463: 311 EP–. Available at: <a href="http://dx.doi.org/10.1038/nature08696">http://dx.doi.org/10.1038/nature08696</a>.
</div>
<div id="ref-Melton2021" class="csl-entry">
<span class="smallcaps">Melton, A.E.</span>, <span class="smallcaps">J. Beck</span>, <span class="smallcaps">S.J. Galla</span>, <span class="smallcaps">J. Jenkins</span>, <span class="smallcaps">L. Handley</span>, <span class="smallcaps">M. Kim</span>, <span class="smallcaps">J. Grimwood</span>, et al. 2021. A draft genome provides hypotheses on drought tolerance in a keystone plant species in western north america threatened by climate change. <em>Ecology and Evolution</em> 11: 15417–15429. Available at: <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.8245">https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.8245</a>.
</div>
<div id="ref-Melton2022" class="csl-entry">
<span class="smallcaps">Melton, A.E.</span>, <span class="smallcaps">A.W. Child</span>, <span class="smallcaps">J. Beard Richard S</span>, <span class="smallcaps">C.D.C. Dumaguit</span>, <span class="smallcaps">J.S. Forbey</span>, <span class="smallcaps">M. Germino</span>, <span class="smallcaps">M.-A. de Graaff</span>, et al. 2022. <span class="nocase">A haploid pseudo-chromosome genome assembly for a keystone sagebrush species of western North American rangelands</span>. <em>G3 Genes|Genomes|Genetics</em> 12: Available at: <a href="https://doi.org/10.1093/g3journal/jkac122">https://doi.org/10.1093/g3journal/jkac122</a>.
</div>
<div id="ref-Russell2017" class="csl-entry">
<span class="smallcaps">Russell, J.J.</span>, <span class="smallcaps">J.A. Theriot</span>, <span class="smallcaps">P. Sood</span>, <span class="smallcaps">W.F. Marshall</span>, <span class="smallcaps">L.F. Landweber</span>, <span class="smallcaps">L. Fritz-Laylin</span>, <span class="smallcaps">J.K. Polka</span>, et al. 2017. Non-model model organisms. <em>BMC Biology</em> 15: 55. Available at: <a href="https://doi.org/10.1186/s12915-017-0391-5">https://doi.org/10.1186/s12915-017-0391-5</a>.
</div>
<div id="ref-Saitou2013" class="csl-entry">
<span class="smallcaps">Saitou, N.</span> 2013. Eukaryote genomes. <em>In</em> Introduction to evolutionary genomics, 193–222. Springer London, London. Available at: <a href="https://doi.org/10.1007/978-1-4471-5304-7_8">https://doi.org/10.1007/978-1-4471-5304-7_8</a>.
</div>
<div id="ref-Satam2023" class="csl-entry">
<span class="smallcaps">Satam, H.</span>, <span class="smallcaps">K. Joshi</span>, <span class="smallcaps">U. Mangrolia</span>, <span class="smallcaps">S. Waghoo</span>, <span class="smallcaps">G. Zaidi</span>, <span class="smallcaps">S. Rawool</span>, <span class="smallcaps">R.P. Thakare</span>, et al. 2023. Next-generation sequencing technology: Current trends and advancements. <em>Biology</em> 12: Available at: <a href="https://www.mdpi.com/2079-7737/12/7/997">https://www.mdpi.com/2079-7737/12/7/997</a>.
</div>
</div>
<hr />
</div>
<div id="appendix-1" class="section level1 appendix" number="21">
<h1><span class="header-section-number">21</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.30. 2025.
<a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.12. 2021.
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>.</p>
<p>[3] M. C. Koohafkan. <em>kfigr: Integrated Code Chunk Anchoring and
Referencing for R Markdown Documents</em>. R package version 1.2.1. 2021.
<a href="https://github.com/mkoohafkan/kfigr" class="uri">https://github.com/mkoohafkan/kfigr</a>.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
<p>[5] H. Wickham, J. Bryan, M. Barrett, et al. <em>usethis: Automate Package
and Project Setup</em>. R package version 3.2.1. 2025.
<a href="https://usethis.r-lib.org" class="uri">https://usethis.r-lib.org</a>.</p>
<p>[6] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.1.4. 2023.
<a href="https://dplyr.tidyverse.org" class="uri">https://dplyr.tidyverse.org</a>.</p>
<p>[7] H. Wickham, J. Hester, W. Chang, et al. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.4.6. 2025.
<a href="https://devtools.r-lib.org/" class="uri">https://devtools.r-lib.org/</a>.</p>
<p>[8] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. Boca Raton, Florida: Chapman and Hall/CRC, 2016. ISBN:
978-1138700109. <a href="https://bookdown.org/yihui/bookdown" class="uri">https://bookdown.org/yihui/bookdown</a>.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.46. 2025.
<a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>.</p>
<p>[10] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[11] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.14. 2023. <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>.</p>
<p>[12] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014.</p>
<p>[13] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[14] Y. Xie and J. Allaire. <em>tufte: Tufte’s Styles for R Markdown
Documents</em>. R package version 0.14.0. 2025.
<a href="https://github.com/rstudio/tufte" class="uri">https://github.com/rstudio/tufte</a>.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. Boca Raton, Florida: Chapman and Hall/CRC, 2018. ISBN:
9781138359338. <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. Boca
Raton, Florida: Chapman and Hall/CRC, 2020. ISBN: 9780367563837.
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>.</p>
<p>[17] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.4.0. 2024.
<a href="http://haozhu233.github.io/kableExtra/" class="uri">http://haozhu233.github.io/kableExtra/</a>.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
