<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki - Boise State University" />

<meta name="date" content="2022-02-24" />

<title>BIOL 497/597 - Genomics &amp; Bioinformatics</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="R_Functions.html">R functions</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">BIOL 497/597 - Genomics &amp; Bioinformatics</h1>
<h3 class="subtitle">Syllabus &amp; Timetable</h3>
<h4 class="author">Sven Buerki - Boise State University</h4>
<h4 class="date">2022-02-24</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div style="text-align: right">
<a href="index.pdf">Download pdf version</a>
</div>
<div style="text-align: right">
<a href="https://github.com/svenbuerki/Genomics-Bioinformatics">Raw data on GitHub</a>
</div>
<div id="instructor" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Instructor</h1>
<ul>
<li>Name: Sven Buerki</li>
<li>Office: Science building, office 114 (ground floor).</li>
<li>Email: <a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a></li>
<li>Office hours: By appointment.</li>
</ul>
</div>
<div id="class-ethos" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Class ethos</h1>
<blockquote>
<p>Everyone here is smart; distinguish yourself by being kind.</p>
<footer>
Kindness in Science is an inclusive approach that fosters diversity, respect, wellbeing &amp; openness leading to better science outcomes.
</footer>
</blockquote>
<p><img src="Images/Kindness.png" width="40%" style="display: block; margin: auto 0 auto auto;" /></p>
</div>
<div id="class-details" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Class details</h1>
<div id="teaching-mode-and-class-location" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Teaching mode and class location</h2>
<p>In-person in <a href="https://maps.boisestate.edu/?id=715#!m/89069">Science Bldg, Rm 149</a>.</p>
</div>
<div id="meeting-days-and-times" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Meeting days and times</h2>
<ul>
<li><strong>Lecture sessions:</strong> Wednesdays from 9:00-10:50 AM.</li>
<li><strong>Laboratory sessions:</strong> Fridays from 9:00-10:30 AM.</li>
</ul>
<p>Please see section <a href="#timetable">20</a> for tentative timetable for this course.</p>
</div>
<div id="health-and-safety-protocols" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Health and safety protocols</h2>
<p><strong>Our priority is your safety!</strong></p>
<p>The following rules apply for in-person classes:</p>
<ul>
<li>Wear masks and maintain physical distancing.</li>
<li>Use sanitizing wipes to clean surfaces (desk and keyboard) at beginning and end of class.</li>
<li>Keep using the same seat throughout the semester. This information should be entered into MyBoiseState.</li>
<li>Do not attend class if you are sick.</li>
</ul>
<p>For more information consult the <a href="https://www.boisestate.edu/coronavirus-response/">BSU campus guidelines</a>.</p>
</div>
</div>
<div id="course-goal-and-description" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Course goal and description</h1>
<p>The goal of this course is to provide students with the theoretical and applied knowledge in genomics and bioinformatics to sequence, assemble and annotate genomes, especially for non-model organisms (see Figure <a href="#fig:flowchart">4.1</a>). To provide an authentic research experience, we will be using examples from the instructor’s research on sagebrush (<em>Artemisia tridentata</em>; Asteraceae) genomics <span class="citation">(e.g. <a href="#ref-Melton2021" role="doc-biblioref">Melton et al., 2021</a>)</span>.</p>
<div class="figure"><span id="fig:flowchart"></span>
<img src="Images/Flowchart_course_crop.png" alt="Overview of the structure of the course. See text for more details." width="100%" />
<p class="caption">
Figure 4.1: Overview of the structure of the course. See text for more details.
</p>
</div>
<p>A suite of unix-based open source software will be used to perform genomic analyses (see Figure <a href="#fig:BioinfoWorkflow">4.2</a>). In addition to learning how to execute these programs, students will also be taught procedures to install them therefore gaining critical expertise for their career. Finally, students will produce scientific reports and present their results therefore promoting developing their dissemination skills. Reports together with documents provided by the instructor will provide students with a framework allowing them to analyze their own genomic data.</p>
<div class="figure"><span id="fig:BioinfoWorkflow"></span>
<img src="Images/Workflow_Bioinformatics.png" alt="Overview of an example of an approach applied to produce a draft genome assembly. In this course, students will become accustomed with such approach and master some specific key steps." width="100%" />
<p class="caption">
Figure 4.2: Overview of an example of an approach applied to produce a draft genome assembly. In this course, students will become accustomed with such approach and master some specific key steps.
</p>
</div>
</div>
<div id="course-format" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Course format</h1>
<p>This course provides a mixture of classic lectures together with more applied, hands-on bioinformatic tutorials (conducted both during lecture and lab sessions). Tutorials are designed to support students in mastering theoretical genomic concepts through their implementations in bioinformatic protocols. As mentioned above, the genomic field heavily relies on bioinformatic expertise, especially related to unix-based software.</p>
<p>We will be working on computers running the <a href="https://ubuntu.com/">Ubuntu Linux operating system</a>. This operating system is widely used in genomic research. Using these computers will provide opportunities for students to become familiar with <code>bash/shell</code>, <code>Python</code> and <code>R</code> and they provide an on-ramp to using research computing clusters (e.g. <a href="https://www.boisestate.edu/rcs/2020/10/12/boise-state-university-launches-borah-a-new-peak-computing-system-for-idaho/">BSU Borah research cluster</a>). Finally, lecture sessions will also serve as a platform to:</p>
<ul>
<li>Work on graded <a href="#minireport">mini-reports</a>.</li>
<li>Study the genomic literature through a <a href="#journalclub">journal club</a>.</li>
</ul>
<p>While reading scientific papers, we will be focusing on mastering methodological concepts allowing students to assess the quality and reproducibility of published genomic studies.</p>
</div>
<div id="content-of-the-course" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Content of the course</h1>
<p>The course is subdivided into nine chapters segregated between lecture and laboratory sessions as follows (see Figure <a href="#fig:flowchart">4.1</a>):</p>
<p><strong>Lecture sessions:</strong></p>
<ul>
<li><strong>Chapter 1</strong>: DNA &amp; RNA.</li>
<li><strong>Chapter 2</strong>: Next-generation sequencing.</li>
<li><strong>Chapter 3</strong>: Genome databases.</li>
<li><strong>Chapter 4</strong>: Genome assembly.</li>
<li><strong>Chapter 5</strong>: Transcriptome assembly.</li>
<li><strong>Chapter 6</strong>: Structural annotation.</li>
<li><strong>Chapter 8</strong>: Comparative genomics.</li>
</ul>
<p><strong>Laboratory sessions:</strong></p>
<ul>
<li><strong>Chapter 9</strong>: Applied bioinformatics.</li>
</ul>
</div>
<div id="course-learning-outcomes" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Course learning outcomes</h1>
<p>Please find below a non-exhaustive list of learning outcomes associated to the chapters presented above (Figure <a href="#fig:flowchart">4.1</a>):</p>
<p><strong>Chapter 1: DNA &amp; RNA</strong></p>
<ul>
<li>Understand the differences between “Genetics” and “Genomics.”</li>
<li>Know the basic dogma that DNA is transcribed to RNA, which is translated to protein.</li>
<li>Appreciate the diversity of genome organization in organelles, prokaryotes and eukaryotes.</li>
<li>Appreciate that organelle genomes have forged eukaryotic chromosomes and that this has to be accounted for in designing genome assembly projects.</li>
<li>Appreciate that eukaryotic genomes contain extensive repetitive regions of several different kinds.</li>
<li>Appreciate the diversity of published genomes and their differences in the level of completeness.</li>
<li>Understand the importance of computer science and bioinformatics in producing the raw sequence data and assembly genomes.</li>
<li>Understand the challenges in sequencing a genome.</li>
</ul>
<p><strong>Chapter 2: Next-generation sequencing</strong></p>
<ul>
<li>Learn the next-generation sequencing (NGS) jargon underpinning genome assembly and annotation.</li>
<li>Understand the general principle of genome sequencing (wet-lab) and assembly (bioinformatics workflow) (see Figure <a href="#fig:BioinfoWorkflow">4.2</a>).</li>
<li>Be familiar and know the limitations of the following NGS platforms:
<ul>
<li>Illumina.</li>
<li>PacBio.</li>
<li>Oxford Nanopore.</li>
</ul></li>
<li>Be accustomed to handle NGS data output, especially FASTA and FASTQ files.</li>
<li>Learn protocols to quantify the quality of the nucleobases generated by NGS platforms using Phred quality score.</li>
</ul>
<p><strong>Chapter 3: Genome databases</strong></p>
<ul>
<li>Understand the importance of computer science and bioinformatics in:
<ul>
<li>Producing raw data.</li>
<li>Creating databases in molecular biology.</li>
<li>Archiving and careful curation of the data.</li>
<li>Distributing them via the Internet.</li>
<li>Creating information-retrieval tools to allow effective mining of the data for research application.</li>
</ul></li>
<li>Gain knowledge on major molecular biology databases, which are key to the assembly and annotation of genomes:
<ul>
<li>Nucleic acid sequences databases.</li>
<li>Protein sequences databases.</li>
<li>Gene ontology databases.</li>
<li>Metabolic pathways databases.</li>
<li>Specialized annotated genome portals.</li>
</ul></li>
<li>Learn protocols to query genomic information and remotely download files from GenBank.</li>
</ul>
<p><strong>Chapter 4: Genome assembly</strong></p>
<ul>
<li>Understand key steps involved in producing a genome assembly and best strategies to get there.</li>
<li>Be able to perform reads quality checks as follows:
<ul>
<li>Read quality assessments.
<ul>
<li>Read trimming.</li>
<li>Contaminant screening.</li>
<li>k-mer counting.</li>
<li>Genome size estimation.</li>
<li>Read error correction.</li>
</ul></li>
</ul></li>
<li>Understand the use of de Bruijn graphs for <em>de novo</em> genome assembly.</li>
<li>Gain an overview of current genome assemblers and their specificity.</li>
<li>Study the benefit of long reads in closing gaps and correcting draft genome assemblies.</li>
</ul>
<p><strong>Chapter 5: Transcriptome assembly</strong></p>
<ul>
<li>Learn about the types of RNA molecules in cell.</li>
<li>Study the link between the transcriptome and proteome via the genetic code.</li>
<li>Understand key steps involved in producing a transcriptome assembly and best strategies to get there. We will mostly compare referenced-based vs. <em>de novo</em> strategies, but also look at the “combined strategy.”</li>
<li>Gain an overview of current transcriptome assemblers and their specificity.</li>
</ul>
<p><strong>Chapter 6: Structural annotation</strong></p>
<ul>
<li>Learn what we exactly mean by the term “genome annotation.”</li>
<li>Understand key steps involved in genome annotation and best strategies to get there.</li>
<li>Be familiar with the following key stages of the structural genome annotation process:
<ul>
<li>Repeat identification.</li>
<li>Evidence alignment.</li>
<li><em>Ab initio</em> gene prediction.</li>
<li>Evidence-driven gene prediction.</li>
<li>Introduction to available structural annotation software.</li>
</ul></li>
</ul>
<p><strong>Chapter 7: Functional annotation</strong></p>
<ul>
<li>Be aware of the challenges to obtain accurate data on gene functions.</li>
<li>Study and compare available pipelines to perform automated annotations of genomes.</li>
<li>Review approaches to assess annotation quality.</li>
<li>Be familiar with functional annotation software.</li>
<li>Overview of the procedure to submit annotated genomes to public databases.</li>
</ul>
<p><strong>Chapter 8: Comparative genomics</strong></p>
<ul>
<li>Learn to read and present a scientific paper reporting new genomic or transcriptomic data.</li>
<li>Complement expertise by reading and discussing scientific publications during the <a href="journalclub">journal club</a> sessions.</li>
</ul>
<p><strong>Chapter 9: Applied Bioinformatics</strong></p>
<p>In this chapter, we will be using data produced by the instructor’s lab to learn protocols to:</p>
<ul>
<li>Analyze Sanger sequence data.</li>
<li>Conduct phylogenetic analyses.</li>
<li>Clean Illumina reads.</li>
<li>Assemble draft genome.</li>
<li>Mine scaffolds in an Illumina draft genome containing target genes.</li>
<li>Validate mining by locating Open Reading Frames (ORFs), reconstructing protein sequences and inferring gene phylogenetic trees.</li>
<li>Annotate validated genes.</li>
</ul>
<p>Please see section <a href="#timetable">20</a> for tentative timetable associated to these chapters.</p>
</div>
<div id="pre-requisite" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Pre-requisite</h1>
<p><strong>BIOL 310 GENETICS (3-0-3)(F,S,SU).</strong> A study of the principles of genetics as they relate to living organisms.
PREREQ: BIOL 191-192 or BIOL 191 and BIOL 320.
PRE/COREQ: CHEM 301 or 307.</p>
</div>
<div id="sharing-teaching-material" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Sharing teaching material</h1>
<p>Teaching material will be shared with students on <a href="https://github.com/svenbuerki/Genomics-Bioinformatics">GitHub</a> and Google drive. Google drive will mostly be used to share:</p>
<ul>
<li>Sensitive material (e.g. IP addresses and passwords of lab computers).</li>
<li>Scientific publications.</li>
<li>Documents related to class assignments.</li>
</ul>
<p>Students will also be uploading their assignments on Google Drive.</p>
<p>The URL to the shared Google drive is (the instructor will grant you access to this folder prior to the start of the semester):</p>
<ul>
<li><a href="https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing" class="uri">https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing</a></li>
</ul>
</div>
<div id="publications-textbooks-and-websites-supporting-this-course" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Publications, textbooks and websites supporting this course</h1>
<p>The reading material at the basis of this class is composed of a mixture of publications and chapters from four textbooks <span class="citation">(<a href="#ref-Dale2012" role="doc-biblioref">Dale et al., 2012</a>; <a href="#ref-deSalle2013" role="doc-biblioref">deSalle and Rosenfeld, 2013</a>; <a href="#ref-Brown2017" role="doc-biblioref">Brown, 2017</a>; <a href="#ref-Lesk2017" role="doc-biblioref">Lesk, 2017</a>)</span>.
The instructor has copies of these textbooks and students are welcome to consult them at anytime.</p>
<p>A list of references used in this course is provided <a href="References.html">here</a>.</p>
<p>In addition, there is also a plethora of website devoted to genomic and transcriptomic data.
Please find a selection below:</p>
<ul>
<li><a href="https://www.ebi.ac.uk/training/online/course/ebi-next-generation-sequencing-practical-course">Next-generation sequencing online course</a>.</li>
<li><a href="https://www.integratedbreeding.net/courses/genomics-and-comparative-genomics/www.generationcp.org/genomics/index390e.html?page=1131">Genomics and comparative genomics website</a>.</li>
</ul>
</div>
<div id="bioinformatic-tools" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Bioinformatic tools</h1>
<p>Although we will be using software installed on the Linux computers, the instructor is advising students to install the following programs on their personal computers:</p>
<ul>
<li>The <strong>R</strong> statistical language that will allow you to gather data and analyze it.</li>
<li>The <strong>LaTeX</strong> and <strong>Markdown</strong> markup languages that you can use to create documents (slideshows, articles, books, webpages) for presenting your findings.</li>
<li>The <em>knitr</em> and <em>rmarkdown</em> <strong>packages</strong> for R and other tools, including <strong>command-line shell programs</strong> like GNU Make and Git version control, for dynamically tiding your data gathering, analysis, and presentation documents together so that they can be easily reproduced.</li>
<li><strong>RStudio</strong>, a program that brings all of these tools together in one place.</li>
</ul>
<div id="installing-the-main-software" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Installing the main software</h2>
<p>As shown above, <strong>R</strong> and <strong>RStudio</strong> are at the core of this course and will have to be installed on your computers. This can be easily done by downloading the software from the following websites:</p>
<ul>
<li><strong>R</strong>: <a href="https://www.r-project.org" class="uri">https://www.r-project.org</a></li>
<li><strong>RStudio</strong>: <a href="https://www.rstudio.com/products/rstudio/download/" class="uri">https://www.rstudio.com/products/rstudio/download/</a></li>
</ul>
<p>There is a preview release of RStudio including a new visual Markdown editing mode, which is handy when you are using this program extensively. You can download it <a href="https://rstudio.github.io/visual-markdown-editing/#/">here</a>, but this version might not be as stable as the other version made available on the main webpage (see above).</p>
<p>The download webpages for these software have comprehensive information on how to install them, so please refer to those pages for more information.</p>
</div>
<div id="installing-markup-languages" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Installing Markup languages</h2>
<p>If you are planning to create <code>LaTeX</code> documents, you will need to install a <code>Tex</code> distribution. Please refer to this website for more details: <a href="https://www.latex-project.org/get/" class="uri">https://www.latex-project.org/get/</a></p>
<p>If you want to create Markdown documents you can separately install the <em>rmarkdown</em> package in R (see <a href="#Rpack">below</a> for more details).</p>
</div>
<div id="installing-r-packages" class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> <a name="Rpack"></a>Installing R packages</h2>
<p>We will be using a number of R packages especially designed to support reproducible research. Many of those packages are not included in the default <code>R</code> installation and will need to be installed separately.
To install key packages used in class, copy the following code and paste it into your R console:</p>
<pre class="r"><code>install.packages(c(&quot;brew&quot;, &quot;countrycode&quot;, &quot;devtools&quot;, &quot;dplyr&quot;, &quot;ggplot2&quot;, &quot;googleVis&quot;, 
    &quot;knitr&quot;, &quot;rmarkdown&quot;, &quot;tidyr&quot;, &quot;xtable&quot;))</code></pre>
<p>Once you enter this code, you may be asked to select a CRAN “mirror” to download the packages from. Simply select the mirror closest to you.</p>
<p>Finally, it is highly likely that we will have to install additional packages. In this case, you can simply install it by using the same R function <code>install.packages()</code> or by using RStudio as follows: Select “Tools” -&gt; “Install Packages …” and then type the name of the package in the window (make sure to tick the “Install dependencies” box).</p>
</div>
<div id="rstudio-cheat-sheets" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> <a name="Cheat"></a>RStudio Cheat Sheets</h2>
<p>RStudio provides a suite of cheat sheets that can be accessed by going to the “Help” menu and selecting “Cheatsheets.”</p>
<p>Five cheat sheets are especially relevant to chapters taught in this course:</p>
<ul>
<li><em>RStudio IDE : Cheat sheet</em></li>
<li><em>Data Manipulation with dplyr, tidyr</em></li>
<li><em>Data Visualization with ggplot2</em></li>
<li><em>R Markdown Cheat Sheet</em></li>
<li><em>R Markdown Reference Guide</em></li>
</ul>
</div>
<div id="r-tutorials" class="section level2" number="11.5">
<h2><span class="header-section-number">11.5</span> R tutorials</h2>
<p>Please find below two documents providing a comprehensive introduction to R:</p>
<ul>
<li>R for beginners (a tutorial by Emmanuel Paradis): <a href="https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf" class="uri">https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf</a></li>
<li>An introduction to R: <a href="https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf" class="uri">https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf</a></li>
</ul>
</div>
</div>
<div id="journalclub" class="section level1" number="12">
<h1><span class="header-section-number">12</span> Journal club</h1>
<p>The aim of the journal club is to train students to read scientific publications related to genomic data.
Journal clubs will also provide opportunities for students to learn how to present their scientific publications.</p>
<p>Students are expected to have read the scientific publications before coming to class.
The instructor will start by briefly summarizing the methodology and main outcomes of the study and the rest of the class will be devoted to general discussion (please come with some questions and have a critical, but positive mind set).</p>
<p>Please find below the list of publications that we will discuss in class (see section <a href="#timetable">20</a> for information on dates):</p>
<ul>
<li><strong>Publication 1:</strong> <span class="citation"><a href="#ref-Li2009" role="doc-biblioref">Li et al.</a> (<a href="#ref-Li2009" role="doc-biblioref">2009</a>)</span> published a draft genome of the giant panda and screened for genes responsible for the detoxification of cyanide produced by bamboos. The presentation slides (in <code>pdf</code> format) of this publication can be downloaded <a href="Lectures/Journal_Club_Giant_panda.pdf" target="_blank">here</a>.</li>
<li><strong>Publication 2:</strong> <span class="citation"><a href="#ref-Ming2015" role="doc-biblioref">Ming et al.</a> (<a href="#ref-Ming2015" role="doc-biblioref">2015</a>)</span> published the draft genome of the pineapple and investigated the origin of CAM photosynthesis in this lineage.</li>
</ul>
<p>If you are not familiar with approaches to read scientific publications, please come and seek advice. For undergraduate students, there is a class BIOL 306 that you could take next semester. Please find below the details of this class:</p>
<p><strong>BIOL 306 COMMUNICATION IN THE BIOLOGICAL SCIENCES (3-0-3)(F,S)(CID)</strong> Development of written and oral communication skills necessary for future careers in the biological sciences.
Skills include summarizing and evaluating scientific research, and communicating scientific information to targeted audiences.
PREREQ: BIOL 304 and ENGL 101, 102.</p>
</div>
<div id="grading-information" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Grading information</h1>
<div id="tests-conducted-during-this-course" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> Tests conducted during this course</h2>
<ul>
<li>3 individual mini-reports (2x 25 points and 1x 50 points, TOTAL: 100 points).</li>
<li>1 individual oral presentation on a scientific paper (50 points).</li>
<li>1 group lab report (150 points).</li>
<li>1 group lab presentation (50 points).</li>
</ul>
<p>Exams are summing to a total of <strong>350 points</strong> and grading will follow the scale presented in Table <a href="#tab:mygrade">13.1</a>. Consult the <a href="timetable">Timetable</a> for information on dates when assignments are due.</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:mygrade">Table 13.1: </span>Grading scale applied in this course.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Percentage
</th>
<th style="text-align:left;">
Grade
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
100-98
</td>
<td style="text-align:left;">
A+
</td>
</tr>
<tr>
<td style="text-align:left;">
97.9-93
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
92.9-90
</td>
<td style="text-align:left;">
A-
</td>
</tr>
<tr>
<td style="text-align:left;">
89.9-88
</td>
<td style="text-align:left;">
B+
</td>
</tr>
<tr>
<td style="text-align:left;">
87.9-83
</td>
<td style="text-align:left;">
B
</td>
</tr>
<tr>
<td style="text-align:left;">
82.9-80
</td>
<td style="text-align:left;">
B-
</td>
</tr>
<tr>
<td style="text-align:left;">
79.9-78
</td>
<td style="text-align:left;">
C+
</td>
</tr>
<tr>
<td style="text-align:left;">
77.9-73
</td>
<td style="text-align:left;">
C
</td>
</tr>
<tr>
<td style="text-align:left;">
72.9-70
</td>
<td style="text-align:left;">
C-
</td>
</tr>
<tr>
<td style="text-align:left;">
69.9-68
</td>
<td style="text-align:left;">
D+
</td>
</tr>
<tr>
<td style="text-align:left;">
67.9-60
</td>
<td style="text-align:left;">
D
</td>
</tr>
<tr>
<td style="text-align:left;">
59.9-0
</td>
<td style="text-align:left;">
F
</td>
</tr>
</tbody>
</table>
</div>
<div id="minireport" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> Mini-reports (TOTAL: 100 points)</h2>
<p>To further gain expertise in this field, students will be producing three mini-reports on the following topics (see Figure <a href="#fig:flowchart">4.1</a>):</p>
<ul>
<li><strong>Mini-Report 1</strong>: Sequencing technologies (25 points; this task is related to Chapter 2).</li>
<li><strong>Mini-Report 2</strong>: Molecular biology databases (25 points; this task is related to Chapter 3).</li>
<li><strong>Mini-Report 3</strong>: Apply a coupled DNA barcoding and phylogenetic approach to identify species and infer their evolutionary history (50 points). This report provides students with an introduction to methods applied in the laboratory sessions and expertise to select their comparative article (<a href="#chap8">Chapter 8</a>).</li>
</ul>
<p>Time will be allocated in class to work on these mini-reports, but the instructor expects students to complete those on their own time.</p>
<p>The webpage dedicated to the Mini Reports is available <a href="Mini_reports.html">here</a>.</p>
</div>
<div id="chap8" class="section level2" number="13.3">
<h2><span class="header-section-number">13.3</span> Individual oral presentation (TOTAL: 50 points)</h2>
<p>Each student will have to present a scientific publication at the end of the semester related to material presented in Chapter 8.
Time will be allocated to select a paper and work on presentations, but the instructor strongly advise students to start this endeavor sooner.
<strong>The presentation should not exceed 4 minutes (plus 2 minutes for questions) and contain a maximum of 5 slides.</strong></p>
</div>
<div id="group-lab-report-total-150-points" class="section level2" number="13.4">
<h2><span class="header-section-number">13.4</span> Group lab report (TOTAL: 150 points)</h2>
<div id="research-topic" class="section level3" number="13.4.1">
<h3><span class="header-section-number">13.4.1</span> Research topic</h3>
<p>Students will be working in groups to produce a report associated to the following research question:</p>
<blockquote>
<p>What Aquaporin genes are encoded in the sagebrush genome (<em>Artemisia tridentata</em>) and what are their functions?</p>
</blockquote>
<p>To gain further insights into Aquaporins and sagebrush and the bioinformatic analyses, students will be reading this companion publication <span class="citation">(<a href="#ref-Melton2021" role="doc-biblioref">Melton et al., 2021</a>)</span>.</p>
</div>
<div id="associated-scripts" class="section level3" number="13.4.2">
<h3><span class="header-section-number">13.4.2</span> Associated scripts</h3>
<p>Students can access <code>bash</code> and <code>R</code> scripts associated to their lab report <a href="Mining_draft_genome.html">here</a>.</p>
</div>
<div id="details-on-reporting" class="section level3" number="13.4.3">
<h3><span class="header-section-number">13.4.3</span> Details on reporting</h3>
<p>Lab reports will be formal documents written in the form of scientific manuscripts (Introduction, Material &amp; Methods, Results, Discussion and References sections).</p>
<p>The instructor does not fix a minimum number of words or pages, but urges students to be succinct and avoid redundancy.</p>
<p>Please make sure that you present all evidence allowing an external reviewer to replicate your analyses. In order to achieve this goal and support reproducible science, students will have to provide associated data and code underpinning their research. The instructor will provide guidance on these protocols during lab sessions.</p>
<p>Please note that if students are working in group then each student should contribute equally to the project. If participation is not judge adequate within a group, then the instructor will grade students individually.</p>
</div>
</div>
<div id="group-lab-presentation-total-50-points" class="section level2" number="13.5">
<h2><span class="header-section-number">13.5</span> Group lab presentation (TOTAL: 50 points)</h2>
<p>Students will use their lab reports to prepare 10 minutes presentations (+ 5 minutes for questions) taking place during the final weeks of the semester.</p>
<p>The instructor expects each student to take part to the presentation and to be involved in answering questions.</p>
<p>Presentations will have to be deposited in advance onto a shared Google drive folder or be sent to the instructor prior to the presentation.</p>
</div>
<div id="extra-credits" class="section level2" number="13.6">
<h2><span class="header-section-number">13.6</span> Extra credits</h2>
<p>For students interested in getting some extra credits, the instructor will provide two opportunities to score a total of 30 extra credits. Students will be able to allocate these extra credits to any of their tests (by sending an email to the instructor with credits allocation). These two non-mandatory assignments have strict deadlines (see <a href="#timetable">Timetable</a>). This means that the instructor will consider that if students have not turned-in their assignment on time that they are not interested in getting extra credits. To facilitate students’ time management, credits allocated to each question will be provided. These opportunities should provide students with the ability to claim extra credits (and therefore improve their overall grade) while also learning about several aspects of genome assembly and annotation. As usual, students can work in groups, but they have to submit individual assignments.</p>
<p>The non-mandatory assignments will be accessible here:</p>
<ul>
<li>Non-mandatory assignment 1 <!-- (Non_mandatory_assignment_1.html).--></li>
<li>Non-mandatory assignment 2 <!--(Non_mandatory_assignment_2.html).--></li>
</ul>
</div>
</div>
<div id="late-work-policy" class="section level1" number="14">
<h1><span class="header-section-number">14</span> Late work policy</h1>
<p>The instructor expects students to deliver their assignments on time and set enough time aside to work on their projects (see above for more details).
However, if you have any issues preventing completion of your work on time, please contact the instructor as soon as possible to find common solutions.</p>
</div>
<div id="engagement-expectations" class="section level1" number="15">
<h1><span class="header-section-number">15</span> Engagement expectations</h1>
<p>The remote teaching mode of this course makes it harder for the instructor to take attendance.
The instructor is expecting students to attend classes (please join on time and for the full duration of the class) and actively engage by asking questions and giving feedback on teaching material and course content.
This course was designed to help students implementing a reproducible approach to their thesis projects.
If you are judging that additional content should be covered in class, please contact the instructor.
The instructor will do his very best to obtain information or seek support from colleagues to cover the requested material.
Finally, the instructor is aware that COVID could potentially impact yourselves and your families.
In the case that you have any issues attending class, please contact the instructor by email (<a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a>) asap and see below for more details.</p>
</div>
<div id="what-you-can-expect-of-me" class="section level1" number="16">
<h1><span class="header-section-number">16</span> What you can expect of me</h1>
<p>The instructor will be prepared for class, on time and not leave early.
He will also be respectful of you and your opinions.
Overall, the instructor wants to foster a kind and respectful class environment where all students can express themselves and share their opinions.
This means that meaningful and constructive dialogue is encouraged in this class and it requires a degree of mutual respect, willingness to listen, and tolerance of opposing points of view.
Respect for individual differences and alternative viewpoints will be maintained at all times in this class.
One’s words and use of language should be temperate and within acceptable bounds of civility and decency.
Finally, the instructor will reply to emails and grade tests as soon as possible (and provide positive criticism) to allow students mastering the material presented in class.</p>
</div>
<div id="this-course-was-designed-with-you-in-mind" class="section level1" number="17">
<h1><span class="header-section-number">17</span> This course was designed with you in mind</h1>
<p>The instructor has developed this course to provide a welcoming environment and effective, equitable learning experience for all students.
If you encounter barriers in this course, please bring them to my attention so that I may work to address them.</p>
<div id="this-classs-community-is-inclusive" class="section level2" number="17.1">
<h2><span class="header-section-number">17.1</span> This class’s community is inclusive</h2>
<p>Students in this class represent a rich variety of backgrounds and perspectives.
The Biological Sciences department is committed to providing an atmosphere for learning that respects diversity and creates inclusive environments in our courses.
While working together to build this community, we ask all members to:</p>
<ul>
<li>Share their unique experiences, values, and beliefs, if comfortable doing so.</li>
<li>Listen deeply to one another.</li>
<li>Honor the uniqueness of their peers.</li>
<li>Appreciate the opportunity we have to learn from each other in this community.</li>
<li>Use this opportunity together to discuss ways in which we can create an inclusive environment in this course and across the campus community.</li>
<li>Recognize opportunities to invite a community member to exhibit more inclusive, equitable speech or behavior—and then also invite them into further conversation. We also expect community members to respond with gratitude and to take a moment of reflection when they receive such an invitation, rather than react immediately from defensiveness.</li>
<li>Keep confidential any discussions that the community has of a personal (or professional) nature, unless the speaker has given explicit permission to share what they have said.</li>
<li>Respect the right of students to be addressed and referred to by the names and pronouns that correspond to their gender identities, including the use of non-binary pronouns.</li>
</ul>
</div>
<div id="we-use-each-others-preferred-names-and-pronouns" class="section level2" number="17.2">
<h2><span class="header-section-number">17.2</span> We use each other’s preferred names and pronouns</h2>
<p>Please let the instructor know of your preferred or adopted name and gender pronoun(s), and he will make those changes to his own records and address you that way in all cases.</p>
<p>To change to a preferred name so that it displays on all BSU sites, including Blackboard and the course roster, contact the Registrar’s Office at (208) 426-4249.
Note that only a legal name change can alter your name on BSU official and legal documents (e.g., your transcript).</p>
</div>
<div id="this-course-is-accessible-to-students-with-disabilities" class="section level2" number="17.3">
<h2><span class="header-section-number">17.3</span> This course is accessible to students with disabilities</h2>
<p>The instructor recognizes that navigating your education and life can often be more difficult if you have disabilities.
I want you to achieve at your highest capacity in this class.
If you have a disability, the instructor needs to know if you encounter inequitable opportunities in this course related to: - Accessing and understanding course materials.
- Engaging with course materials and other students in the course.
- Demonstrating your skills and knowledge on assignments and exams.</p>
<p>If you have a documented disability, you may be eligible for accommodations in all of your courses.
To learn more, make an appointment with the university’s <a href="https://www.boisestate.edu/eac/">Educational Access Center</a>.</p>
</div>
<div id="for-students-responsible-for-children" class="section level2" number="17.4">
<h2><span class="header-section-number">17.4</span> For students responsible for children</h2>
<p>The instructor recognizes the unique challenges that can arise for students who are also parents or guardians of children.
If you have any specific needs related to this topic, please contact the instructor asap.</p>
</div>
</div>
<div id="academic-integrity" class="section level1" number="18">
<h1><span class="header-section-number">18</span> Academic integrity</h1>
<p>To create a welcoming, engaging, and effective learning environment, the instructor expects all of us to exhibit behavior that reflects Boise State’s <a href="https://www.boisestate.edu/president/values/statement-of-shared-values/">Statement of Shared Values</a>.
The Shared Values emphasize academic excellence, caring, citizenship, fairness, respect, responsibility, and trustworthiness.
In keeping with these values, the instructor expects students in this course to uphold the standards outlined in the Boise State University <a href="https://www.boisestate.edu/policy/student-affairs/code-of-conduct/">Student Code of Conduct</a>.</p>
</div>
<div id="student-well-being" class="section level1" number="19">
<h1><span class="header-section-number">19</span> Student well-being</h1>
<p>If you are struggling for any reason (COVID, relationship, family, or life’s stresses) and believe these may impact your performance in the course, the instructor is encouraging you to contact the Dean of Students at (208) 426-1527 or email <a href="mailto:deanofstudents@boisestate.edu" class="email">deanofstudents@boisestate.edu</a> for support.
Additionally, if you are comfortable doing so, please reach out to me and I will provide any resources or accommodations that I can.
If you notice a significant change in your mood, sleep, feelings of hopelessness or a lack of self worth, consider connecting immediately with Counseling Services (1529 Belmont Street, Norco Building) at (208) 426-1459 or email <a href="mailto:healthservices@boisestate.edu" class="email">healthservices@boisestate.edu</a>.</p>
<div id="what-do-you-need" class="section level2" number="19.1">
<h2><span class="header-section-number">19.1</span> What do you need?</h2>
<p>The university has many resources designed to support you as a learner and human being.
Among these are:</p>
<ul>
<li><p><a href="https://library.boisestate.edu/">Albertsons Library</a> provides a treasure trove of physical and electronic resources.</p>
<ul>
<li>As you enter the library, straight ahead you’ll find the Reference Desk, where librarians can help you find the information and resources you need.</li>
<li>The Circulation Desk lets students borrow various technologies.</li>
<li>The MakerLab on the second floor offers tools for student use, and there are friendly staff in the MakerLab to help you learn how.</li>
</ul></li>
<li><p>The <a href="https://www.boisestate.edu/writingcenter/">Writing Center</a> offers individual consultations tailored to your needs, including making sense of writing assignment instructions, brainstorming, crafting a thesis, organizing an essay, revisions, citations, and more.</p></li>
<li><p><a href="https://www.boisestate.edu/healthservices/counseling">Counseling Services</a> helps you tap into your strengths and find resources to deal more effectively with concerns that impact your pursuit of personal and academic goals. It emphasizes prevention and early detection and provides a broad spectrum of short-term counseling, consultative, evaluative, teaching, and training functions. Counseling staff consists of licensed counselors, psychologists, and closely supervised trainees/post-graduate interns.</p></li>
<li><p>The <a href="https://www.boisestate.edu/genderequity/">Gender Equity Center</a> provides specialized, brief, no-cost, confidential support for campus members affected by:</p>
<ul>
<li>Identity-related harassment.</li>
<li>Sexual harassment, sexual assault, unhealthy relationships, and stalking.</li>
<li>Life experiences related to gender, sexual orientation, and allyship.</li>
<li>Life transitions and crisis.</li>
<li>Personal, financial, or academic roadblocks.</li>
</ul></li>
<li><p><a href="https://www.boisestate.edu/mss/">Multicultural Student Services</a> works with all students interested in equity and social justice, students from different cultures, countries and ethnicities. The staff understands cultural differences and perspectives, and knows what it’s like living in a dominant culture. The training and the programs Multicultural Student Services provides can help you navigate through political correctness and into respectful dialogue. This center also provides:</p>
<ul>
<li>Support and guidance for planning cultural festivals and events.</li>
<li>In-center tutoring and studying lounge.</li>
<li>Volunteer opportunities.</li>
<li>Mentoring and Training.</li>
<li>Programs and events.</li>
</ul></li>
<li><p>Food assistance: If you are hungry and cannot afford to purchase food, the campus has some resources to help you. You can visit the <a href="https://www.boisestate.edu/deanofstudents/student-support/food-help/food-pantry/">campus food pantry</a> or <a href="https://www.boisestate.edu/deanofstudents/student-support/food-help/meal-assistance/">get free meals in the campus dining hall</a>.</p></li>
</ul>
</div>
</div>
<div id="timetable" class="section level1" number="20">
<h1><span class="header-section-number">20</span> Timetable</h1>
<p>A tentative timetable for the Spring semester 2022 is presented in Table <a href="#tab:timetab">20.1</a>.</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:timetab">Table 20.1: </span>Tentative timetable for the Genomics and Bioinformatics course during the Spring semester 2022.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
WeekID
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Weekday
</th>
<th style="text-align:left;">
Topic
</th>
<th style="text-align:left;">
Task / Deadline
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2022-01-12
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Syllabus
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2022-01-14
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Chapter 1
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2022-01-19
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Chapter 1 / Chapter 2
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2022-01-21
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini1">Mini-Report 1</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
2022-01-26
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Chapter 2 / <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini1">Mini-Report 1</a>
</td>
<td style="text-align:left;">
Read Giant panda <a href="http://dx.doi.org/10.1038/nature08696">publication</a> for Journal Club (on <a href="https://drive.google.com/drive/folders/1-Fkz0NnlqqeSq5DbSeDGn43xbturaPnu?usp=sharing">Google Drive</a>)
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
2022-01-28
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/index.html#12_Journal_club">Journal Club: Pub. #1</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
2022-02-02
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Chapter 3
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
2022-02-04
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Chapter 3 / <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini2">Mini-Report 2</a>
</td>
<td style="text-align:left;">
Turn-in Mini-Report 1 (by 5PM MT)
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
2022-02-09
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini3">Mini-Report 3 - PART1</a>
</td>
<td style="text-align:left;">
<a href="https://digitalworldbiology.com/FinchTV">Install FinchTV</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
2022-02-11
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini3">Mini-Report 3 - PART1</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
2022-02-16
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini3">Mini-Report 3 - PARTS 2-3</a>
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#4102_Bioinformatic_tools">Install R, RStudio and rentrez</a> and <a href="https://www.megasoftware.net">Install MEGA</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
2022-02-18
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini3">Mini-Report 3 - PARTS 2-3</a>
</td>
<td style="text-align:left;">
Turn-in Mini-Report 2 (by 5PM MT)
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
2022-02-23
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini3">Mini-Report 3 - PART3</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
2022-02-25
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#mini3">Mini-Report 3 - PART3</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
2022-03-02
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Mini-Report 3 - Work on <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Mini_reports.html#46_Writing_your_report">Reports</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
2022-03-04
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Chapter 4 - Introduction to <em>de novo</em> genome assembly
</td>
<td style="text-align:left;">
If you have a PC: <a href="http://www.putty.org">Install Putty</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
2022-03-09
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
<a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Tutorials.html">Group accounts &amp; Introduction to Linux OS</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
2022-03-11
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Mining genome: Set-up project &amp; BLAST analysis
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
2022-03-16
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Mining genome: Extract scaffolds &amp; identify ORFs
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
2022-03-18
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Mining genome: Annotating ORFs &amp; identifying ORFs associated to target genes
</td>
<td style="text-align:left;">
Turn-in Mini-Report 3 (by 5PM MT)
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
2022-03-30
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Mining genome: Annotating ORFs &amp; identifying ORFs associated to target genes
</td>
<td style="text-align:left;">
Groups submit preliminary lab reports for comment
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
2022-04-01
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Mining genome: Validate AQP protein sequences and infer their function(s)
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
2022-04-06
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Chapter 4 - PARTS 1-2
</td>
<td style="text-align:left;">
Students submit their publications for approval by instructor
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
2022-04-08
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Chapter 4 - PART3 &amp; Chapter 7
</td>
<td style="text-align:left;">
Read pubs. for chapters 5 &amp; 6
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
2022-04-13
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Mining genome: Group work
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
2022-04-15
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Mining genome: Group work
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
2022-04-20
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Finalize group reports &amp; work on ind. oral presentations (chapter 8)
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
2022-04-22
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Students oral presentations (on comparative genomics; chapter 8)
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
2022-04-27
</td>
<td style="text-align:left;">
Wednesday
</td>
<td style="text-align:left;">
Students oral presentations (on comparative genomics; chapter 8)
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
2022-04-29
</td>
<td style="text-align:left;">
Friday
</td>
<td style="text-align:left;">
Mining genome - Oral group presentations
</td>
<td style="text-align:left;">
Turn-in lab reports (by 5PM MT)
</td>
</tr>
</tbody>
</table>
</div>
<div id="references" class="section level1" number="21">
<h1><span class="header-section-number">21</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Brown2017" class="csl-entry">
<span class="smallcaps">Brown, T.A.</span> 2017. Genomes 4. Garland Science.
</div>
<div id="ref-Dale2012" class="csl-entry">
<span class="smallcaps">Dale, J.W.</span>, <span class="smallcaps">M. von Schantz</span>, and <span class="smallcaps">N. Plant</span>. 2012. From genes to genomes: Concepts and applications of DNA technology, 3rd edition. Wiley.
</div>
<div id="ref-deSalle2013" class="csl-entry">
<span class="smallcaps">deSalle, R.</span>, and <span class="smallcaps">J. Rosenfeld</span>. 2013. Phylogenomics: A primer 1st edition. Garland Science.
</div>
<div id="ref-Lesk2017" class="csl-entry">
<span class="smallcaps">Lesk, A.</span> 2017. Introduction to genomics, third edition. Oxford University Press.
</div>
<div id="ref-Li2009" class="csl-entry">
<span class="smallcaps">Li, R.</span>, <span class="smallcaps">W. Fan</span>, <span class="smallcaps">G. Tian</span>, <span class="smallcaps">H. Zhu</span>, <span class="smallcaps">L. He</span>, <span class="smallcaps">J. Cai</span>, <span class="smallcaps">Q. Huang</span>, et al. 2009. The sequence and de novo assembly of the giant panda genome. <em>Nature</em> 463: 311 EP–. Available at: <a href="http://dx.doi.org/10.1038/nature08696">http://dx.doi.org/10.1038/nature08696</a>.
</div>
<div id="ref-Melton2021" class="csl-entry">
<span class="smallcaps">Melton, A.E.</span>, <span class="smallcaps">J. Beck</span>, <span class="smallcaps">S.J. Galla</span>, <span class="smallcaps">J. Jenkins</span>, <span class="smallcaps">L. Handley</span>, <span class="smallcaps">M. Kim</span>, <span class="smallcaps">J. Grimwood</span>, et al. 2021. A draft genome provides hypotheses on drought tolerance in a keystone plant species in western north america threatened by climate change. <em>Ecology and Evolution</em> 11: 15417–15429. Available at: <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.8245">https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.8245</a>.
</div>
<div id="ref-Ming2015" class="csl-entry">
<span class="smallcaps">Ming, R.</span>, <span class="smallcaps">R. VanBuren</span>, <span class="smallcaps">C.M. Wai</span>, <span class="smallcaps">H. Tang</span>, <span class="smallcaps">M.C. Schatz</span>, <span class="smallcaps">J.E. Bowers</span>, <span class="smallcaps">E. Lyons</span>, et al. 2015. <span class="nocase">The pineapple genome and the evolution of CAM photosynthesis</span>. <em><span>Nature Genetics</span></em> 47: 1435.
</div>
</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="22">
<h1><span class="header-section-number">22</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.6. 2020. &lt;URL:
<a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.10. 2019. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] M. C. Koohafkan. <em>kfigr: Integrated Code Chunk Anchoring and
Referencing for R Markdown Documents</em>. R package version 1.2. 2015.
&lt;URL: <a href="https://github.com/mkoohafkan/kfigr" class="uri">https://github.com/mkoohafkan/kfigr</a>&gt;.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[5] H. Wickham and J. Bryan. <em>usethis: Automate Package and Project
Setup</em>. R package version 2.0.0. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>&gt;.</p>
<p>[6] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of
Data Manipulation</em>. R package version 1.0.2. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>&gt;.</p>
<p>[7] H. Wickham, J. Hester, and W. Chang. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.3.2. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>&gt;.</p>
<p>[8] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. &lt;URL: <a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.21. 2020. &lt;URL:
<a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[10] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[11] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.7. 2019. &lt;URL: <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>&gt;.</p>
<p>[12] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R.” In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. &lt;URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[13] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.30. 2020. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[14] Y. Xie and J. Allaire. <em>tufte: Tufte’s Styles for R Markdown
Documents</em>. R package version 0.9. 2020. &lt;URL:
<a href="https://github.com/rstudio/tufte" class="uri">https://github.com/rstudio/tufte</a>&gt;.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>&gt;.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020. &lt;URL:
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
<p>[17] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.2.1. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=kableExtra" class="uri">https://CRAN.R-project.org/package=kableExtra</a>&gt;.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
