<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-01-25" />

<title>Lexicon</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.24/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.11.3/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="R_Functions.html">R functions</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Lexicon</h1>
<h4 class="date">2023-01-25</h4>

</div>


<p>A non-exclusive list of terms and their definitions used in this course is provided in the table below. Please feel free to contact the instructor to add terms to this list or make any corrections.</p>
<div id="htmlwidget-dc46843822bd6a44c949" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-dc46843822bd6a44c949">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["Amino acid (AA)","Bioinformatics","BLAST","Clade","Cluster","Complementary DNA sequence","Cryptic species","Distance/similarity (Genetic)","DNA","Exome sequencing","Genetic code ","Genome","Genomics","Heterozygosity","Heterozygote","High molecular weight DNA","Holoploid 1C-value (abbreviated to 1C-value)","k-mer","Mate pair sequencing","Maximum Likelihood (ML)","Monoisolate","Monophyly","N50","Next-Generation Sequencing (NGS)","Open Reading Frame (ORF) ","Orthologous genes","Paired-end sequencing","Paralogue","Phred quality score","Phylogenetic species concept","Phylogenetic support","Phylogenetic tree","Plesiomorphic","Polymerase chain reaction (PCR)","Promoter","Read trimming","Reverse DNA sequence","RNA","RNA-Seq","Single-end sequencing","Species circumscription/delimitation/delineation","Specimen identification/determination","Synteny","Targeted re-sequencing","Umami, or savory taste","Whole-genome sequencing (WGS)"],["One of the constituents of proteins; proteins are polymers of amino acids.","The use of computer methods in studies of genomes.","The Basic Local Alignment Search Tool uses a query sequence and large database to find regions of local similarity between sequences. The\nprogram is at the heart of the National Center for Biotechnology Information’s sequence search engine, compares nucleotide or protein sequences to the\never-growing sequence databases and estimates the statistical significance of matches.","This term refers to a monophyletic (natural) group, namely a hypothetical common ancestor and all its descendants, as identified by uniquely derived\nand unreversed synapomorphies. A clade is visualized on a cladogram as a node and all its subtended terminals.","A group of individuals or genes visualized as terminals on a tree or dendrogram and used in place of “clade” whenever analyses are conducted\nbelow the species level. A group of organisms is said to cluster in an analysis when they share an exclusive node. Because clustering algorithms may be\napplied below the species level where relationships are not strictly nested, clusters are not monophyletic in the strict sense, only a graphic one. Cluster is also\na term used to define closely related organisms in principal components analysis or STRUCTURE.","Nucleic acid sequence of bases that can form a double- stranded structure by matching base pairs. For example, the complementary sequence to C-A-T-G (where each letter stands for one of the bases in DNA) is G-T-A-C.","Difficult to detect and, in reference to species, referring to difficulty in diagnosing or recognizing morphologically\nindistinguishable species without DNA barcode data. One of the explicitly targeted applications of DNA barcoding is that of detecting cryptic species.","A phenetic measure of comparison which represents the overall similarity of two organisms. Operationally, a pairwise measure\ngenerated from sequence data, most commonly via the Kimura two parameter (K2P) model which specifies probabilities of different kinds of character state\n(base pair) change. The lack of equiprobability is used to correct the distance measure for rate heterogeneity of sequence change.","Deoxyribonucleic acid.","Sequencing only the portions of a genome corresponding to expressed proteins.","The genetic code specifies how the nucleotide sequence of an mRNA is translated into the amino acid (AA) sequence of a protein. Different genetic codes depending on lineages. 20 amino acids: high redundancy (except Met &amp; Trp). Start codon: Met (sometimes Trp in cpDNA).","A genome is the entire genetic complement of a living organism. Each organism possesses a genome containing the biological information needed to construct and maintain a living example of that organism. Most genomes are made of DNA (deoxyribonucleic acid), but few viruses have RNA (ribonucleic acid) genomes.","The study of genomes, the complete set of genetic material within an organism (i.e. nuclear, mitochondrial and chloroplastic genomes). Genomics uses an approach combining next-generation sequencing (NGS) and bioinformatics to assemble and annotate entire genomes. ","The condition of having two different alleles at a locus.","A diploid organism that carries two different versions of a specific gene.","High molecular weight DNA corresponds to DNA fragments between 100-300kb in length","This term refers to the\namount of DNA in the unreplicated gametic nucleus (e.g.,\npollen or egg cell of angiosperms) regardless of the ploidy\nlevel of the cell.","In bioinformatics, k-mers are subsequences of length k contained within a biological sequence. ","Technique in which sequence is reported from two ends of a DNA fragment, usually several thousand base-pairs long.","A class of parameterized tree-building approaches that incorporates probabilities of character state change based on\nfrequentist statistics among different classes of character data (e.g., transitions vs. transversions, codon positions, etc.). The likelihood of the data given a\ntree and a model is computed to find an optimal tree for a dataset.","a single animal, cultured cell-line, inbred population,.or possibly a heterogeneous population when a single genome assembly is generated from a pooled sample because multiple individuals are needed to collect enough material and an inbred line is not available; however, this situation is not preferred.","In cladistics for a group of organisms, monophyly is the condition of being a clade—that is, a group of taxa composed only of a common ancestor and all of its lineal descendants.","Measure to describe the quality of assembled genomes that are fragmented in contigs/scaffolds of different length. The N50 is defined as the minimum contig/scaffold length needed to cover 50% of the genome. ","Next-Generation Sequncing (NGS) is the term applied to methods enabling thousands of millions of DNA fragments to be sequenced in parallel in a single experiment. NGS outperformed Sanger sequencing, which was able to sequence only individual DNA fragments, each fragment obtained by a different PCR.","In molecular genetics, an open reading frame (ORF) is the part of a reading frame that has the ability to be translated. An ORF is a continuous stretch of codons that begins with a start codon (usually AUG) and ends at a stop codon (usually UAA, UAG or UGA).","Genes diverging after evolution gives rise to different species (i.e. they follow speciation events).","Technique in which sequence is reported from both ends of a fragment (with a number of undetermined bases between the reads that is known only approximately).","Genes or proteines with similar sequences, but different (although possibly related) functions, arising from a common ancestral gene.","Measure of the quality of the identification of the nucleobases generated by automated DNA sequencing.","The concept of a species as an irreducible group whose members are descended from a common ancestor and who all possess a combination of certain defining, or derived, traits. Hence, this concept defines a species as a group having a shared and unique evolutionary history.","The strength of inference for nodes in a phylogenetic tree are assessed using support measures. Higher the support measures\nconnote greater reliability for a given hypothesized relationship. Bremer support (maximum parsimony based), bootstrap (distance, parsimony, likelihood) and\nBayesian posteriors are all different kinds of support measures used in phylogenetic analysis.","Any bifurcating graphic or dendrogram intended to summarize comparative data and interpreted to reflect common ancestry. Since\n“tree” refers to the graphic, it is not strictly synonymous with “phylogeny” but may be treated equivalently under the explicit assumptions of an underlying\nnested hierarchy generated by descent with modification. Trees based on recombinant elements of individual conspecific organisms may violate these\nassumptions but are still be used as provisional tools for approximating species boundaries. Phylogenetic trees can be generated using any number of\nmethods as described above; the term “clade” is properly used with reference to derived or diagnostic characters and thus “cladogram” is generally reserved\nfor trees generated under parsimony.","A primitive or ancestral character state is called plesiomorphy (plesiomorphic character), and a shared plesiomorphy is called a symplesiomorphy. For example, hair is a unique mammalian character that evolved with the evolution of mammals."," Method widely used to rapidly make millions to billions of copies of a specific DNA sample, allowing scientists to take a very small sample of DNA and amplify it to a large enough amount to study in detail.","Promoter sequences are DNA sequences that define where transcription of a gene by RNA polymerase begins. Promoter sequences are typically located directly upstream or at the 5' end of the transcription initiation site.","Sequencing data is often provided as raw reads which are processed prior to analysis 1 of the most used preprocessing procedures is read trimming, which aims at removing low quality portions while preserving the longest high quality part of a NGS read.","It is the sequence of the upper strand in the direction from its 3′- to its 5′-end. The reverse complement sequence is the sequence of the lower strand in the direction of its 5′- to its 3′-end.","Ribonucleic acid.","Sequencing RNAs in a sample. RNA sequencing is carried out by reverse transcribing the RNAs, and sequencing the complementary DNA (cDNA).","Technique in which sequence is reported from only one end of a fragment.","The iterative process of collating potentially diagnostic character data to proscribe observational\nboundaries between two or more species. Species delimitation methods are broad and require a criterion specified a priori. Delimitation is\nused interchangeably with delineation, circumscription and demarcation.","The process of using morphological or molecular diagnostics or other organismal attributes to assign biological\nspecimens taxonomic names. Not to be confused with species delimitation or discovery.","Conservation of blocks of genes/DNA between two or more sets of chromosomes belonging to different species.","Sequencing of selected regions of a genome; for instance, a subset of genes (using either baits or amplicons). This approach is now commonly applied for phylogenetic inferences. ","One of the five basic tastes, which has been described as brothy or meaty and is associated with glutamate.","The process of determining the complete DNA sequence of an organism's genome at a single time."]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Term<\/th>\n      <th>Definition<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-weight':'bold'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
