<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki - Boise State University" />

<meta name="date" content="2022-01-18" />

<title>BIOL 497/597 - Genomics &amp; Bioinformatics</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">BIOL 497/597 - Genomics &amp; Bioinformatics</h1>
<h3 class="subtitle">Mini-reports</h3>
<h4 class="author">Sven Buerki - Boise State University</h4>
<h4 class="date">2022-01-18</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>To further gain expertise in the field of genomics, students are producing three mini-reports on the following topics:</p>
<ul>
<li><a href="#mini1"><strong>Mini-Report 1</strong></a>: Sequencing technologies (25 points; this task is related to Chapter 2).</li>
<li><a href="#mini2"><strong>Mini-Report 2</strong></a>: Molecular biology databases (25 points; this task is related to Chapter 3).</li>
<li><a href="#mini3"><strong>Mini-Report 3</strong></a>: Apply a coupled DNA barcoding and phylogenetic approach to identify species and infer their evolutionary history (50 points). This report provides students with an introduction to methods applied in the laboratory sessions and expertise to select a comparative article (Chapter 8).</li>
</ul>
<p>Time will be allocated in class to work on these mini-reports, but the instructor expects students to complete those on their own time.</p>
</div>
<div id="mini1" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Mini-report 1 – Sequencing platforms &amp; technologies</h1>
<p>To have a full overview of available sequencing platforms, students are tasked to produce individual mini-reports on the following sequencing platforms and their associated technologies:</p>
<ul>
<li><strong>Sequencing platform 1:</strong> Illumina.</li>
<li><strong>Sequencing platform 2:</strong> PacBio.</li>
<li><strong>Sequencing platform 3:</strong> Oxford Nanopore.</li>
</ul>
<div id="background-information" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Background information</h2>
<p>Each student was assigned a sequencing platform to work on (please see <a href="https://docs.google.com/spreadsheets/d/1cRM1yasUfDa98ecyz_-RxUOw8cU4wkQx4O5Wxw3y4yU/edit?usp=sharing">Google spreadsheet</a> for more information).</p>
<p>The assignment is mandatory and will be graded. Please see this task as an opportunity to consolidate your knowledge on next-generation sequencing technologies and writing scientific reports.</p>
<p>Time will be allocated in class on the 21st of January 2022 to allow students to work on their assignments. Reports are due on <strong>the 4th of February 2022</strong> and should be uploaded on the shared <a href="https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing">Google Drive</a> in the <code>Mini_Reports</code> folder.</p>
</div>
<div id="structure" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Structure of mini-reports</h2>
<p>Please structure your reports to ensure covering the following aspects:</p>
<ul>
<li>Introduce the sequencing technology (incl. library preparation and DNA requirements).</li>
<li>Provide an overview of available sequencing systems.</li>
<li>Outline the research potential/applicability of the sequencing platform.</li>
<li>Provide any additional information such as e.g. price, accessibility, technical support, that you feel are relevant to your sequencing platform.</li>
</ul>
</div>
<div id="reporting-format" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Reporting format</h2>
<p>Maximum two pages, single spaced, using either Arial 11 pt or Times New Roman 12 pt and with all margins set to 1 inch. The two pages do not include a title page and references. The title page should include: name of the sequencing platform as title and name of student contributing the report. Please keep in mind that a figure can convey a lot more information than a long text! Your reports should follow the structure presented in section <a href="#structure">2.2</a>.</p>
<p>Upload your reports on the shared Google drive by the 4th of February before 5 pm. Name your report documents following this pattern: <code>Sequencing platform_Surname</code>.</p>
</div>
<div id="resources" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Resources</h2>
<p>The instructor has provided <code>pdf</code> documents devoted to each sequencing platform below to support students in starting their assignments.</p>
<p>In addition, please use information available from manufacturers’ websites (see below), publications, online NGS training sites (e.g. EMBL-EBI Training), Google, YouTube and Wikipedia to prepare your reports and presentations. Do not forget to provide citations supporting your claims in your reports. I let you decide which reference style you want to use (look at your favorite journals for examples).</p>
<p>Here are a few additional online references to support your assignment:</p>
<ul>
<li><strong>An Overview of Next-Generation Sequencing</strong>: <a href="https://www.technologynetworks.com/genomics/articles/an-overview-of-next-generation-sequencing-346532" class="uri">https://www.technologynetworks.com/genomics/articles/an-overview-of-next-generation-sequencing-346532</a></li>
<li><strong>Illumina official website</strong>: <a href="https://www.illumina.com/" class="uri">https://www.illumina.com/</a></li>
<li><strong>PacBio official website</strong>: <a href="https://www.pacb.com/" class="uri">https://www.pacb.com/</a></li>
<li><strong>Oxford Nanopore official website</strong>: <a href="https://nanoporetech.com/" class="uri">https://nanoporetech.com/</a></li>
</ul>
<button class="btn btn-primary" data-toggle="collapse" data-target="#Illumina">
Data on Illumina
</button>
<div id="Illumina" class="collapse">
<object data="Images/Seq_1_Illumina.pdf" width="750px" height="750px">
<embed src="Images/Seq_1_Illumina.pdf">
</embed>
</object>
</div>
<button class="btn btn-primary" data-toggle="collapse" data-target="#PacBio">
Data on PacBio
</button>
<div id="PacBio" class="collapse">
<object data="Images/Seq_2_PacBio.pdf" width="750px" height="750px">
<embed src="Images/Seq_2_PacBio.pdf">
</embed>
</object>
</div>
<button class="btn btn-primary" data-toggle="collapse" data-target="#Nanopore">
Data on Nanopore
</button>
<div id="Nanopore" class="collapse">
<object data="Images/Seq_3_Oxford_Nanopore.pdf" width="750px" height="750px">
<embed src="Images/Seq_3_Oxford_Nanopore.pdf">
</embed>
</object>
</div>
</div>
</div>
<div id="mini2" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Mini-report 2 – Molecular biology databases</h1>
<p>To have a full overview of available molecular biology databases and assess their contributions to genome annotation, students are tasked to produce individual mini-reports on the following molecular databases:</p>
<ul>
<li><strong>Molecular database 1:</strong> <a href="#prot">Protein sequences databases</a>.</li>
<li><strong>Molecular database 2:</strong> <a href="#GO">Gene ontology databases</a>.</li>
<li><strong>Molecular database 3:</strong> <a href="#KEGG">Metabolic pathways databases</a>.</li>
</ul>
<div id="background-information-1" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Background information</h2>
<p>Each student was assigned a molecular database to work on (please see <a href="https://docs.google.com/spreadsheets/d/1cRM1yasUfDa98ecyz_-RxUOw8cU4wkQx4O5Wxw3y4yU/edit?usp=sharing">Google spreadsheet</a> for more information).</p>
<p>The assignment is mandatory and will be graded. Please see this task as an opportunity to consolidate your knowledge on molecular databases and writing scientific reports.</p>
<p>Time will be allocated in class on the 10th of February 2021 to allow students to work on their assignments. Reports are due on <strong>the 19th of February 2021</strong> and should be uploaded on the shared <a href="https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing">Google Drive</a> in the <code>Mini_Report_2</code> folder.</p>
</div>
<div id="structure2" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Structure of mini-reports</h2>
<p>Please structure your reports to cover the following aspects:</p>
<ul>
<li>Provide overview of available databases falling under your major type of molecular biology data.</li>
<li>Highlight potential contributions of these databases in supporting annotation of genomes.</li>
<li>Provide a survey of available bioinformatic tools and websites to query/access these databases.</li>
<li>Complement reports with information relevant/specific to your data repositories.</li>
</ul>
</div>
<div id="reporting-format-1" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Reporting format</h2>
<p>Maximum two pages, single spaced, using either Arial 11 pt or Times New Roman 12 pt and with all margins set to 1 inch. The two pages do not include a title page and references. The title page should include: name of the sequencing platform as title and name of student contributing the report. Please keep in mind that a figure can convey a lot more information than a long text! Your reports should follow the structure presented in section <a href="#structure2">3.2</a>.</p>
<p>Upload your reports on the shared Google drive by the 19th of February before 5 pm. Name your report documents following this pattern: <code>Molecular database_Surname</code>.</p>
</div>
<div id="resources-1" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Resources</h2>
<p>The instructor provides below short descriptions of each of the major types of molecular biology databases. Use the information provided here to prime your research and carve your reports. Please do not forget to provide citations and references in your reports. The instructor let you decide which reference style/formatting you prefer using (look at your favorite journals for examples).</p>
<div id="prot" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Proteins sequences databases</h3>
<p>There are three major protein sequences databases:</p>
<ul>
<li><strong>PIR</strong> (Protein Information Resource in Washington, USA): <a href="http://pir.georgetown.edu">http://pir.georgetown.edu</a></li>
<li><strong>SWISS-PROT</strong> (in Geneva, Switzerland): <a href="http://www.uniprot.org/statistics/Swiss-Prot">http://www.uniprot.org/statistics/Swiss-Prot</a></li>
<li><strong>TrEMBL</strong> (in Hinxton, UK): <a href="http://www.uniprot.org/statistics/TrEMBL">http://www.uniprot.org/statistics/TrEMBL</a></li>
</ul>
<p>In 2002, these three databases coordinated their efforts to form the <strong>UniProt</strong> consortium. The partners in this enterprise share the database, but continue to offer separate information-retrieval tools for access. The UniProtKB database is available at this URL: <a href="http://www.uniprot.org">http://www.uniprot.org</a>. This latter database will be especially important to retrieve information on gene annotations (chapters 6, 7) using the online tool available at this URL: <a href="http://www.uniprot.org/uploadlists/">http://www.uniprot.org/uploadlists/</a>. This will be done by specifically looking at Gene ontology (see <a href="#GO">GeneOntology</a>).</p>
</div>
<div id="GO" class="section level3" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Gene ontology databases</h3>
<p>Gene Ontology (GO) (<a href="http://www.geneontology.org">http://www.geneontology.org</a>): the framework for the model of biology. The GO defines concepts/classes used to describe gene function, and relationships between these concepts. It classifies functions along three aspects:</p>
<ul>
<li><strong>Cellular component:</strong> This term describes a component of a cell that is part of a larger object, such as an anatomical structure (e.g. rough endoplasmic reticulum or nucleus) or a gene product group (e.g. ribosome, proteasome or a protein dimer).</li>
<li><strong>Biological process:</strong> A biological process term describes a series of events accomplished by one or more organized assemblies of molecular functions. Examples of broad biological process terms are “cellular physiological process” or “signal transduction.” Examples of more specific terms are “pyrimidine metabolic process” or “alpha-glucoside transport.” The general rule to assist in distinguishing between a biological process and a molecular function is that a process must have more than one distinct steps.
A biological process is not equivalent to a pathway. At present, the GO does not try to represent the dynamics or dependencies that would be required to fully describe a pathway.</li>
<li><strong>Molecular function:</strong> Molecular function terms are describing activities occurring at the molecular level, such as “catalytic activity” or “binding activity.” GO molecular function terms represent activities rather than the entities (molecules or complexes) that perform the actions, and do not specify where, when, or in what context the action takes place. Molecular functions generally correspond to activities that can be performed by individual gene products, but some activities are performed by assembled complexes of gene products. Examples of broad functional terms are “catalytic activity” and “transporter activity”; examples of narrower functional terms are “adenylate cyclase activity” or “toll receptor binding.”
It is easy to confuse a gene product name with its molecular function; for that reason GO molecular functions are often appended with the word “activity.”</li>
</ul>
<p>GO is also available for all the protein sequences deposited on the UniProt database.
Other useful tool to perform gene annotations: </p>
</div>
<div id="KEGG" class="section level3" number="3.4.3">
<h3><span class="header-section-number">3.4.3</span> Metabolic pathways databases</h3>
<p>The Kyoto Encyclopedia of Genes and Genomes (KEGG) (<a href="http://www.genome.jp/kegg/">http://www.genome.jp/kegg/</a>) collects individual genomes, gene products, and their functions, but its special strength lies in its integration of biochemical and genetic information. KEGG focuses on interactions: molecular assemblies, and metabolic and regulatory networks. KEGG organizes five data types into a comprehensive system:</p>
<ul>
<li>Catalogs of chemical compounds in living cells.</li>
<li>Gene catalogs.</li>
<li>Genome maps.</li>
<li>Pathway maps.</li>
<li>Orthologue tables.</li>
</ul>
<p>The catalogs of chemical compounds and genes contain information about particular molecules or sequences. Genome maps integrate the genes themselves according to their chromosomal position. Pathway maps describe potential networks of molecular activities, both metabolic and regulatory. One enzyme in one organism would be referred to in KEGG in its orthologue tables, which link the enzyme to related ones in other organisms. This permits analysis of relationships between the metabolic pathways of different organisms.</p>
</div>
</div>
</div>
<div id="mini3" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Mini-report 3 – Apply a coupled DNA barcoding and phylogenetic approach to identify species and infer their evolutionary history</h1>
<div id="background-information-2" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Background Information</h2>
<div id="how-many-species-are-there-on-earth" class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> How many species are there on Earth?</h3>
<p>Projections of global biodiversity have ranged from 2 to 100 million species <span class="citation">(<a href="#ref-Larsen2017" role="doc-biblioref">Larsen et al., 2017</a>)</span>.</p>
<p>But these estimations are without accounting for cryptic species! See for instance <span class="citation"><a href="#ref-dentinger2014s" role="doc-biblioref">Dentinger and Suz</a> (<a href="#ref-dentinger2014s" role="doc-biblioref">2014</a>)</span> for an example on porcini. In this study, the authors used DNA-sequencing to identify three species of mushroom contained within a commercial packet of dried Chinese porcini purchased in London. Surprisingly, all three have never been formally described by science and required new scientific names.</p>
<p><span class="citation"><a href="#ref-Larsen2017" role="doc-biblioref">Larsen et al.</a> (<a href="#ref-Larsen2017" role="doc-biblioref">2017</a>)</span> recently published a keystone paper where they predicted 1 to 6 billion species on Earth! This prediction was based on an average of 6 cryptic species per described species.</p>
<p>Overall, data presented here demonstrate that most species on this planet are either poorly known or pending description.</p>
<p>In this context, the fields of genetics/genomics (more specifically <a href="#barcode">DNA barcoding</a>) and phylogenetics have the potential to contribute to:</p>
<ul>
<li>species identifications and naming (taxonomy).</li>
<li>inferring evolutionary frameworks allowing developing working hypotheses on species relationships and their spatio-temporal histories (e.g. knowing how did species cope with past climatic conditions will provide insights into their adaptive capacity to forecasted conditions).</li>
</ul>
<p>Approaches combining these two objectives were applied across lineages. See for instance, an article describing a new species belonging to the soapberry family (Sapindaceae) endemic to Fiji <span class="citation">(<a href="#ref-buerki2017alectryon" role="doc-biblioref">Buerki et al., 2017</a>)</span>. In this study, the phylogeny is used to confirm the new taxon and place it within a larger evolutionary and biogeographical framework. Finally, this evidence is coupled with occurrence data to infer a species extinction risk following IUCN guidelines.</p>
</div>
<div id="barcode" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> DNA barcoding: Species identification</h3>
<p>The Consortium for the Barcode of Life (<a href="https://www.ibol.org/phase1/cbol/">CBOL</a>) is an international initiative devoted to developing DNA barcoding as a global standard for the identification of biological species. CBOL has more than 130 Member Organizations from more than 40 countries <span class="citation">(<a href="#ref-CBOL" role="doc-biblioref">CBOL, 2021</a>)</span>.</p>
<p><strong>DNA barcoding</strong> is a method of species identification using a short section of DNA from a specific gene or genes. The premise of DNA barcoding is that, by comparison with a reference library of such DNA sections, an individual DNA sequence can be used to uniquely identify an organism to species, in the same way that a supermarket scanner uses the familiar black stripes of the UPC barcode to identify an item in its stock against its reference database <span class="citation">(<a href="#ref-CBOL" role="doc-biblioref">CBOL, 2021</a>)</span>. These DNA barcodes are sometimes used in an effort to identify unknown species, parts of an organism, or simply to catalog as many taxa as possible, or to compare with traditional taxonomy in an effort to determine species boundaries. For more details on DNA barcodes applied to plants, please see section <a href="#vanbarcode">below</a> dedicated to vanilla.</p>
<p>DNA barcoding has a wide range of applications from biodiversity surveys <span class="citation">(e.g. <a href="#ref-telfer2015biodiversity" role="doc-biblioref">Telfer et al., 2015</a>)</span>, illegal trade of wildlife <span class="citation">(e.g. <a href="#ref-gonccalves2015dna" role="doc-biblioref">Gonçalves et al., 2015</a>)</span> to food security <span class="citation">(e.g. <a href="#ref-quinto2016dna" role="doc-biblioref">Quinto et al., 2016</a>)</span>. The DNA barcoding approach has been recently reviewer by <span class="citation"><a href="#ref-deSalle2019" role="doc-biblioref">DeSalle and Goldstein</a> (<a href="#ref-deSalle2019" role="doc-biblioref">2019</a>)</span>.</p>
<div id="procedure" class="section level4" number="4.1.2.1">
<h4><span class="header-section-number">4.1.2.1</span> Procedure</h4>
<p>A DNA barcoding approach is subdivided into 4 steps:</p>
<ol style="list-style-type: decimal">
<li>Isolate DNA from the target sample.</li>
<li>Amplify the target DNA barcode region using PCR.</li>
<li>Sequence the PCR products using either Sanger sequencing or NGS (mostly on Illumina platform).</li>
<li>Compare the resulting sequences against reference databases to find the matching species. Here we will be using DNA sequences from <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a> as our reference.</li>
</ol>
</div>
</div>
<div id="phylogenetic-inference" class="section level3" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> Phylogenetic inference</h3>
<p>As described in <span class="citation"><a href="#ref-Masters2017" role="doc-biblioref">Masters and Pozzi</a> (<a href="#ref-Masters2017" role="doc-biblioref">2017</a>)</span>, phylogenetic inference is the practice of reconstructing the evolutionary history of related species by grouping them in successively more inclusive sets based on shared ancestry. Homologous characters in independent lineages are similar because they have been inherited from a common ancestor, and they alone should be used in phylogenetic reconstructions. Homoplasies are characters that appear similar, but have evolved from different ancestral states. They may mislead interpretations of evolutionary history. Both molecular and morphological datasets are subject to obfuscation by homoplasy. Methods of phylogenetic inference aim to distinguish between homologous (signal) and homoplastic (noise) resemblance. Molecular datasets tend to be very large and are analyzed using statistical techniques that fit the data to models of molecular evolution. These methods are not well suited to morphological data, and combined analyses including both kinds of data tend to obscure the morphological signal. Rates of molecular change may be used to estimate divergence ages.</p>
<p>In this mini-report, we will be focusing on inferring phylogenetic trees based on DNA sequences obtained through the DNA barcoding approach described <a href="#barcode">above</a>.</p>
<div id="procedure-1" class="section level4" number="4.1.3.1">
<h4><span class="header-section-number">4.1.3.1</span> Procedure</h4>
<p>A phylogenetic analysis is subdivided into 4 steps:</p>
<ol style="list-style-type: decimal">
<li>Produce DNA sequences following steps 1 to 3 as described in <a href="#barcode">DNA barcoding section</a>.</li>
<li>Produce DNA multiple sequence alignments.</li>
<li>Test for best-fit model reflecting evolution of DNA region.</li>
<li>Infer phylogenetic tree analysis (usually based either on Maximum Likelihood and/or Bayesian criteria).</li>
</ol>
</div>
</div>
</div>
<div id="learning-outcomes" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Learning outcomes</h2>
<p>The following learning outcomes will be taught as part of this Mini-report:</p>
<ol style="list-style-type: decimal">
<li>Process and clean DNA sequence electropherograms generated using the <a href="https://dnalc.cshl.edu/resources/animations/cycseq.html">Sanger sequencing approach</a> (based on PCR amplifications).</li>
<li>Validate DNA sequence and provide species working hypothesis by conducting a BLAST similarity DNA sequences analysis coupled with a distance tree inference. This will be based on DNA sequences deposited on GenBank.</li>
<li>Query GenBank and download DNA sequences for phylogenetic analyses using R packages.</li>
<li>Produce DNA multiple sequence alignments (containing your target DNA sequences and DNA sequences deposited on GenBank) using freely available algorithms.</li>
<li>Infer phylogenetic analysis based on Maximum Likelihood (ML) criterion as implemented in RAxML <span class="citation">(<a href="#ref-Stamatakis2014" role="doc-biblioref">Stamatakis, 2014</a>)</span>. This phylogenetic framework will be used to test and refine species working hypotheses (as inferred in learning outcome #2).</li>
<li>Visualize and interpret phylogenetic analysis.</li>
<li>Learn the terminology associated to DNA barcoding and phylogenetic analyses. Several definitions associated to the most used terms is available in the <a href="Lexicon.html">Lexicon</a>.</li>
</ol>
<p>To support learning these skills, we will be using a DNA dataset focusing on the <a href="#vanilla">vanilla</a> spice (from the Orchidaceae family).</p>
</div>
<div id="vanilla" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Vanilla as model for this report</h2>
<p>Like porcini <span class="citation">(<a href="#ref-dentinger2014s" role="doc-biblioref">Dentinger and Suz, 2014</a>)</span>, vanilla is one of our most beloved spice, but we still know very little about its biology, taxonomy and evolution <span class="citation">(see <a href="#ref-Ellestadinpress" role="doc-biblioref">Ellestad et al., 2021</a> for a review)</span>. As part of a study aiming at inferring domestication processes in the genus <em>Vanilla</em> and species adaptation to drought, fieldwork was conducted in Mexico by Paige Ellestad (an EEB Ph.D. student at BSU) and her in-country partners.</p>
<p>For her Ph.D. dissertation, Paige is applying a DNA barcoding approach coupled with phylogenetic inference (and ecological niche modeling) to answer the following questions:</p>
<blockquote>
<p>Q1: What species of vanilla are cultivated in Mexico?</p>
<p>Q2: How are those species related and is there evidence of gene flow/hybridization?</p>
<p>Q3: What level of genetic diversity is there among each species?</p>
</blockquote>
<div id="vanbarcode" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> DNA barcodes generated for vanilla project</h3>
<p>To answer these questions, she sampled &gt;30 locations (mostly plantations, but also wild populations) representing &gt;60 samples and sequenced two DNA barcode regions widely used in plants: one chloroplastic (the coding <em>rbcL</em> region) and one nuclear ribosomal (the ITS region, which stands for for “Internal Transcribed Spacer”). These two DNA regions are traditionally used by researchers as DNA barcodes supporting species delimitation and identifications. Since you are already familiar with <em>rbcL</em> <span class="citation">(see <a href="#ref-Hollingsworth2009" role="doc-biblioref">Hollingsworth et al., 2009</a> for more details)</span>, we will be providing more details on the ITS region. The nuclear ribosomal ITS region includes part of 18S, ITS1, 5.8S, ITS2 and part of 26S <span class="citation">(see Figure <a href="#fig:ITS">4.1</a>, <a href="#ref-Cheng2016" role="doc-biblioref">Cheng et al., 2016</a>)</span>. This region is repeated in tandem 1000 of times and duplicated across chromosomes in order to produce ribosomes, which are key to produce proteins. This latter DNA region is shared across kingdoms and plant specific primers have to be used otherwise we may be at risk of amplifying this region for symbiotic organisms <span class="citation">(e.g. fungi; see <a href="#ref-Cheng2016" role="doc-biblioref">Cheng et al., 2016</a>)</span>.</p>
<div class="figure" style="text-align: center"><span id="fig:ITS"></span>
<img src="Images/ITS.png" alt="Map of the ITS nuclear ribsomal region with primers (from Cheng et al, 2016)." width="100%" />
<p class="caption">
Figure 4.1: Map of the ITS nuclear ribsomal region with primers (from Cheng et al, 2016).
</p>
</div>
<p>DNA extractions of leaf material were conducted at BSU using the Qiagen Plant Mini kit followed by PCR amplifications using plant specific primers <span class="citation">(for instance, the IT4p and ITS5p primers for the ITS region; see <a href="#ref-Cheng2016" role="doc-biblioref">Cheng et al., 2016</a>)</span>. Sequencing was outsourced to <a href="https://www.genewiz.com">GENEWIZ</a> and performed using Sanger sequencing technology. PCR amplicons/fragments were only sequenced using forward primers. As per provider, we are expecting high quality DNA sequencing read lengths of up to ca. 1000 bases. The company also mentions that a typical read would provide 800 bases with Phred score of 20.</p>
</div>
</div>
<div id="sciq" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Scientific questions associated to the mini-report</h2>
<p>For this assignment, students will investigate the following question:</p>
<blockquote>
<p>Q1: To what species of vanilla do the individuals presented in Figure <a href="#fig:van">4.2</a> belong to?</p>
</blockquote>
<p>To answer these questions, <strong>we will be focusing on four samples/individuals of vanilla (see Figure <a href="#fig:van">4.2</a>) and their associated ITS sequence data as well as ITS sequences deposited on GenBank</strong>. See the <a href="#sciprocess">scientific process section</a> below for more details.</p>
<div class="figure" style="text-align: center"><span id="fig:van"></span>
<img src="Images/Vanilla_samples.png" alt="Images of the four samples of vanilla collected in Mexico studied in this project." width="100%" />
<p class="caption">
Figure 4.2: Images of the four samples of vanilla collected in Mexico studied in this project.
</p>
</div>
<p>Throughout this project, we will be referring to these vanilla samples/individuals as follows (see Figure <a href="#fig:van">4.2</a>):</p>
<ul>
<li><em>PE25</em>, <em>PE44</em>, <em>PE49</em> and <em>PE50</em>.</li>
</ul>
</div>
<div id="sciprocess" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Scientific process</h2>
<p>The scientific process associated to this assignment can be summarized as follows (see Figure <a href="#fig:sciproc">4.3</a>):</p>
<div class="figure" style="text-align: center"><span id="fig:sciproc"></span>
<img src="Images/Scientific_process.png" alt="The scientific process." width="80%" />
<p class="caption">
Figure 4.3: The scientific process.
</p>
</div>
<ol style="list-style-type: decimal">
<li>To what species of vanilla do the individuals presented in Figure <a href="#fig:van">4.2</a> belong to?
<ul>
<li><strong>Hypothesis:</strong> Individuals belong to the same species (<em>V. planifolia</em>) and observed phenotypic variation is due to phenotypic plasticity (exhibiting responses to contrasting climates).</li>
</ul></li>
<li>To test the hypothesis we are applying the <strong>phylogenetic species concept</strong> <span class="citation">(<a href="#ref-wheeler1999phylogenetic" role="doc-biblioref">Wheeler, 1999</a>)</span> (mainly criterion of monophyly) based on DNA sequence data analyzed using a phylogenetic approach.</li>
<li>ITS barcode data are sequenced for individuals in Figure <a href="#fig:van">4.2</a> and compared to data from GenBank (= reference data).</li>
<li>Analyses are done in 4 steps:
<ul>
<li>PCR and Sanger sequencing of ITS barcodes for target individuals.</li>
<li>Assembly of reference ITS data available on GenBank using R.</li>
<li>BLAST and RAxML Maximum likelihood phylogenetic analysis (including bootstrap analysis).</li>
<li>Produce figures using R.</li>
</ul></li>
<li>Intersect evidence from analyses (especially phylogenetic clustering combined with node supports and taxonomy of ITS sequences from GenBank) to infer to what species do target individuals belong to. Use criterion of monophyly to help circumscribe species.</li>
<li>See section <a href="#report">4.6</a> for more details.</li>
</ol>
</div>
<div id="report" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Writing your report</h2>
<p>The instructor provides below information to complete the Mini-Report 3. All the material, data, code and references required to complete this report are provided here and were studies in class. In this context, students will have to focus on formatting their reports following guideline presented here as well as making sure that their <code>R</code> code is working and ready to be shared.</p>
<div id="apply-the-imrad-format-and-supply-r-code" class="section level3" number="4.6.1">
<h3><span class="header-section-number">4.6.1</span> Apply the IMRAD format and supply R code</h3>
<p>Your reports will be structured and organized following the <strong>IMMRAD</strong> format: <strong>I</strong>ntroduction, <strong>M</strong>aterial &amp; <strong>M</strong>ethods, <strong>R</strong>esults, and <strong>D</strong>iscussion. This format is widely used to report experimental research in many scientific disciplines. In addition, you will be complementing your reports with an <strong>Abstract</strong> (see below). Finally, since this research relies heavily on bioinformatics, students will complete their reports by supplying their commented <code>R</code> scripts.</p>
</div>
<div id="structure-of-the-mini-report-3" class="section level3" number="4.6.2">
<h3><span class="header-section-number">4.6.2</span> Structure of the Mini-Report 3</h3>
<p>The instructor provides guidance on content for each section of your individual reports:</p>
<ol style="list-style-type: decimal">
<li><strong>Title &amp; author</strong>: Please supply a first page containing the title and author information.</li>
<li><strong>Abstract</strong>: Provide a short text summarizing the i) objective(s), ii) methods (incl. sampling), iii) most important results and iv) major conclusions and significance of your research (maximum: 300 words).</li>
<li><strong>Introduction</strong>: Summarize challenges associated to biodiversity research and benefit of DNA barcoding approach in tackling those issues (e.g. by rapidly identifying species, this approach supports species descriptions, monitoring and development of conservation strategies). Then, move on to talk about vanilla, your model organism, and provide the scientific question and hypothesis investigated in this report (see section <a href="#sciq">4.4</a>). Finally, finish this section with an overview of the applied methodology used to answer your questions.</li>
<li><strong>Material &amp; Methods</strong>: This section should be subdivided into four subsections. Subsections a. and b. are associated to wet-lab analyses, whereas subsections c. and d. correspond to bioinformatic analyses. Please, make sure to cite references supporting your methods and report used bioinformatic tools (incl. references) together with their applied settings. Finally, you can also refer to your <code>R</code> code for further details. The subsections are as follows:
<ol style="list-style-type: lower-alpha">
<li>Sampling (see section <a href="#sciq">4.4</a>).</li>
<li>DNA extraction and sequencing (see section <a href="#vanbarcode">4.3.1</a>).</li>
<li>BLAST analysis.</li>
<li>Phylogenetic inference.</li>
</ol></li>
<li><strong>Results</strong>: Results should be presented in the same order as presented in the Material &amp; Methods section. Tables, Figures and code are reported in this section (by directly embedding them in the text and providing captions). To speed-up the writing of this report, you don’t have to report on results associated to Sampling (a.) and DNA extraction and sequencing (b.) subsections, but rather focus on the BLAST analysis (c.) and phylogenetic inference (d.) subsections.<br />
</li>
<li><strong>Discussion</strong>: In your discussion, we are expecting you to collate all the evidence presented in the study to answer your scientific questions. In this context, please subdivide this section into three subsections named as follows:
<ol style="list-style-type: lower-alpha">
<li>To what species of vanilla do the individuals studied here belong to?</li>
<li>How are those individuals related to each other?</li>
<li>Conclusions and Perspectives.</li>
</ol></li>
<li><strong>References</strong>: Add a references section containing full citations of references cited in the text. Students can pick their preferred citation format, but references need to be consistently edited in the same fashion.</li>
<li><strong>Appendix</strong>: Your <code>R</code> code has to be provided as an Appendix either by directly coping your code in the main document under this section or by providing the name and location of the <code>R</code> script in the Appendix section (e.g. by providing a text similar to this: “All the R code associated to this research can be downloaded at this path.”).</li>
</ol>
</div>
<div id="submission-deadline" class="section level3" number="4.6.3">
<h3><span class="header-section-number">4.6.3</span> Submission &amp; deadline</h3>
<p>Individual reports should be submitted in either Google Docs, Word or Rmarkdown formats (and if you decide to submit an additional file containing your R script, submit this file as an <code>.R</code> format). These files should be uploaded on the shared Google Drive in the <code>Mini_Report_3</code> folder. Name your report documents following this pattern: <code>Mini_Report_3_Surname</code>.</p>
<p><strong>Upload your reports on the shared Google drive by the 10th of March before 5 PM.</strong></p>
</div>
</div>
<div id="schedule" class="section level2" number="4.7">
<h2><span class="header-section-number">4.7</span> Schedule</h2>
<p>Analyses for this report are subdivided into three parts:</p>
<ul>
<li><strong>Part 1:</strong> Process and clean ITS DNA sequence electropherograms and infer species working hypothesis.</li>
<li><strong>Part 2:</strong> DNA sequence retrieval and prepare data for analyses.</li>
<li><strong>Part 3:</strong> DNA multiple sequence alignment and phylogenetic inference.</li>
</ul>
<p>Please see the <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/index.html#21_Timetable">Timetable</a> for more details on the schedule.</p>
<p><strong>Individual reports are due on Wed 10th of March at 5 PM</strong> (on Google Drive). Please see section <a href="#report">4.6</a> for more details.</p>
</div>
<div id="data-and-project-structure" class="section level2" number="4.8">
<h2><span class="header-section-number">4.8</span> Data and project structure</h2>
<p>The data are deposited on the shared <a href="https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing">Google Drive</a> in a folder entitled <code>DNA_barcoding</code>. All students enrolled in this class have been granted access to this folder. The folder is subdivided into four sub-folders (see Figure <a href="#fig:datastr">4.4</a>) and contains a master spreadsheet <code>Vanilla_samples_records.xlsx</code> at its root:</p>
<ul>
<li><code>01_Field_images</code>: <code>jpeg</code> images of samples.</li>
<li><code>02_Raw_ITS_data_ab1</code>: <code>.ab1</code> sequence electropherograms of ITS sequences.</li>
<li><code>03_Processed_ITS_data_FASTA</code>: Folder where cleaned ITS sequences will be saved in <code>FASTA</code> format.</li>
<li><code>04_Data_analyses</code>: Folder where we will be saving outputs of analyses conducted in this project.</li>
<li><code>Vanilla_samples_records.xlsx</code> contains meta-data information about the samples. We will also update this file with information on species hypotheses.</li>
<li><code>PART2_Vanilla.R</code> is an R script containing code for the part 2 of this mini-report.</li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:datastr"></span>
<img src="Images/DNA_barcoding_str.png" alt="Screenshot of the DNA_barcoding folder showing data structure for this project." width="90%" />
<p class="caption">
Figure 4.4: Screenshot of the DNA_barcoding folder showing data structure for this project.
</p>
</div>
</div>
<div id="part-1-process-and-clean-its-dna-sequence-electropherograms-and-infer-species-working-hypothesis" class="section level2" number="4.9">
<h2><span class="header-section-number">4.9</span> Part 1: Process and clean ITS DNA sequence electropherograms and infer species working hypothesis</h2>
<div id="bioinformatic-tools" class="section level3" number="4.9.1">
<h3><span class="header-section-number">4.9.1</span> Bioinformatic tools</h3>
<p>The bioinformatic tools used in part 1 are as follows:</p>
<ol style="list-style-type: decimal">
<li><code>FinchTV</code>: A popular desktop software developed by Geospiza, Inc. for viewing trace data from Sanger DNA Sequencing. <code>FinchTV</code> is freely available and operates on Windows and Mac platforms. Start by downloading and installing the software on your computers at this URL: <a href="https://digitalworldbiology.com/FinchTV" class="uri">https://digitalworldbiology.com/FinchTV</a></li>
<li><code>BLAST</code>: The Basic Local Alignment Search Tool <span class="citation">(BLAST, <a href="#ref-ALTSCHUL1990403" role="doc-biblioref">Altschul et al., 1990</a>)</span> will be applied onto cleaned DNA sequences to:
<ul>
<li>Confirm that the DNA sequences correspond to the correct DNA region (here ITS region).</li>
<li>Validate that DNA sequences belong to the right taxon (here belonging to the genus <em>Vanilla</em>) and are therefore not contaminated.</li>
<li>Provide species working hypotheses using the distance-tree approach implemented on the online version of BLAST.</li>
</ul></li>
</ol>
<p>Although <code>BLAST</code> can be run locally, we will be using the web portal available here: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="uri">https://blast.ncbi.nlm.nih.gov/Blast.cgi</a></p>
</div>
<div id="overview-of-the-dna-sequence-data-cleaning-procedure" class="section level3" number="4.9.2">
<h3><span class="header-section-number">4.9.2</span> Overview of the DNA sequence data cleaning procedure</h3>
<p>When evaluating <code>.ab1</code> files, you should first see the electropherogram and come to a conclusion whether your data can be considered of good quality or not.</p>
<p>Good quality sequencing data/positions are characterized by:</p>
<ul>
<li>Well-defined peak resolution (bad resolution of the first 10-25 bases is acceptable; see Figure <a href="#fig:finchtv">4.5</a>). The Phred score is displayed at top of the window (see bars on Figure <a href="#fig:finchtv">4.5</a>).</li>
<li>Uniform peak spacing.</li>
<li>High signal-to-noise ratios.</li>
</ul>
<p>Bad quality sequencing data/positions are characterized by:</p>
<ul>
<li>Presence of “N”s in the sequence. Indeed, when the base-calling software is unable to accurately identify a nucleotide, it will score it as “N” (meaning that it can be any base; see Figure <a href="#fig:finchtv">4.5</a>). In this tutorial, we will open files individually and search for peaks scored as “N”s. If we can confidently correct those peaks/positions (to either "A/T/C/G or any other IUPAC code; see below) then we will edit the sequence accordingly. To keep track of changes, it is standard procedure to identify edited peaks/positions by using lower cases letters (instead of capital letters as set by default). See below for more details on cases of base polymorphism.</li>
</ul>
<p>We will be discussing this topic further during our Zoom meeting.</p>
<div class="figure" style="text-align: center"><span id="fig:finchtv"></span>
<img src="Images/FinchTV_snapshot.png" alt="Screenshot of FinchTV app." width="100%" />
<p class="caption">
Figure 4.5: Screenshot of FinchTV app.
</p>
</div>
</div>
<div id="iupac-codes" class="section level3" number="4.9.3">
<h3><span class="header-section-number">4.9.3</span> IUPAC codes</h3>
<p>Nuclear DNA regions such as ITS could show evidence of recombination. This means that there could be polymorphism at a specific base <span class="citation">(also know as single-nucleotide polymorphism or SNP; see <a href="#ref-Poplin2018" role="doc-biblioref">Poplin et al., 2018</a> for bioinformatics technics to identify SNPs based on NGS data)</span>. The signature of recombination in an electropherogram would be recognized by the occurrence of “peak under peak” (Figure <a href="#fig:peakunderpeak">4.6</a>). The International Union of Pure and Applied Chemistry (IUPAC) has defined a standard representation of DNA bases by single characters that specify either a single base (e.g. G for guanine, A for adenine) or a set of bases (e.g. R for either G or A). UCSC uses these single character codes to represent multiple observed alleles of single-base polymorphisms (Table <a href="#tab:IUPAC">4.1</a>).</p>
<div class="figure" style="text-align: center"><span id="fig:peakunderpeak"></span>
<img src="Images/peak_under_peak.png" alt="Screenshot of DNA sequence electropherogram showing signature of peak under peak suggesting recombination." width="20%" />
<p class="caption">
Figure 4.6: Screenshot of DNA sequence electropherogram showing signature of peak under peak suggesting recombination.
</p>
</div>
<table>
<caption><span id="tab:IUPAC">Table 4.1: </span>IUPAC codes used to score base polymorphism. This is especially useful when analyzing nuclear DNA regions.</caption>
<thead>
<tr class="header">
<th align="left">IUPAC nucleotide code</th>
<th align="left">Base</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">Adenine</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">Cytosine</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">Guanine</td>
</tr>
<tr class="even">
<td align="left">T (or U)</td>
<td align="left">Thymine (or Uracil)</td>
</tr>
<tr class="odd">
<td align="left">R</td>
<td align="left">A or G</td>
</tr>
<tr class="even">
<td align="left">Y</td>
<td align="left">C or T</td>
</tr>
<tr class="odd">
<td align="left">S</td>
<td align="left">G or C</td>
</tr>
<tr class="even">
<td align="left">W</td>
<td align="left">A or T</td>
</tr>
<tr class="odd">
<td align="left">K</td>
<td align="left">G or T</td>
</tr>
<tr class="even">
<td align="left">M</td>
<td align="left">A or C</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="left">C or G or T</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">A or G or T</td>
</tr>
<tr class="odd">
<td align="left">H</td>
<td align="left">A or C or T</td>
</tr>
<tr class="even">
<td align="left">V</td>
<td align="left">A or C or G</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">any base</td>
</tr>
<tr class="even">
<td align="left">. or -</td>
<td align="left">gap</td>
</tr>
</tbody>
</table>
</div>
<div id="step-by-step-protocol" class="section level3" number="4.9.4">
<h3><span class="header-section-number">4.9.4</span> Step-by-step protocol</h3>
<p>Here, we will be using <code>its25-ITSp4.ab1</code> as an example for the analysis. This file is located in <code>DNA_barcoding/02_Raw_ITS_data_ab1</code>.</p>
<ol style="list-style-type: decimal">
<li>Copy the <code>DNA_barcoding/</code> folder onto your computers.</li>
<li>Launch <code>FinchTV</code>.</li>
<li>Open <code>.ab1</code> file (one at a time) using the <code>Open</code> tab or the folder icon.</li>
<li>Make sure that the <code>Base Position Numbers</code>, <code>Base Calls</code> and <code>Quality Values</code> settings are ticked using the <code>View</code> tab (Figure <a href="#fig:finchtv">4.5</a>).</li>
<li>Trim the first 10-20 bp of the sequence. This is done by selecting bases using the <code>Shift</code> command and then pressing <code>Delete</code> (Figure <a href="#fig:finchtvtrim">4.7</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:finchtvtrim"></span>
<img src="Images/Trimming_seq.png" alt="Screenshot of FinchTV app showing trimming procedure." width="90%" />
<p class="caption">
Figure 4.7: Screenshot of FinchTV app showing trimming procedure.
</p>
</div>
<ol start="6" style="list-style-type: decimal">
<li>Scroll through the sequence and edit “N” peaks/positions by replacing those with lower cases “a/t/c/g.” If you are unable to call the nucleotide, do not edit the position. In the example, position 10 was edited to “g” (Figure <a href="#fig:finchtedit">4.8</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:finchtedit"></span>
<img src="Images/Edit_sequence.png" alt="Screenshot of FinchTV app showing editing procedure." width="90%" />
<p class="caption">
Figure 4.8: Screenshot of FinchTV app showing editing procedure.
</p>
</div>
<ol start="7" style="list-style-type: decimal">
<li>Peaks at the end of the sequences will become more rounded and harder to call. It is therefore standard procedure to trim the last 10-30 bp/positions. Please trim these bp following the procedure explained above. In the example, the quality drops around position 670 (see Figure <a href="#fig:finchtvtrimend">4.9</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:finchtvtrimend"></span>
<img src="Images/Trim_end_sequence.png" alt="Screenshot of FinchTV app showing trimming procedure at the end of the sequence." width="90%" />
<p class="caption">
Figure 4.9: Screenshot of FinchTV app showing trimming procedure at the end of the sequence.
</p>
</div>
<ol start="8" style="list-style-type: decimal">
<li>Save cleaned sequence in <code>FASTA</code> format in <code>03_Processed_ITS_data_FASTA</code>. Exporting the cleaned sequence is done by pressing <code>File -&gt; Export -&gt; DNA Sequence: FASTA</code>. Please do not rename file, leave it as proposed by <code>FinchTV</code>. The file is saved in interleaved FASTA format as shown in Figure <a href="#fig:fasta">4.10</a>.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:fasta"></span>
<img src="Images/FASTA.png" alt="Screenshot of cleaned FASTA sequence as outputted by FinchTV." width="80%" />
<p class="caption">
Figure 4.10: Screenshot of cleaned FASTA sequence as outputted by FinchTV.
</p>
</div>
<ol start="9" style="list-style-type: decimal">
<li>Open fasta file in a text editor and copy DNA sequence (including header starting with <code>&gt;</code>; see Figure <a href="#fig:blast1">4.11</a>).</li>
<li>Go on the BLAST website by clicking <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&amp;PAGE_TYPE=BlastSearch&amp;LINK_LOC=blasthome">here</a> and copy your DNA sequence as shown in Figure <a href="#fig:blast1">4.11</a>. Click on the <code>BLAST</code> button to start your query.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:blast1"></span>
<img src="Images/BLAST1.png" alt="Screenshot of BLAST form where you copy content of its25-ITSp4.seq" width="90%" />
<p class="caption">
Figure 4.11: Screenshot of BLAST form where you copy content of its25-ITSp4.seq
</p>
</div>
<ol start="11" style="list-style-type: decimal">
<li>Inspect the BLAST output to make sure that the top hits are associated to <em>Vanilla</em> and refer to the right DNA region (Figure <a href="#fig:blastres">4.12</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:blastres"></span>
<img src="Images/BLAST.png" alt="Screenshot of BLAST search based on its25-ITSp4. Note that top hits are ITS sequences of Vanilla species." width="90%" />
<p class="caption">
Figure 4.12: Screenshot of BLAST search based on its25-ITSp4. Note that top hits are ITS sequences of Vanilla species.
</p>
</div>
<ol start="12" style="list-style-type: decimal">
<li><p>Click on the <code>Distance tree of results</code> link to perform phylogenetic distance analysis showing position of your sequence compared to sequences available on GenBank (see Figure <a href="#fig:blastres">4.12</a>). This will open a new window.</p></li>
<li><p>Expand tree to locate your DNA sequence by following procedure in Figure <a href="#fig:expandtree">4.13</a>.</p></li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:expandtree"></span>
<img src="Images/ExpandTree.png" alt="Procedure to expand tree to show position of your DNA sequence in phylogeny." width="90%" />
<p class="caption">
Figure 4.13: Procedure to expand tree to show position of your DNA sequence in phylogeny.
</p>
</div>
<ol start="14" style="list-style-type: decimal">
<li>Use the Zoom toggle to locate your DNA sequence on the tree (see Figure <a href="#fig:DNAtree">4.14</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:DNAtree"></span>
<img src="Images/DNAtree.png" alt="Position of your DNA sequence on tree." width="90%" />
<p class="caption">
Figure 4.14: Position of your DNA sequence on tree.
</p>
</div>
<ol start="15" style="list-style-type: decimal">
<li><p>Open <code>Vanilla_samples_records.xlsx</code> and update <code>Species_BLAST</code> column with a species name (your first working hypothesis). Don’t forget to save the file.</p></li>
<li><p>Repeat this procedure until you analyzed all the <code>ab1</code> files.</p></li>
</ol>
</div>
</div>
<div id="part-2-dna-sequence-retrieval-and-prepare-data-for-analyses" class="section level2" number="4.10">
<h2><span class="header-section-number">4.10</span> Part 2: DNA sequence retrieval and prepare data for analyses</h2>
<div id="bioinformatic-tools-1" class="section level3" number="4.10.1">
<h3><span class="header-section-number">4.10.1</span> Bioinformatic tools</h3>
<p>To execute Part 2, you need to install the following software and R packages on your computer:</p>
<ul>
<li><code>R</code>: <a href="https://www.r-project.org" class="uri">https://www.r-project.org</a></li>
<li><code>RStudio</code>: <a href="https://rstudio.com" class="uri">https://rstudio.com</a>
<ul>
<li>An overview of RStudio environment is available <a href="Images/RStudio_overview.pdf">here</a>.</li>
</ul></li>
<li><code>R</code> package:
<ul>
<li><em>rentrez</em> <span class="citation">(<a href="#ref-rentrez" role="doc-biblioref">Winter, 2017</a>)</span>: <a href="https://cran.r-project.org/web/packages/rentrez/index.html" class="uri">https://cran.r-project.org/web/packages/rentrez/index.html</a></li>
</ul></li>
</ul>
<p>If you don’t know how to install an R package, don’t worry, this topic is covered <a href="#query">here</a>.</p>
</div>
<div id="analytical-workflow" class="section level3" number="4.10.2">
<h3><span class="header-section-number">4.10.2</span> Analytical workflow</h3>
<p>To infer the ML phylogenetic analysis, the following workflow will be executed: implemented:</p>
<ol style="list-style-type: decimal">
<li>Expand sampling for phylogenetic analyses by downloading ITS DNA sequences of <em>Vanilla</em> species deposited on GenBank. This will be done by using the <em>rentrez</em> <code>R</code> package. A table with species taxonomy associated to GenBank accessions will also be built. We will also clean the data and save outputs in <code>FASTA</code> and <code>csv</code> formats.</li>
<li>Format and merge individual <code>.seq</code> files from Part 1 into a <code>FASTA</code> object/file. This will be done by creating a list of all <code>.seq</code> files, opening them individually and merging them into one object using a <code>for</code> loop.</li>
<li>Merge <code>FASTA</code> objects/files from steps 1 and 2 to perform a multiple sequence alignment (msa).</li>
</ol>
<p>Protocols associated to each step within the workflow are detailed below.</p>
</div>
<div id="download-and-write-dna-sequences-and-retrieve-taxonomy" class="section level3" number="4.10.3">
<h3><span class="header-section-number">4.10.3</span> Download and write DNA sequences and retrieve taxonomy</h3>
<p>Here, we are demonstrating an approach to search GenBank for ITS sequences of <em>Vanilla</em> species using <em>rentrez</em> <span class="citation">(<a href="#ref-rentrez" role="doc-biblioref">Winter, 2017</a>)</span>. Students are tasked to study/examine and execute the code and finally adapt it to search for ITS sequences. The code is subdivided into the following steps:</p>
<ol style="list-style-type: decimal">
<li>Build a GenBank query. But prior to that we will load R package and set our working directory.</li>
<li>Retrieve DNA sequences matching query deposited on GenBank.</li>
<li>Fetch meta-data associated to sequences.</li>
<li>Tidy dataset based on meta-data.</li>
<li>Write results in <code>FASTA</code> and <code>csv</code> formats.</li>
</ol>
<p>Before delving into the code, start by creating an <code>.R</code> file in <code>RStudio</code> and save it in the same working directory as where you want to save the output files (e.g. <code>DNA_barcoding/04_Data_analyses</code>).</p>
</div>
<div id="query" class="section level3" number="4.10.4">
<h3><span class="header-section-number">4.10.4</span> Build a GenBank query</h3>
<pre class="r"><code>###~~~
#Check if package is installed if not then install it
###~~~
if(&quot;rentrez&quot; %in% rownames(installed.packages()) == FALSE){
  print(&quot;Install rentrez&quot;)
  install.packages(&quot;rentrez&quot;)
  }else{
    print(&quot;rentrez is installed!&quot;)
}

###~~~
#Load package
###~~~
library(rentrez)

###~~~
#Set working directory
###~~~
#Set working directory to path leading to DNA_barcoding folder
# WARNING: This path as to be adapted to match your computer
setwd(&quot;~/Documents/Class_Genomics&amp;Bioinfo_Spring/DNA_barcoding/&quot;)

#Check that working directory is set correctly
getwd()

###~~~
#Build a query
###~~~

#Taxon
sp &lt;- &quot;Vanilla&quot;

#DNA region: here rbcL
DNA &lt;- &quot;internal transcribed spacer&quot;

#Organism
org &lt;- &quot;Plants&quot;

#Build query: sp AND DNA region
query &lt;- paste0(sp,&quot;[All Fields] AND &quot;, DNA,&quot;[All Fields]&quot;, org, &quot;[filter]&quot;)</code></pre>
</div>
<div id="retrieve-genbank-dna-sequences-matching-query" class="section level3" number="4.10.5">
<h3><span class="header-section-number">4.10.5</span> Retrieve GenBank DNA sequences matching query</h3>
<pre class="r"><code>###~~~
#Retrieve DNA accessions in GenBank
###~~~
GBresults &lt;- rentrez::entrez_search(db=&quot;nuccore&quot;,term = query, retmax=50000)</code></pre>
</div>
<div id="fetch-meta-data-associated-to-sequences" class="section level3" number="4.10.6">
<h3><span class="header-section-number">4.10.6</span> Fetch meta-data associated to sequences</h3>
<p>Even when you execute a query directly on the GenBank portal, there will always be sequences that do neither match your target taxon (here <em>Vanilla</em>) nor your target DNA region. In this context, it is paramount to retrieve meta-data associated to the DNA accessions (stored in <code>GBresults</code>) in order to clean-up your dataset prior to analyses (see next step).</p>
<p>For each DNA sequence the following meta-data are gathered using functions implemented in <em>rentrez</em>:</p>
<ul>
<li>GenBank DNA accession number.</li>
<li>Taxonomy.</li>
<li>Sequence definition line as displayed on GenBank.</li>
<li>Sequence length (in bp).</li>
<li>DNA sequence.</li>
</ul>
<p>Please see the <code>R</code> code below for more details on the procedure to retrieve the meta-data.</p>
<pre class="r"><code>###~~~
#Fetch meta-data associated to sequences
###~~~

#Use loop to automatically retrieve species, 
# seq definition line, seq length and DNA sequence associated 
# to each DNA accession

#Create empty matrix to be populated
OUT &lt;- matrix(ncol=5, nrow=length(GBresults$ids))
colnames(OUT) &lt;- c(&quot;GenBankID&quot;, &quot;Species&quot;, &quot;Definition&quot;, &quot;Seq_length&quot;, &quot;Sequence&quot;)

#Add GenBank ID
OUT[,1] &lt;- GBresults$ids
print(&quot;Processing sequences: fetching meta-data&quot;)
#Set a progress bar
pb &lt;- txtProgressBar(min = 0, max = length(GBresults$ids), style = 3)
for(i in 1:length(GBresults$ids)){
  #print(paste(i, &quot;in&quot;, length(GBresults$ids), sep= &#39; &#39;))
  #Download sequence
  seq &lt;- entrez_fetch(db=&#39;nuccore&#39;, id=GBresults$ids[i], rettype=&#39;fasta&#39;, retmode = &quot;text&quot;)
  
  #Extract definition line
  OUT[i,3] &lt;- strsplit(seq, split=&quot;\n&quot;)[[1]][1]
  
  #Infer seq length
  nbp &lt;- strsplit(seq, split=&quot;\n&quot;)
  OUT[i,4] &lt;- as.numeric(length(strsplit(paste(nbp[[1]][2:length(nbp[[1]])], 
                                               collapse=&#39;&#39;),&quot;&quot;)[[1]]))
  
  #Extract sequence
  OUT[i,5] &lt;- as.vector(paste(nbp[[1]][2:length(nbp[[1]])], collapse=&#39;&#39;)[1])
  
  #Fetch taxon ID associated to GenBank accessions
  taxID &lt;- entrez_link(dbfrom=&#39;nuccore&#39;, id=GBresults$ids[i], db=&#39;taxonomy&#39;)

  #Extract taxonomy: genus and species epithet
  tmp &lt;- strsplit(
    strsplit(entrez_fetch(db=&#39;taxonomy&#39;, id=taxID$links, rettype = &quot;native&quot;), 
             split = &#39;\n&#39;)[[1]][1]
    , split = &#39; &#39;)
  
  OUT[i,2] &lt;- paste(tmp[[1]][2:length(tmp[[1]])], collapse = &#39; &#39;)
  
  # update progress bar
  setTxtProgressBar(pb, i)
}
close(pb)

SEQ &lt;- as.data.frame(OUT)

###~~~
#Write raw data in 04_Data_analyses
###~~~
FileIDRawcsv &lt;- paste(sp, DNA, gsub(&quot;-&quot;, &quot;_&quot;, Sys.Date()), &quot;Raw_GenBank.csv&quot;, sep=&#39;_&#39;)
#Write FASTA file
write.table(SEQ, 
            paste(&quot;04_Data_analyses/CSV/&quot;, 
                  FileIDRawcsv, sep=&#39;&#39;), row.names = F, col.names=T, quote = T)</code></pre>
<p>Let’s have a look at the data downloaded from GenBank:</p>
<pre><code>## [1] &quot;GenBank query retrieved 185 DNA sequences.&quot;</code></pre>
<pre><code>##    GenBankID            Species
## 1 1708599397 Vanilla planifolia
##                                                                                                                                                                                                                                                                                                      Definition
## 1 &gt;MN221421.1 Vanilla planifolia voucher PDBKTMA2013-1860 external transcribed spacer, partial sequence; small subunit ribosomal RNA gene, internal transcribed spacer 1, 5.8S ribosomal RNA gene, and internal transcribed spacer 2, complete sequence; and large subunit ribosomal RNA gene, partial sequence
##   Seq_length
## 1       9463
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Sequence
## 1 CGCGGCTGAGGGCAACGCCACGCCGCGCGGGGCGTGCGGTCGTGGCCAATGATTAGGCGCGCGCGGCGCGCCGTGCAGGGCACAACGTTGCAACCCCGGGCGCGCGGTCGGGGGCACCGCGCCCCTGCGCACCGCCGCGGGCACCGCGCACCTGCGTGCGGTCCTGTGCACCGCGCAGGTTGGTTATGTTGGCTGATGAGGGGACAAAAAAGTGCAACTTTTTTCGGGATTTTTAGCGCTGCGGGCTGCTTCTGCACGGATGCGGCGACAACCAATTGGTAGTCTTGCCTCGGCGCATGATGGCATCGTCGCAAAAAAAAAAATCCGAGTTGCGGGCGTGTGCATCGCGCGCCTGACCCGGGCGTGGGCACCGCGCTCCGGCGTGGGGGCATGTTCATCGAGCTCCTGCTTGCGTGCTTTCGAACCGCGCTCCTGTTGGCTTGGCTATCCGCGCGTTCTCGGCGACTAGGTGGGCCGGCATCGTTGCGCTGCAGAACAGAGCAACTTTCGTCGGCATCTTTAGCCCTGTGTTCTCGCGAGGGGACCCAAAAGTGTAACTTTTTTTGGGATTTTTAGCGCCGCGGGCTGCTTCTGCTTGGATATGGTGACAACCCATTGGTAGTCTGCCTCGTCCCAAGATGCGATTGTCGCAATACACGATCGGAAATGTGGKCATGTTCATAACGCACCTGCGGGCGGGCATGTGCACGGCGCGCCGGCSCGTGGGCATTGGCACCGCGCTCCCGCGGGCCGTGGAGTGCACCGCCCCCCGCGCGCGGCGCGCGTCCGTGGGCACGGCTCGGGTGCGTGCGGCGCGACGTGAATGGCACCGCACCGCCGCGCGGATCGTCGATGGCACCGCGCCGCCGCGAGGGTCGCGTGGTCGTGTGCACCTCGCAGGCGGGCGCGGCGCGCCGTGGATTGAACCGCACCGTCGCCCGGGTCGCGCGGCGGTGTGCACCTCGGAGGCTCGCGCCGCGCTTGGAGCGCGATCGAGGGCGCCGCGCCGCGGCGCGCGGTCGTGCGACCCGCGTAAGCGCGCGAGGCGCGTCGGCGCAGGCACCGCGCCCCTACGCGCCCGGCGCGGCCGCGGGCACCGCGCGTCTGGGTGCTGGCCTTTGCGCCTCGCTCCTGCGCACGGGCATGTGCACCGCGCCCCAGCGCGCGGGCATCTCCCCGCGCCCCAGTGCGTGTGCATGTGGACTGCTCTCCCGCGCGCGGCAGCGTGCAACGCGCTCCTGCTGTTTTTTTTGTGGAGTGCGATGTGGTGCACTGTCTGAACTGCTGCCCTCTTCCATCTGCTTTGGGCTTTGTTTCTTAGTGGTGGTTCGTTTTGTCAATTTGGTTGGAGAAATGTGCAAACTCTTTTCGATGTTATGTCATATCTCCGGAGGGCCAATGTAGGGATGAGGTGTTGCTCAAATCGTGTCGTTTAAGCGGTGCAATTTTTGCTTGGCTCTTTGTCGGTTGGGCGTGATTTTCGCTTTGCATGTGGTGCGAGGTATCGCTTGTCGTGCTATGCGGGTGTTAGCCAATGCCTTCATCCCTAATTAGACCCTTTGCGCTTCCTGTTTTAAACATCGTTGAGGCAATTTGGGCTAATGGAGTTGGATTGGGGAGATGCCGTTTTTGCACAAGAGATGTTTGTTGTGGCGCTACTCTGTATGGCTTGGTCACCTCGTGGTGTGGTCTGTCAATACATGAGTAGGGCTCTGTCACTTCGTTGAACCTTCATACGGAGGAAGTCCGCATGTTGTTTTGCCTTTCTATCGGATGGGTAGGTTCCGTTCTATTATTTTTTATGGCGCCGCACGACTACGGTGTTGGAATGGTGGCGTTACGCGATGCCGTCGAACGCATGATCGTACATGCTCCCCTGTTGGCAAGGGCTCGGAAATCACACGTTTCGGTCATATCGTTCTCCAACAAGGAGGGTTGTCCCTGAGGAATGTTATTTGTCGAGAGGCAATGTTTCGGAACTTGGGCGATGGTATCCCAACCTGGGAGCATGCGCTCTTTCCTTTGTGGAGTATCGCCAGACACGACGAACATGTCAAATTACGACATGGGGTTTTGCTTGGCGTTTCATCATTGTATTGCTACCTGGTTGATCCTGCCAGTAGTCATATGCTTGTCTCAAAGATTAAGCCATGCATGTGTAAGTATGAACAATTTCAGACTGTGAAACTGCGAATGGCTCATTAAATCAGTTATAGTTTGTTTGATGGTACTTGCTACTCGGATAACCGTAGTAATTCTAGAGCTAATACGTGCACCAAACCCCGACTTTTGGAAGGGATGCATTTATTAGGTAAAAGGTCGACGCGGGCTTTTGCCCGGCTCCTTGACGATTCATGATAACTTGTCGGATCGCACGGCCTTCGTGCCGGCGATGCATCATTCGAATTTCTGCCCTATCAACTTTCGATGGTAGGATAGGGGCCTACCATGGTGGTGACGGGTGACGGAGAATTAGGGTTCGATTCCGGAGAGGGAGCCTGAGAGACGGCTACCACATCCAAGGAAGGCAGCAGGCGCGCAAATTACCCAATCCTGACACGGGGAGGTAGTGACAATAAATAACAATACCGGGCTCCACGAGTCTGGTAATTGGAATGAGTACAATCTAAATCCCTTAACGAGGATCCATTGGAGGGCAAGTCTGGTGCCAGCAGCCGCGGTAATTCCAGCTCCAATAGCGTATATTTAAGTTGTTGCAGTTAAAAAGCTCGTAGTTGGACCTTGGTTTGGGTCGGTCGGTCCGCCTTTTGGTGTGCACCGCCCGCCCTGATCCTTTTGTCGACGATGCGGTCTGGCCTTAGCTGGCCGGGTCGTGCCCTCGGCGTTGTTACTTTGAAGAAATTAGAGTGCTCAAAGCAAGCCCACGCTCTGGATACATTAGCATGGGATAACATCACAGGATTTCGATCCTATTGTGTTGGCCTTCGGGATCGGAGTAATGATTAAGAGGGACAGTCGTGGGCATTCGTATTTCATAGTCAGAGGTGAAATTCTTGGATTTATGAAAGACGAACCACTGCGAAAGCATTTGCCAAGGATGTTTTCATTAATCAAGAACGAAAGTTGGGGGCTCGAAGACGATCAGATACCGTCCTAGTCTCAACCATAAACGATGCCGACCAGGGATTGGCGGATGTTGCTCTTTGGACTCCGTCAGCACCTTATGAGAAATCAAAGTCTTTGGGTTCCGGGGGGAGTATGGTCGCAAGGCTGAAACTTAAAGGAATTGACGGAAGGGCACCACCAGGAGTGGAGCCTGCGGCTTAATTTGACTCAACACGGGAAAGCTTACCAGGTCCAGACATAGCAAGGATTGACAGATTGAGAGCTCTTTCTTGATTCTATGGGTGGTGGTGCATGGCCGTTCTTAGTTGGTGGAGCGATTTGTCTGGTTAATTCCGTTAACGAACGAGACCTCAGCTTGCTAACTAGCTATGCGGGGTGCAAGCCCTGTGGCCAGCTTCTTAGAGGGACTATGGCCGCTTAGGCCATGGAAGTTTGAGGCAATAACAGGTCTGTGATGCCCTTAGATGTTCTGGGCCGCACGCGCGCTACACTGATGTATTCAACGAGTCCATTGCCTTGGTCGAAAGGCCTGGGTAATCTTATGAAAATTTCATCGTGATGGGGATAGATCATTGCAATTGTTGGTCTTCAACGAGGAATTCCTAGTAAGCGCGAGTCATCAGCTCGCGTTGACTACGTCCCTGCCCTTTGTACACACCGCCCGTCGCTCCTACCGATTGAATGGTCCGGTGAAGTGTTCGGATCGCTGCGATGCGGGCGGTTTGCCGCGTGCGACTCGGCGAGAAGTCCACTGAACCTTATCATTTAGAGGAAGGAGAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGACGAGAGGTTCGAACGATGGAACGATCTGTCCAACGCGTGGGAGTGCGACGGTTGTTCGATGTCGCGTTCTTTCGTAGCGCGTGCTCTCGCGATCACGCGGAGCTCGACGTTATGGGGGATAAACAAAAGCTTATGGGCGTGGTCAGGCGCCAAGGGAGAGCAAATGTTAGGCTGGCAAACGAGTATGCCGTGCGTCGTCAGGTCCATTGAGTCCTGGCAATCGAACGACTCTCGACAACGGATATCTTGGCTCTCGCATCGATGAAGAACGCAGCGAAATGCGATACGTGTTGTGAATTGTAGAATCCCGTGAACCATCCATTTTTTGAACGCAAGTTGCGCCCGAGGATGCAAGCCGAGGGCACGCCTGCATGGGCGTAATGCGACCCGTCGCTCCCTGCGGAAGCCTGGAATCTTTCGGTTTGGTTCCTCGGCCCCGTCGCGGAGGCGATTGATGGCACCCCGTGCGATGGCACGGCGTGTTGAAGCTTGGGGCGACGGTCGACTTCGACACGATACGAGGTGGACGCCACTGGCTGTCGTGGTGTTGGCCAGCAAGAATCGATGTTGTTGTGCGACGAGCAGATGCCCCGCATAGATCCTGCTCCGTCCTCCACGGTGTGGAATCGTGACCCCATGTTAGGTGAGGCTACCCGCCGAGTTTAAGCATATAAATAAGCGGAGGAGAAGGAACTTACAAGGATTCCCTTAGTAACGGCGAGCGAAACGGGACCAGCCCAGTTTGGAAATCGGGCAGCCTGAAGCCTGAATTGTAGTCTGGAGAGGCGTCCTCAGCGACGGATCGGGATCAAGTCCCCTGGAAAGGGGCGCCGGGGAGGGTGAGAGCCCCGTTCGGCCCGTACCCTGCTGCTACACGAGGCGCCGTCAACGAGTCGGGTTGTTTGGGAATGCAGCCCAAATTGGGTGGTAAATTCCGTCCAAGGCTAAATAATTGCGAGAGACCGATAGCGAACAAGTACCGCGAGGGAAAGATGAAAAGGACTTTGAAAAGAGAGTCAAAGAGTGCTTGAAATTGTCGGGAGGGAAGCAGATGGGGGCCGGCGGTGCGCCTCGGCTGGATGCAGAACGTCGAATGACGGTTTGCTGCACGGCTCGAGGAGCGGACCGTCTCGGGCCATCGCGGCGACCGGAGCCCGGGCGCACGTCGCTCGTGGAGAAATCGTCGGCGTGGCCGATCGCAATGCCCGCGCCATCGAGGCGTGCCACGCGGCACCGCGTGCATTGGTGATGGCCAGTGGGCTCCCCATCTGACCCGTCTTGAAACACGGACCAAGGAGTCTGACATGCTTGCGAGTCGACGGGTGGGCAAGCCCGGAAGGCGCAAGGAAGCTGATTGGTGGGATCCCCGTTTGAGGGGTGAGTCATCGACCGACCCAGATCTTTTGTGAAGGGTTCGAGTGAGAGCATGCCTGTCGGGACCCGAAAGATGGTGAACTATGCCTGAGCGGGGCGAAGCCAGAGGAAACTCTGGTGGAGGCCCGCAGCGATACTGACGTGCAAATCGTTCGTCTGACTTGGGTATAGGGGCGAAAGACTAATCGAACCATCTAGTAGCTGGTTCCCTTCGAAGTTTCCCTCAGGATAGCTGGAGCCACAGTCGGAGTTCTATCGGGTAAAGCCAATGATTAGAGGCATCGGGGGCACAATGCCCTCGACCTATTCTCAAACTTTAAATAGGTAGGAGGGCTCGGCTGCTTTGATGAGTTGAGCCAAGGAATCCAGGCTCCAAGTGGGCCATTTTTGGTAAGCAGAACTGGCGATGCGGGATGAACCGAAAGCCGGGTTACGGTGTCCAACTGCGCGCTAACCTAGAGCCCACAAAGGGTGTTGGTCGATTAAGACAGCAGGACGGTGGTCATGGAAGTTGAAATCCGCTAAGGAGTGTGTAACAACTCACCTGCCGAATCAACTAGCCCCGAAAATGGATGGCGCTTAAGCGTGCGACCCACACCTGGCCGTCGGTGCAATGCAAGGCCCCGACGAGTAGGAGGGTGCAACGGTCGCTGCAAAACGTGGGGCGTGAGCCCGCGTGGAGCGTCTGTTGGTGCAGATCTTGGTGGTAGTAGCAAATATTCAAATGAGAACTTTGAAGGCCGAAGGGGGGAAAGGTTCCATGTGAACGGCACTTGCACATGGGTTAGCCGATCCTAAGAGACGGGGGAAACCCGTCGGATAGTGCCATTCGGCACGAGCTTCGAAAGGGAGTCGAGTTAAAATTCTCGAGCCGGGATGTGGCGGTCGATGGCAACGTCAAGTGGTCCGGAGACGTCGGTGGGGGCCTCGGGAAGAGTTATCTTTTCTGCTTAACAGCCCATCGGCCCTGGAAACGGCTCAGCCGGAGGTAGGGCCAAGCGGTTGGAAGAGCACCGCATGTTGAGCGGTGTCCGATGCGCCCCTGGCGACCCTTGAAAATCCGGATGGCCGAGTGCCTTCCACGCCCGGTCGTACTCATAACCGCATCAGGTCTCCAAGGTGAACAGCCTCTGGTCGATGGAGCAATGTAGGCAAGGGAAGTCGGCAAAATGGATCCGTAACTTCGGGAAAAGGATTGGCTCTGAGGGTTGGGCACGGGGGTCCCAATCCCGAACCCATCGGCTGTCGGCGGACTGCTCGAGCTGCTTGCGCGGCGAGAGCGGGTCGACGCGTGCCGGCTGGGGGACGAATTGGGAGCGGCCCCTTCGGGGGCCTTCCCCGGGCGTCGAACAATCGACTCAGAACTGGTACGGACATGGGGAATCCGACTGTTTAATTAAAACAAAGCATTGCGATGGTCCTCGAGGATGCTCACGCAATGTGATTTCTGCCCAGTGCTCTGAATGTCAAAGTGAAGAAATTCAACCAAGCGCGGGTAAACGGCGGGAGTAACTATGACTCTCTTAAGGTAGCCAAATGCCTCGTCATCTAATTAGTGACGCGCATGAATGGATTAACGAGATTCCCACTGTCCCTGTCTACTATCCAGCGAAACCACAGCCAAGGGAACGGGCTTGGCAGAATCAGCGGGGAAAGAAGACCCTGTTGAGCTTGACTCTAGTCCGACTTTGTGAAATGACTTGAGAGGTGTAGCATAAGTGGGAGTCGGCGTGCCGACGGAATTGAAATACCACTACTTTTAACGTTATTTTACTTATTCCGTGAGACGGAGGCGGGGCCCAGCCCCTCCTTTTGGCTCCAAGTCTCGCCTCGGCGAGTCGATCCGGGCGGAAGACATTGTCAGGTGGGGAGTTTGGCTGGGGCGGCACATCTGTTAAAAGATAACGCAGGTGTCCTAAGATGAGCTCAACGAGAACAGAAATCTCGTGTGGAACAAAAGGGTAAAAGCTCGTTTGATTCTGATTTCCAGTACGAATACGAACCGTGAAAGCGTGGCCTATCGATCCTTTAGGCTTTCAGAATTTGAAGCTAGAGGTGTCAGAAAAGTTACCACAGGGATAACTGGCTTGTGGCAGCCAAGCGTTCATAGCGACGTTGCTTTTTGATCCTTCGATGTCGGCTCTTCCTATCATTGTGAAGCAGAATTCACCAAGTGTTGGATTGTTCACCCACCAATAGGGAACGTGAGCTGGGTTTAGACCGTCGTGAGACAGGTTAGTTTTACCCTACTGATGAATGTGTCGTGATAGTAATTCAACCTAGTACGAGAGGAACCGTTGATTCACACAATTGGTCATCGCGCTTGGTTGAAAAGCCAGTGGCGCGAAGCTATCGTGTGTAGGATTATGACTGAACGCCTCTAAGTCAGAATCCACGCTAAGATGCGACGCTTAGGCCTATCGTTCGCCTGTTGGCCTAAAGTAGGGGCTTGGCCCCCAAGGGCACGCGACCACGGGCTAGTCTGGTGTCATAGAAGGTTGGATGGCATGGGCCCCGTAAGAAATGATAGTTAAGAACGAACGATGGGTAGAATCCTTTGCAGACGACTTAAATTTGCGACGGGGCATTGTAAGTGGCAGAGTGGCCTTGCTGCCACGATCCACTGAGATCCAACCCTATGTTGCAGTAGATTCGTCCCCTTCTGCCGCAGAACACGAAGAGCTTGGTTTTTGGTTTTCTAAAGGGGGACCAAGCTCGCTTGCTCTACAGCTCTACTTATTGCGGTACTTTGTGCGGGTCCTGTGCAACTGGCCTTGCGATGCGTCCCTCGGTGGCCCTACCAATGTGTAAGGCTTAAGCCTGACATTGCACGATTCAGTTATCACAAACGTGAAATAGCTATGAAATGCTGAAAAAGGGTTTGAAAAATATCTGCTGGCTCTGTCTACGTGCACATAGAGGCTTGTCTGCACGCACTTTGGGGCTTGTTTGCGTGCACAGCAGCAACCTACGCGCGCTCCATGTTGGCACCAGAAAAGTGTAACTTTTTTCGGGATTTTCAGCGCTGCGGGCTGCTTCTGCATGGATGCGGTGACAACCTATTGGTAGTCTTACCTTGTGCTGCGTGCTGTCTGTGCACCGGGCCTGTTTGCGTGCACGGCAGCAACCTACGCGCGCTCCGTGTTGGCACCAGAAAAGTGTAACTTTTTTCGGGATTTTTAGCGCTGCGGGCTGCTTCTGCATGGATGCGGTGACAACCTATCGGTAGTCTTGCCTCAGCCCAAGATGGCATTGTCGCAAAACACGATCCGGATGGCGTGCATGGGCATCACGCTCCTGCACGCGGGCATGCGCATCCCGCCCGCCTGCGCCGCGCTCCTGCGTGGGGGAAAGTGTGCTTCTCGCTCCTGCGTGTTGGCCTATCCCCGCTTCTTCGGTCGCAAGGTGCGTCCCCCGCATCATGGTCTTGGAGAATCCTGTAACTTTCACCGAGATCATTAGCTATGCGGTCTTCATGAGTGGACCAAAAAGTGTAACTTTTTTCGGGATTTTTAGCGCTGCGGGCTGCGTCTGCATGGATGCGGTGGCAACCTATTGGTAGTCTTGCCTCTTCCCGAGGTGCGACCATCGCAATACACGATCCGGATGGCGTGCATGTGCATCACGCTCCAGCGATTGGGCCGGTGCGGCGATCCGTCGCGGGCACCGTTCCCTTGCTCGGAACGCGCGGCCGTGGGCACCGCGCTCTCGCGCTCGGTCCTGTGCGCCGCGCAAGTTAGCTCTGCGGGCTCGTGAGGGGACCGAAATTTGTAACTTTTTTCCGGATTTTTAGGGCCGCGGGCTGCTTCAGCATGGTTACGGTGACAACCTATTGGTAGTATTTGCACGTCCCGAGATGCGATCGTCTCAAAACACGTTCCGGATGGCGGTCGTGGTCATCACGCTCCTGCGTGTGGGCAAGCGCGCCGCGCGGCCCCCCGACGCACGAGCGCGCGGCGCGATCATTGGCTCTCCGGTCTCCACGAGGGGACAAAAAAGTGCAACTTTTTTCGGGATTTTCAGCGCTGCGGGCTGCTTCTGCATGGATGCGGTGACAGCCTTTTGGTAGTCTTACCTTGTGCACCGGGCCTGTTTGCGTGCACCGCTGCAACCAACGCGCGCTCCATGTTGGCACCAGAAAAGTGTAACTTTTTTCGGGACCTTTCAGCGCGGCGGG</code></pre>
</div>
<div id="tidy-dataset-based-on-meta-data" class="section level3" number="4.10.7">
<h3><span class="header-section-number">4.10.7</span> Tidy dataset based on meta-data</h3>
<p>Here we apply filters to discard DNA sequences that are:</p>
<ul>
<li>Not belonging to the genus <em>Vanilla</em> OR to the target DNA barcode (here the ITS region).</li>
<li>Contaminated. Previous analysis showed that the <em>V. mexicana</em> ITS sequence is corrupted. We are therefore discarding it from our dataset.</li>
<li>Identified at genus level. Since these sequences cannot be used as reference to identify species and infer relationships with confidence.</li>
<li>Either too short (&lt; 500 bp) or too long (&gt;= 1000 bp).</li>
</ul>
<p>Finally, we are preparing/formatting data for the production of the <code>FASTA</code> file.</p>
<pre class="r"><code>###~~~
#Tidy dataset
###~~~

##
#1. The search retrieved sequences that are NEITHER ITS, NOR Vanilla
##
# Use grep to search for internal (more used than ITS) in definition
# Use grep to search for Vanilla in species
gene &lt;- SEQ[grep(&quot;internal&quot;, SEQ$Definition),]
gene &lt;- gene[grep(&quot;Vanilla&quot;, gene$Species),]

##
#2. Previous analysis showed that the V. mexicana sequence is contaminated/corrupted.
##
# We are therefore discarding it from our dataset
gene &lt;- gene[-which(gene$Species == &quot;Vanilla mexicana&quot;),]

##
#3. Discard DNA sequences identified at genus level
##
#Create vector with names of taxa in dataset
taxaVan &lt;- unique(as.vector(gene$Species))
#Find DNA accessions identified at genus level
spVan &lt;- taxaVan[grep(&quot;sp.&quot;, taxaVan)]
#Subset gene to only keep DNA sequences id at species level
gene &lt;- subset(gene, !(gene$Species %in% spVan))

##
#4. Discard DNA sequences &lt; 500 OR &gt;= 1000 bp
##
gene &lt;- gene[-which(as.numeric(as.vector(gene$Seq_length)) &lt; 500 | as.numeric(as.vector(gene$Seq_length)) &gt;= 1000),]

###~~~
#Prepare dataset for FASTA format
###~~~
# FASTA first line contains GenBank ID and species. 
# Want these fields to be separated by &quot;_&quot; 
# and need to include those in species field
gene$Species &lt;- gsub(&quot; &quot;, &quot;_&quot;, gene$Species)</code></pre>
</div>
<div id="write-results-of-query-in-a-fasta-and-csv-formats" class="section level3" number="4.10.8">
<h3><span class="header-section-number">4.10.8</span> Write results of query in a <code>FASTA</code> and <code>csv</code> formats</h3>
<p>We are using <code>R</code> to generate a <code>FASTA</code> file with the DNA sequences from our GenBank query. To do that we are concatenating information from 3 columns in the <code>gene</code> object:
- <code>gene$GenBankID</code>: Contains unique GenBank ID for DNA sequence.
- <code>gene$Species</code>: Contains species taxonomy associated to sequence.
- <code>gene$Sequence</code>: Contains the DNA sequence.</p>
<pre class="r"><code>###~~~
#Create FASTA
###~~~
FASTAGB &lt;- paste(paste(&quot;&gt;&quot;, as.vector(gene$GenBankID), &quot;_&quot;, 
                         as.vector(gene$Species), sep=&quot;&quot;), 
                   as.vector(gene$Sequence), sep=&#39;\n&#39;)

###~~~
#Write FASTA &amp; table with meta-data
###~~~

##File name FASTA
FileIDFASTA &lt;- paste(sp, DNA, gsub(&quot;-&quot;, &quot;_&quot;, Sys.Date()), &quot;GenBank.fasta&quot;, sep=&#39;_&#39;)
#Write FASTA file in DNA_barcoding/04_Data_analyses/
write.table(FASTAGB, 
            paste(&quot;04_Data_analyses/FASTA/&quot;, 
                  FileIDFASTA, sep=&#39;&#39;), row.names = F, col.names=F, quote = F)

#File name CSV
FileIDcsv &lt;- paste(sp, DNA, gsub(&quot;-&quot;, &quot;_&quot;, Sys.Date()), &quot;GenBank.csv&quot;, sep=&#39;_&#39;)
#Write CSV file
write.table(gene, 
            paste(&quot;04_Data_analyses/CSV/&quot;, 
                  FileIDcsv, sep=&#39;&#39;), row.names = F, col.names=T, quote = T)</code></pre>
<p>Let’s have a look at the tidy data:</p>
<pre><code>## [1] &quot;After cleaning 148 DNA sequences remain.&quot;</code></pre>
<pre><code>##    GenBankID              Species
## 1 1789804163 Vanilla_trigonocarpa
##                                                                                                                                                                                                Definition
## 1 &gt;MN902067.1 Vanilla trigonocarpa voucher W. Stern s.n. internal transcribed spacer 1, partial sequence; 5.8S ribosomal RNA gene, complete sequence; and internal transcribed spacer 2, partial sequence
##   Seq_length
## 1        617
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Sequence
## 1 AGAGGCGTGAATGATGGAACGATCTGTCCAACATGTGGGAGTGCGACAGTTCGATGTCGCCTTCTTCCGTAGCGCGTGCTCTTGCTTCGACGTGGAGCTCGACGCTACGGGGGATAAACAAAAGCTTATGGGCGTTGTCTGGCGCCAAGGGAGAGCAAATGTTCAAGCTGGCAAACGAGTGTGTTGTCGTCAGGTCCATTGAGTCCTGGCAATCGAACGACTCTCGACAACGGATATCTTGGCTCTCGCATCGATGAAGAACGCAGCTTGAAATGCGATACGTGTTGTGAATTGTAGAATCCCGTGAACCATCCATTTTTTGAACGCAAGTTGCGCCCGAGGATGCAAGCCGAGGGCACGCCTGCATGGGTGTAATGCGACCCGTCGCTCCTTGCGGAAGGCTGGAATCTTTGGTTTGGTTCCTCGTCCCCGTTGTGGAGGCGATTGATGGCACCCCGTGCAATAGCATGGCGTGTCGAAGTGTGGGGCGACGGTCGACTGTCGACATGATAAGAGGTGGGCAGCCACCGGCTGTTGTGGTGTTGGCCAGCAATAATCGATGTTGTCGTGCGACAAGCAGGTGCCCCGCATAGATCCAACTCCGTCCTCGATGGTGT</code></pre>
</div>
<div id="format-and-merge-individual-.seq-files-from-part-1-into-a-fasta-file" class="section level3" number="4.10.9">
<h3><span class="header-section-number">4.10.9</span> Format and merge individual <code>.seq</code> files from Part 1 into a <code>FASTA</code> file</h3>
<p>In this section, we are focusing on developing an <code>R</code> code to format and merge our clean sequences from Part 1 by using the following approach:</p>
<ol style="list-style-type: decimal">
<li>Establish a list of all <code>.seq</code> files in folder <code>03_Processed_ITS_data_FASTA/</code> using <code>list.files()</code>.</li>
<li>Open/read files individually using <code>readLines()</code>,</li>
<li>Convert format of <code>FASTA</code> file from interleave to sequential.</li>
<li>Infer reverse complement DNA sequences to be formatted in the same manner as GenBank sequences. For more details on this protocol click <a href="Images/Reverse_complement_DNA_seq.pdf">here</a>.</li>
<li>Merge individual <code>FASTA</code> objects using <code>rbind()</code>.</li>
<li>Write output <code>FASTA</code> object into file using <code>write.table()</code>.</li>
</ol>
<p>Please notice that steps 2 to 4 will take place within a <code>for</code> loop.</p>
<pre class="r"><code>###~~~
#List of .seq files
###~~~

#Please adapt path to your working directory/project
Files &lt;- list.files(&quot;03_Processed_ITS_data_FASTA/&quot;, pattern=&#39;.seq&#39;, full.names = T)

###~~~
#Format and merge all files into one file
###~~~

#This is done by using a loop
seqOUT &lt;- NULL
for(i in 1:length(Files)){
  #Read FASTA
  tmp &lt;- readLines(Files[i])
  #Extract and concatenate DNA sequence
  DNAseq &lt;- paste(tmp[2:length(tmp)], collapse = &#39;&#39;)
  #Infer complementary sequence
  DNAseqcomp &lt;- unname(sapply(strsplit(DNAseq,&quot;&quot;)[[1]], switch,  &quot;A&quot;=&quot;T&quot;, &quot;T&quot;=&quot;A&quot;,&quot;G&quot;=&quot;C&quot;,&quot;C&quot;=&quot;G&quot;))
  #Infer reverse and complement sequence (and collapse into one element)
  DNAseqcomprev &lt;- paste(rev(DNAseqcomp), collapse=&#39;&#39;)
  #Edit FASTA to have only one object/line (=sequential format)
  tmp &lt;- paste(tmp[1], DNAseqcomprev, sep=&#39;\n&#39;)
  
  #Merge objects
  seqOUT &lt;- rbind(seqOUT, tmp)
}

###~~~
#Write data
###~~~

#File name
FileIDfastaSeq &lt;- paste(sp, DNA, gsub(&quot;-&quot;, &quot;_&quot;, Sys.Date()), &quot;msa_input.fasta&quot;, sep=&#39;_&#39;)
#Write file (= input for msa analysis)
write.table(seqOUT, file = 
              paste(&quot;04_Data_analyses/FASTA/&quot;, 
                    FileIDfastaSeq, sep=&#39;&#39;), col.names = F, row.names = F, quote = F)</code></pre>
<p>Please check that the format of your output file is as expected using a text editor (see Figure <a href="#fig:fastaall">4.15</a>).</p>
<div class="figure" style="text-align: center"><span id="fig:fastaall"></span>
<img src="Images/FASTA_single-gene.png" alt="Screenshot of merged FASTA file used as input for msa analysis." width="90%" />
<p class="caption">
Figure 4.15: Screenshot of merged FASTA file used as input for msa analysis.
</p>
</div>
</div>
<div id="merge-fasta-objectsfiles-to-perform-multiple-sequence-alignment" class="section level3" number="4.10.10">
<h3><span class="header-section-number">4.10.10</span> Merge <code>FASTA</code> objects/files to perform multiple sequence alignment</h3>
<p>Here, we are aiming at merging <code>FASTA</code> outputs to produce the input of the multiple sequence alignment. This step is pretty straightforward and involves using the <em>c()</em> function and writing the output.</p>
<p>The objects containg the GenBank DNA sequences and your DNA sequences are <code>FASTAGB</code> and <code>seqOUT</code>, respectively.</p>
<pre class="r"><code>###~~~
#Start by merging datasets (GenBank and your newly produced seq.)
###~~~

#If your FASTA objects are still in R
# merge FASTA into one object
FASTAall &lt;- c(FASTAGB, seqOUT)

#Else, you will have to load fasta files using readLines() before merging them

###~~~
#Write data
###~~~

#File name
FileIDfastaALL &lt;- paste(sp, DNA, gsub(&quot;-&quot;, &quot;_&quot;, Sys.Date()), &quot;GenBank_seq_msa_input.fasta&quot;, sep=&#39;_&#39;)

#Write file (= input for msa analysis)
write.table(FASTAall, file = 
              paste(&quot;04_Data_analyses/FASTA/&quot;, 
                    FileIDfastaALL, sep=&#39;&#39;), col.names = F, row.names = F, quote = F)</code></pre>
</div>
</div>
<div id="part-3-dna-multiple-sequence-alignment-and-phylogenetic-inference" class="section level2" number="4.11">
<h2><span class="header-section-number">4.11</span> Part 3: DNA multiple sequence alignment and phylogenetic inference</h2>
<div id="bioinformatic-tools-2" class="section level3" number="4.11.1">
<h3><span class="header-section-number">4.11.1</span> Bioinformatic tools</h3>
<p>To execute Part 3, you need to install the following software and R packages on your computer:</p>
<ul>
<li><code>MEGA</code> <span class="citation">(<a href="#ref-kumar2018mega" role="doc-biblioref">Kumar et al., 2018</a>)</span>: Please download the GUI version of the software associated to your operating system at this URL:
<ul>
<li><a href="https://www.megasoftware.net/" class="uri">https://www.megasoftware.net/</a></li>
</ul></li>
<li><code>FigTree</code> (a software to visualize and manipulate trees): <a href="https://github.com/rambaut/figtree/releases" class="uri">https://github.com/rambaut/figtree/releases</a></li>
<li><code>R</code> package: <em>ape</em> <span class="citation">(<a href="#ref-ape" role="doc-biblioref">Paradis et al., 2004</a>)</span>.</li>
</ul>
<p>If you don’t remember how to install an R package, don’t worry, this topic was covered <a href="#query">here</a>.</p>
</div>
<div id="analytical-workflow-1" class="section level3" number="4.11.2">
<h3><span class="header-section-number">4.11.2</span> Analytical workflow</h3>
<p>To infer the ML phylogenetic analysis, the following workflow will be executed: implemented:</p>
<ol style="list-style-type: decimal">
<li>Conduct the msa analysis using the <code>MUSCLE</code> algorithm <span class="citation">(<a href="#ref-edgar2004muscle" role="doc-biblioref">Edgar, 2004</a>)</span>. This algorithm is implemented in <code>MEGA</code>. The input data for this analysis is <code>Vanilla_internal transcribed spacer_2021_02_09_GenBank_seq_msa_input.fasta</code>, which is stored in <code>04_Data_analyses/FASTA/</code>.</li>
<li>Check and manually edit msa. This will be done in <code>MEGA</code>.</li>
<li>Infer ML phylogenetic tree based on msa file. This will be done using the <code>RAxML</code> algorithm (available on a web platform).</li>
<li>Visualize phylogenetic tree and interpret results.</li>
</ol>
</div>
<div id="conduct-msa-analysis" class="section level3" number="4.11.3">
<h3><span class="header-section-number">4.11.3</span> Conduct msa analysis</h3>
<div id="disclaimer" class="section level4" number="4.11.3.1">
<h4><span class="header-section-number">4.11.3.1</span> Disclaimer</h4>
<p>The analysis described below relies on <code>MEGA</code>; however you might be experiencing issues with the <code>.fasta</code> file outputted by this program (especially on Windows operating system). These issues might compromise downstream analyses, more specifically the RAxML inference and R code used to generate figures. Those issues are associated to changes in file encryption and changing formatting of samples names (i.e. replacing "_" by " "). In the event that it happens to you, please switch over and use the online <code>MUSCLE</code> platform available at this URL: <a href="https://www.ebi.ac.uk/Tools/msa/muscle/" class="uri">https://www.ebi.ac.uk/Tools/msa/muscle/</a></p>
<p><strong>WARNING:</strong> When you submit your job on the portal, do not forget to select <code>Pearson/FASTA</code> as output file format in step 2. In addition, the output will only be made available in a window and you will have to copy and past the whole content in a new file using your favorite text editor. The file should be saved and named as detailed in the step 8 of the section below.</p>
</div>
<div id="step-by-step-protocol-1" class="section level4" number="4.11.3.2">
<h4><span class="header-section-number">4.11.3.2</span> Step-by-step protocol</h4>
<p>To conduct a multiple sequence alignment in <code>MEGA</code> do the following:</p>
<ol style="list-style-type: decimal">
<li>Launch <code>MEGA</code>.</li>
<li>Load your input file (<code>Vanilla_internal transcribed spacer_2021_02_09_GenBank_seq_msa_input.fasta</code>) by clicking <code>File -&gt; Open A File/Session...</code> and selecting the right file.</li>
<li>The program will ask you <code>How would you like...</code> and you will then click on the <code>Align</code> button. This will open a new window with your data (corresponding to an unaligned DNA matrix; see Figure <a href="#fig:unaligned">4.16</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:unaligned"></span>
<img src="Images/UnalignedMEGA.png" alt="Screenshot of DNA matrix in MEGA." width="100%" />
<p class="caption">
Figure 4.16: Screenshot of DNA matrix in MEGA.
</p>
</div>
<p>4.To start a <code>MUSCLE</code> analysis do has shown in Figure <a href="#fig:muscle">4.17</a> (<code>Alignment -&gt; Align by MUSCLE</code>. A window will open saying <code>Nothing selected for alignment. Select all?</code>, click the <code>OK</code> button. Further details on the <code>MUSCLE</code> algorithm can be found in <span class="citation"><a href="#ref-edgar2004muscle" role="doc-biblioref">Edgar</a> (<a href="#ref-edgar2004muscle" role="doc-biblioref">2004</a>)</span>.</p>
<div class="figure" style="text-align: center"><span id="fig:muscle"></span>
<img src="Images/MUSCLEMega.png" alt="Screenshot showing how to start a MUSCLE analysis in MEGA." width="100%" />
<p class="caption">
Figure 4.17: Screenshot showing how to start a MUSCLE analysis in MEGA.
</p>
</div>
<ol start="5" style="list-style-type: decimal">
<li>A window showing settings associated with the analysis will appear as shown on Figure <a href="#fig:muscleset">4.18</a>. Please use the parameters set by default and click <code>OK</code> to start the analysis. The analysis should take ca. 5-10 minutes to complete.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:muscleset"></span>
<img src="Images/MUSCLEMega_settings.png" alt="Screenshot showing MUSCLE settings in MEGA." width="100%" />
<p class="caption">
Figure 4.18: Screenshot showing MUSCLE settings in MEGA.
</p>
</div>
<ol start="6" style="list-style-type: decimal">
<li>Once the analysis is completed, you will be able to inspect and edit the alignment. Please see Figure <a href="#fig:muscledone">4.19</a> for an example. Notice the gaps (<code>-</code>) that had to be incorporated to accommodate for differences in DNA sequences between samples. Your DNA sequences of <em>Vanilla</em> are at the bottom of the file.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:muscledone"></span>
<img src="Images/MUSCLEMega_done.png" alt="Screenshot showing MUSCLE alignment in MEGA." width="100%" />
<p class="caption">
Figure 4.19: Screenshot showing MUSCLE alignment in MEGA.
</p>
</div>
<ol start="7" style="list-style-type: decimal">
<li>Save the analysis in <code>FASTA</code> format by clicking <code>Data -&gt; Export Alignment -&gt; FASTA Format</code>. This procedure is also shown in Figure <a href="#fig:muscleexport">4.20</a>.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:muscleexport"></span>
<img src="Images/MUSCLEMega_export.png" alt="Screenshot showing how to export MUSCLE alignment into a FASTA format in MEGA." width="100%" />
<p class="caption">
Figure 4.20: Screenshot showing how to export MUSCLE alignment into a FASTA format in MEGA.
</p>
</div>
<ol start="8" style="list-style-type: decimal">
<li>Save the file in <code>04_Data_analyses/FASTA/</code> and adjust file name as follows: <code>_GenBank_seq_msa_output.fasta</code>. This procedure is shown in Figure <a href="#fig:musclefile">4.21</a>. Make sure to have your file extension set as <code>.fasta</code> otherwise you won’t be able to load your file during the <code>RAxML</code> analysis.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:musclefile"></span>
<img src="Images/MUSCLEMega_filename.png" alt="Screenshot showing how to save file in MEGA." width="100%" />
<p class="caption">
Figure 4.21: Screenshot showing how to save file in MEGA.
</p>
</div>
</div>
</div>
<div id="infer-ml-phylogenetic-tree-based-on-msa-file" class="section level3" number="4.11.4">
<h3><span class="header-section-number">4.11.4</span> Infer ML phylogenetic tree based on msa file</h3>
<p>The <code>RAxML</code> algorithm will be used to conduct the ML phylogenetic analysis and infer node statistical supports using the bootstrap procedure <span class="citation">(<a href="#ref-RaxML" role="doc-biblioref">Kozlov et al., 2019</a>)</span>. This algorithm has been implemented on a web service platform accessible at this URL:
- <a href="https://raxml-ng.vital-it.ch/#/" class="uri">https://raxml-ng.vital-it.ch/#/</a></p>
<p>To perform the ML phylogenetic analysis:</p>
<ol style="list-style-type: decimal">
<li>Go on the <a href="https://raxml-ng.vital-it.ch/#/">RAxML portal</a>.</li>
<li>Select the output of the <code>MEGA</code> analysis (<code>*_GenBank_seq_msa_output.fasta</code>) stored in <code>04_Data_analyses/FASTA/</code> as input data.</li>
<li>For the rest of the settings use default parameters with the exception of the <code>Bootstrapping settings</code> (see below). Settings have to be set as shown in Figure <a href="#fig:RAxML">4.22</a>.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:RAxML"></span>
<img src="Images/RAxML_online.png" alt="Screenshot showing Bootstrapping settings for the RAxML analysis." width="100%" />
<p class="caption">
Figure 4.22: Screenshot showing Bootstrapping settings for the RAxML analysis.
</p>
</div>
<ol start="4" style="list-style-type: decimal">
<li><p>Finally, don’t forget to provide your email before submitting the analysis. The analysis should take 1-5 hours to run. You will receive an email to download results or you can access results by using your unique URL.</p></li>
<li><p>When your analysis is completed, download results (see Figure <a href="#fig:RAxMLresults">4.23</a>) and save it in <code>04_Data_analyses/Phylogenetic_analyses/</code>.</p></li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:RAxMLresults"></span>
<img src="Images/RAxML_results.png" alt="Screenshot of RAxML portal showing results of your analysis. The best ML phylogenetic tree is also displayed." width="100%" />
<p class="caption">
Figure 4.23: Screenshot of RAxML portal showing results of your analysis. The best ML phylogenetic tree is also displayed.
</p>
</div>
<ol start="6" style="list-style-type: decimal">
<li>Unzip the file and inspect output files (use a text editor to open files). Descriptions of key files is provided here:
<ul>
<li><code>raxmlArg.txt</code>: Contains RAxML command line used to conduct analysis.</li>
<li><code>result.raxml.bestModel</code>: Contains estimated parameters for model used for ML analysis.</li>
<li><code>result.raxml.bestTree</code>: Best ML tree in newick format.</li>
<li><code>result.raxml.bootstraps</code>: Bootstrap trees in newick format.</li>
<li><code>result.raxml.support</code>: Best ML tree (same tree than in <code>result.raxml.bestTree</code>) with node supports (inferred from data in <code>result.raxml.bootstraps</code>) in newick format.</li>
<li><code>sequenceAlignment.fasta</code>: Your input aligned FASTA file.</li>
</ul></li>
<li>You can open <code>result.raxml.support</code> in <code>FigTree</code> to visualize it and locate the four samples of vanilla from Mexico. We will learn how to process phylogenetic trees in R in the next section.</li>
</ol>
</div>
<div id="visualize-phylogenetic-tree-and-interpret-results" class="section level3" number="4.11.5">
<h3><span class="header-section-number">4.11.5</span> Visualize phylogenetic tree and interpret results</h3>
<div id="introduction-to-newick-tree-format" class="section level4" number="4.11.5.1">
<h4><span class="header-section-number">4.11.5.1</span> Introduction to Newick tree format</h4>
<p>The Newick Standard for representing trees in computer-readable form makes use of the correspondence between trees and nested parentheses. We owe this format to the British mathematician <a href="https://mathshistory.st-andrews.ac.uk/Biographies/Cayley/">Arthur Cayley</a>. Here we will be providing an overview of the Newick format using the R package <em>ape</em> <span class="citation">(<a href="#ref-ape" role="doc-biblioref">Paradis et al., 2004</a>)</span>.</p>
<p>To produce the “dummy” phylogenetic tree displayed in Figure <a href="#fig:intronewick">4.24</a>, the following syntax should be applied:</p>
<div class="figure" style="text-align: center"><span id="fig:intronewick"></span>
<img src="Mini_reports_files/figure-html/intronewick-1.png" alt="Rooted tree showing relationsip between 4 species." width="100%" />
<p class="caption">
Figure 4.24: Rooted tree showing relationsip between 4 species.
</p>
</div>
<pre class="r"><code>###~~~
#Check if package is installed if not then install it
###~~~

if(&quot;ape&quot; %in% rownames(installed.packages()) == FALSE){
  print(&quot;Install ape&quot;)
  install.packages(&quot;ape&quot;)
  }else{
    print(&quot;ape is installed!&quot;)
}

###~~~
#Load package
###~~~
library(ape)

###~~~
#Set working directory
###~~~
#Set working directory to path leading to DNA_barcoding folder
# WARNING: This path as to be adapted to match your computer
setwd(&quot;~/Documents/Class_Genomics&amp;Bioinfo_Spring/DNA_barcoding/&quot;)

#Check that working directory is set correctly
getwd()

###~~~
#Your first tree in Newick format
###~~~
tr1 &lt;- &quot;(A._speciosa,(B._dimorpha,(C._elegans,D._viridis)));&quot;

###~~~
#Plot your tree with ape package
###~~~
plot(ape::read.tree(text=tr1))</code></pre>
<p>Some considerations associated to the syntax:</p>
<ul>
<li>The tree ends with a semicolon (<code>;</code>).</li>
<li>The bottommost node in the tree is an interior node, not a tip.</li>
<li>Interior nodes are represented by a pair of matched parentheses (<code>()</code>). Between them are representations of the nodes that are immediately descended from that node, separated by commas (<code>,</code>).</li>
<li>Tips (or samples) are represented by their names. A name can be any string of printable characters except blanks, colons, semicolons, parentheses, and square brackets.</li>
<li>Because you may want to include a blank in a tip name, it is assumed that an underscore character (<code>_</code>) stands for a blank; any of these in a name will be converted to a blank when it is read in (see code above and Figure <a href="#fig:intronewick">4.24</a> for more details).</li>
</ul>
<p>Branch lengths can be incorporated into a tree by putting a real number, with or without decimal point, after a node and preceded by a colon (<code>:</code>). This represents the length of the branch immediately below that node. Thus, the above tree might have lengths represented as in Figure <a href="#fig:newickbrlens">4.25</a>:</p>
<pre class="r"><code>###~~~
#Tree in Newick format with branch lengths
###~~~
tr2 &lt;- &quot;(A._speciosa:2.0,(B._dimorpha:4.2,(C._elegans:2.0,D._viridis:3.0):1):4.0);&quot;

###~~~
#Plot your tree with ape package
###~~~
plot(ape::read.tree(text=tr2))</code></pre>
<div class="figure" style="text-align: center"><span id="fig:newickbrlens"></span>
<img src="Mini_reports_files/figure-html/newickbrlens-1.png" alt="Rooted tree showing relationsip between 4 species with branch lengths." width="100%" />
<p class="caption">
Figure 4.25: Rooted tree showing relationsip between 4 species with branch lengths.
</p>
</div>
<p>Now, we can finally add node supports by simply adding values after each closing parenthesis. This is done as follows and displayed in Figure <a href="#fig:newickbrlensnode">4.26</a>:</p>
<pre class="r"><code>###~~~
#Tree in Newick format with branch lengths and node supports
###~~~
tr3 &lt;- &quot;(A._speciosa:2.0,(B._dimorpha:4.2,(C._elegans:2.0,D._viridis:3.0)60:1)90:4.0)100;&quot;

###~~~
#Plot your tree with ape package
###~~~
# Note that to display node labels, we have to set show.node.label = TRUE 
plot(ape::read.tree(text=tr3), show.node.label = TRUE)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:newickbrlensnode"></span>
<img src="Mini_reports_files/figure-html/newickbrlensnode-1.png" alt="Rooted tree showing relationsip between 4 species with branch lengths and node supports." width="100%" />
<p class="caption">
Figure 4.26: Rooted tree showing relationsip between 4 species with branch lengths and node supports.
</p>
</div>
</div>
<div id="phylo-class-and-phylogenetic-tree-in-r" class="section level4" number="4.11.5.2">
<h4><span class="header-section-number">4.11.5.2</span> <code>phylo</code> class and phylogenetic tree in R</h4>
<p>Here, we will be introducing the <code>phylo</code> class implemented in R packages dealing with phylogenetic analyses. To study this topic, we will be using the phylogenetic tree with branch lengths and node supports (see Figure <a href="#fig:newickbrlensnode">4.26</a>).</p>
<pre class="r"><code>###~~~
#Tree in Newick format with branch lengths and node supports
###~~~
tr3 &lt;- &quot;(A._speciosa:2.0,(B._dimorpha:4.2,(C._elegans:2.0,D._viridis:3.0)60:1)90:4.0)100;&quot;

###~~~
#Create and load tree
###~~~
tr &lt;- ape::read.tree(text=tr3)
tr</code></pre>
<pre><code>## 
## Phylogenetic tree with 4 tips and 3 internal nodes.
## 
## Tip labels:
##   A._speciosa, B._dimorpha, C._elegans, D._viridis
## Node labels:
##   100, 90, 60
## 
## Rooted; includes branch lengths.</code></pre>
<pre class="r"><code>###~~~
#Check class
###~~~
class(tr)</code></pre>
<pre><code>## [1] &quot;phylo&quot;</code></pre>
<p><code>phylo</code> class objects (here <code>tr</code>) are lists allowing to access multiple attributes associated to the phylogenetic tree. Three of these lists are especially useful to us:</p>
<ul>
<li><code>tr$tip.label</code>: Vector with tip labels.</li>
<li><code>tr$node.label</code>: Vector with node labels.</li>
<li><code>tr$edge.length</code>: Vector with branch lengths.</li>
</ul>
<p>Here, are the details with our example:</p>
<pre class="r"><code>###~~~
#To access tip labels
###~~~
tr$tip.label</code></pre>
<pre><code>## [1] &quot;A._speciosa&quot; &quot;B._dimorpha&quot; &quot;C._elegans&quot;  &quot;D._viridis&quot;</code></pre>
<pre class="r"><code>###~~~
#To access node labels
###~~~
tr$node.label</code></pre>
<pre><code>## [1] &quot;100&quot; &quot;90&quot;  &quot;60&quot;</code></pre>
<pre class="r"><code>###~~~
#To access branch lengths
###~~~
tr$edge.length</code></pre>
<pre><code>## [1] 2.0 4.0 4.2 1.0 2.0 3.0</code></pre>
</div>
<div id="vanilla-phylogenetic-tree" class="section level4" number="4.11.5.3">
<h4><span class="header-section-number">4.11.5.3</span> Vanilla phylogenetic tree</h4>
<p>In this section, we will be analyzing the output of the RAxML analysis focusing on <code>result.raxml.support</code>, which contains the phylogenetic tree with node supports (here obtained through the bootstrap method). The file is stored in <code>04_Data_analyses/Phylogenetic_analysis/result_raxml_45818</code>.</p>
<p>Our analyses are divided into seven steps:</p>
<ol style="list-style-type: decimal">
<li>Load ITS tree using the <em>ape::read.tree()</em> function.</li>
<li>Discard bootstrap values &lt; 50% since these nodes are not statistically supported. This is done by finding nodes matching this criterion (in <code>ITS$node.label</code>) and replacing their values by nothing (<code>""</code>).</li>
<li>Rename tips to facilitate reading. Since all accessions belong to same genus, we will replace <em>Vanilla</em> by <em>V.</em>.</li>
<li>Root phylogenetic tree with outgroup taxon (using the <em>ape::root()</em> function). Analyses conducted by Paige showed that <em>Vanilla inodora</em> (represented by the sample <code>1789804116_Vanilla_inodora</code>) is a suitable outgroup taxon.</li>
<li>Ladderize and plot phylogenetic tree (using <em>ape::ladderize()</em> and <em>ape::plot.phylo()</em> functions; see Figure <a href="#fig:Vantreeall">4.27</a>). We will also color in red the tips associated to your vanilla samples.</li>
<li>Extract subtree containing our vanilla samples (using <em>ape::getMRCA()</em> and <em>ape::extract.clade()</em> functions) and plot tree (this time using the radial plotting mode; see Figure <a href="#fig:Vantreesub">4.28</a>). To visualize your vanilla samples we will plot a red circle next to their tips (using the <em>ape::tiplabels()</em> function) and rename them based on Figure <a href="#fig:van">4.2</a>.</li>
<li>Export phylogenetic tree from step 6 in <code>pdf</code> format.</li>
</ol>
<pre class="r"><code>###~~~
#1. Load Vanilla RAxML ITS tree
###~~~
#Please adjust path based on your computer
ITS &lt;- ape::read.tree(file=&quot;Data/04_Data_analyses/Phylogenetic_analyses/result_raxml_45818/result.raxml.support&quot;)

###~~~
#2. Discard bootstrap values &lt; 50%
###~~~
#Find nodes with low statistical supports
ITS$node.label[which(as.numeric(ITS$node.label) &lt; 50)]</code></pre>
<pre><code>##  [1] &quot;17&quot; &quot;12&quot; &quot;14&quot; &quot;10&quot; &quot;4&quot;  &quot;1&quot;  &quot;3&quot;  &quot;11&quot; &quot;3&quot;  &quot;1&quot;  &quot;1&quot;  &quot;1&quot;  &quot;1&quot;  &quot;4&quot;  &quot;25&quot;
## [16] &quot;3&quot;  &quot;0&quot;  &quot;0&quot;  &quot;3&quot;  &quot;3&quot;  &quot;43&quot; &quot;10&quot; &quot;1&quot;  &quot;1&quot;  &quot;0&quot;  &quot;4&quot;  &quot;11&quot; &quot;42&quot; &quot;48&quot; &quot;41&quot;
## [31] &quot;40&quot; &quot;24&quot; &quot;42&quot; &quot;49&quot; &quot;45&quot; &quot;8&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot; 
## [46] &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;0&quot;  &quot;1&quot;  &quot;11&quot; &quot;48&quot; &quot;14&quot; &quot;18&quot; &quot;33&quot; &quot;44&quot; &quot;20&quot; &quot;15&quot;
## [61] &quot;26&quot; &quot;24&quot; &quot;10&quot; &quot;5&quot;  &quot;41&quot; &quot;46&quot; &quot;20&quot; &quot;19&quot; &quot;38&quot; &quot;42&quot; &quot;28&quot; &quot;13&quot; &quot;2&quot;</code></pre>
<pre class="r"><code>#Replace/overwrite values by &quot;&quot;
ITS$node.label[which(as.numeric(ITS$node.label) &lt; 50)] &lt;- &quot;&quot;

###~~~
#3. Rename tips: Vanilla by V.
###~~~
#Replace/overwrite values
ITS$tip.label &lt;- gsub(&quot;Vanilla&quot;, &quot;V.&quot;, ITS$tip.label)

###~~~
#4. Root phylogenetic tree with outgroup taxon
###~~~
#Re-root tree with suitable sample
ITS &lt;- ape::root(ITS, outgroup = &quot;1789804116_V._inodora&quot;, resolve.root=T)
#This procedure add a node label &quot;Root&quot;, which needs to be discarded.
ITS$node.label[which(ITS$node.label == &quot;Root&quot;)] &lt;- &quot;&quot;

###~~~
#5. Ladderize and plot tree by sorting nodes
###~~~
#Ladderize
ITS &lt;- ape::ladderize(ITS)

#Color tips: your samples in red
tipcol &lt;- rep(&quot;black&quot;, length(ITS$tip.label))
#Find which tips are your samples and replace color by red 
tipcol[grep(&quot;-ITSp4&quot;, ITS$tip.label)] &lt;- &quot;red&quot;

#Plot
ape::plot.phylo(ITS, cex=.3, tip.color = tipcol, show.node.label = TRUE, align.tip.label = T, no.margin = TRUE)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:Vantreeall"></span>
<img src="Mini_reports_files/figure-html/Vantreeall-1.png" alt="RaxML ITS rooted phylogenetic tree of species of Vanilla." width="100%" />
<p class="caption">
Figure 4.27: RaxML ITS rooted phylogenetic tree of species of Vanilla.
</p>
</div>
<pre class="r"><code>###~~~
#6. Extract subtree and plot tree
###~~~
#Find node corresponding to MRCA of our vanilla samples
MRCAsamples &lt;- ape::getMRCA(ITS, tip=ITS$tip.label[grep(&quot;-ITSp4&quot;, ITS$tip.label)])

#Extract subtree
VanITS &lt;- ape::extract.clade(ITS, node=MRCAsamples)

#Rename vanilla samples using apply and switch as learned before
VanITS$tip.label[grep(&quot;-ITSp4&quot;, VanITS$tip.label)] &lt;- unname(sapply(VanITS$tip.label[grep(&quot;-ITSp4&quot;, VanITS$tip.label)], switch,  &quot;its25-ITSp4&quot;=&quot;PE25&quot;, &quot;its50-ITSp4&quot;=&quot;PE50&quot;,&quot;its44-ITSp4&quot;=&quot;PE44&quot;,&quot;its49-ITSp4&quot;=&quot;PE49&quot;))

#Color tips: your samples in red
tipcolVan &lt;- rep(&quot;black&quot;, length(VanITS$tip.label))
#Find which tips are your samples and replace color by red 
# ^ means that the search has to begin with PE
tipcolVan[grep(&quot;^PE&quot;, VanITS$tip.label)] &lt;- &quot;red&quot;

#Plot tree in radial mode
ape::plot.phylo(VanITS, cex = .4, use.edge.length = F, tip.color = tipcolVan, type = &quot;radial&quot;, label.offset = 0.04, no.margin = TRUE)
#Add bootstrap supports
ape::nodelabels(text = VanITS$node.label, adj = c(0.5,0.5), frame = &quot;none&quot;, cex = 0.5)
#Add circles next to your vanilla samples
ape::tiplabels(tip = grep(&quot;^PE&quot;, VanITS$tip.label), pch = 16, col = &quot;red&quot;, offset = 0.02)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:Vantreesub"></span>
<img src="Mini_reports_files/figure-html/Vantreesub-1.png" alt="RAxML ITS phylogenetic tree with focus on clade containing all accessions of vanilla studied here." width="100%" />
<p class="caption">
Figure 4.28: RAxML ITS phylogenetic tree with focus on clade containing all accessions of vanilla studied here.
</p>
</div>
<pre class="r"><code>###~~~
#7. Export tree from step 6 in pdf format 
###~~~
#Use the pdf function to create pdf
pdf(&quot;Data/04_Data_analyses/Phylogenetic_analyses/result_raxml_45818/VanSubTree.pdf&quot;)
#Plot tree in radial mode
ape::plot.phylo(VanITS, cex = .4, use.edge.length = F, tip.color = tipcolVan, type = &quot;radial&quot;, label.offset = 0.04, no.margin = TRUE)
#Add bootstrap supports
ape::nodelabels(text = VanITS$node.label, adj = c(0.5,0.5), frame = &quot;none&quot;, cex = 0.5)
#Add circles next to your vanilla samples
ape::tiplabels(tip = grep(&quot;^PE&quot;, VanITS$tip.label), pch = 16, col = &quot;red&quot;, offset = 0.02)
#This function closes the pdf
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<p>Click on the button below to inspect the <em>Vanilla</em> subtree in <code>pdf</code> format. Ask yourself this question:</p>
<blockquote>
<p>What species do the samples of vanilla analyzed here belong to (see Figure <a href="#fig:van">4.2</a>)?</p>
</blockquote>
<button class="btn btn-primary" data-toggle="collapse" data-target="#VanSub">
Vanilla subtree
</button>
<div id="VanSub" class="collapse">
<object data="Data/04_Data_analyses/Phylogenetic_analyses/result_raxml_45818/VanSubTree.pdf" width="750px" height="750px">
<embed src="Data/04_Data_analyses/Phylogenetic_analyses/result_raxml_45818/VanSubTree.pdf">
</embed>
</object>
</div>
</div>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-ALTSCHUL1990403" class="csl-entry">
<span class="smallcaps">Altschul, S.F.</span>, <span class="smallcaps">W. Gish</span>, <span class="smallcaps">W. Miller</span>, <span class="smallcaps">E.W. Myers</span>, and <span class="smallcaps">D.J. Lipman</span>. 1990. Basic local alignment search tool. <em>Journal of Molecular Biology</em> 215: 403–410. Available at: <a href="http://www.sciencedirect.com/science/article/pii/S0022283605803602">http://www.sciencedirect.com/science/article/pii/S0022283605803602</a>.
</div>
<div id="ref-buerki2017alectryon" class="csl-entry">
<span class="smallcaps">Buerki, S.</span>, <span class="smallcaps">P.P. Lowry II</span>, <span class="smallcaps">J. Munzinger</span>, <span class="smallcaps">M. Tuiwawa</span>, <span class="smallcaps">A. Naikatini</span>, and <span class="smallcaps">M.W. Callmander</span>. 2017. Alectryon vitiensis: A new species of sapindaceae endemic to fiji. <em>Novon: A Journal for Botanical Nomenclature</em> 25: 421–429.
</div>
<div id="ref-CBOL" class="csl-entry">
<span class="smallcaps">CBOL</span>. 2021. International barcode of life: DNA barcoding. Available at: <a href="https://ibol.org/about/dna-barcoding/">https://ibol.org/about/dna-barcoding/</a>.
</div>
<div id="ref-Cheng2016" class="csl-entry">
<span class="smallcaps">Cheng, T.</span>, <span class="smallcaps">C. Xu</span>, <span class="smallcaps">L. Lei</span>, <span class="smallcaps">C. Li</span>, <span class="smallcaps">Y. Zhang</span>, and <span class="smallcaps">S. Zhou</span>. 2016. Barcoding the kingdom plantae: New PCR primers for ITS regions of plants with improved universality and specificity. <em>Molecular Ecology Resources</em> 16: 138–149. Available at: <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.12438">https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.12438</a>.
</div>
<div id="ref-dentinger2014s" class="csl-entry">
<span class="smallcaps">Dentinger, B.T.</span>, and <span class="smallcaps">L.M. Suz</span>. 2014. What’s for dinner? Undescribed species of porcini in a commercial packet. <em>PeerJ</em> 2: e570.
</div>
<div id="ref-deSalle2019" class="csl-entry">
<span class="smallcaps">DeSalle, R.</span>, and <span class="smallcaps">P. Goldstein</span>. 2019. Review and interpretation of trends in DNA barcoding. <em>Frontiers in Ecology and Evolution</em> 7: 302. Available at: <a href="https://www.frontiersin.org/article/10.3389/fevo.2019.00302">https://www.frontiersin.org/article/10.3389/fevo.2019.00302</a>.
</div>
<div id="ref-edgar2004muscle" class="csl-entry">
<span class="smallcaps">Edgar, R.C.</span> 2004. MUSCLE: Multiple sequence alignment with high accuracy and high throughput. <em>Nucleic acids research</em> 32: 1792–1797.
</div>
<div id="ref-Ellestadinpress" class="csl-entry">
<span class="smallcaps">Ellestad, P.</span>, <span class="smallcaps">F. Forest</span>, <span class="smallcaps">M. Serpe</span>, <span class="smallcaps">S.J. Novak</span>, and <span class="smallcaps">S. Buerki</span>. 2021. <span class="nocase">Harnessing large-scale biodiversity data to infer the current distribution of Vanilla planifolia (Orchidaceae)</span>. <em>Botanical Journal of the Linnean Society</em>. Available at: <a href="https://doi.org/10.1093/botlinnean/boab005">https://doi.org/10.1093/botlinnean/boab005</a>.
</div>
<div id="ref-gonccalves2015dna" class="csl-entry">
<span class="smallcaps">Gonçalves, P.F.</span>, <span class="smallcaps">A.R. Oliveira-Marques</span>, <span class="smallcaps">T.E. Matsumoto</span>, and <span class="smallcaps">C.Y. Miyaki</span>. 2015. DNA barcoding identifies illegal parrot trade. <em>Journal of Heredity</em> 106: 560–564.
</div>
<div id="ref-Hollingsworth2009" class="csl-entry">
<span class="smallcaps">Hollingsworth, P.M.</span>, <span class="smallcaps">L.L. Forrest</span>, <span class="smallcaps">J.L. Spouge</span>, <span class="smallcaps">M. Hajibabaei</span>, <span class="smallcaps">S. Ratnasingham</span>, <span class="smallcaps">M. van der Bank</span>, <span class="smallcaps">M.W. Chase</span>, et al. 2009. A DNA barcode for land plants. <em>Proceedings of the National Academy of Sciences</em> 106: 12794–12797. Available at: <a href="https://www.pnas.org/content/106/31/12794">https://www.pnas.org/content/106/31/12794</a>.
</div>
<div id="ref-RaxML" class="csl-entry">
<span class="smallcaps">Kozlov, A.M.</span>, <span class="smallcaps">D. Darriba</span>, <span class="smallcaps">T. Flouri</span>, <span class="smallcaps">B. Morel</span>, and <span class="smallcaps">A. Stamatakis</span>. 2019. <span class="nocase">RAxML-NG: a fast, scalable and user-friendly tool for maximum likelihood phylogenetic inference</span>. <em>Bioinformatics</em> 35: 4453–4455. Available at: <a href="https://doi.org/10.1093/bioinformatics/btz305">https://doi.org/10.1093/bioinformatics/btz305</a>.
</div>
<div id="ref-kumar2018mega" class="csl-entry">
<span class="smallcaps">Kumar, S.</span>, <span class="smallcaps">G. Stecher</span>, <span class="smallcaps">M. Li</span>, <span class="smallcaps">C. Knyaz</span>, and <span class="smallcaps">K. Tamura</span>. 2018. MEGA x: Molecular evolutionary genetics analysis across computing platforms. <em>Molecular biology and evolution</em> 35: 1547–1549.
</div>
<div id="ref-Larsen2017" class="csl-entry">
<span class="smallcaps">Larsen, B.B.</span>, <span class="smallcaps">E.C. Miller</span>, <span class="smallcaps">M.K. Rhodes</span>, and <span class="smallcaps">J.J. Wiens</span>. 2017. Inordinate fondness multiplied and redistributed: The number of species on earth and the new pie of life. <em>The Quarterly Review of Biology</em> 92: 229–265.
</div>
<div id="ref-Masters2017" class="csl-entry">
<span class="smallcaps">Masters, J.C.</span>, and <span class="smallcaps">L. Pozzi</span>. 2017. Phylogenetic inference. <em>In</em> The international encyclopedia of primatology, 1–6. American Cancer Society. Available at: <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/9781119179313.wbprim0419">https://onlinelibrary.wiley.com/doi/abs/10.1002/9781119179313.wbprim0419</a>.
</div>
<div id="ref-ape" class="csl-entry">
<span class="smallcaps">Paradis, E.</span>, <span class="smallcaps">J. Claude</span>, and <span class="smallcaps">K. Strimmer</span>. 2004. A<span>PE</span>: Analyses of phylogenetics and evolution in <span>R</span> language. <em>Bioinformatics</em> 20: 289–290.
</div>
<div id="ref-ape" class="csl-entry">
<span class="smallcaps">Paradis, E.</span>, <span class="smallcaps">J. Claude</span>, and <span class="smallcaps">K. Strimmer</span>. 2004. A<span>PE</span>: Analyses of phylogenetics and evolution in <span>R</span> language. <em>Bioinformatics</em> 20: 289–290.
</div>
<div id="ref-Poplin2018" class="csl-entry">
<span class="smallcaps">Poplin, R.</span>, <span class="smallcaps">P.-C. Chang</span>, <span class="smallcaps">D. Alexander</span>, <span class="smallcaps">S. Schwartz</span>, <span class="smallcaps">T. Colthurst</span>, <span class="smallcaps">A. Ku</span>, <span class="smallcaps">D. Newburger</span>, et al. 2018. A universal SNP and small-indel variant caller using deep neural networks. <em>Nature Biotechnology</em> 36: 983–987. Available at: <a href="https://doi.org/10.1038/nbt.4235">https://doi.org/10.1038/nbt.4235</a>.
</div>
<div id="ref-quinto2016dna" class="csl-entry">
<span class="smallcaps">Quinto, C.A.</span>, <span class="smallcaps">R. Tinoco</span>, and <span class="smallcaps">R.S. Hellberg</span>. 2016. DNA barcoding reveals mislabeling of game meat species on the US commercial market. <em>Food Control</em> 59: 386–392.
</div>
<div id="ref-Stamatakis2014" class="csl-entry">
<span class="smallcaps">Stamatakis, A.</span> 2014. RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. <em>Bioinformatics (Oxford, England)</em> 30: 1312–1313. Available at: <a href="https://doi.org/10.1093/bioinformatics/btu033">https://doi.org/10.1093/bioinformatics/btu033</a>.
</div>
<div id="ref-telfer2015biodiversity" class="csl-entry">
<span class="smallcaps">Telfer, A.C.</span>, <span class="smallcaps">M.R. Young</span>, <span class="smallcaps">J. Quinn</span>, <span class="smallcaps">K. Perez</span>, <span class="smallcaps">C.N. Sobel</span>, <span class="smallcaps">J.E. Sones</span>, <span class="smallcaps">V. Levesque-Beaudin</span>, et al. 2015. Biodiversity inventories in high gear: DNA barcoding facilitates a rapid biotic survey of a temperate nature reserve. <em>Biodiversity data journal</em>.
</div>
<div id="ref-wheeler1999phylogenetic" class="csl-entry">
<span class="smallcaps">Wheeler, Q.D.</span> 1999. Why the phylogenetic species concept?—elementary. <em>Journal of nematology</em> 31: 134.
</div>
<div id="ref-rentrez" class="csl-entry">
<span class="smallcaps">Winter, D.J.</span> 2017. <span class="nocase">rentrez</span>: An r package for the NCBI eUtils API. <em>The R Journal</em> 9: 520–526.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
