<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki &amp; Anthony Melton - Boise State University" />

<meta name="date" content="2021-03-16" />

<title>BIOL 497/597 - Genomics &amp; Bioinformatics</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">BIOL 497/597 - Genomics &amp; Bioinformatics</h1>
<h3 class="subtitle">Lab: Mining sagebrush draft genome</h3>
<h4 class="author">Sven Buerki &amp; Anthony Melton - Boise State University</h4>
<h4 class="date">2021-03-16</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>This webpage supports analyses aiming at mining the sagebrush draft genome for Aquaporin genes. It is subdivided into 4 modules as follows:</p>
<ul>
<li><a href="#mod1"><strong>Module 1:</strong></a> Set-up environment and conduct BLAST analysis.</li>
<li><a href="#mod2"><strong>Module 2:</strong></a> Extract scaffolds and find <strong>O</strong>pen <strong>R</strong>eading <strong>F</strong>rames.</li>
<li><strong>Module 3:</strong> Annotate ORFs and assemble proteins.</li>
<li><strong>Module 4:</strong> Alignment and phylogenetic reconstruction.</li>
</ul>
<div id="project-structure" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Project structure</h2>
<div id="where-are-the-data-located" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Where are the data located?</h3>
<p>All the data associated to this project are located on your individual account in the <code>DraftGenomeMineR/</code> folder. This folder is located in <code>/home/bio_11/DraftGenomeMineR</code>. Please replace <code>bio_11</code> by your <a href="https://docs.google.com/document/d/1t0DdQBes42Gzo8XWgc9JL2xEGjv1kbfoVE7Vznx6C9I/edit">account detail</a>.</p>
<p><strong>Key files:</strong></p>
<ul>
<li><code>Draft_Genome_Assembly.fasta</code>: Sagebrush draft genome assembly (at scaffold level).</li>
<li><code>FASTAs/PIP1_3.fa</code>: Reference containing <em>PIP1</em> gene (AA) for BLAST analysis. This data will be used to find Aquaporin genes in the draft genome.</li>
<li>Scripts related to each module are located in <code>Lesson_Modules/</code>.</li>
</ul>
</div>
<div id="how-do-i-access-the-data-and-run-analyses" class="section level3" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> How do I access the data and run analyses?</h3>
<p>To access the data and run the analyses, please remotely connect to your dedicated computer using <code>ssh</code> protocol. Your credentials are available <a href="https://docs.google.com/document/d/1t0DdQBes42Gzo8XWgc9JL2xEGjv1kbfoVE7Vznx6C9I/edit">here</a>.</p>
</div>
</div>
</div>
<div id="mod1" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Module 1: Set-up environment and conduct BLAST analysis</h1>
<div id="remotely-connect-to-computer" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Remotely connect to computer</h2>
<p>Start by using <code>ssh</code> protocol to connect to your Linux computer. This is done as follows (enter password when prompted to):</p>
<pre class="bash"><code>#SSH with bio_11 as example
ssh bio_11@132.178.142.214</code></pre>
</div>
<div id="copy-project-folder-to-remote-computer" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Copy project folder to remote computer</h2>
<p>After downloading the <code>zip</code> file from our shared Google Drive to your personal computer, please copy this file to your Linux computer (on your personal account) as follows:</p>
<pre class="bash"><code>#scp file: general syntax
scp file user@host:dir

#Example
scp DraftGenomeMineR.zip bio_11@132.178.142.214:/home/bio_11

#unzip file
unzip DraftGenomeMineR.zip</code></pre>
<p><strong>Disclaimer:</strong> This procedure only works if you have a unix-based computer. If you don’t have a unix-based computer, please ask instructors to transfer this file for you to your account.</p>
</div>
<div id="start-r-session" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Start R session</h2>
<p>Navigate to <code>DraftGenomeMineR/</code> and start a new <code>R</code> session:</p>
<pre class="bash"><code>#Navigate to DraftGenomeMineR/
cd DraftGenomeMineR/

#Start R session
R-4</code></pre>
</div>
<div id="load-r-packages-and-user-defined-functions" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Load R packages and user-defined functions</h2>
<p>Create an object with all the requiered R packages and load them:</p>
<pre class="r"><code># This will make a list of packages that another function will use to make sure they are all installed and loaded.
list.of.packages &lt;- c(&quot;ape&quot;,
                      &quot;Biostrings&quot;,
                      &quot;dplyr&quot;,
                      &quot;FastaUtils&quot;,
                      &quot;ORFik&quot;,
                      &quot;readr&quot;,
                      &quot;tidyr&quot;,
                      &quot;rBLAST&quot;,
                      &quot;seqinr&quot;,
                      &quot;stringr&quot;)

# Use lapply to load all of the packages in the list.
lapply(list.of.packages, require, character.only = TRUE)</code></pre>
<p>Load all the user-defined function located in <code>Functions/</code>:</p>
<pre class="r"><code># Load all of the functions written for DraftGenomeMineR     
files.sources &lt;- list.files(&quot;Functions&quot;, full.names=T)
sapply(files.sources, source)</code></pre>
</div>
<div id="setting-environment-for-blast-analysis" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Setting environment for BLAST analysis</h2>
<p>To be able to navigate between folders within our project, we will set a object entitled <code>project.folder</code>, which contains the path to the root of the folder.</p>
<p>To create this object, do as follows:</p>
<pre class="r"><code>#Copy the output of getwd() in the object as follows:
project.folder &lt;- getwd()</code></pre>
<p>Call <code>project.folder</code> to check that it is right. It should be:
<code>"/home/bio_11/DraftGenomeMineR"</code> (with <code>bio_11</code> replaced by your account ID).</p>
<p>Set working directory to <code>project.folder</code> as follows:</p>
<pre class="r"><code>setwd(project.folder)</code></pre>
</div>
<div id="running-blast-analysis" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Running BLAST analysis</h2>
<p>Set general parameters for the BLAST analysis:</p>
<pre class="r"><code># There are a few parameters that need to be set. Let&#39;s create some R objects to store file paths and parameters for the BLAST search.

# We need a query (a fasta file of a gene you want to find in the draft genome), a draft genome assembly, BLAST databases, paths to other required folders (described in the README), and other parameters that can be used to filter results.
query.file.path &lt;- &quot;FASTAs/PIP1_3.fa&quot;
genome.file.name &lt;- &quot;Draft_Genome_Assembly.fasta&quot;
genome.path &lt;- &quot;FASTAs/Draft_Genome_Assembly.fasta&quot;
blast.db.path &lt;- &quot;BlastDBs/Draft_Genome_Assembly.fasta&quot;
AA.BlastDB.folder &lt;- &quot;AA_BlastDB/&quot;
AA.ORF.folder &lt;- &quot;AA_ORFs/&quot;
max.e &lt;- 5E-50
perc.ident &lt;- 90.000
query.type &lt;- &quot;AA&quot;
blast.type &lt;- &quot;tblastn&quot;
make.BlastDB &lt;- T
BlastDB.type &lt;- &quot;nucl&quot;</code></pre>
<p>Read in the sagebrush draft genome assembly in <code>FASTA</code> format:</p>
<pre class="r"><code># Read in the genome assembly file
genome &lt;- readLines(con = genome.path)
head(genome) # Print the top 6 lines of the fasta. There should be no spaces in what is printed. Each header should be on one line, followed by its entire scaffold on the next.</code></pre>
<p>Make a BLAST database for query:</p>
<pre class="r"><code># Do you need to make a BLAST database or use an existing one?
if(make.BlastDB == TRUE){
  setwd(&quot;BlastDBs/&quot;)
  makeblastdb(file = genome.file.name, dbtype = BlastDB.type)
}</code></pre>
<p>Read in the query file in <code>FASTA</code> format:</p>
<pre class="r"><code># Read in the query. What type of molecule is the query? DNA (or RNA) or amino acid sequences?
setwd(project.folder) # Previous lines changed the wd, so we need to go back to the project folder.
if (query.type == &quot;DNA&quot;) {
  query &lt;- readDNAStringSet(filepath = query.file.path,
                            format = &quot;fasta&quot;)
} else {
  query &lt;- readAAStringSet(filepath = query.file.path,
                           format = &quot;fasta&quot;)
}</code></pre>
<p>Perform BLAST analysis (<code>BLAST</code> needs to be installed on the computer):</p>
<pre class="r"><code># Now we have everything set up and can perform a BLAST search.
bl &lt;- blast(db = blast.db.path, type = blast.type) # Create an object with the BLAST database and the type of BLAST search to perform.
cl &lt;- predict(bl, query) # Perform the BLAST search
head(cl) # Look at the top BLAST hits</code></pre>
<p>Filter output of BLAST analysis:</p>
<pre class="r"><code># We can now filter out bad BLAST hits using a few thresholds and parameters. This leaves us with only the best matches to our query.
# What parameters do you think would give you just the best candidates to be your gene of interest?
cl.filt &lt;- subset(x = cl, Perc.Ident &gt;= perc.ident &amp; E &lt;= max.e)
cl.filt.unique &lt;- cl.filt[!duplicated(cl.filt[,c(&#39;SubjectID&#39;)]),] # SubjectID is the column that contains the scaffold names
cl.filt.unique
nrow(cl)
nrow(cl.filt)
nrow(cl.filt.unique)
blast.hits.to.extract &lt;- subset(x = cl.filt.unique, SubjectID == cl.filt.unique[1,2]) </code></pre>
<p>Write output of BLAST analysis in <code>csv</code> file:</p>
<pre class="r"><code># We have evaluated the top BLAST hits and can see a clear best hit. Let&#39;s save this data so we can extract just that scaffold later.
write.csv(x = blast.hits.to.extract, file = &quot;Unique_Filtered_Blast_Hit_Info.csv&quot;, row.names = F)</code></pre>
</div>
</div>
<div id="mod2" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Module 2: Extract scaffolds and find <strong>O</strong>pen <strong>R</strong>eading <strong>F</strong>rames</h1>
<p>Here, we are:</p>
<ol style="list-style-type: decimal">
<li>Extracting scaffold(s) identified in <a href="#mod1">module 1</a> from the draft genome assembly (in <code>FASTAs/Draft_Genome_Assembly.fasta</code>) and saving the output as a <code>FASTA</code> object/file.</li>
<li>Finding ORFs along scaffold(s) using the user-defined function <em>findORFsTranslateDNA2AA()</em> and saving the output into a <code>FASTA</code> object/file. In molecular genetics, an <strong>O</strong>pen <strong>R</strong>eading <strong>F</strong>rame (ORF) is the part of a reading frame that has the ability to be translated. An ORF is a continuous stretch of codons that begins with a start codon (usually AUG) and ends at a stop codon (usually UAA, UAG or UGA). We will study the code implemented in <em>findORFsTranslateDNA2AA()</em> to fully understand the applied approach.</li>
</ol>
<div id="getting-ready" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Getting ready!</h2>
<p>Students have to:</p>
<ol style="list-style-type: decimal">
<li>Remotely connect to their computers (using their individual accounts) with <code>ssh</code> protocol.</li>
<li>Navigate to project directory (<code>DraftGenomeMineR/</code>).</li>
<li>Create a new folder (<code>Output_FASTAs/</code>) using <code>mkdir</code> to store results of scaffold analysis.</li>
<li>Create a new folder (<code>ORFs_report/</code>) using <code>mkdir</code> to store results of ORFs analysis.</li>
<li>Start a new <code>R</code> session using <code>R-4</code> command.</li>
<li>Load R packages and user-defined functions.</li>
<li>Set working directory to <code>DraftGenomeMineR/</code>.</li>
</ol>
</div>
<div id="extracting-scaffolds-identified-in-module-1" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Extracting scaffold(s) identified in module 1</h2>
<p>Open <code>Unique_Filtered_Blast_Hit_Info.csv</code> containing results of BLAST analysis and extract name of target scaffold(s).</p>
<pre class="r"><code>#Open BLAST file
cl.filt.unique &lt;- read.csv(file = &quot;Unique_Filtered_Blast_Hit_Info.csv&quot;) # Output of module 1

#Scaffold IDs
cl.filt.unique$SubjectID</code></pre>
<p>Extract scaffold(s) sequences from draft genome file (<code>FASTAs/Draft_Genome_Assembly.fasta</code>).</p>
<pre class="r"><code>#Open draft genome file
genome &lt;- readLines(&quot;FASTAs/Draft_Genome_Assembly.fasta&quot;)

#Check formatting of file
head(genome)

#How many scaffolds?
Nscaff &lt;- length(genome)/2

#Extract scaffolds in a loop

#Create data frame to store scaffold data
scaffold &lt;- data.frame(&quot;scaffoldID&quot; = character(length(cl.filt.unique$SubjectID)), &quot;scaffoldSeq&quot; = character(length(cl.filt.unique$SubjectID)))
#Start populating data frame
scaffold$scaffoldID &lt;- cl.filt.unique$SubjectID

for(i in 1:nrow(scaffold)){
  # Find and Extract seq of each scaffold from draft genome FASTA file
  scaffold$scaffoldSeq[i] &lt;- genome[match(paste0(&quot;&gt;&quot;,scaffold$scaffoldID[i]), genome)+1]
}

#Convert into FASTA format
scaffoldFASTA &lt;- apply(scaffold, 1, paste, collapse=&quot;\n&quot;)

#Save/export FASTA file
write.table(scaffoldFASTA, &quot;Output_FASTAs/Scaffold151535.fa&quot;, row.names = F, col.names=F, quote = F)</code></pre>
</div>
<div id="finding-orfs-along-scaffolds" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Finding ORFs along scaffold(s)</h2>
<p>We will now find ORFs in scaffold(s) using <em>findORFsTranslateDNA2AA()</em>. This user-defined function relies on the <em>ORFik</em> R package. The output will be saved in <code>ORFs_report/</code>.</p>
<pre class="r"><code>scaffold &lt;- readLines(&quot;Output_FASTAs/Scaffold151535.fa&quot;)
scaffoldID &lt;- grep(pattern = &quot;^&gt;&quot;, x = scaffold, value = T)
scaffoldID &lt;- gsub(pattern = &quot;&gt;&quot;, replacement = &quot;&quot;, x = scaffoldID)
tryCatch(
  {
    for(i in 1:length(scaffoldID)){
      findORFsTranslateDNA2AA(scaffold = scaffold, scaffoldID = scaffoldID[i], MinLen = 40)
    }
  })</code></pre>
<p>We should now have ORF reports and FASTA files of translated ORFs. Let’s check these outputs.</p>
<pre class="r"><code>orf.report &lt;- read.csv(&quot;ORFs_report/Scaffold151535_ORFs.csv&quot;)

#How many ORFs were found?
nrow(orf.report) 

#What are the longest ORFs?
orf.report[order(-orf.report$width),] 

#Look at translated ORFs (=AA sequences)
translated.orfs &lt;- readLines(&quot;AA_ORFs/Scaffold151535_ORFs.fa&quot;)
head(translated.orfs)</code></pre>
</div>
</div>
<div id="references" class="section level1" number="4">
<h1><span class="header-section-number">4</span> References</h1>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
