---
title: "BIOL 497/597 - Genomics & Bioinformatics"
subtitle: "Lab: Mining sagebrush draft genome"
author: "Sven Buerki & Anthony Melton - Boise State University"
date: "`r Sys.Date()`"
output:
  bookdown::html_document2: 
    toc: TRUE
    toc_float: TRUE
    toc_depth: 4
link-citations: yes
fontsize: 12pt
urlcolor: blue
csl: AmJBot.csl
bibliography: References.bib
editor_options: 
  markdown: 
    wrap: sentence
---

```{js logo-js, echo=FALSE}
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
```

# Introduction

This webpage supports analyses aiming at mining the sagebrush draft genome for Aquaporin genes. It is subdivided into 4 modules as follows:

- [**Module 1:**](#mod1) Set-up environment and conduct BLAST analysis.
- [**Module 2:**](#mod2) Extract scaffolds and find **O**pen **R**eading **F**rames.
- **Module 3:** Annotate ORFs and assemble proteins.
- **Module 4:** Alignment and phylogenetic reconstruction.

## Project structure

### Where are the data located?

All the data associated to this project are located on your individual account in the `DraftGenomeMineR/` folder. This folder is located in `/home/bio_11/DraftGenomeMineR`. Please replace `bio_11` by your [account detail](https://docs.google.com/document/d/1t0DdQBes42Gzo8XWgc9JL2xEGjv1kbfoVE7Vznx6C9I/edit).

**Key files:**

- `Draft_Genome_Assembly.fasta`: Sagebrush draft genome assembly (at scaffold level).
- `FASTAs/PIP1_3.fa`: Reference containing *PIP1* gene (AA) for BLAST analysis. This data will be used to find Aquaporin genes in the draft genome.
- Scripts related to each module are located in `Lesson_Modules/`.

### How do I access the data and run analyses?

To access the data and run the analyses, please remotely connect to your dedicated computer using `ssh` protocol. Your credentials are available [here](https://docs.google.com/document/d/1t0DdQBes42Gzo8XWgc9JL2xEGjv1kbfoVE7Vznx6C9I/edit).

# Module 1: Set-up environment and conduct BLAST analysis {#mod1}

## Remotely connect to computer

Start by using `ssh` protocol to connect to your Linux computer. This is done as follows (enter password when prompted to):

```{bash, eval=F}
#SSH with bio_11 as example
ssh bio_11@132.178.142.214
```

## Copy project folder to remote computer

After downloading the `zip` file from our shared Google Drive to your personal computer, please copy this file to your Linux computer (on your personal account) as follows:

```{bash, eval=F}
#scp file: general syntax
scp file user@host:dir

#Example
scp DraftGenomeMineR.zip bio_11@132.178.142.214:/home/bio_11

#unzip file
unzip DraftGenomeMineR.zip
```

**Disclaimer:** This procedure only works if you have a unix-based computer. If you don't have a unix-based computer, please ask instructors to transfer this file for you to your account. However, students can `cp` this file by copying it from the `bioinformatics` account onto their Linux computer as follows:

```{r, echo=F, eval=F}
#Navigate where the zip file should reside (adjust path to your account ID)
cd /home/bio_11

#Copy the zip file (containing all data for project) on your Linux account
cp /home/bioinformatics/GenomeMining_Bioinformatics_2021/DraftGenomeMineR.zip .

#unzip file
unzip DraftGenomeMineR.zip
```

## Start R session

Navigate to `DraftGenomeMineR/` and start a new `R` session:

```{bash, eval=F}
#Navigate to DraftGenomeMineR/
cd DraftGenomeMineR/

#Start R session
R-4
```

## Load R packages and user-defined functions

Create an object with all the required R packages and load them:

```{r, eval=F}
# This will make a list of packages that another function will use to make sure they are all installed and loaded.
list.of.packages <- c("ape",
                      "Biostrings",
                      "dplyr",
                      "FastaUtils",
                      "ORFik",
                      "readr",
                      "tidyr",
                      "rBLAST",
                      "seqinr",
                      "stringr")

# Use lapply to load all of the packages in the list.
lapply(list.of.packages, require, character.only = TRUE)
```

Load all the user-defined function located in `Functions/`:

```{r, eval=F}
# Load all of the functions written for DraftGenomeMineR     
files.sources <- list.files("Functions", full.names=T)
sapply(files.sources, source)
```

## Setting environment for BLAST analysis

To be able to navigate between folders within our project, we will set a object entitled `project.folder`, which contains the path to the root of the folder.

To create this object, do as follows:

```{r, eval=F}
#Copy the output of getwd() in the object as follows:
project.folder <- getwd()
```

Call `project.folder` to check that it is right. It should be:
`"/home/bio_11/DraftGenomeMineR"` (with `bio_11` replaced by your account ID).

Set working directory to `project.folder` as follows:

```{r, eval=F}
setwd(project.folder)
```

## Running BLAST analysis

Set general parameters for the BLAST analysis:

```{r, eval=F}
# There are a few parameters that need to be set. Let's create some R objects to store file paths and parameters for the BLAST search.

# We need a query (a fasta file of a gene you want to find in the draft genome), a draft genome assembly, BLAST databases, paths to other required folders (described in the README), and other parameters that can be used to filter results.
query.file.path <- "FASTAs/PIP1_3.fa"
genome.file.name <- "Draft_Genome_Assembly.fasta"
genome.path <- "FASTAs/Draft_Genome_Assembly.fasta"
blast.db.path <- "BlastDBs/Draft_Genome_Assembly.fasta"
AA.BlastDB.folder <- "AA_BlastDB/"
AA.ORF.folder <- "AA_ORFs/"
max.e <- 5E-50
perc.ident <- 90.000
query.type <- "AA"
blast.type <- "tblastn"
make.BlastDB <- T
BlastDB.type <- "nucl"
```

Read in the sagebrush draft genome assembly in `FASTA` format:

```{r, eval=F}
# Read in the genome assembly file
genome <- readLines(con = genome.path)
head(genome) # Print the top 6 lines of the fasta. There should be no spaces in what is printed. Each header should be on one line, followed by its entire scaffold on the next.
```

Make a BLAST database for query:

```{r, eval=F}
# Do you need to make a BLAST database or use an existing one?
if(make.BlastDB == TRUE){
  setwd("BlastDBs/")
  makeblastdb(file = genome.file.name, dbtype = BlastDB.type)
}
```

Read in the query file in `FASTA` format:

```{r, eval=F}
# Read in the query. What type of molecule is the query? DNA (or RNA) or amino acid sequences?
setwd(project.folder) # Previous lines changed the wd, so we need to go back to the project folder.
if (query.type == "DNA") {
  query <- readDNAStringSet(filepath = query.file.path,
                            format = "fasta")
} else {
  query <- readAAStringSet(filepath = query.file.path,
                           format = "fasta")
}
```

Perform BLAST analysis (`BLAST` needs to be installed on the computer):

```{r, eval=F}
# Now we have everything set up and can perform a BLAST search.
bl <- blast(db = blast.db.path, type = blast.type) # Create an object with the BLAST database and the type of BLAST search to perform.
cl <- predict(bl, query) # Perform the BLAST search
head(cl) # Look at the top BLAST hits
```

Filter output of BLAST analysis:

```{r, eval=F}
# We can now filter out bad BLAST hits using a few thresholds and parameters. This leaves us with only the best matches to our query.
# What parameters do you think would give you just the best candidates to be your gene of interest?
cl.filt <- subset(x = cl, Perc.Ident >= perc.ident & E <= max.e)
cl.filt.unique <- cl.filt[!duplicated(cl.filt[,c('SubjectID')]),] # SubjectID is the column that contains the scaffold names
cl.filt.unique
nrow(cl)
nrow(cl.filt)
nrow(cl.filt.unique)
blast.hits.to.extract <- subset(x = cl.filt.unique, SubjectID == cl.filt.unique[1,2]) 
```

Write output of BLAST analysis in `csv` file:

```{r, eval=F}
# We have evaluated the top BLAST hits and can see a clear best hit. Let's save this data so we can extract just that scaffold later.
write.csv(x = blast.hits.to.extract, file = "Unique_Filtered_Blast_Hit_Info.csv", row.names = F)
```


# Module 2: Extract scaffolds and find Open Reading Frames {#mod2}

In this module, we are conducting the following tasks:

- Extracting scaffold(s) identified in [module 1](#mod1) from the draft genome assembly (in `FASTAs/Draft_Genome_Assembly.fasta`) and saving the output as a `FASTA` object/file.
- Finding ORFs along scaffold(s) using the user-defined function *findORFsTranslateDNA2AA()* and saving the output into a `FASTA` object/file. In molecular genetics, an **O**pen **R**eading **F**rame (ORF) is the part of a reading frame that has the ability to be translated. An ORF is a continuous stretch of codons that begins with a start codon (usually AUG) and ends at a stop codon (usually UAA, UAG or UGA). We will study the code implemented in *findORFsTranslateDNA2AA()* to fully understand the applied approach. 

## Getting ready!

Before starting analyses in `R`, students have to complete these following tasks:

1. Remotely connect to their computers (using their individual accounts) with `ssh` protocol.
2. Navigate to project directory (`DraftGenomeMineR/`) using `cd`.
3. Create a new folder (`Output_FASTAs/`) using `mkdir` to store results of BLAST scaffold analysis.
4. Create a new folder (`ORFs_report/`) using `mkdir` to store results of ORFs analysis.
5. Start a new `R` session using `R-4` command.
6. Load R packages and user-defined functions.
7. Set working directory to `DraftGenomeMineR/`.

For an example of code, see below:

```{r, eval=F}
#1. ssh
ssh svenbuerki@132.178.142.214

#scp file
#scp DraftGenomeMineR.zip svenbuerki@132.178.142.214:/home/svenbuerki

#unzip file
#unzip DraftGenomeMineR.zip

#2. Navigate to DraftGenomeMineR/
cd DraftGenomeMineR/

#3. & 4. Create new folders 
mkdir Output_FASTAs/
mkdir ORFs_report/
  
#5. Start R session
R-4

#6. This will make a list of packages that another function will use to make sure they are all installed and loaded.
list.of.packages <- c("ape",
                      "Biostrings",
                      "dplyr",
                      "FastaUtils",
                      "ORFik",
                      "readr",
                      "tidyr",
                      "rBLAST",
                      "seqinr",
                      "stringr")

# Use lapply to load all of the packages in the list.
lapply(list.of.packages, require, character.only = TRUE)

# Load all of the functions written for DraftGenomeMineR     
files.sources <- list.files("Functions", full.names=T)
sapply(files.sources, source)

#7. Copy the output of getwd() in the object as follows:
project.folder <- getwd()

#Set wd
setwd(project.folder)
```

## Extracting scaffold(s) identified in module 1

Open `Unique_Filtered_Blast_Hit_Info.csv` containing results of BLAST analysis and extract name of target scaffold(s).

```{r, eval=F}
#Open BLAST file
cl.filt.unique <- read.csv(file = "Unique_Filtered_Blast_Hit_Info.csv") # Output of module 1

#Scaffold IDs
cl.filt.unique$SubjectID
```

Extract scaffold(s) sequences from draft genome file (`FASTAs/Draft_Genome_Assembly.fasta`).

```{r, eval=F}
#Open draft genome file
genome <- readLines("FASTAs/Draft_Genome_Assembly.fasta")

#Check formatting of file
head(genome)

#How many scaffolds?
Nscaff <- length(genome)/2

#Extract scaffolds in a loop

#Create data frame to store scaffold data
scaffold <- data.frame("scaffoldID" = character(length(cl.filt.unique$SubjectID)), "scaffoldSeq" = character(length(cl.filt.unique$SubjectID)))
#Start populating data frame
scaffold$scaffoldID <- cl.filt.unique$SubjectID

for(i in 1:nrow(scaffold)){
  # Find and Extract seq of each scaffold from draft genome FASTA file
  scaffold$scaffoldSeq[i] <- genome[match(paste0(">",scaffold$scaffoldID[i]), genome)+1]
}

#Convert into FASTA format
scaffoldFASTA <- apply(scaffold, 1, paste, collapse="\n")

#Save/export FASTA file
write.table(scaffoldFASTA, "Output_FASTAs/Scaffold151535.fa", row.names = F, col.names=F, quote = F)
```

## Finding ORFs along scaffold(s)

We will now find ORFs in scaffold(s) using *findORFsTranslateDNA2AA()*. This user-defined function relies on the *ORFik* R package. The output will be saved in `ORFs_report/`.

```{r, eval=F}
scaffold <- readLines("Output_FASTAs/Scaffold151535.fa")
scaffoldID <- grep(pattern = "^>", x = scaffold, value = T)
scaffoldID <- gsub(pattern = ">", replacement = "", x = scaffoldID)
tryCatch(
  {
    for(i in 1:length(scaffoldID)){
      findORFsTranslateDNA2AA(scaffold = scaffold, scaffoldID = scaffoldID[i], MinLen = 40)
    }
  })
```

We should now have ORFs reports and `FASTA` files of translated ORFs. Let's check these outputs.

```{r, eval=F}
orf.report <- read.csv("ORFs_report/Scaffold151535_ORFs.csv")

#How many ORFs were found?
nrow(orf.report) 

#What are the longest ORFs?
orf.report[order(-orf.report$width),] 

#Look at translated ORFs (=AA sequences)
translated.orfs <- readLines("AA_ORFs/Scaffold151535_ORFs.fa")
head(translated.orfs)
```

# Module 3: Annotate ORFs and assemble proteins

# Module 4: Alignment and phylogenetic reconstruction

# References