---
title: "BIOL 497/597 - Genomics & Bioinformatics"
subtitle: "Mini-reports"
author: "Sven Buerki - Boise State University"
date: "`r Sys.Date()`"
output:
  bookdown::html_document2: 
    toc: TRUE
    toc_float: TRUE
link-citations: yes
fontsize: 12pt
urlcolor: blue
csl: AmJBot.csl
bibliography: References.bib
editor_options: 
  markdown: 
    wrap: sentence
---

```{js logo-js, echo=FALSE}
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
```

# Introduction

To further gain expertise in the field of genomics, students are producing three mini-reports on the following topics:

- [**Mini-Report 1**](#mini1): Sequencing technologies (25 points; this task is related to Chapter 2).
- [**Mini-Report 2**](#mini2): Molecular biology databases (25 points; this task is related to Chapter 3).
- **Mini-Report 3**: Use knowledge from the previous mini-reports to review plant genomes available on GenBank database (50 points). Here, we will be using `R` to mine and analyze data.

Time will be allocated in class to work on these mini-reports, but the instructor expects students to complete those on their own time.

# Mini-report 1 -- Sequencing platforms & technologies {#mini1}

To have a full overview of available sequencing platforms, students are tasked to produce individual mini-reports on the following sequencing platforms and their associated technologies:

- **Sequencing platform 1:** Illumina.
- **Sequencing platform 2:** PacBio.
- **Sequencing platform 3:** Oxford Nanopore.

## Background information

Each student was assigned a sequencing platform to work on (please see [Google spreadsheet](https://docs.google.com/spreadsheets/d/1cRM1yasUfDa98ecyz_-RxUOw8cU4wkQx4O5Wxw3y4yU/edit?usp=sharing) for more information). 

The assignment is mandatory and will be graded. Please see this task as an opportunity to consolidate your knowledge on next-generation sequencing technologies and writing scientific reports.   

Time will be allocated in class on the 27th of January 2021 to allow students to work on their assignments. Reports are due on **the 5th of February 2021** and should be uploaded on the shared [Google Drive](https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing) in the `Mini_Reports` folder.


## Structure of mini-reports {#structure}

Please structure your reports to ensure covering the following aspects:

- Introduce the sequencing technology (incl. library preparation and DNA requirements).
- Provide an overview of available sequencing systems.
- Outline the research potential/applicability of the sequencing platform. 
- Provide any additional information such as e.g. price, accessibility, technical support, that you feel are relevant to your sequencing platform. 

## Reporting format

Maximum two pages, single spaced, using either Arial 11 pt or Times New Roman 12 pt and with all margins set to 1 inch. The two pages do not include a title page and references. The title page should include: name of the sequencing platform as title and name of student contributing the report. Please keep in mind that a figure can convey a lot more information than a long text! Your reports should follow the structure presented in section \@ref(structure). 

Upload your reports on the shared Google drive by the 5th of February before 5 pm. Name your report documents following this pattern: `Sequencing platform_Surname`.

## Resources

The instructor has provided `pdf` documents devoted to each sequencing platform provided below to support students in starting their assignments. 

In addition, please use information available from manufacturers' websites, publications, online NGS training sites (e.g. EMBL-EBI Training), Google, YouTube and Wikipedia to prepare your reports and presentations. Do not forget to provide citations supporting your claims in your reports. I let you decide which reference style you want to use (look at your favorite journals for examples).  

<button class="btn btn-primary" data-toggle="collapse" data-target="#Illumina">Data on Illumina</button>
<div id="Illumina" class="collapse">    
<object data="Images/Seq_1_Illumina.pdf" width="750px" height="750px">
    <embed src="Images/Seq_1_Illumina.pdf">
    </embed>
</object>
</div>

<button class="btn btn-primary" data-toggle="collapse" data-target="#PacBio">Data on PacBio</button>
<div id="PacBio" class="collapse">    
<object data="Images/Seq_2_PacBio.pdf" width="750px" height="750px">
    <embed src="Images/Seq_2_PacBio.pdf">
    </embed>
</object>
</div>

<button class="btn btn-primary" data-toggle="collapse" data-target="#Nanopore">Data on Nanopore</button>
<div id="Nanopore" class="collapse">    
<object data="Images/Seq_3_Oxford_Nanopore.pdf" width="750px" height="750px">
    <embed src="Images/Seq_3_Oxford_Nanopore.pdf">
    </embed>
</object>
</div>



# Mini-report 2 -- Molecular biology databases {#mini2}

To have a full overview of available molecular biology databases and assess their contributions to genome annotation, students are tasked to produce individual mini-reports on the following molecular databases:

- **Molecular database 1:** [Protein sequences databases](#prot).
- **Molecular database 2:** [Gene ontology databases](#GO).
- **Molecular database 3:** [Metabolic pathways databases](#KEGG).

## Background information

Each student was assigned a molecular database to work on (please see [Google spreadsheet](https://docs.google.com/spreadsheets/d/1cRM1yasUfDa98ecyz_-RxUOw8cU4wkQx4O5Wxw3y4yU/edit?usp=sharing) for more information). 

The assignment is mandatory and will be graded. Please see this task as an opportunity to consolidate your knowledge on molecular databases and writing scientific reports.   

Time will be allocated in class on the 10th of February 2021 to allow students to work on their assignments. Reports are due on **the 19th of February 2021** and should be uploaded on the shared [Google Drive](https://drive.google.com/drive/folders/1BFZFJeR8tW8N7QLfngqIVbjqEkDuIyWj?usp=sharing) in the `Mini_Reports` folder.

## Structure of mini-reports {#structure2}

Please structure your reports to cover the following aspects:

- Provide overview of available databases falling under your major type of molecular biology data.
- Highlight potential contributions of these databases in supporting annotation of genomes.
- Provide a survey of available bioinformatic tools and websites to query/access these databases. 
- Complement reports with information relevant/specific to your data repositories. 

## Reporting format

Maximum two pages, single spaced, using either Arial 11 pt or Times New Roman 12 pt and with all margins set to 1 inch. The two pages do not include a title page and references. The title page should include: name of the sequencing platform as title and name of student contributing the report. Please keep in mind that a figure can convey a lot more information than a long text! Your reports should follow the structure presented in section \@ref(structure2). 

Upload your reports on the shared Google drive by the 19th of February before 5 pm. Name your report documents following this pattern: `Molecular database_Surname`.

## Resources

The instructor provides below short descriptions of each of the major types of molecular biology databases. Use the information provided here to prime your research and carve your reports. Please do not forget to provide citations and references in your reports. The instructor let you decide which reference style/formatting you prefer using (look at your favorite journals for examples).

### Proteins sequences databases {#prot}

There are three major protein sequences databases:

- **PIR** (Protein Information Resource in Washington, USA): [http://pir.georgetown.edu](http://pir.georgetown.edu)
- **SWISS-PROT** (in Geneva, Switzerland): [http://www.uniprot.org/statistics/Swiss-Prot](http://www.uniprot.org/statistics/Swiss-Prot)
- **TrEMBL** (in Hinxton, UK): [http://www.uniprot.org/statistics/TrEMBL](http://www.uniprot.org/statistics/TrEMBL)  

In 2002, these three databases coordinated their efforts to form the **UniProt** consortium. The partners in this enterprise share the database, but continue to offer separate information-retrieval tools for access. The UniProtKB database is available at this URL: [http://www.uniprot.org](http://www.uniprot.org). This latter database will be especially important to retrieve information on gene annotations (chapters 6, 7) using the online tool available at this URL: [http://www.uniprot.org/uploadlists/](http://www.uniprot.org/uploadlists/). This will be done by specifically looking at Gene ontology (see [GeneOntology](#GO)).

### Gene ontology databases {#GO}

Gene Ontology (GO) ([http://www.geneontology.org](http://www.geneontology.org)): the framework for the model of biology. The GO defines concepts/classes used to describe gene function, and relationships between these concepts. It classifies functions along three aspects:

- **Cellular component:** This term describes a component of a cell that is part of a larger object, such as an anatomical structure (e.g. rough endoplasmic reticulum or nucleus) or a gene product group (e.g. ribosome, proteasome or a protein dimer).
- **Biological process:** A biological process term describes a series of events accomplished by one or more organized assemblies of molecular functions. Examples of broad biological process terms are "cellular physiological process" or "signal transduction". Examples of more specific terms are "pyrimidine metabolic process" or "alpha-glucoside transport". The general rule to assist in distinguishing between a biological process and a molecular function is that a process must have more than one distinct steps.
A biological process is not equivalent to a pathway. At present, the GO does not try to represent the dynamics or dependencies that would be required to fully describe a pathway.
- **Molecular function:** Molecular function terms are describing activities occurring at the molecular level, such as "catalytic activity" or "binding activity". GO molecular function terms represent activities rather than the entities (molecules or complexes) that perform the actions, and do not specify where, when, or in what context the action takes place. Molecular functions generally correspond to activities that can be performed by individual gene products, but some activities are performed by assembled complexes of gene products. Examples of broad functional terms are "catalytic activity" and "transporter activity"; examples of narrower functional terms are "adenylate cyclase activity" or "toll receptor binding".
It is easy to confuse a gene product name with its molecular function; for that reason GO molecular functions are often appended with the word "activity".

GO is also available for all the protein sequences deposited on the UniProt database. 
Other useful tool to perform gene annotations: \url{http://amigo.geneontology.org/amigo/search/annotation} 

### Metabolic pathways databases {#KEGG}

The Kyoto Encyclopedia of Genes and Genomes (KEGG) ([http://www.genome.jp/kegg/](http://www.genome.jp/kegg/)) collects individual genomes, gene products, and their functions, but its special strength lies in its integration of biochemical and genetic information. KEGG focuses on interactions: molecular assemblies, and metabolic and regulatory networks. KEGG organizes five data types into a comprehensive system:

- Catalogs of chemical compounds in living cells.
- Gene catalogs.
- Genome maps.
- Pathway maps.
- Orthologue tables.

The catalogs of chemical compounds and genes contain information about particular molecules or sequences. Genome maps integrate the genes themselves according to their chromosomal position. Pathway maps describe potential networks of molecular activities, both metabolic and regulatory. One enzyme in one organism would be referred to in KEGG in its orthologue tables, which link the enzyme to related ones in other organisms. This permits analysis of relationships between the metabolic pathways of different organisms. 
