<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-02-17" />

<title>Documentation on R functions</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.24/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.11.3/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Timetable.html">Timetable</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="R_Functions.html">R functions</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Documentation on R functions</h1>
<h4 class="date">2023-02-17</h4>

</div>


<p>The table below contains details on R functions used in this course. Please use the <code>Search</code> field to make a query and learn more about a function.</p>
<div id="htmlwidget-6ea8a791aeb742c32a0a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6ea8a791aeb742c32a0a">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["-","-.Date","-.POSIXt",":","::",":::","!","!.hexmode","!.octmode","!=","(","[","[.AsIs","[.data.frame","[.Date","[.difftime","[.Dlist","[.DLLInfoList","[.factor","[.hexmode","[.listof","[.noquote","[.numeric_version","[.octmode","[.POSIXct","[.POSIXlt","[.simple.list","[.table","[.warnings","[[","[[.data.frame","[[.Date","[[.factor","[[.numeric_version","[[.POSIXct","[[.POSIXlt","{","@","*","*.difftime","/","/.difftime","&amp;","&amp;.hexmode","&amp;.octmode","&amp;&amp;","%*%","%/%","%%","%in%","%o%","%x%","^","+","+.Date","+.POSIXt","&lt;","&lt;=","=","==","&gt;","&gt;=","|","|.hexmode","|.octmode","||","~","$","$.DLLInfo","$.package_version","abbreviate","abs","acos","acosh","addNA","addTaskCallback","agrep","agrepl","alist","all","all.equal","all.equal.character","all.equal.default","all.equal.environment","all.equal.envRefClass","all.equal.factor","all.equal.formula","all.equal.language","all.equal.list","all.equal.numeric","all.equal.POSIXt","all.equal.raw","all.names","all.vars","allowInterrupts","any","anyDuplicated","anyDuplicated.array","anyDuplicated.data.frame","anyDuplicated.default","anyDuplicated.matrix","anyNA","anyNA.numeric_version","anyNA.POSIXlt","aperm","aperm.default","aperm.table","append","apply","Arg","args","array","arrayInd","as.array","as.array.default","as.call","as.character","as.character.condition","as.character.Date","as.character.default","as.character.error","as.character.factor","as.character.hexmode","as.character.numeric_version","as.character.octmode","as.character.POSIXt","as.character.srcref","as.complex","as.data.frame","as.data.frame.array","as.data.frame.AsIs","as.data.frame.character","as.data.frame.complex","as.data.frame.data.frame","as.data.frame.Date","as.data.frame.default","as.data.frame.difftime","as.data.frame.factor","as.data.frame.integer","as.data.frame.list","as.data.frame.logical","as.data.frame.matrix","as.data.frame.model.matrix","as.data.frame.noquote","as.data.frame.numeric","as.data.frame.numeric_version","as.data.frame.ordered","as.data.frame.POSIXct","as.data.frame.POSIXlt","as.data.frame.raw","as.data.frame.table","as.data.frame.ts","as.data.frame.vector","as.Date","as.Date.character","as.Date.default","as.Date.factor","as.Date.numeric","as.Date.POSIXct","as.Date.POSIXlt","as.difftime","as.double","as.double.difftime","as.double.POSIXlt","as.environment","as.expression","as.expression.default","as.factor","as.function","as.function.default","as.hexmode","as.integer","as.list","as.list.data.frame","as.list.Date","as.list.default","as.list.environment","as.list.factor","as.list.function","as.list.numeric_version","as.list.POSIXct","as.list.POSIXlt","as.logical","as.logical.factor","as.matrix","as.matrix.data.frame","as.matrix.default","as.matrix.noquote","as.matrix.POSIXlt","as.name","as.null","as.null.default","as.numeric","as.numeric_version","as.octmode","as.ordered","as.package_version","as.pairlist","as.POSIXct","as.POSIXct.Date","as.POSIXct.default","as.POSIXct.numeric","as.POSIXct.POSIXlt","as.POSIXlt","as.POSIXlt.character","as.POSIXlt.Date","as.POSIXlt.default","as.POSIXlt.factor","as.POSIXlt.numeric","as.POSIXlt.POSIXct","as.qr","as.raw","as.single","as.single.default","as.symbol","as.table","as.table.default","as.vector","as.vector.factor","asin","asinh","asNamespace","asplit","asS3","asS4","assign","atan","atan2","atanh","attach","attachNamespace","attr","attr.all.equal","attributes","autoload","autoloader","backsolve","baseenv","basename","besselI","besselJ","besselK","besselY","beta","bindingIsActive","bindingIsLocked","bindtextdomain","bitwAnd","bitwNot","bitwOr","bitwShiftL","bitwShiftR","bitwXor","body","bquote","break","browser","browserCondition","browserSetDebug","browserText","builtins","by","by.data.frame","by.default","bzfile","c","c.Date","c.difftime","c.noquote","c.numeric_version","c.POSIXct","c.POSIXlt","c.warnings","call","callCC","capabilities","casefold","cat","cbind","cbind.data.frame","ceiling","char.expand","character","charmatch","charToRaw","chartr","check_tzones","chkDots","chol","chol.default","chol2inv","choose","class","clearPushBack","close","close.connection","close.srcfile","close.srcfilealias","closeAllConnections","col","colMeans","colnames","colSums","commandArgs","comment","complex","computeRestarts","conditionCall","conditionCall.condition","conditionMessage","conditionMessage.condition","conflictRules","conflicts","Conj","contributors","cos","cosh","cospi","crossprod","Cstack_info","cummax","cummin","cumprod","cumsum","curlGetHeaders","cut","cut.Date","cut.default","cut.POSIXt","data.class","data.frame","data.matrix","date","debug","debuggingState","debugonce","default.stringsAsFactors","delayedAssign","deparse","det","detach","determinant","determinant.matrix","dget","diag","diff","diff.Date","diff.default","diff.difftime","diff.POSIXt","difftime","digamma","dim","dim.data.frame","dimnames","dimnames.data.frame","dir","dir.create","dir.exists","dirname","do.call","dontCheck","double","dput","dQuote","drop","droplevels","droplevels.data.frame","droplevels.factor","dump","duplicated","duplicated.array","duplicated.data.frame","duplicated.default","duplicated.matrix","duplicated.numeric_version","duplicated.POSIXlt","duplicated.warnings","dyn.load","dyn.unload","dynGet","eapply","eigen","emptyenv","enc2native","enc2utf8","encodeString","Encoding","endsWith","enquote","env.profile","environment","environmentIsLocked","environmentName","errorCondition","eval","eval.parent","evalq","exists","exp","expand.grid","expm1","expression","extSoftVersion","F","factor","factorial","fifo","file","file.access","file.append","file.choose","file.copy","file.create","file.exists","file.info","file.link","file.mode","file.mtime","file.path","file.remove","file.rename","file.show","file.size","file.symlink","Filter","Find","find.package","findInterval","findPackageEnv","findRestart","floor","flush","flush.connection","for","force","forceAndCall","formals&lt;-","format","format.AsIs","format.data.frame","format.Date","format.default","format.difftime","format.factor","format.hexmode","format.info","format.libraryIQR","format.numeric_version","format.octmode","format.packageInfo","format.POSIXct","format.POSIXlt","format.pval","format.summaryDefault","formatC","formatDL","forwardsolve","function","gamma","gc","gc.time","gcinfo","gctorture","gctorture2","get","get0","getAllConnections","getCallingDLL","getCallingDLLe","getConnection","getDLLRegisteredRoutines","getDLLRegisteredRoutines.character","getDLLRegisteredRoutines.DLLInfo","getElement","geterrmessage","getExportedValue","getHook","getLoadedDLLs","getNamespace","getNamespaceExports","getNamespaceImports","getNamespaceInfo","getNamespaceName","getNamespaceUsers","getNamespaceVersion","getNativeSymbolInfo","getOption","getRversion","getSrcLines","getTaskCallbackNames","gettext","gettextf","getwd","gl","globalenv","gregexpr","grep","grepl","grepRaw","grouping","gsub","gzcon","gzfile","I","iconv","iconvlist","icuGetCollate","icuSetCollate","identical","identity","if","ifelse","Im","importIntoEnv","inherits","integer","interaction","interactive","intersect","intToBits","intToUtf8","inverse.rle","invisible","invokeRestart","invokeRestartInteractively","is.array","is.atomic","is.call","is.character","is.complex","is.data.frame","is.double","is.element","is.environment","is.expression","is.factor","is.finite","is.function","is.infinite","is.integer","is.language","is.list","is.loaded","is.logical","is.matrix","is.na","is.na.data.frame","is.na.numeric_version","is.na&lt;-.numeric_version","is.name","is.nan","is.null","is.numeric","is.numeric_version","is.numeric.Date","is.numeric.difftime","is.numeric.POSIXt","is.object","is.ordered","is.package_version","is.pairlist","is.primitive","is.qr","is.R","is.raw","is.recursive","is.single","is.symbol","is.table","is.unsorted","is.vector","isatty","isBaseNamespace","isdebugged","isFALSE","isIncomplete","isNamespace","isNamespaceLoaded","ISOdate","ISOdatetime","isOpen","isRestart","isS4","isSeekable","isSymmetric","isSymmetric.matrix","isTRUE","jitter","julian","julian.Date","julian.POSIXt","kappa","kappa.default","kappa.lm","kappa.qr","kronecker","l10n_info","La_library","La_version","La.svd","labels","labels.default","lapply","last.warning","lazyLoad","lazyLoadDBexec","lazyLoadDBfetch","lbeta","lchoose","length&lt;-.POSIXct","length&lt;-.POSIXlt","lengths","levels","levels.default","levels&lt;-","levels&lt;-.factor","lfactorial","lgamma","libcurlVersion","library","library.dynam","library.dynam.unload","licence","license","list","list.dirs","list.files","list2env","load","loadedNamespaces","loadingNamespaceInfo","loadNamespace","local","lockBinding","lockEnvironment","log","log10","log1p","log2","logb","logical","lower.tri","ls","make.names","make.unique","makeActiveBinding","Map","mapply","margin.table","mat.or.vec","match","match.arg","match.call","match.fun","Math.data.frame","Math.Date","Math.difftime","Math.factor","Math.POSIXt","matrix","max","max.col","mean","mean.Date","mean.default","mean.difftime","mean.POSIXct","mean.POSIXlt","mem.limits","mem.maxNSize","mem.maxVSize","memCompress","memDecompress","memory.profile","merge","merge.data.frame","merge.default","message","mget","missing","Mod","mode","month.abb","month.name","months.POSIXt","mostattributes&lt;-","names","names.POSIXlt","names&lt;-","names&lt;-.POSIXlt","namespaceExport","namespaceImport","namespaceImportClasses","namespaceImportFrom","namespaceImportMethods","nargs","nchar","ncol","NCOL","Negate","new.env","next","NextMethod","ngettext","nlevels","noquote","norm","normalizePath","nrow","NROW","numeric","numeric_version","nzchar","objects","oldClass","oldClass&lt;-","OlsonNames","on.exit","open","open.connection","open.srcfile","open.srcfilealias","open.srcfilecopy","Ops.data.frame","Ops.Date","Ops.difftime","Ops.factor","Ops.numeric_version","Ops.ordered","Ops.POSIXt","options","order","ordered","outer","package_version","packageEvent","packageNotFoundError","packageStartupMessage","packBits","pairlist","parent.env","parent.env&lt;-","parent.frame","parse","parseNamespaceFile","paste","paste0","path.expand","path.package","pcre_config","pi","pipe","pmatch","pmax","pmax.int","pmin","pmin.int","polyroot","pos.to.env","Position","pretty","pretty.default","prettyNum","print","print.AsIs","print.by","print.condition","print.connection","print.data.frame","print.Date","print.default","print.difftime","print.Dlist","print.DLLInfo","print.DLLInfoList","print.DLLRegisteredRoutines","print.eigen","print.factor","print.function","print.hexmode","print.libraryIQR","print.listof","print.NativeRoutineList","print.noquote","print.numeric_version","print.octmode","print.packageInfo","print.POSIXct","print.POSIXlt","print.proc_time","print.restart","print.rle","print.simple.list","print.srcfile","print.srcref","print.summary.table","print.summary.warnings","print.summaryDefault","print.table","print.warnings","prmatrix","proc.time","prod","prop.table","provideDimnames","psigamma","pushBack","pushBackLength","q","qr","qr.coef","qr.default","qr.fitted","qr.Q","qr.qty","qr.qy","qr.R","qr.resid","qr.solve","qr.X","quarters","quarters.Date","quarters.POSIXt","quit","quote","R_system_version","R.home","R.version","R.Version","R.version.string","range","range.default","rank","rapply","raw","rawConnection","rawConnectionValue","rawShift","rawToBits","rawToChar","rbind","rbind.data.frame","rcond","Re","read.dcf","readBin","readChar","readline","readLines","readRDS","Recall","Reduce","reg.finalizer","regexec","regexpr","registerS3method","registerS3methods","regmatches","regmatches&lt;-","remove","removeTaskCallback","rep","rep_len","rep.Date","rep.factor","rep.int","rep.numeric_version","rep.POSIXct","rep.POSIXlt","repeat","replace","replicate","require","requireNamespace","restartDescription","restartFormals","retracemem","return","returnValue","rev","rev.default","rle","rm","round.Date","round.POSIXt","row.names","row.names.data.frame","row.names.default","row.names&lt;-","row.names&lt;-.data.frame","row.names&lt;-.default","rowMeans","rownames","rownames&lt;-","rowsum","rowsum.data.frame","rowsum.default","rowSums","sample","sample.int","sapply","save","save.image","saveRDS","scale","scale.default","scan","search","searchpaths","seek","seek.connection","seq","seq_along","seq_len","seq.Date","seq.default","seq.int","seq.POSIXt","sequence","serialize","set.seed","setdiff","setequal","setHook","setNamespaceInfo","setSessionTimeLimit","setTimeLimit","setwd","showConnections","shQuote","sign","signalCondition","signif","simpleCondition","simpleError","simpleMessage","simpleWarning","simplify2array","sin","single","sinh","sink","sink.number","sinpi","slice.index","socketConnection","socketSelect","solve","solve.default","solve.qr","sort","sort.default","source","split","split.data.frame","split.Date","split.default","split.POSIXct","split&lt;-","split&lt;-.data.frame","split&lt;-.default","sprintf","sqrt","sQuote","srcfile","srcfilealias","srcfilecopy","standardGeneric","startsWith","stderr","stdin","stdout","stop","stopifnot","storage.mode","storage.mode&lt;-","str2expression","str2lang","strftime","strptime","strrep","strsplit","strtoi","strtrim","strwrap","subset","subset.data.frame","subset.default","subset.matrix","substitute","substr","substr&lt;-","substring","substring&lt;-","sum","summary","summary.connection","summary.data.frame","Summary.data.frame","summary.Date","Summary.Date","summary.default","Summary.difftime","summary.factor","Summary.factor","summary.matrix","Summary.numeric_version","Summary.ordered","summary.POSIXct","Summary.POSIXct","summary.POSIXlt","Summary.POSIXlt","summary.proc_time","summary.srcfile","summary.srcref","summary.table","summary.warnings","suppressMessages","suppressPackageStartupMessages","suppressWarnings","suspendInterrupts","svd","sweep","switch","sys.call","sys.calls","Sys.chmod","Sys.Date","sys.frame","sys.frames","sys.function","Sys.getenv","Sys.getlocale","Sys.getpid","Sys.glob","Sys.info","sys.load.image","Sys.localeconv","sys.nframe","sys.on.exit","sys.parent","sys.parents","Sys.readlink","sys.save.image","Sys.setenv","Sys.setFileTime","Sys.setlocale","Sys.sleep","sys.source","sys.status","Sys.time","Sys.timezone","Sys.umask","Sys.unsetenv","Sys.which","system","system.file","system.time","system2","t","T","t.data.frame","t.default","table","tabulate","tan","tanh","tanpi","tapply","taskCallbackManager","tcrossprod","tempdir","tempfile","textConnection","textConnectionValue","tolower","topenv","toString","toString.default","toupper","trace","traceback","tracemem","tracingState","transform","transform.data.frame","transform.default","trigamma","trimws","trunc","trunc.Date","trunc.POSIXt","truncate","truncate.connection","try","tryCatch","typeof","unclass","undebug","union","unique.data.frame","unique.default","unique.matrix","unique.numeric_version","unique.POSIXlt","unique.warnings","units","units.difftime","units&lt;-","units&lt;-.difftime","unix.time","unlink","unlist","unloadNamespace","unlockBinding","unname","unserialize","unsplit","untrace","untracemem","unz","upper.tri","url","UseMethod","utf8ToInt","validEnc","validUTF8","vapply","vector","Vectorize","version","warning","warningCondition","warnings","weekdays","weekdays.Date","weekdays.POSIXt","which","which.max","which.min","while","with","with.default","withAutoprint","withCallingHandlers","within","within.data.frame","within.list","withRestarts","withVisible","write","write.dcf","writeBin","writeChar","writeLines","xor","xpdrows.data.frame","xtfrm","xtfrm.AsIs","xtfrm.Date","xtfrm.default","xtfrm.difftime","xtfrm.factor","xtfrm.numeric_version","xtfrm.POSIXct","xtfrm.POSIXlt","xzfile","zapsmall","?","adist","alarm","apropos","aregexec","argsAnywhere","as.person","as.personList","as.relistable","as.roman","asDateBuilt","askYesNo","aspell","aspell_package_C_files","aspell_package_R_files","aspell_package_Rd_files","aspell_package_vignettes","aspell_write_personal_dictionary_file","assignInMyNamespace","assignInNamespace","available.packages","bibentry","browseEnv","browseURL","browseVignettes","bug.report","capture.output","changedFiles","checkCRAN","chooseBioCmirror","chooseCRANmirror","citation","cite","citeNatbib","citEntry","citFooter","de.ncols","de.restore","debugcall","debugger","demo","download.file","download.packages","dump.frames","edit","emacs","example","file_test","file.edit","fileSnapshot","find","findLineNum","fix","fixInNamespace","flush.console","formatOL","formatUL","getFromNamespace","getParseData","getParseText","getS3method","getSrcDirectory","getSrcFilename","getSrcLocation","getSrcref","getTxtProgressBar","glob2rx","globalVariables","hasName","head","help.request","help.search","help.start","history","hsearch_db","hsearch_db_concepts","hsearch_db_keywords","install.packages","installed.packages","is.relistable","isS3method","isS3stdGeneric","limitedLabels","loadhistory","localeToCharset","ls.str","lsf.str","maintainer","make.packages.html","make.socket","makeRweaveLatexCodeRunner","memory.limit","memory.size","menu","methods","mirror2html","modifyList","new.packages","news","nsl","object.size","old.packages","osVersion","package.skeleton","packageDate","packageDescription","packageName","packageStatus","packageVersion","page","person","personList","pico","process.events","prompt","promptData","promptImport","promptPackage","rc.getOption","rc.options","rc.settings","rc.status","read.csv","read.csv2","read.delim","read.delim2","read.DIF","read.fortran","read.fwf","read.socket","read.table","readCitationFile","recover","relist","remove.packages","removeSource","Rprof","Rprofmem","RShowDoc","RSiteSearch","rtags","Rtangle","RtangleSetup","RtangleWritedoc","RweaveChunkPrefix","RweaveEvalWithOpt","RweaveLatex","RweaveLatexFinish","RweaveLatexOptions","RweaveLatexSetup","RweaveLatexWritedoc","RweaveTryStop","savehistory","select.list","sessionInfo","setBreakpoint","setRepositories","setTxtProgressBar","stack","Stangle","str","strcapture","strOptions","summaryRprof","suppressForeignCheck","Sweave","SweaveHooks","SweaveSyntaxLatex","SweaveSyntaxNoweb","SweaveSyntConv","tail","tail.matrix","tar","timestamp","toBibtex","toLatex","txtProgressBar","type.convert","undebugcall","unstack","untar","unzip","update.packages","upgrade","url.show","URLdecode","URLencode","vi","View","vignette","warnErrList","write.csv","write.csv2","write.socket","write.table","xedit","xemacs","zip","[.DNAbin","AAsubst","abbreviateGenus","ace","add.scale.bar","additive","alex","all.equal.DNAbin","all.equal.phylo","alview","arecompatible","as.AAbin","as.AAbin.character","as.alignment","as.bitsplits","as.bitsplits.prop.part","as.character.AAbin","as.character.DNAbin","as.DNAbin","as.DNAbin.alignment","as.DNAbin.character","as.DNAbin.list","as.evonet","as.evonet.phylo","as.hclust.phylo","as.igraph.evonet","as.igraph.phylo","as.list.AAbin","as.list.DNAbin","as.matching","as.matching.phylo","as.matrix.DNAbin","as.network.evonet","as.network.phylo","as.networx.evonet","as.phyDat.AAbin","binaryPGLMM","binaryPGLMM.sim","bionj","bionjs","biplot.pcoa","birthdeath","bitsplits","boot.phylo","BOTHlabels","branching.times","bydir","c.DNAbin","CADM.global","CADM.post","cbind.DNAbin","checkAlignment","checkLabel","checkValidPhylo","cherry","chronoMPL","chronopl","circular.plot","cladewise","cladogram.plot","clustal","clustalomega","coalescent.intervals","collapse.singles","collapsed.intervals","compar.cheverud","compar.gee","compar.lynch","compar.ou","comparePhylo","compute.brtime","consensus","cophenetic.phylo","cophyloplot","corBlomberg","corBrownian","corGrafen","corMartins","corPagel","corphylo","correlogram.formula","countBipartitions","dbd","dbdTime","def","del.colgapsonly","del.gaps","del.rowgapsonly","delta.plot","deviance.ace","di2multi","di2multi.multiPhylo","di2multi.phylo","dist.aa","dist.dna","dist.gene","dist.nodes","dist.topo","diversi.gof","diversi.time","diversity.contrast.test","DNAbin2indel","dnds","drawSupportOnEdges","drop.fossil","drop.tip","dyule","edgelabels","edges","editFileExtensions","estimate.dates","estimate.mu","evonet","ewLasso","extract.clade","extract.popsize","fancyarrows","fastme.bal","fastme.ols","find.skyline.epsilon","floating.pie.asp","Ftab","gammaStat","GC.content","getAnnotationsGenBank","getMRCA","has.singles","howmanytrees","image.AAbin","image.DNAbin","is.binary","is.binary.multiPhylo","is.binary.phylo","is.binary.tree","is.compatible","is.compatible.bitsplits","is.monophyletic","is.rooted","is.rooted.multiPhylo","is.rooted.phylo","is.ultrametric","is.ultrametric.multiPhylo","is.ultrametric.phylo","keep.tip","kronoviz","label2table","labels.DNAbin","ladderize","latag2n","lmorigin","LTT","ltt.coplot","ltt.lines","ltt.plot","ltt.plot.coords","makeChronosCalib","makeLabel","makeLabel.character","makeNodeLabel","mantel.test","matexpo","mcconwaysims.test","mcmc.popsize","mixedFontLabel","mltt.plot","Moran.I","MPR","mrca","mst","multi2di","multi2di.multiPhylo","multi2di.phylo","muscle","mvr","mvrs","Nedge","Nedge.evonet","Nedge.multiPhylo","Nedge.phylo","new2old.phylo","nexus2DNAbin","nj","njs","Nnode","Nnode.multiPhylo","Nnode.phylo","node_depth","node_depth_edgelength","node_height","node_height_clado","node.depth","node.depth.edgelength","node.height","nodelabels","nodepath","Ntip","Ntip.multiPhylo","Ntip.phylo","old2new.phylo","ONEwise","parafit","pcoa","perm.rowscols","phydataplot","phylogram.plot","phymltest","pic","pic.ortho","plot.evonet","plot.multiPhylo","plot.phylo","plotBreakLongEdges","plotPhyloCoor","plotTreeTime","polar2rect","postorder","postprocess.prop.part","print.AAbin","print.DNAbin","print.phylo","prop.clades","prop.part","rbdtree","rbind.DNAbin","rcoal","rDNAbin","read.caic","read.dna","read.evonet","read.FASTA","read.fastq","read.GenBank","read.gff","read.nexus.data","reconstruct","rect2polar","reorder.evonet","reorder.multiPhylo","reorder.phylo","reorderRcpp","richness.yule.test","ring","rlineage","rmtree","root","root.multiPhylo","root.phylo","rotate","rotateConstr","rphylo","rTraitCont","rTraitDisc","rTraitMult","rtree","SDM","seg.sites","seq_root2tip","skyline","skylineplot","skylineplot.deluxe","slowinskiguyer.test","speciesTree","stree","stripLabel","subtreeplot","subtrees","summary.phylo","tcoffee","tiplabels","trans","treePop","trex","triangMtd","triangMtds","ultrametric","unique.multiPhylo","unroot","unroot.multiPhylo","unroot.phylo","unrooted.xy","updateLabel","varcomp","varCompPhylip","vcv","vcv.corPhyl","vcv.phylo","vcv2phylo","weight.taxo","weight.taxo2","where","which.edge","write.dna","write.evonet","write.nexus","write.nexus.data","write.tree","Xplor","Xplorefiles","yule","yule.cov","yule.time","zoom","entrez_citmatch","entrez_db_links","entrez_db_searchable","entrez_db_summary","entrez_dbs","entrez_fetch","entrez_global_query","entrez_info","entrez_link","entrez_post","entrez_search","entrez_summary","extract_from_esummary","linkout_urls","parse_pubmed_xml","set_entrez_key"],["base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","base","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","utils","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","ape","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez","rentrez"],["Arithmetic Operators","Operators on the Date Class","Date-Time Classes","Colon Operator","Double Colon and Triple Colon Operators","Double Colon and Triple Colon Operators","Logical Operators","Display Numbers in Hexadecimal","Display Numbers in Octal","Relational Operators","Parentheses and Braces","Extract or Replace Parts of an Object","Inhibit Interpretation/Conversion of Objects","Extract or Replace Parts of a Data Frame","Date Class","Time Intervals / Differences","Extract or Replace Parts of an Object","Get DLLs Loaded in Current Session","Extract or Replace Parts of a Factor","Display Numbers in Hexadecimal","Extract or Replace Parts of an Object","Class for 'no quote' Printing of Character Strings","Numeric Versions","Display Numbers in Octal","Date-Time Classes","Date-Time Classes","Extract or Replace Parts of an Object","Cross Tabulation and Table Creation","Print Warning Messages","Extract or Replace Parts of an Object","Extract or Replace Parts of a Data Frame","Date Class","Extract or Replace Parts of a Factor","Numeric Versions","Date-Time Classes","Date-Time Classes","Parentheses and Braces","Extract or Replace A Slot","Arithmetic Operators","Time Intervals / Differences","Arithmetic Operators","Time Intervals / Differences","Logical Operators","Display Numbers in Hexadecimal","Display Numbers in Octal","Logical Operators","Matrix Multiplication","Arithmetic Operators","Arithmetic Operators","Value Matching","Outer Product of Arrays","Kronecker Products on Arrays","Arithmetic Operators","Arithmetic Operators","Operators on the Date Class","Date-Time Classes","Relational Operators","Relational Operators","Assignment Operators","Relational Operators","Relational Operators","Relational Operators","Logical Operators","Display Numbers in Hexadecimal","Display Numbers in Octal","Logical Operators","Tilde Operator","Extract or Replace Parts of an Object","Get DLLs Loaded in Current Session","Numeric Versions","Abbreviate Strings","Miscellaneous Mathematical Functions","Trigonometric Functions","Hyperbolic Functions","Factors","Add or Remove a Top-Level Task Callback","Approximate String Matching (Fuzzy Matching)","Approximate String Matching (Fuzzy Matching)","Lists - Generic and Dotted Pairs","Are All Values True?","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Test if Two Objects are (Nearly) Equal","Find All Names in an Expression","Find All Names in an Expression","Condition Handling and Recovery","Are Some Values True?","Determine Duplicate Elements","Determine Duplicate Elements","Determine Duplicate Elements","Determine Duplicate Elements","Determine Duplicate Elements","'Not Available' / Missing Values","Numeric Versions","Date-Time Classes","Array Transposition","Array Transposition","Array Transposition","Vector Merging","Apply Functions Over Array Margins","Complex Numbers and Basic Functionality","Argument List of a Function","Multi-way Arrays","Which indices are TRUE?","Multi-way Arrays","Multi-way Arrays","Function Calls","Character Vectors","Condition Handling and Recovery","Date Conversion Functions to and from Character","Character Vectors","Condition Handling and Recovery","Character Vectors","Display Numbers in Hexadecimal","Numeric Versions","Display Numbers in Octal","Date-time Conversion Functions to and from Character","References to Source Files and Code","Complex Numbers and Basic Functionality","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Date Class","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Coerce to a Data Frame","Numeric Versions","Coerce to a Data Frame","Date-Time Classes","Date-Time Classes","Coerce to a Data Frame","Cross Tabulation and Table Creation","Coerce to a Data Frame","Coerce to a Data Frame","Date Conversion Functions to and from Character","Date Conversion Functions to and from Character","Date Conversion Functions to and from Character","Date Conversion Functions to and from Character","Date Conversion Functions to and from Character","Date Conversion Functions to and from Character","Date Conversion Functions to and from Character","Time Intervals / Differences","Double-Precision Vectors","Time Intervals / Differences","Date-time Conversion Functions","Coerce to an Environment Object","Unevaluated Expressions","Unevaluated Expressions","Factors","Convert Object to Function","Convert Object to Function","Display Numbers in Hexadecimal","Integer Vectors","Lists - Generic and Dotted Pairs","Lists - Generic and Dotted Pairs","Date Class","Lists - Generic and Dotted Pairs","Lists - Generic and Dotted Pairs","Lists - Generic and Dotted Pairs","Lists - Generic and Dotted Pairs","Numeric Versions","Date-Time Classes","Date-Time Classes","Logical Vectors","Logical Vectors","Matrices","Matrices","Matrices","Class for 'no quote' Printing of Character Strings","Date-Time Classes","Names and Symbols","The Null Object","The Null Object","Numeric Vectors","Numeric Versions","Display Numbers in Octal","Factors","Numeric Versions","Lists - Generic and Dotted Pairs","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","Date-time Conversion Functions","The QR Decomposition of a Matrix","Raw Vectors","Double-Precision Vectors","Double-Precision Vectors","Names and Symbols","Cross Tabulation and Table Creation","Cross Tabulation and Table Creation","Vectors","Vectors","Trigonometric Functions","Hyperbolic Functions","Namespace Internals","Split Array/Matrix By Its Margins","Test for an S4 object","Test for an S4 object","Assign a Value to a Name","Trigonometric Functions","Trigonometric Functions","Hyperbolic Functions","Attach Set of R Objects to Search Path","Loading and Unloading Name Spaces","Object Attributes","Test if Two Objects are (Nearly) Equal","Object Attribute Lists","On-demand Loading of Packages","On-demand Loading of Packages","Solve an Upper or Lower Triangular System","Environment Access","Manipulate File Paths","Bessel Functions","Bessel Functions","Bessel Functions","Bessel Functions","Special Functions of Mathematics","Binding and Environment Locking","Binding and Environment Locking","Translate Text Messages","Bitwise Logical Operations","Bitwise Logical Operations","Bitwise Logical Operations","Bitwise Logical Operations","Bitwise Logical Operations","Bitwise Logical Operations","Access to and Manipulation of the Body of a Function","Partial substitution in expressions","Control Flow","Environment Browser","Functions to Retrieve Values Supplied by Calls to the Browser","Functions to Retrieve Values Supplied by Calls to the Browser","Functions to Retrieve Values Supplied by Calls to the Browser","Returns the Names of All Built-in Objects","Apply a Function to a Data Frame Split by Factors","Apply a Function to a Data Frame Split by Factors","Apply a Function to a Data Frame Split by Factors","Functions to Manipulate Connections (Files","Combine Values into a Vector or List","Date Class","Time Intervals / Differences","Class for 'no quote' Printing of Character Strings","Numeric Versions","Date-Time Classes","Date-Time Classes","Print Warning Messages","Function Calls","Call With Current Continuation","Report Capabilities of this Build of R","Character Translation and Casefolding","Concatenate and Print","Combine R Objects by Rows or Columns","Combine R Objects by Rows or Columns","Rounding of Numbers","Expand a String with Respect to a Target Table","Character Vectors","Partial String Matching","Convert to or from Raw Vectors","Character Translation and Casefolding","Date-Time Classes","Warn About Extraneous Arguments in the ... of Its Caller","The Choleski Decomposition","The Choleski Decomposition","Inverse from Choleski (or QR) Decomposition","Special Functions of Mathematics","Object Classes","Push Text Back on to a Connection","Functions to Manipulate Connections (Files","Functions to Manipulate Connections (Files","References to Source Files and Code","References to Source Files and Code","Display Connections","Column Indexes","Form Row and Column Sums and Means","Row and Column Names","Form Row and Column Sums and Means","Extract Command Line Arguments","Query or Set a 'comment' Attribute","Complex Numbers and Basic Functionality","Condition Handling and Recovery","Condition Handling and Recovery","Condition Handling and Recovery","Condition Handling and Recovery","Condition Handling and Recovery","Loading/Attaching and Listing of Packages","Search for Masked Objects on the Search Path","Complex Numbers and Basic Functionality","R Project Contributors","Trigonometric Functions","Hyperbolic Functions","Trigonometric Functions","Matrix Crossproduct","Report Information on C Stack Size and Usage","Cumulative Sums","Cumulative Sums","Cumulative Sums","Cumulative Sums","Retrieve Headers from URLs","Convert Numeric to Factor","Convert a Date or Date-Time Object to a Factor","Convert Numeric to Factor","Convert a Date or Date-Time Object to a Factor","Object Classes","Data Frames","Convert a Data Frame to a Numeric Matrix","System Date and Time","Debug a Function","Debug a Function","Debug a Function","Data Frames","Delay Evaluation","Expression Deparsing","Calculate the Determinant of a Matrix","Detach Objects from the Search Path","Calculate the Determinant of a Matrix","Calculate the Determinant of a Matrix","Write an Object to a File or Recreate it","Matrix Diagonals","Lagged Differences","Lagged Differences","Lagged Differences","Time Intervals / Differences","Lagged Differences","Time Intervals / Differences","Special Functions of Mathematics","Dimensions of an Object","Dimensions of an Object","Dimnames of an Object","Dimnames of an Object","List the Files in a Directory/Folder","Manipulation of Directories and File Permissions","Manipulation of Directories and File Permissions","Manipulate File Paths","Execute a Function Call","Identity Function to Suppress Checking","Double-Precision Vectors","Write an Object to a File or Recreate it","Quote Text","Drop Redundant Extent Information","Drop Unused Levels from Factors","Drop Unused Levels from Factors","Drop Unused Levels from Factors","Text Representations of R Objects","Determine Duplicate Elements","Determine Duplicate Elements","Determine Duplicate Elements","Determine Duplicate Elements","Determine Duplicate Elements","Numeric Versions","Date-Time Classes","Print Warning Messages","Foreign Function Interface","Foreign Function Interface","Return the Value of a Named Object","Apply a Function Over Values in an Environment","Spectral Decomposition of a Matrix","Environment Access","Read or Set the Declared Encodings for a Character Vector","Read or Set the Declared Encodings for a Character Vector","Encode Character Vector as for Printing","Read or Set the Declared Encodings for a Character Vector","Does String Start or End With Another String?","Substituting and Quoting Expressions","Environment Access","Environment Access","Binding and Environment Locking","Environment Access","Condition Handling and Recovery","Evaluate an (Unevaluated) Expression","Evaluate an (Unevaluated) Expression","Evaluate an (Unevaluated) Expression","Is an Object Defined?","Logarithms and Exponentials","Create a Data Frame from All Combinations of Factor Variables","Logarithms and Exponentials","Unevaluated Expressions","Report Versions of Third-Party Software","Logical Vectors","Factors","Special Functions of Mathematics","Functions to Manipulate Connections (Files","Functions to Manipulate Connections (Files","Ascertain File Accessibility","File Manipulation","Choose a File Interactively","File Manipulation","File Manipulation","File Manipulation","Extract File Information","File Manipulation","Extract File Information","Extract File Information","Construct Path to File","File Manipulation","File Manipulation","Display One or More Text Files","Extract File Information","File Manipulation","Common Higher-Order Functions in Functional Programming Languages","Common Higher-Order Functions in Functional Programming Languages","Find Packages","Find Interval Numbers or Indices","Internal Objects in Package 'base'","Condition Handling and Recovery","Rounding of Numbers","Functions to Manipulate Connections (Files","Functions to Manipulate Connections (Files","Control Flow","Force Evaluation of an Argument","Call a function with Some Arguments Forced","Access to and Manipulation of the Formal Arguments","Encode in a Common Format","Encode in a Common Format","Encode in a Common Format","Date Conversion Functions to and from Character","Encode in a Common Format","Time Intervals / Differences","Encode in a Common Format","Display Numbers in Hexadecimal","format(.) Information","Loading/Attaching and Listing of Packages","Numeric Versions","Display Numbers in Octal","Loading/Attaching and Listing of Packages","Date-time Conversion Functions to and from Character","Date-time Conversion Functions to and from Character","Format P Values","Object Summaries","Formatting Using C-style Formats","Format Description Lists","Solve an Upper or Lower Triangular System","Function Definition","Special Functions of Mathematics","Garbage Collection","Report Time Spent in Garbage Collection","Garbage Collection","Torture Garbage Collector","Torture Garbage Collector","Return the Value of a Named Object","Is an Object Defined?","Display Connections","Compute DLL for Native Interface Call","Compute DLL for Native Interface Call","Display Connections","Reflectance Information for C/Fortran routines in a DLL","Reflectance Information for C/Fortran routines in a DLL","Reflectance Information for C/Fortran routines in a DLL","Extract or Replace Parts of an Object","Stop Function Execution","Namespace Reflection Support","Functions to Get and Set Hooks for Load","Get DLLs Loaded in Current Session","Namespace Reflection Support","Namespace Reflection Support","Namespace Reflection Support","Namespace Internals","Namespace Reflection Support","Namespace Reflection Support","Namespace Reflection Support","Obtain a Description of one or more Native (C/Fortran) Symbols","Options Settings","Numeric Versions","References to Source Files and Code","Query the Names of the Current Internal Top-Level Task Callbacks","Translate Text Messages","Use C-style String Formatting Commands","Get or Set Working Directory","Generate Factor Levels","Environment Access","Pattern Matching and Replacement","Pattern Matching and Replacement","Pattern Matching and Replacement","Pattern Matching for Raw Vectors","Grouping Permutation","Pattern Matching and Replacement","(De)compress I/O Through Connections","Functions to Manipulate Connections (Files","Inhibit Interpretation/Conversion of Objects","Convert Character Vector between Encodings","Convert Character Vector between Encodings","Setup Collation by ICU","Setup Collation by ICU","Test Objects for Exact Equality","Identity Function","Control Flow","Conditional Element Selection","Complex Numbers and Basic Functionality","Namespace Internals","Object Classes","Integer Vectors","Compute Factor Interactions","Is R Running Interactively?","Set Operations","Convert to or from Raw Vectors","Convert Integer Vectors to or from UTF-8-encoded Character Vectors","Run Length Encoding","Change the Print Mode to Invisible","Condition Handling and Recovery","Condition Handling and Recovery","Multi-way Arrays","Is an Object Atomic or Recursive?","Function Calls","Character Vectors","Complex Numbers and Basic Functionality","Coerce to a Data Frame","Double-Precision Vectors","Set Operations","Environment Access","Unevaluated Expressions","Factors","Finite","Is an Object of Type (Primitive) Function?","Finite","Integer Vectors","Is an Object a Language Object?","Lists - Generic and Dotted Pairs","Foreign Function Interface","Logical Vectors","Matrices","'Not Available' / Missing Values","'Not Available' / Missing Values","Numeric Versions","Numeric Versions","Names and Symbols","Finite","The Null Object","Numeric Vectors","Numeric Versions","Numeric Vectors","Time Intervals / Differences","Numeric Vectors","Is an Object 'internally classed'?","Factors","Numeric Versions","Lists - Generic and Dotted Pairs","Is an Object of Type (Primitive) Function?","The QR Decomposition of a Matrix","Are we using R","Raw Vectors","Is an Object Atomic or Recursive?","Is an Object of Single Precision Type?","Names and Symbols","Cross Tabulation and Table Creation","Test if an Object is Not Sorted","Vectors","Display Connections","Namespace Internals","Debug a Function","Logical Operators","Functions to Manipulate Connections (Files","Namespace Internals","Loading and Unloading Name Spaces","Date-time Conversion Functions from Numeric Representations","Date-time Conversion Functions from Numeric Representations","Functions to Manipulate Connections (Files","Condition Handling and Recovery","Test for an S4 object","Functions to Reposition Connections","Test if a Matrix or other Object is Symmetric (Hermitian)","Test if a Matrix or other Object is Symmetric (Hermitian)","Logical Operators","'Jitter' (Add Noise) to Numbers","Extract Parts of a POSIXt or Date Object","Extract Parts of a POSIXt or Date Object","Extract Parts of a POSIXt or Date Object","Compute or Estimate the Condition Number of a Matrix","Compute or Estimate the Condition Number of a Matrix","Compute or Estimate the Condition Number of a Matrix","Compute or Estimate the Condition Number of a Matrix","Kronecker Products on Arrays","Localization Information","LAPACK Library","LAPACK Version","Singular Value Decomposition of a Matrix","Find Labels from Object","Find Labels from Object","Apply a Function over a List or Vector","Print Warning Messages","Lazy Load a Database of R Objects","Lazy Load a Database of R Objects","Internal Objects in Package 'base'","Special Functions of Mathematics","Special Functions of Mathematics","Date-Time Classes","Date-Time Classes","Lengths of List or Vector Elements","Levels Attributes","Levels Attributes","Levels Attributes","Levels Attributes","Special Functions of Mathematics","Special Functions of Mathematics","Report Version of libcurl","Loading/Attaching and Listing of Packages","Loading DLLs from Packages","Loading DLLs from Packages","The R License Terms","The R License Terms","Lists - Generic and Dotted Pairs","List the Files in a Directory/Folder","List the Files in a Directory/Folder","From A List","Reload Saved Datasets","Loading and Unloading Name Spaces","Namespace Internals","Loading and Unloading Name Spaces","Evaluate an (Unevaluated) Expression","Binding and Environment Locking","Binding and Environment Locking","Logarithms and Exponentials","Logarithms and Exponentials","Logarithms and Exponentials","Logarithms and Exponentials","Logarithms and Exponentials","Logical Vectors","Lower and Upper Triangular Part of a Matrix","List Objects","Make Syntactically Valid Names","Make Character Strings Unique","Binding and Environment Locking","Common Higher-Order Functions in Functional Programming Languages","Apply a Function to Multiple List or Vector Arguments","Compute table margin","Create a Matrix or a Vector","Value Matching","Argument Verification Using Partial Matching","Argument Matching","Extract a Function Specified by Name","S3 Group Generic Functions","Date Class","Time Intervals / Differences","Factors","Date-Time Classes","Matrices","Maxima and Minima","Find Maximum Position in Matrix","Arithmetic Mean","Date Class","Arithmetic Mean","Time Intervals / Differences","Date-Time Classes","Date-Time Classes","Defunct Functions in Package 'base'","Query and Set Heap Size Limits","Query and Set Heap Size Limits","In-memory Compression and Decompression","In-memory Compression and Decompression","Profile the Usage of Cons Cells","Merge Two Data Frames","Merge Two Data Frames","Merge Two Data Frames","Diagnostic Messages","Return the Value of a Named Object","Does a Formal Argument have a Value?","Complex Numbers and Basic Functionality","The (Storage) Mode of an Object","Built-in Constants","Built-in Constants","Extract Parts of a POSIXt or Date Object","Object Attribute Lists","The Names of an Object","Date-Time Classes","The Names of an Object","Date-Time Classes","Namespace Internals","Namespace Internals","Namespace Internals","Namespace Internals","Namespace Internals","The Number of Arguments to a Function","Count the Number of Characters (or Bytes or Width)","The Number of Rows/Columns of an Array","The Number of Rows/Columns of an Array","Common Higher-Order Functions in Functional Programming Languages","Environment Access","Control Flow","Class Methods","Translate Text Messages","The Number of Levels of a Factor","Class for 'no quote' Printing of Character Strings","Compute the Norm of a Matrix","Express File Paths in Canonical Form","The Number of Rows/Columns of an Array","The Number of Rows/Columns of an Array","Numeric Vectors","Numeric Versions","Count the Number of Characters (or Bytes or Width)","List Objects","Object Classes","Object Classes","Time Zones","Function Exit Code","Functions to Manipulate Connections (Files","Functions to Manipulate Connections (Files","References to Source Files and Code","References to Source Files and Code","References to Source Files and Code","S3 Group Generic Functions","Operators on the Date Class","Time Intervals / Differences","Factors","Numeric Versions","Factors","Date-Time Classes","Options Settings","Ordering Permutation","Factors","Outer Product of Arrays","Numeric Versions","Functions to Get and Set Hooks for Load","Find Packages","Diagnostic Messages","Convert to or from Raw Vectors","Lists - Generic and Dotted Pairs","Environment Access","Environment Access","Functions to Access the Function Call Stack","Parse R Expressions","Namespace Internals","Concatenate Strings","Concatenate Strings","Expand File Paths","Find Packages","Report Configuration Options for PCRE","Built-in Constants","Functions to Manipulate Connections (Files","Partial String Matching","Maxima and Minima","Maxima and Minima","Maxima and Minima","Maxima and Minima","Find Zeros of a Real or Complex Polynomial","Convert Positions in the Search Path to Environments","Common Higher-Order Functions in Functional Programming Languages","Pretty Breakpoints","Pretty Breakpoints","Formatting Using C-style Formats","Print Values","Inhibit Interpretation/Conversion of Objects","Apply a Function to a Data Frame Split by Factors","Condition Handling and Recovery","Functions to Manipulate Connections (Files","Printing Data Frames","Date Class","Default Printing","Time Intervals / Differences","Print Values","Get DLLs Loaded in Current Session","Get DLLs Loaded in Current Session","Reflectance Information for C/Fortran routines in a DLL","Spectral Decomposition of a Matrix","Print Values","Print Values","Display Numbers in Hexadecimal","Loading/Attaching and Listing of Packages","Print Values","Reflectance Information for C/Fortran routines in a DLL","Class for 'no quote' Printing of Character Strings","Numeric Versions","Display Numbers in Octal","Loading/Attaching and Listing of Packages","Date-Time Classes","Date-Time Classes","Running Time of R","Condition Handling and Recovery","Run Length Encoding","Print Values","References to Source Files and Code","References to Source Files and Code","Cross Tabulation and Table Creation","Print Warning Messages","Object Summaries","Print Values","Print Warning Messages","Print Matrices","Running Time of R","Product of Vector Elements","Express Table Entries as Fraction of Marginal Table","Dimnames of an Object","Special Functions of Mathematics","Push Text Back on to a Connection","Push Text Back on to a Connection","Terminate an R Session","The QR Decomposition of a Matrix","The QR Decomposition of a Matrix","The QR Decomposition of a Matrix","The QR Decomposition of a Matrix","Reconstruct the Q","The QR Decomposition of a Matrix","The QR Decomposition of a Matrix","Reconstruct the Q","The QR Decomposition of a Matrix","The QR Decomposition of a Matrix","Reconstruct the Q","Extract Parts of a POSIXt or Date Object","Extract Parts of a POSIXt or Date Object","Extract Parts of a POSIXt or Date Object","Terminate an R Session","Substituting and Quoting Expressions","Numeric Versions","Return the R Home Directory","Version Information","Version Information","Version Information","Range of Values","Range of Values","Sample Ranks","Recursively Apply a Function to a List","Raw Vectors","Raw Connections","Raw Connections","Convert to or from Raw Vectors","Convert to or from Raw Vectors","Convert to or from Raw Vectors","Combine R Objects by Rows or Columns","Combine R Objects by Rows or Columns","Compute or Estimate the Condition Number of a Matrix","Complex Numbers and Basic Functionality","Read and Write Data in DCF Format","Transfer Binary Data To and From Connections","Transfer Character Strings To and From Connections","Read a Line from the Terminal","Read Text Lines from a Connection","Serialization Interface for Single Objects","Recursive Calling","Common Higher-Order Functions in Functional Programming Languages","Finalization of Objects","Pattern Matching and Replacement","Pattern Matching and Replacement","Namespace Internals","Namespace Internals","Extract or Replace Matched Substrings","Extract or Replace Matched Substrings","Remove Objects from a Specified Environment","Add or Remove a Top-Level Task Callback","Replicate Elements of Vectors and Lists","Replicate Elements of Vectors and Lists","Replicate Elements of Vectors and Lists","Replicate Elements of Vectors and Lists","Replicate Elements of Vectors and Lists","Numeric Versions","Replicate Elements of Vectors and Lists","Replicate Elements of Vectors and Lists","Control Flow","Replace Values in a Vector","Apply a Function over a List or Vector","Loading/Attaching and Listing of Packages","Loading and Unloading Name Spaces","Condition Handling and Recovery","Condition Handling and Recovery","Trace Copying of Objects","Function Definition","Interactive Tracing and Debugging of Calls to a Function or Method","Reverse Elements","Reverse Elements","Run Length Encoding","Remove Objects from a Specified Environment","Round / Truncate Data-Time Objects","Round / Truncate Data-Time Objects","Get and Set Row Names for Data Frames","Get and Set Row Names for Data Frames","Get and Set Row Names for Data Frames","Get and Set Row Names for Data Frames","Get and Set Row Names for Data Frames","Get and Set Row Names for Data Frames","Form Row and Column Sums and Means","Row and Column Names","Row and Column Names","Give Column Sums of a Matrix or Data Frame","Give Column Sums of a Matrix or Data Frame","Give Column Sums of a Matrix or Data Frame","Form Row and Column Sums and Means","Random Samples and Permutations","Random Samples and Permutations","Apply a Function over a List or Vector","Save R Objects","Save R Objects","Serialization Interface for Single Objects","Scaling and Centering of Matrix-like Objects","Scaling and Centering of Matrix-like Objects","Read Data Values","Give Search Path for R Objects","Give Search Path for R Objects","Functions to Reposition Connections","Functions to Reposition Connections","Sequence Generation","Sequence Generation","Sequence Generation","Generate Regular Sequences of Dates","Sequence Generation","Sequence Generation","Generate Regular Sequences of Times","Create A Vector of Sequences","Simple Serialization Interface","Random Number Generation","Set Operations","Set Operations","Functions to Get and Set Hooks for Load","Namespace Internals","Set CPU and/or Elapsed Time Limits","Set CPU and/or Elapsed Time Limits","Get or Set Working Directory","Display Connections","Quote Strings for Use in OS Shells","Sign Function","Condition Handling and Recovery","Rounding of Numbers","Condition Handling and Recovery","Condition Handling and Recovery","Condition Handling and Recovery","Condition Handling and Recovery","Apply a Function over a List or Vector","Trigonometric Functions","Double-Precision Vectors","Hyperbolic Functions","Send R Output to a File","Send R Output to a File","Trigonometric Functions","Slice Indexes in an Array","Functions to Manipulate Connections (Files","Wait on Socket Connections","Solve a System of Equations","Solve a System of Equations","The QR Decomposition of a Matrix","Sorting or Ordering Vectors","Sorting or Ordering Vectors","Read R Code from a File","Divide into Groups and Reassemble","Divide into Groups and Reassemble","Date Class","Divide into Groups and Reassemble","Date-Time Classes","Divide into Groups and Reassemble","Divide into Groups and Reassemble","Divide into Groups and Reassemble","Use C-style String Formatting Commands","Miscellaneous Mathematical Functions","Quote Text","References to Source Files and Code","References to Source Files and Code","References to Source Files and Code","Formal Method System - Dispatching S4 Methods","Does String Start or End With Another String?","Display Connections","Display Connections","Display Connections","Stop Function Execution","Ensure the Truth of R Expressions","The (Storage) Mode of an Object","The (Storage) Mode of an Object","Parse R Expressions","Parse R Expressions","Date-time Conversion Functions to and from Character","Date-time Conversion Functions to and from Character","Repeat the Elements of a Character Vector","Split the Elements of a Character Vector","Convert Strings to Integers","Trim Character Strings to Specified Display Widths","Wrap Character Strings to Format Paragraphs","Subsetting Vectors","Subsetting Vectors","Subsetting Vectors","Subsetting Vectors","Substituting and Quoting Expressions","Substrings of a Character Vector","Substrings of a Character Vector","Substrings of a Character Vector","Substrings of a Character Vector","Sum of Vector Elements","Object Summaries","Functions to Manipulate Connections (Files","Object Summaries","S3 Group Generic Functions","Date Class","Date Class","Object Summaries","Time Intervals / Differences","Object Summaries","Factors","Object Summaries","Numeric Versions","Factors","Date-Time Classes","Date-Time Classes","Date-Time Classes","Date-Time Classes","Running Time of R","References to Source Files and Code","References to Source Files and Code","Cross Tabulation and Table Creation","Print Warning Messages","Diagnostic Messages","Diagnostic Messages","Warning Messages","Condition Handling and Recovery","Singular Value Decomposition of a Matrix","Sweep out Array Summaries","Select One of a List of Alternatives","Functions to Access the Function Call Stack","Functions to Access the Function Call Stack","Manipulation of Directories and File Permissions","Get Current Date and Time","Functions to Access the Function Call Stack","Functions to Access the Function Call Stack","Functions to Access the Function Call Stack","Get Environment Variables","Query or Set Aspects of the Locale","Get the Process ID of the R Session","Wildcard Expansion on File Paths","Extract System and User Information","Internal Objects in Package 'base'","Find Details of the Numerical and Monetary Representations in the","Functions to Access the Function Call Stack","Functions to Access the Function Call Stack","Functions to Access the Function Call Stack","Functions to Access the Function Call Stack","Read File Symbolic Links","Internal Objects in Package 'base'","Set or Unset Environment Variables","Set File Time","Query or Set Aspects of the Locale","Suspend Execution for a Time Interval","Parse and Evaluate Expressions from a File","Functions to Access the Function Call Stack","Get Current Date and Time","Time Zones","Manipulation of Directories and File Permissions","Set or Unset Environment Variables","Find Full Paths to Executables","Invoke a System Command","Find Names of R System Files","CPU Time Used","Invoke a System Command","Matrix Transpose","Logical Vectors","Matrix Transpose","Matrix Transpose","Cross Tabulation and Table Creation","Tabulation for Vectors","Trigonometric Functions","Hyperbolic Functions","Trigonometric Functions","Apply a Function Over a Ragged Array","Create an R-level Task Callback Manager","Matrix Crossproduct","Create Names for Temporary Files","Create Names for Temporary Files","Text Connections","Text Connections","Character Translation and Casefolding","Top Level Environment","Convert an R Object to a Character String","Convert an R Object to a Character String","Character Translation and Casefolding","Interactive Tracing and Debugging of Calls to a Function or Method","Get and Print Call Stacks","Trace Copying of Objects","Interactive Tracing and Debugging of Calls to a Function or Method","Transform an Object","Transform an Object","Transform an Object","Special Functions of Mathematics","Remove Leading/Trailing Whitespace","Rounding of Numbers","Round / Truncate Data-Time Objects","Round / Truncate Data-Time Objects","Functions to Reposition Connections","Functions to Reposition Connections","Try an Expression Allowing Error Recovery","Condition Handling and Recovery","The Type of an Object","Object Classes","Debug a Function","Set Operations","Extract Unique Elements","Extract Unique Elements","Extract Unique Elements","Numeric Versions","Date-Time Classes","Print Warning Messages","Time Intervals / Differences","Time Intervals / Differences","Time Intervals / Differences","Time Intervals / Differences","CPU Time Used","Delete Files and Directories","Flatten Lists","Loading and Unloading Name Spaces","Binding and Environment Locking","Remove 'names' or 'dimnames'","Simple Serialization Interface","Divide into Groups and Reassemble","Interactive Tracing and Debugging of Calls to a Function or Method","Trace Copying of Objects","Functions to Manipulate Connections (Files","Lower and Upper Triangular Part of a Matrix","Functions to Manipulate Connections (Files","Class Methods","Convert Integer Vectors to or from UTF-8-encoded Character Vectors","Check if a Character Vector is Validly Encoded","Check if a Character Vector is Validly Encoded","Apply a Function over a List or Vector","Vectors","Vectorize a Scalar Function","Version Information","Warning Messages","Condition Handling and Recovery","Print Warning Messages","Extract Parts of a POSIXt or Date Object","Extract Parts of a POSIXt or Date Object","Extract Parts of a POSIXt or Date Object","Which indices are TRUE?","Where is the Min() or Max() or first TRUE or FALSE ?","Where is the Min() or Max() or first TRUE or FALSE ?","Control Flow","Evaluate an Expression in a Data Environment","Evaluate an Expression in a Data Environment","Read R Code from a File","Condition Handling and Recovery","Evaluate an Expression in a Data Environment","Evaluate an Expression in a Data Environment","Evaluate an Expression in a Data Environment","Condition Handling and Recovery","Return both a Value and its Visibility","Write Data to a File","Read and Write Data in DCF Format","Transfer Binary Data To and From Connections","Transfer Character Strings To and From Connections","Write Lines to a Connection","Logical Operators","Data Frame Auxiliary Functions","Auxiliary Function for Sorting and Ranking","Auxiliary Function for Sorting and Ranking","Auxiliary Function for Sorting and Ranking","Auxiliary Function for Sorting and Ranking","Auxiliary Function for Sorting and Ranking","Auxiliary Function for Sorting and Ranking","Numeric Versions","Auxiliary Function for Sorting and Ranking","Auxiliary Function for Sorting and Ranking","Functions to Manipulate Connections (Files","Rounding of Numbers: Zapping Small Ones to Zero","Documentation Shortcuts","Approximate String Distances","Alert the User","Find Objects by (Partial) Name","Approximate String Match Positions","Retrieve an R Object","Persons","Persons","Allow Re-Listing an unlist()ed Object","Roman Numerals","Package Description","Ask a Yes/No Question","Spell Check Interface","Spell Check Utilities","Spell Check Utilities","Spell Check Utilities","Spell Check Utilities","Spell Check Utilities","Utility Functions for Developing Namespaces","Utility Functions for Developing Namespaces","List Available Packages at CRAN-like Repositories","Bibliography Entries","Browse Objects in Environment","Load URL into an HTML Browser","List Vignettes in an HTML Browser","Send a Bug Report","Send Output to a Character String or File","Detect which Files Have Changed","Managing Repository Mirrors","Select a Bioconductor Mirror","Select a CRAN Mirror","Citing R and R Packages in Publications","Cite a Bibliography Entry","Cite a Bibliography Entry","Bibliography Entries (Older Interface)","Bibliography Entries (Older Interface)","Spreadsheet Interface for Entering Data","Spreadsheet Interface for Entering Data","Debug a Call","Post-Mortem Debugging","Demonstrations of R Functionality","Download File from the Internet","Download Packages from CRAN-like Repositories","Post-Mortem Debugging","Invoke a Text Editor","Invoke a Text Editor","Run an Examples Section from the Online Help","Shell-style Tests on Files","Edit One or More Files","Detect which Files Have Changed","Find Objects by (Partial) Name","Find the Location of a Line of Source Code","Fix an Object","Utility Functions for Developing Namespaces","Flush Output to a Console","Format Unordered and Ordered Lists","Format Unordered and Ordered Lists","Utility Functions for Developing Namespaces","Get Detailed Parse Information from Object","Get Detailed Parse Information from Object","Get an S3 Method","Source Reference Utilities","Source Reference Utilities","Source Reference Utilities","Source Reference Utilities","Text Progress Bar","Change Wildcard or Globbing Pattern into Regular Expression","Declarations Used in Checking a Package","Check for Name","Return the First or Last Part of an Object","Send a Post to R-help","Search the Help System","Hypertext Documentation","Load or Save or Display the Commands History","Help Search Utilities","Help Search Utilities","Help Search Utilities","Install Packages from Repositories or Local Files","Find Installed Packages","Allow Re-Listing an unlist()ed Object","Is 'method' the Name of an S3 Method?","Check if a Function Acts as an S3 Generic","Browsing after an Error","Load or Save or Display the Commands History","Select a Suitable Encoding Name from a Locale Name","List Objects and their Structure","List Objects and their Structure","Show Package Maintainer","Update HTML Package List","Create a Socket Connection","Sweave Driver Utilities","Report on Memory Allocation (on Windows)","Report on Memory Allocation (on Windows)","Menu Interaction Function","List Methods for S3 Generic Functions or Classes","Managing Repository Mirrors","Recursively Modify Elements of a List","Compare Installed Packages with CRAN-like Repositories","Build and Query R or Package News Information","Look up the IP Address by Hostname (on Unix-alikes)","Report the Space Allocated for an Object","Compare Installed Packages with CRAN-like Repositories","Collect Information About the Current R Session","Create a Skeleton for a New Source Package","Package Description","Package Description","Find Package Associated with an Environment","Package Management Tools","Package Description","Invoke a Pager on an R Object","Persons","Persons","Invoke a Text Editor","Trigger Event Handling","Produce Prototype of an R Documentation File","Generate Outline Documentation for a Data Set","Produce Prototype of an R Documentation File","Generate a Shell for Documentation of a Package","A Completion Generator for R","A Completion Generator for R","A Completion Generator for R","A Completion Generator for R","Data Input","Data Input","Data Input","Data Input","Data Input from Spreadsheet","Read Fixed-Format Data in a Fortran-like Style","Read Fixed Width Format Files","Read from or Write to a Socket","Data Input","Citing R and R Packages in Publications","Browsing after an Error","Allow Re-Listing an unlist()ed Object","Remove Installed Packages","Remove Stored Source from a Function or Language Object","Enable Profiling of R's Execution","Enable Profiling of R's Memory Use","Show R Manuals and Other Documentation","Search for Key Words or Phrases in Documentation","An Etags-like Tagging Utility for R","R Driver for Stangle","R Driver for Stangle","Sweave Driver Utilities","Sweave Driver Utilities","Sweave Driver Utilities","R/LaTeX Driver for Sweave","Sweave Driver Utilities","Sweave Driver Utilities","R/LaTeX Driver for Sweave","Sweave Driver Utilities","Sweave Driver Utilities","Load or Save or Display the Commands History","Select Items from a List","Collect Information About the Current R Session","Find the Location of a Line of Source Code","Select Package Repositories","Text Progress Bar","Stack or Unstack Vectors from a Data Frame or List","Automatic Generation of Reports","Compactly Display the Structure of an Arbitrary R Object","Capture String Tokens into a data.frame","Compactly Display the Structure of an Arbitrary R Object","Summarise Output of R Sampling Profiler","Declarations Used in Checking a Package","Automatic Generation of Reports","Sweave Driver Utilities","Automatic Generation of Reports","Automatic Generation of Reports","Convert Sweave Syntax","Return the First or Last Part of an Object","Return the First or Last Part of an Object","Create a Tar Archive","Load or Save or Display the Commands History","Converting R Objects to BibTeX or LaTeX","Converting R Objects to BibTeX or LaTeX","Text Progress Bar","Convert Data to Appropriate Type","Debug a Call","Stack or Unstack Vectors from a Data Frame or List","Extract or List Tar Archives","Extract or List Zip Archives","Compare Installed Packages with CRAN-like Repositories","Package Management Tools","Display a Text URL","Encode or Decode a (partial) URL","Encode or Decode a (partial) URL","Invoke a Text Editor","Invoke a Data Viewer","View","Collect and Summarize Errors From List","Data Output","Data Output","Read from or Write to a Socket","Data Output","Invoke a Text Editor","Invoke a Text Editor","Create Zip Archives","Manipulate DNA Sequences in Bit-Level Format","Amino Acid Sequences","Label Management","Ancestral Character Estimation","Add a Scale Bar to a Phylogeny Plot","Incomplete Distance Matrix Filling","Alignment Explorer With Multiple Devices","Compare DNA Sets","Global Comparison of two Phylogenies","Print DNA or AA Sequence Alignement","Check Compatibility of Splits","Amino Acid Sequences","Amino Acid Sequences","Conversion Among DNA Sequence Internal Formats","Split Frequencies and Conversion Among Split Classes","Split Frequencies and Conversion Among Split Classes","Amino Acid Sequences","Conversion Among DNA Sequence Internal Formats","Conversion Among DNA Sequence Internal Formats","Conversion Among DNA Sequence Internal Formats","Conversion Among DNA Sequence Internal Formats","Conversion Among DNA Sequence Internal Formats","Evolutionary Networks","Evolutionary Networks","Conversion Among Tree and Network Objects","Evolutionary Networks","Conversion Among Tree and Network Objects","Amino Acid Sequences","Manipulate DNA Sequences in Bit-Level Format","Conversion Between Phylo and Matching Objects","Conversion Between Phylo and Matching Objects","Manipulate DNA Sequences in Bit-Level Format","Evolutionary Networks","Conversion Among Tree and Network Objects","Evolutionary Networks","Amino Acid Sequences","Phylogenetic Generalized Linear Mixed Model for Binary Data","Phylogenetic Generalized Linear Mixed Model for Binary Data","Tree Estimation Based on an Improved Version of the NJ Algorithm","Tree Reconstruction from Incomplete Distances With NJ* or bio-NJ*","Principal Coordinate Analysis","Estimation of Speciation and Extinction Rates With Birth-Death Models","Split Frequencies and Conversion Among Split Classes","Tree Bipartition and Bootstrapping Phylogenies","Internal Ape Functions","Branching Times of a Phylogenetic Tree","Tools to Explore Files","Manipulate DNA Sequences in Bit-Level Format","Congruence among distance matrices","Congruence among distance matrices","Manipulate DNA Sequences in Bit-Level Format","Check DNA Alignments","Checking Labels","Check the Structure of a phylo Object","Number of Cherries and Null Models of Trees","Molecular Dating With Mean Path Lengths","Molecular Dating With Penalized Likelihood","Internal Ape Functions","Internal Reordering of Trees","Internal Ape Functions","Multiple Sequence Alignment with External Applications","Multiple Sequence Alignment with External Applications","Coalescent Intervals","Collapse Single Nodes","Collapsed Coalescent Intervals","Cheverud's Comparative Method","Comparative Analysis with GEEs","Lynch's Comparative Method","Ornstein-Uhlenbeck Model for Continuous Characters","Compare Two phylo Objects","Compute and Set Branching Times","Concensus Trees","Pairwise Distances from a Phylogenetic Tree","Plots two phylogenetic trees face to face with links between the tips.","Blomberg et al.'s Correlation Structure","Brownian Correlation Structure","Grafen's (1989) Correlation Structure","Martins's (1997) Correlation Structure","Pagel's lambda Correlation Structure","Correlations among Multiple Traits with Phylogenetic Signal","Phylogenetic Correlogram","Split Frequencies and Conversion Among Split Classes","Probability Density Under Birth-Death Models","Probability Density Under Birth-Death Models","Definition of Vectors for Plotting or Annotating","Delete Alignment Gaps in DNA Sequences","Delete Alignment Gaps in DNA Sequences","Delete Alignment Gaps in DNA Sequences","Delta Plots","Ancestral Character Estimation","Collapse and Resolve Multichotomies","Collapse and Resolve Multichotomies","Collapse and Resolve Multichotomies","Amino Acid Sequences","Pairwise Distances from DNA Sequences","Pairwise Distances from Genetic Data","Pairwise Distances from a Phylogenetic Tree","Topological Distances Between Two Trees","Tests of Constant Diversification Rates","Analysis of Diversification with Survival Models","Diversity Contrast Test","Recode Blocks of Indels","dN/dS Ratio","Extra Fuctions to Plot and Annotate Phylogenies","Tree Simulation Under the Time-Dependent Birth-Death Models","Remove Tips in a Phylogenetic Tree","Probability Density Under Birth-Death Models","Labelling the Nodes","Draw Additional Edges on a Plotted Tree","Tools to Explore Files","node.dating","node.dating","Evolutionary Networks","Incomplete distances and edge weights of unrooted topology","Remove Tips in a Phylogenetic Tree","Reversible Jump MCMC to Infer Demographic History","Draw Additional Edges on a Plotted Tree","Tree Estimation Based on the Minimum Evolution Algorithm","Tree Estimation Based on the Minimum Evolution Algorithm","Skyline Plot Estimate of Effective Population Size","Internal Ape Functions","Base frequencies from DNA Sequences","Gamma-Statistic of Pybus and Harvey","Base frequencies from DNA Sequences","Read Annotations from GenBank","Find Most Recent Common Ancestors Between Pairs","Collapse Single Nodes","Calculate Numbers of Phylogenetic Trees","Amino Acid Sequences","Plot of DNA Sequence Alignement","Test for Binary Tree","Test for Binary Tree","Test for Binary Tree","Test for Binary Tree","Check Compatibility of Splits","Check Compatibility of Splits","Is Group Monophyletic","Roots Phylogenetic Trees","Roots Phylogenetic Trees","Roots Phylogenetic Trees","Test if a Tree is Ultrametric","Test if a Tree is Ultrametric","Test if a Tree is Ultrametric","Remove Tips in a Phylogenetic Tree","Plot Multiple Chronograms on the Same Scale","Label Management","Manipulate DNA Sequences in Bit-Level Format","Ladderize a Tree","Leading and Trailing Alignment Gaps to N","Multiple regression through the origin","Theoretical Lineage-Through Time Plots","Lineages Through Time Plot","Lineages Through Time Plot","Lineages Through Time Plot","Lineages Through Time Plot","Molecular Dating by Penalised Likelihood and Maximum Likelihood","Label Management","Label Management","Makes Node Labels","Mantel Test for Similarity of Two Matrices","Matrix Exponential","McConway-Sims Test of Homogeneous Diversification","Reversible Jump MCMC to Infer Demographic History","Mixed Font Labels for Plotting","Lineages Through Time Plot","Moran's I Autocorrelation Index","Most Parsimonious Reconstruction","Find Most Recent Common Ancestors Between Pairs","Minimum Spanning Tree","Collapse and Resolve Multichotomies","Collapse and Resolve Multichotomies","Collapse and Resolve Multichotomies","Multiple Sequence Alignment with External Applications","Minimum Variance Reduction","Minimum Variance Reduction","Print Summary of a Phylogeny","Evolutionary Networks","Print Summary of a Phylogeny","Print Summary of a Phylogeny","Conversion Among Tree and Network Objects","Read Character Data In NEXUS Format","Neighbor-Joining Tree Estimation","Tree Reconstruction from Incomplete Distances With NJ* or bio-NJ*","Print Summary of a Phylogeny","Print Summary of a Phylogeny","Print Summary of a Phylogeny","Internal Ape Functions","Internal Ape Functions","Internal Ape Functions","Internal Ape Functions","Depth and Heights of Nodes and Tips","Depth and Heights of Nodes and Tips","Depth and Heights of Nodes and Tips","Labelling the Nodes","Find Paths of Nodes","Print Summary of a Phylogeny","Print Summary of a Phylogeny","Print Summary of a Phylogeny","Conversion Among Tree and Network Objects","Internal Ape Functions","Test of host-parasite coevolution","Principal Coordinate Analysis","Internal Ape Functions","Tree Annotation","Internal Ape Functions","Fits a Bunch of Models with PhyML","Phylogenetically Independent Contrasts","Phylogenetically Independent Orthonormal Contrasts","Evolutionary Networks","Plot Phylogenies","Plot Phylogenies","Extra Fuctions to Plot and Annotate Phylogenies","Internal Ape Functions","Plot Tree With Time Axis","Internal Ape Functions","Internal Reordering of Trees","Internal Ape Functions","Amino Acid Sequences","Manipulate DNA Sequences in Bit-Level Format","Compact Display of a Phylogeny","Tree Bipartition and Bootstrapping Phylogenies","Tree Bipartition and Bootstrapping Phylogenies","Tree Simulation Under the Time-Dependent Birth-Death Models","Manipulate DNA Sequences in Bit-Level Format","Generates Random Trees","Random DNA Sequences","Read Tree File in CAIC Format","Read DNA Sequences in a File","Evolutionary Networks","Read DNA Sequences in a File","Read DNA Sequences in a File","Read DNA Sequences from GenBank via Internet","Read GFF Files","Read Character Data In NEXUS Format","Continuous Ancestral Character Estimation","Internal Ape Functions","Evolutionary Networks","Internal Reordering of Trees","Internal Reordering of Trees","Internal Ape Functions","Test of Diversification-Shift With the Yule Process","Tree Annotation","Tree Simulation Under the Time-Dependent Birth-Death Models","Generates Random Trees","Roots Phylogenetic Trees","Roots Phylogenetic Trees","Roots Phylogenetic Trees","Swapping Sister Clades","Swapping Sister Clades","Tree Simulation Under the Time-Dependent Birth-Death Models","Continuous Character Simulation","Discrete Character Simulation","Multivariate Character Simulation","Generates Random Trees","Construction of Consensus Distance Matrix With SDM","Find Segregating Sites in DNA Sequences","Internal Ape Functions","Skyline Plot Estimate of Effective Population Size","Drawing Skyline Plot Graphs","Drawing Skyline Plot Graphs","Slowinski-Guyer Test of Homogeneous Diversification","Species Tree Estimation","Generates Systematic Regular Trees","Label Management","Zoom on a Portion of a Phylogeny by Successive Clicks","All subtrees of a Phylogenetic Tree","Print Summary of a Phylogeny","Multiple Sequence Alignment with External Applications","Labelling the Nodes","Translation from DNA to Amino Acid Sequences","Tree Popping","Tree Explorer With Multiple Devices","Tree Reconstruction Based on the Triangles Method","Tree Reconstruction Based on the Triangles Method","Incomplete Distance Matrix Filling","Revomes Duplicate Trees","Roots Phylogenetic Trees","Roots Phylogenetic Trees","Roots Phylogenetic Trees","Internal Ape Functions","Update Labels","Compute Variance Component Estimates","Variance Components with Orthonormal Contrasts","Phylogenetic Variance-covariance or Correlation Matrix","Phylogenetic Variance-covariance or Correlation Matrix","Phylogenetic Variance-covariance or Correlation Matrix","Variance-Covariance Matrix to Tree","Define Similarity Matrix","Define Similarity Matrix","Find Patterns in DNA Sequences","Identifies Edges of a Tree","Write DNA Sequences in a File","Evolutionary Networks","Write Tree File in Nexus Format","Write Character Data in NEXUS Format","Write Tree File in Parenthetic Format","Tools to Explore Files","Tools to Explore Files","Fits the Yule Model to a Phylogenetic Tree","Fits the Yule Model With Covariates","Fits the Time-Dependent Yule Model","Zoom on a Portion of a Phylogeny","Fetch pubmed ids matching specially formatted citation strings","List available links for records from a given NCBI database","List available search fields for a given database","Retrieve summary information about an NCBI database","List databases available from the NCBI","Download data from NCBI databases","Find the number of records that match a given term across all NCBI","Get information about EUtils databases","Get links to datasets related to records from an NCBI database","Post IDs to Eutils for later use","Search the NCBI databases using EUtils","Get summaries of objects in NCBI datasets from a unique ID","Extract elements from a list of esummary records","Extract URLs from an elink object","Summarize an XML record from pubmed.","Set the ENTREZKEY variable to be used by all rentrez functions"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Function<\/th>\n      <th>Package<\/th>\n      <th>Definition<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":25,"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-weight':'italic'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
