<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-01-25" />

<title>References</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Mining_draft_genome.html">Lab.</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="Lexicon.html">Lexicon</a>
</li>
<li>
  <a href="R_Functions.html">R functions</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">References</h1>
<h4 class="date">2023-01-25</h4>

</div>


<p>References (web resources and publications) used in the BIOL 497/597 - Genomics &amp; Bioinformatics course are cited here. The citation of references supports transparency in science as well as provides opportunities for students to further their knowledge by accessing articles and books used to develop this course. The <code>BibTex</code> document is accessible <a href="References.bib">here</a>. Please contact the instructor (<a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a>) to add missing references to this document.</p>
<div id="web-resources" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Web resources</h1>
<ul>
<li><strong>Human Genome Project</strong>: This <a href="https://www.nature.com/scitable/topicpage/dna-sequencing-technologies-key-to-the-human-828/">webpage</a> contain details on the human genome project, especially associated to the applied sequencing strategy.</li>
<li><strong>Completing the human genome</strong>: <a href="https://www.science.org/toc/science/376/6588">Special Issue</a> published in <em>Science</em> in April 2022.</li>
<li><strong>An Overview of Next-Generation Sequencing</strong>: This <a href="https://www.technologynetworks.com/genomics/articles/an-overview-of-next-generation-sequencing-346532">webpage</a> contains information related to next-generation sequencing.</li>
<li><strong>Long-read sequencing is method of the year</strong>: To large-scale projects and individual labs, long-read sequencing has delivered new vistas and long wish lists for this technology’s future. Read this <a href="https://www.nature.com/articles/s41592-022-01730-w">article</a> published in <em>Nature Methods</em> telling you more why long-read sequencing is revolutionizing the field of genomics.</li>
<li><strong>The Earth BioGenome Project</strong>: A <a href="https://www.pnas.org/cc/the-earth-biogenome-project">special issue</a> published in <em>PNAS</em> with a collection of articles on this project.</li>
<li><strong>NCBI BioProject Help Book</strong>: This <a href="https://www.ncbi.nlm.nih.gov/books/NBK54364/">book</a> provides definitions of terms associated to submitted data to NCBI. This information is pertinent to Chapter 3.</li>
</ul>
</div>
<div id="videos" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Videos</h1>
<ul>
<li><strong>Sanger sequencing</strong>: A <a href="https://www.youtube.com/watch?v=e2G5zx-OJIw">video</a> explaining the Sanger sequencing technology.</li>
</ul>
</div>
<div id="publications" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Publications</h1>
<p>[1] S. F. Altschul, W. Gish, W. Miller, et al. “Basic local alignment
search tool”. In: <em>Journal of Molecular Biology</em> 215.3 (1990), pp. 403
- 410. ISSN: 0022-2836. DOI:
<a href="https://doi.org/10.1016/S0022-2836(05)80360-2" class="uri">https://doi.org/10.1016/S0022-2836(05)80360-2</a>.
<a href="http://www.sciencedirect.com/science/article/pii/S0022283605803602" class="uri">http://www.sciencedirect.com/science/article/pii/S0022283605803602</a>.</p>
<p>[2] S. F. Altschul, T. L. Madden, A. A. Schäffer, et al. “Gapped BLAST
and PSI-BLAST: a new generation of protein database search programs”.
In: <em>Nucleic acids research</em> 25.17 (1997), pp. 3389-3402.</p>
<p>[3] S. Andrews. <em>FastQC A Quality Control tool for High Throughput
Sequence Data</em>. 2012.
<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>.</p>
<p>[4] S. Asaf, A. L. Khan, A. R. Khan, et al. “Complete Chloroplast
Genome of Nicotiana otophora and its Comparison with Related Species”.
In: <em>Frontiers in Plant Science</em> 7 (2016), p. 843. ISSN: 1664-462X.
DOI: 10.3389/fpls.2016.00843.
<a href="https://www.frontiersin.org/article/10.3389/fpls.2016.00843" class="uri">https://www.frontiersin.org/article/10.3389/fpls.2016.00843</a>.</p>
<p>[5] A. Bankevich, S. Nurk, D. Antipov, et al. “SPAdes: A New Genome
Assembly Algorithm and Its Applications to Single-Cell Sequencing”. In:
<em>J Comput Biol</em> 19.5 (May. 2012). 22506599[pmid], pp. 455-477. ISSN:
1066-5277. DOI: 10.1089/cmb.2012.0021[PII].
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3342519/" class="uri">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3342519/</a>.</p>
<p>[6] A. Beckerman, D. Childs, and O. Petchey. <em>Getting Started with R:
An Introduction for Biologists</em>. Oxford University Press, 2017.</p>
<p>[7] D. A. Benson, I. Karsch-Mizrachi, D. J. Lipman, et al. “GenBank”.
In: <em>Nucleic Acids Research</em> 33 (2005), pp. D34-D38.</p>
<p>[8] C. Bleidorn. <em>Phylogenomics</em>. Springer International Publishing,
2017.</p>
<p>[9] U. Bodenhofer, E. Bonatesta, C. Horejš-Kainrath, et al. “msa: an R
package for multiple sequence alignment”. In: <em>Bioinformatics</em> 31.24
(2015), pp. 3997-3999. DOI: 10.1093/bioinformatics/btv494. eprint:
/oup/backfile/content_public/journal/bioinformatics/31/24/10.1093/bioinformatics/btv494/2/btv494.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btv494&gt;.</p>
<p>[10] R. E. Bone, J. A. C. Smith, N. Arrigo, et al. “A macro-ecological
perspective on crassulacean acid metabolism (CAM) photosynthesis
evolution in Afro-Madagascan drylands: Eulophiinae orchids as a case
study”. In: <em>New Phytologist</em> 208.2 (2015). 2015-19192, pp. 469-481.
ISSN: 1469-8137. DOI: 10.1111/nph.13572.
<a href="http://dx.doi.org/10.1111/nph.13572" class="uri">http://dx.doi.org/10.1111/nph.13572</a>.</p>
<p>[11] S. Bory, O. Catrice, S. Brown, et al. “Natural polyploidy in
Vanilla planifolia (Orchidaceae)”. In: <em>Genome</em> 51.10 (2008). PMID:
18923533, pp. 816-826. DOI: 10.1139/G08-068. eprint:
<a href="https://doi.org/10.1139/G08-068" class="uri">https://doi.org/10.1139/G08-068</a></p>
<p>. <a href="https://doi.org/10.1139/G08-068" class="uri">https://doi.org/10.1139/G08-068</a>.</p>
<p>[12] S. Bory, M. Grisoni, M. Duval, et al. “Biodiversity and
preservation of vanilla: present state of knowledge”. In: <em>Genetic
Resources and Crop Evolution</em> 55.4 (Jun. 2008), pp. 551-571. ISSN:
1573-5109. DOI: 10.1007/s10722-007-9260-3.
<a href="https://doi.org/10.1007/s10722-007-9260-3" class="uri">https://doi.org/10.1007/s10722-007-9260-3</a>.</p>
<p>[13] A. Bouetard, P. Lefeuvre, R. Gigant, et al. “Evidence of
transoceanic dispersion of the genus Vanilla based on plastid DNA
phylogenetic analysis”. In: <em>Molecular Phylogenetics and Evolution</em>
55.2 (2010), pp. 621 - 630. ISSN: 1055-7903. DOI:
<a href="https://doi.org/10.1016/j.ympev.2010.01.021" class="uri">https://doi.org/10.1016/j.ympev.2010.01.021</a>.
<a href="http://www.sciencedirect.com/science/article/pii/S1055790310000230" class="uri">http://www.sciencedirect.com/science/article/pii/S1055790310000230</a>.</p>
<p>[14] K. R. Bradnam, J. N. Fass, A. Alexandrov, et al. “Assemblathon 2:
evaluating de novo methods of genome assembly in three vertebrate
species”. In: <em>GigaScience</em> 2.1 (2013), pp. 1-31. DOI:
10.1186/2047-217X-2-10. eprint:
/oup/backfile/content_public/journal/gigascience/2/1/10.1186_2047-217x-2-10/7/13742_2013_article_29.pdf.
&lt;+%20http://dx.doi.org/10.1186/2047-217X-2-10&gt;.</p>
<p>[15] T. A. Brown. <em>Genomes 4</em>. Garland Science, 2017.</p>
<p>[16] S. Buerki, F. Forest, N. Alvarez, et al. “An evaluation of new
parsimony-based versus parametric inference methods in biogeography: a
case study using the globally distributed plant family Sapindaceae”.
In: <em>Journal of Biogeography</em> 38.3 (2011), pp. 531-550. ISSN:
1365-2699. DOI: 10.1111/j.1365-2699.2010.02432.x.
<a href="http://dx.doi.org/10.1111/j.1365-2699.2010.02432.x" class="uri">http://dx.doi.org/10.1111/j.1365-2699.2010.02432.x</a>.</p>
<p>[17] S. Buerki, P. P. Lowry II, J. Munzinger, et al. “Alectryon
vitiensis: A new species of Sapindaceae endemic to Fiji”. In: <em>Novon: A
Journal for Botanical Nomenclature</em> 25.4 (2017), pp. 421-429.</p>
<p>[18] J. Burnier, S. Buerki, N. Arrigo, et al. “Genetic structure and
evolution of Alpine polyploid complexes: Ranunculus kuepferi
(Ranunculaceae) as a case study”. In: <em>Molecular Ecology</em> 18.17 (2009),
pp. 3730-3744. ISSN: 1365-294X. DOI: 10.1111/j.1365-294X.2009.04281.x.
<a href="http://dx.doi.org/10.1111/j.1365-294X.2009.04281.x" class="uri">http://dx.doi.org/10.1111/j.1365-294X.2009.04281.x</a>.</p>
<p>[19] M. A. Butler and A. A. King. “Phylogenetic comparative analysis: a
modeling approach for adaptive evolution”. In: <em>American Naturalist</em>
164 (2004), pp. 683-695.
<a href="http://www.press.uchicago.edu/ucp/journals/journal/an.html" class="uri">http://www.press.uchicago.edu/ucp/journals/journal/an.html</a>.</p>
<p>[20] J. Cai, X. Liu, K. Vanneste, et al. “The genome sequence of the
orchid Phalaenopsis equestris”. In: <em>Nature Genetics</em> 47.1 (2015), pp.
65-+. ISSN: 1061-4036. DOI: 10.1038/ng.3149.
&lt;%3CGo%20to%20ISI%3E://WOS:000346990400013&gt;.</p>
<p>[21] CBOL. <em>International Barcode of Life: DNA barcoding</em>. 2021.
<a href="https://ibol.org/about/dna-barcoding/" class="uri">https://ibol.org/about/dna-barcoding/</a>.</p>
<p>[22] S. Chamberlain. <em>rgbif: Interface to the Global ‘Biodiversity’
Information Facility ‘API’</em>. 2016.
<a href="https://CRAN.R-project.org/package=rgbif" class="uri">https://CRAN.R-project.org/package=rgbif</a>.</p>
<p>[23] D. Charif and J. Lobry. “SeqinR 1.0-2: a contributed package to
the R project for statistical computing devoted to biological sequences
retrieval and analysis.” In: <em>Structural approaches to sequence
evolution: Molecules, networks, populations</em>. Ed. by U. Bastolla, M.
Porto, H. Roman and M. Vendruscolo. Biological and Medical Physics,
Biomedical Engineering. ISBN : 978-3-540-35305-8. New York: Springer
Verlag, 2007, pp. 207-232.</p>
<p>[24] F. Chaumont, M. Moshelion, and M. J. Daniels. “Regulation of plant
aquaporin activity”. In: <em>Biology of the Cell</em> 97.10 (2005), pp.
749-764.</p>
<p>[25] T. Cheng, C. Xu, L. Lei, et al. “Barcoding the kingdom Plantae:
new PCR primers for ITS regions of plants with improved universality
and specificity”. In: <em>Molecular Ecology Resources</em> 16.1 (2016), pp.
138-149. DOI: 10.1111/1755-0998.12438. eprint:
<a href="https://onlinelibrary.wiley.com/doi/pdf/10.1111/1755-0998.12438" class="uri">https://onlinelibrary.wiley.com/doi/pdf/10.1111/1755-0998.12438</a>.
<a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.12438" class="uri">https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.12438</a>.</p>
<p>[26] J. Claverie and C. Notredame. <em>Bioinformatics For Dummies, 2nd
Edition</em>. Wiley, 2006.</p>
<p>[27] A. Coghlan, T. J. Fiedler, S. J. McKay, et al. “nGASP - the
nematode genome annotation assessment project”. In: <em>BMC
Bioinformatics</em> 9.1 (Dec. 2008), p. 549. ISSN: 1471-2105. DOI:
10.1186/1471-2105-9-549. <a href="https://doi.org/10.1186/1471-2105-9-549" class="uri">https://doi.org/10.1186/1471-2105-9-549</a>.</p>
<p>[28] M. J. Crawley. <em>The R Book, 2nd Edition</em>. Wiley, 2012.</p>
<p>[29] M. Crusoe, H. Alameldin, S. Awad, et al. “The khmer software
package: enabling efficient nucleotide sequence analysis [version 1;
referees: 2 approved, 1 approved with reservations]”. In:
<em>F1000Research</em> 4.900 (2015). DOI: 10.12688/f1000research.6924.1.</p>
<p>[30] A. Dai. “Drought under global warming: a review”. In: <em>Wiley
Interdisciplinary Reviews: Climate Change</em> 2.1 (2011), pp. 45-65. ISSN:
1757-7799. DOI: 10.1002/wcc.81. <a href="http://dx.doi.org/10.1002/wcc.81" class="uri">http://dx.doi.org/10.1002/wcc.81</a>.</p>
<p>[31] J. W. Dale, M. von Schantz, and N. Plant. <em>From Genes to Genomes:
Concepts and Applications of DNA Technology, 3rd Edition</em>. Wiley, 2012.</p>
<p>[32] J. . Danielson and U. Johanson. “Unexpected complexity of the
aquaporin gene family in the moss Physcomitrella patens”. In: <em>BMC
plant biology</em> 8.1 (2008), pp. 1-15.</p>
<p>[33] B. E. Davidson, S. J. Novak, and M. D. Serpe. “Consequences of
inoculation with native arbuscular mycorrhizal fungi for root
colonization and survival of Artemisia tridentata ssp. wyomingensis
seedlings after transplanting”. In: <em>Mycorrhiza</em> 26.6 (Aug. 2016), pp.
595-608. ISSN: 1432-1890. DOI: 10.1007/s00572-016-0696-1.
<a href="https://doi.org/10.1007/s00572-016-0696-1" class="uri">https://doi.org/10.1007/s00572-016-0696-1</a>.</p>
<p>[34] B. T. Dentinger and L. M. Suz. “What’s for dinner? Undescribed
species of porcini in a commercial packet”. In: <em>PeerJ</em> 2 (2014), p.
e570.</p>
<p>[35] R. DeSalle and P. Goldstein. “Review and Interpretation of Trends
in DNA Barcoding”. In: <em>Frontiers in Ecology and Evolution</em> 7 (2019),
p. 302. ISSN: 2296-701X. DOI: 10.3389/fevo.2019.00302.
<a href="https://www.frontiersin.org/article/10.3389/fevo.2019.00302" class="uri">https://www.frontiersin.org/article/10.3389/fevo.2019.00302</a>.</p>
<p>[36] R. deSalle and J. Rosenfeld. <em>Phylogenomics: A Primer 1st
Edition</em>. Garland Science, 2013.</p>
<p>[37] S. Dodsworth, M. W. Chase, L. J. Kelly, et al. “Genomic Repeat
Abundances Contain Phylogenetic Signal”. In: <em>Systematic Biology</em> 64.1
(2015), pp. 112-126. DOI: 10.1093/sysbio/syu080. eprint:
/oup/backfile/content_public/journal/sysbio/64/1/10.1093_sysbio_syu080/1/syu080.pdf.
&lt;+%20http://dx.doi.org/10.1093/sysbio/syu080&gt;.</p>
<p>[38] S. Dray and A. Dufour. “The ade4 package: implementing the duality
diagram for ecologists”. In: <em>Journal of Statistical Software</em> 22.4
(2007), pp. 1-20.</p>
<p>[39] R. C. Edgar. “MUSCLE: multiple sequence alignment with high
accuracy and high throughput”. In: <em>Nucleic acids research</em> 32.5
(2004), pp. 1792-1797.</p>
<p>[40] R. Ekblom and J. B. W. Wolf. “A field guide to whole-genome
sequencing, assembly and annotation”. In: <em>Evolutionary Applications</em>
7.9 (2014), pp. 1026-1042. ISSN: 1752-4571. DOI: 10.1111/eva.12178.
<a href="http://dx.doi.org/10.1111/eva.12178" class="uri">http://dx.doi.org/10.1111/eva.12178</a>.</p>
<p>[41] P. Ellestad, F. Forest, M. Serpe, et al. “Harnessing large-scale
biodiversity data to infer the current distribution of Vanilla
planifolia (Orchidaceae)”. In: <em>Botanical Journal of the Linnean
Society</em> (Feb. 2021). boab005. ISSN: 0024-4074. DOI:
10.1093/botlinnean/boab005. eprint:
<a href="https://academic.oup.com/botlinnean/advance-article-pdf/doi/10.1093/botlinnean/boab005/36210432/boab005.pdf" class="uri">https://academic.oup.com/botlinnean/advance-article-pdf/doi/10.1093/botlinnean/boab005/36210432/boab005.pdf</a>.
<a href="https://doi.org/10.1093/botlinnean/boab005" class="uri">https://doi.org/10.1093/botlinnean/boab005</a>.</p>
<p>[42] J. Flexas, J. M. Escalona, S. Evain, et al. “Steady-state
chlorophyll fluorescence (Fs) measurements as a tool to follow
variations of net CO2 assimilation and stomatal conductance during
water-stress in C3 plants”. In: <em>Physiologia Plantarum</em> 114.2 (2002),
pp. 231-240. ISSN: 1399-3054. DOI: 10.1034/j.1399-3054.2002.1140209.x.
<a href="http://dx.doi.org/10.1034/j.1399-3054.2002.1140209.x" class="uri">http://dx.doi.org/10.1034/j.1399-3054.2002.1140209.x</a>.</p>
<p>[43] J. Gaffney, R. Tibebu, R. Bart, et al. “Open access to genetic
sequence data maximizes value to scientists, farmers, and society”. In:
<em>Global Food Security</em> 26 (2020), p. 100411. ISSN: 2211-9124. DOI:
<a href="https://doi.org/10.1016/j.gfs.2020.100411" class="uri">https://doi.org/10.1016/j.gfs.2020.100411</a>.
<a href="https://www.sciencedirect.com/science/article/pii/S2211912420300651" class="uri">https://www.sciencedirect.com/science/article/pii/S2211912420300651</a>.</p>
<p>[44] H. Gehrig, K. Faist, and M. Kluge. “Identification of
phosphoenolpyruvate carboxylase isoforms in leaf, stem and roots of the
obligate CAM plant Vanilla planifolia Salib. (Orchidaceae): a
physiological and molecular approach”. In: <em>Plant Molecular Biology</em>
38.6 (Dec. 1998), pp. 1215-1223. ISSN: 1573-5028. DOI:
10.1023/A:1006006331011. <a href="https://doi.org/10.1023/A:1006006331011" class="uri">https://doi.org/10.1023/A:1006006331011</a>.</p>
<p>[45] S. Gnerre, I. MacCallum, D. Przybylski, et al. “High-quality draft
assemblies of mammalian genomes from massively parallel sequence data”.
In: <em>Proceedings of the National Academy of Sciences</em> 108.4 (2011), pp.
1513-1518. ISSN: 0027-8424. DOI: 10.1073/pnas.1017351108. eprint:
<a href="http://www.pnas.org/content/108/4/1513.full.pdf" class="uri">http://www.pnas.org/content/108/4/1513.full.pdf</a>.
<a href="http://www.pnas.org/content/108/4/1513" class="uri">http://www.pnas.org/content/108/4/1513</a>.</p>
<p>[46] S. Gnerre, I. MacCallum, D. Przybylski, et al. “High-quality draft
assemblies of mammalian genomes from massively parallel sequence data”.
In: <em>Proceedings of the National Academy of Sciences</em> 108.4 (2011), pp.
1513-1518. DOI: 10.1073/pnas.1017351108. eprint:
<a href="http://www.pnas.org/content/108/4/1513.full.pdf" class="uri">http://www.pnas.org/content/108/4/1513.full.pdf</a>.
<a href="http://www.pnas.org/content/108/4/1513.abstract" class="uri">http://www.pnas.org/content/108/4/1513.abstract</a>.</p>
<p>[47] P. F. Gonçalves, A. R. Oliveira-Marques, T. E. Matsumoto, et al.
“DNA barcoding identifies illegal parrot trade”. In: <em>Journal of
Heredity</em> 106.S1 (2015), pp. 560-564.</p>
<p>[48] M. G. Grabherr, B. J. Haas, M. Yassour, et al. “Full-length
transcriptome assembly from RNA-Seq data without a reference genome”.
In: <em>Nature Biotechnology</em> 29 (May. 2011), pp. 644 EP -.
<a href="http://dx.doi.org/10.1038/nbt.1883" class="uri">http://dx.doi.org/10.1038/nbt.1883</a>.</p>
<p>[49] A. Gurevich, V. Saveliev, N. Vyahhi, et al. “QUAST: quality
assessment tool for genome assemblies”. In: <em>Bioinformatics</em> 29.8
(2013), pp. 1072-1075. DOI: 10.1093/bioinformatics/btt086. eprint:
/oup/backfile/content_public/journal/bioinformatics/29/8/10.1093_bioinformatics_btt086/2/btt086.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btt086&gt;.</p>
<p>[50] C. Hahn, L. Bachmann, and B. Chevreux. “Reconstructing
mitochondrial genomes directly from genomic next-generation sequencing
reads—a baiting and iterative mapping approach”. In: <em>Nucleic Acids
Research</em> 41.13 (2013), p. e129. DOI: 10.1093/nar/gkt371. eprint:
/oup/backfile/content_public/journal/nar/41/13/10.1093_nar_gkt371/3/gkt371.pdf.
&lt;+%20http://dx.doi.org/10.1093/nar/gkt371&gt;.</p>
<p>[51] D. L. Hartl. “Fly meets shotgun: shotgun wins”. In: <em>Nature
Genetics</em> 24 (Apr. 2000), p. 327 EP. <a href="http://dx.doi.org/10.1038/74125" class="uri">http://dx.doi.org/10.1038/74125</a>.</p>
<p>[52] R. J. Hijmans, S. E. Cameron, J. L. Parra, et al. “Very high
resolution interpolated climate surfaces for global land areas”. In:
<em>International Journal of Climatology</em> 25.15 (2005), pp. 1965-1978.
ISSN: 1097-0088. DOI: 10.1002/joc.1276.
<a href="http://dx.doi.org/10.1002/joc.1276" class="uri">http://dx.doi.org/10.1002/joc.1276</a>.</p>
<p>[53] P. M. Hollingsworth, L. L. Forrest, J. L. Spouge, et al. “A DNA
barcode for land plants”. In: <em>Proceedings of the National Academy of
Sciences</em> 106.31 (2009), pp. 12794-12797. ISSN: 0027-8424. DOI:
10.1073/pnas.0905845106. eprint:
<a href="https://www.pnas.org/content/106/31/12794.full.pdf" class="uri">https://www.pnas.org/content/106/31/12794.full.pdf</a>.
<a href="https://www.pnas.org/content/106/31/12794" class="uri">https://www.pnas.org/content/106/31/12794</a>.</p>
<p>[54] C. Holt and M. Yandell. “MAKER2: an annotation pipeline and
genome-database management tool for second-generation genome projects”.
In: <em>BMC Bioinformatics</em> 12.1 (Dec. 2011), p. 491. ISSN: 1471-2105.
DOI: 10.1186/1471-2105-12-491.
<a href="https://doi.org/10.1186/1471-2105-12-491" class="uri">https://doi.org/10.1186/1471-2105-12-491</a>.</p>
<p>[55] T. A. G. Initiative. “Analysis of the genome sequence of the
flowering plant Arabidopsis thaliana”. In: <em>Nature</em> 408.6814 (Dec.
2000), pp. 796-815. ISSN: 1476-4687. DOI: 10.1038/35048692.
<a href="https://doi.org/10.1038/35048692" class="uri">https://doi.org/10.1038/35048692</a>.</p>
<p>[56] K. Ishibashi. “Aquaporin subfamily with unusual NPA boxes”. In:
<em>Biochimica et Biophysica Acta (BBA)-Biomembranes</em> 1758.8 (2006), pp.
989-993.</p>
<p>[57] S. D. Jackman, B. P. Vandervalk, H. Mohamadi, et al. “ABySS 2.0:
resource-efficient assembly of large genomes using a Bloom filter”. In:
<em>Genome Research</em> 27.5 (2017), pp. 768-777. DOI: 10.1101/gr.214346.116.
eprint: <a href="http://genome.cshlp.org/content/27/5/768.full.pdf+html" class="uri">http://genome.cshlp.org/content/27/5/768.full.pdf+html</a>.
<a href="http://genome.cshlp.org/content/27/5/768.abstract" class="uri">http://genome.cshlp.org/content/27/5/768.abstract</a>.</p>
<p>[58] J. Jersáková, P. Trávníček, B. Kubátová, et al. “Genome size
variation in Orchidaceae subfamily Apostasioideae: filling the
phylogenetic gap”. In: <em>Botanical Journal of the Linnean Society</em> 172.1
(2013), pp. 95-105. ISSN: 1095-8339. DOI: 10.1111/boj.12027.
<a href="http://dx.doi.org/10.1111/boj.12027" class="uri">http://dx.doi.org/10.1111/boj.12027</a>.</p>
<p>[59] K. Katoh, J. Rozewicki, and K. D. Yamada. “MAFFT online service:
multiple sequence alignment, interactive sequence choice and
visualization”. In: <em>Briefings in Bioinformatics</em> 20.4 (Sep. 2017), pp.
1160-1166. ISSN: 1477-4054. DOI: 10.1093/bib/bbx108. eprint:
<a href="https://academic.oup.com/bib/article-pdf/20/4/1160/30119669/bbx108.pdf" class="uri">https://academic.oup.com/bib/article-pdf/20/4/1160/30119669/bbx108.pdf</a>.
<a href="https://doi.org/10.1093/bib/bbx108" class="uri">https://doi.org/10.1093/bib/bbx108</a>.</p>
<p>[60] M. Kearse, R. Moir, A. Wilson, et al. “Geneious Basic: An
integrated and extendable desktop software platform for the
organization and analysis of sequence data”. In: <em>Bioinformatics</em> 28.12
(2012), pp. 1647-1649. DOI: 10.1093/bioinformatics/bts199. eprint:
/oup/backfile/content_public/journal/bioinformatics/28/12/10.1093/bioinformatics/bts199/2/bts199.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/bts199&gt;.</p>
<p>[61] J. M. Keith. <em>Bioinformatics, Volume I: Data, Sequence Analysis,
and Evolution</em>. Springer Science+Business Media New York, 2017.</p>
<p>[62] D. R. Kelley, M. C. Schatz, and S. L. Salzberg. “Quake:
quality-aware detection and correction of sequencing errors”. In:
<em>Genome Biology</em> 11.11 (2010), p. R116. ISSN: 1474-760X. DOI:
10.1186/gb-2010-11-11-r116.
<a href="https://doi.org/10.1186/gb-2010-11-11-r116" class="uri">https://doi.org/10.1186/gb-2010-11-11-r116</a>.</p>
<p>[63] L. A. Kelley, S. Mezulis, C. M. Yates, et al. “The Phyre2 web
portal for protein modeling, prediction and analysis”. In: <em>Nature
protocols</em> 10.6 (2015), pp. 845-858.</p>
<p>[64] S. Kembel, P. Cowan, M. Helmus, et al. “Picante: R tools for
integrating phylogenies and ecology”. In: <em>Bioinformatics</em> 26 (2010),
pp. 1463-1464.</p>
<p>[65] A. A. Kolesnikov and E. S. Gerasimov. “Diversity of mitochondrial
genome organization”. English. In: <em>BIOCHEMISTRY-MOSCOW</em> 77.13 (), pp.
1424-1435. ISSN: 0006-2979. DOI: 10.1134/S0006297912130020.</p>
<p>[66] S. Koren, B. P. Walenz, K. Berlin, et al. “Canu: scalable and
accurate long-read assembly via adaptive k-mer weighting and repeat
separation”. In: <em>Genome Research</em> 27.5 (2017), pp. 722-736. DOI:
10.1101/gr.215087.116. eprint:
<a href="http://genome.cshlp.org/content/27/5/722.full.pdf+html" class="uri">http://genome.cshlp.org/content/27/5/722.full.pdf+html</a>.
<a href="http://genome.cshlp.org/content/27/5/722.abstract" class="uri">http://genome.cshlp.org/content/27/5/722.abstract</a>.</p>
<p>[67] A. M. Kozlov, D. Darriba, T. Flouri, et al. “RAxML-NG: a fast,
scalable and user-friendly tool for maximum likelihood phylogenetic
inference”. In: <em>Bioinformatics</em> 35.21 (May. 2019), pp. 4453-4455.
ISSN: 1367-4803. DOI: 10.1093/bioinformatics/btz305. eprint:
<a href="https://academic.oup.com/bioinformatics/article-pdf/35/21/4453/30330793/btz305.pdf" class="uri">https://academic.oup.com/bioinformatics/article-pdf/35/21/4453/30330793/btz305.pdf</a>.
<a href="https://doi.org/10.1093/bioinformatics/btz305" class="uri">https://doi.org/10.1093/bioinformatics/btz305</a>.</p>
<p>[68] A. Krogh, B. Larsson, G. Von Heijne, et al. “Predicting
transmembrane protein topology with a hidden Markov model: application
to complete genomes”. In: <em>Journal of molecular biology</em> 305.3 (2001),
pp. 567-580.</p>
<p>[69] S. Kumar, G. Stecher, M. Li, et al. “MEGA X: Molecular
Evolutionary Genetics Analysis across Computing Platforms”. In:
<em>Molecular Biology and Evolution</em> 35.6 (May. 2018), pp. 1547-1549.
ISSN: 0737-4038. DOI: 10.1093/molbev/msy096. eprint:
<a href="https://academic.oup.com/mbe/article-pdf/35/6/1547/25091958/msy096.pdf" class="uri">https://academic.oup.com/mbe/article-pdf/35/6/1547/25091958/msy096.pdf</a>.
<a href="https://doi.org/10.1093/molbev/msy096" class="uri">https://doi.org/10.1093/molbev/msy096</a>.</p>
<p>[70] S. Kumar, G. Stecher, M. Li, et al. “MEGA X: molecular
evolutionary genetics analysis across computing platforms”. In:
<em>Molecular biology and evolution</em> 35.6 (2018), pp. 1547-1549.</p>
<p>[71] S. Kurtz, A. Phillippy, A. L. Delcher, et al. “Versatile and open
software for comparing large genomes”. In: <em>Genome Biology</em> 5.2 (Jan.
2004), p. R12. ISSN: 1474-760X. DOI: 10.1186/gb-2004-5-2-r12.
<a href="https://doi.org/10.1186/gb-2004-5-2-r12" class="uri">https://doi.org/10.1186/gb-2004-5-2-r12</a>.</p>
<p>[72] B. B. Larsen, E. C. Miller, M. K. Rhodes, et al. “Inordinate
Fondness Multiplied and Redistributed: the Number of Species on Earth
and the New Pie of Life”. In: <em>The Quarterly Review of Biology</em> 92.3
(2017), pp. 229-265. DOI: 10.1086/693564.</p>
<p>[73] A. Larsson. “AliView: a fast and lightweight alignment viewer and
editor for large datasets”. In: <em>Bioinformatics</em> 30.22 (Aug. 2014), pp.
3276-3278. ISSN: 1367-4803. DOI: 10.1093/bioinformatics/btu531. eprint:
<a href="https://academic.oup.com/bioinformatics/article-pdf/30/22/3276/17146391/btu531.pdf" class="uri">https://academic.oup.com/bioinformatics/article-pdf/30/22/3276/17146391/btu531.pdf</a>.
<a href="https://doi.org/10.1093/bioinformatics/btu531" class="uri">https://doi.org/10.1093/bioinformatics/btu531</a>.</p>
<p>[74] M. Lawrence, W. Huber, H. Pagès, et al. “Software for Computing
and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8
2013). DOI: 10.1371/journal.pcbi.1003118.
<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118%7D." class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118}.</a></p>
<p>[75] P. Lemey, M. Salemi, and A. Vandamme. <em>The Phylogenetic Handbook:
A Practical Approach to Phylogenetic Analysis and Hypothesis Testing,
Edition 2</em>. Cambridge University Press, 2009.</p>
<p>[76] A. Lesk. <em>Introduction to Genomics, Third edition</em>. Oxford
University Press, 2017.</p>
<p>[77] G. Li, V. Santoni, and C. Maurel. “Plant aquaporins: roles in
plant physiology”. In: <em>Biochimica et Biophysica Acta (BBA)-General
Subjects</em> 1840.5 (2014), pp. 1574-1582.</p>
<p>[78] H. Li and R. Durbin. “Fast and accurate short read alignment with
Burrows–Wheeler transform”. In: <em>Bioinformatics</em> 25.14 (2009), pp.
1754-1760. DOI: 10.1093/bioinformatics/btp324. eprint:
/oup/backfile/content_public/journal/bioinformatics/25/14/10.1093/bioinformatics/btp324/2/btp324.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btp324&gt;.</p>
<p>[79] H. Li, B. Handsaker, A. Wysoker, et al. “The Sequence
Alignment/Map format and SAMtools”. In: <em>Bioinformatics</em> 25.16 (2009),
pp. 2078-2079. DOI: 10.1093/bioinformatics/btp352. eprint:
/oup/backfile/content_public/journal/bioinformatics/25/16/10.1093/bioinformatics/btp352/2/btp352.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btp352&gt;.</p>
<p>[80] R. Li, W. Fan, G. Tian, et al. “The sequence and de novo assembly
of the giant panda genome”. In: <em>Nature</em> 463 (Dec. 2009). Article, pp.
311 EP -. <a href="http://dx.doi.org/10.1038/nature08696" class="uri">http://dx.doi.org/10.1038/nature08696</a>.</p>
<p>[81] C. Lin, J. J. W. Chen, Y. Huang, et al. “The location and
translocation of ndh genes of chloroplast origin in the Orchidaceae
family”. In: <em>Scientific Reports</em> 5 (Mar. 2015). Article, pp. 9040 EP
-. <a href="http://dx.doi.org/10.1038/srep09040" class="uri">http://dx.doi.org/10.1038/srep09040</a>.</p>
<p>[82] R. Luo, B. Liu, Y. Xie, et al. “SOAPdenovo2: an empirically
improved memory-efficient short-read de novo assembler”. In:
<em>GigaScience</em> 1.1 (Dec. 2012), p. 18. ISSN: 2047-217X. DOI:
10.1186/2047-217X-1-18. <a href="https://doi.org/10.1186/2047-217X-1-18" class="uri">https://doi.org/10.1186/2047-217X-1-18</a>.</p>
<p>[83] J. Males and H. Griffiths. “Stomatal biology of CAM plants”. In:
<em>Plant Physiology</em> (2017). ISSN: 0032-0889. DOI: 10.1104/pp.17.00114.
eprint:
<a href="http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114.full.pdf" class="uri">http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114.full.pdf</a>.
<a href="http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114" class="uri">http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114</a>.</p>
<p>[84] D. Mapleson, G. Garcia Accinelli, G. Kettleborough, et al. “KAT: a
K-mer analysis toolkit to quality control NGS datasets and genome
assemblies”. In: <em>Bioinformatics</em> 33.4 (2017), pp. 574-576. DOI:
10.1093/bioinformatics/btw663. eprint:
/oup/backfile/content_public/journal/bioinformatics/33/4/10.1093_bioinformatics_btw663/4/btw663.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btw663&gt;.</p>
<p>[85] G. Marçais and C. Kingsford. “A fast, lock-free approach for
efficient parallel counting of occurrences of k-mers”. In:
<em>Bioinformatics</em> 27.6 (2011), pp. 764-770. DOI:
10.1093/bioinformatics/btr011. eprint:
/oup/backfile/content_public/journal/bioinformatics/27/6/10.1093_bioinformatics_btr011/2/btr011.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btr011&gt;.</p>
<p>[86] F. Markowetz. “Five selfish reasons to work reproducibly”. In:
<em>Genome Biology</em> 16.1 (Dec. 2015), p. 274. ISSN: 1474-760X. DOI:
10.1186/s13059-015-0850-7. <a href="https://doi.org/10.1186/s13059-015-0850-7" class="uri">https://doi.org/10.1186/s13059-015-0850-7</a>.</p>
<p>[87] J. A. Martin and Z. Wang. “Next-generation transcriptome
assembly”. In: <em>Nature Reviews Genetics</em> 12 (Sep. 2011). Review
Article, pp. 671 EP -. <a href="http://dx.doi.org/10.1038/nrg3068" class="uri">http://dx.doi.org/10.1038/nrg3068</a>.</p>
<p>[88] V. Marx. “Method of the year: long-read sequencing”. In: <em>Nature
Methods</em> 20.1 (Jan. 2023), pp. 6-11. ISSN: 1548-7105. DOI:
10.1038/s41592-022-01730-w.
<a href="https://doi.org/10.1038/s41592-022-01730-w" class="uri">https://doi.org/10.1038/s41592-022-01730-w</a>.</p>
<p>[89] J. C. Masters and L. Pozzi. “Phylogenetic Inference”. In: <em>The
International Encyclopedia of Primatology</em>. American Cancer Society,
2017, pp. 1-6. ISBN: 9781119179313. DOI:
<a href="https://doi.org/10.1002/9781119179313.wbprim0419" class="uri">https://doi.org/10.1002/9781119179313.wbprim0419</a>. eprint:
<a href="https://onlinelibrary.wiley.com/doi/pdf/10.1002/9781119179313.wbprim0419" class="uri">https://onlinelibrary.wiley.com/doi/pdf/10.1002/9781119179313.wbprim0419</a>.
<a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/9781119179313.wbprim0419" class="uri">https://onlinelibrary.wiley.com/doi/abs/10.1002/9781119179313.wbprim0419</a>.</p>
<p>[90] N. J. Matzke. <em>BioGeoBEARS: BioGeography with Bayesian (and
Likelihood) Evolutionary Analysis in R Scripts</em>. The
rexpokit/cladoRcpp/BioGeoBEARS packages were first published in:
Matzke, Nicholas J. (2012). Founder-event speciation in BioGeoBEARS
package dramatically improves likelihoods and alters parameter
inference in Dispersal-Extinction-Cladogenesis (DEC) analyses.
Frontiers of Biogeography, 4(suppl. 1): 210, December 2012. Link to
abstract and PDF of poster:
<a href="http://phylo.wikidot.com/matzke-2013-international-biogeography-society-poster" class="uri">http://phylo.wikidot.com/matzke-2013-international-biogeography-society-poster</a>
. (Poster abstract published in the Conference Program and Abstracts of
the International Biogeography Society 6th Biannual Meeting, Miami,
Florida. Poster Session P10: Historical and Paleo-Biogeography. Poster
129B. January 11, 2013.). University of California, Berkeley. Berkeley,
CA, 2013. <a href="http://CRAN.R-project.org/package=BioGeoBEARS" class="uri">http://CRAN.R-project.org/package=BioGeoBEARS</a>.</p>
<p>[91] K. Maxwell and G. N. Johnson. “Chlorophyll fluorescence—a
practical guide”. In: <em>Journal of Experimental Botany</em> 51.345 (2000),
pp. 659-668. DOI: 10.1093/jexbot/51.345.659. eprint:
/oup/backfile/content_public/journal/jxb/51/345/10.1093_jexbot_51.345.659/1/510659.pdf.
&lt;+%20http://dx.doi.org/10.1093/jexbot/51.345.659&gt;.</p>
<p>[92] A. E. Melton, J. Beck, S. J. Galla, et al. “A draft genome
provides hypotheses on drought tolerance in a keystone plant species in
Western North America threatened by climate change”. In: <em>Ecology and
Evolution</em> 11.21 (2021), pp. 15417-15429. DOI:
<a href="https://doi.org/10.1002/ece3.8245" class="uri">https://doi.org/10.1002/ece3.8245</a>. eprint:
<a href="https://onlinelibrary.wiley.com/doi/pdf/10.1002/ece3.8245" class="uri">https://onlinelibrary.wiley.com/doi/pdf/10.1002/ece3.8245</a>.
<a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.8245" class="uri">https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.8245</a>.</p>
<p>[93] A. E. Melton, A. W. Child, J. Beard Richard S, et al. “A haploid
pseudo-chromosome genome assembly for a keystone sagebrush species of
western North American rangelands”. In: <em>G3 Genes|Genomes|Genetics</em>
12.7 (May. 2022). jkac122. ISSN: 2160-1836. DOI:
10.1093/g3journal/jkac122. eprint:
<a href="https://academic.oup.com/g3journal/article-pdf/12/7/jkac122/44473065/jkac122.pdf" class="uri">https://academic.oup.com/g3journal/article-pdf/12/7/jkac122/44473065/jkac122.pdf</a>.
<a href="https://doi.org/10.1093/g3journal/jkac122" class="uri">https://doi.org/10.1093/g3journal/jkac122</a>.</p>
<p>[94] R. Ming, R. VanBuren, C. M. Wai, et al. “The pineapple genome and
the evolution of CAM photosynthesis”. In: <em>Nature Genetics</em> 47.12 (Dec.
2015), p. 1435. ISSN: 1061-4036. DOI: 10.1038/ng.3435.</p>
<p>[95] H. Mizuno, Y. Kawahara, H. Sakai, et al. “Massive parallel
sequencing of mRNA in identification of unannotated salinity
stress-inducible transcripts in rice (Oryza sativa L.)”. In: <em>BMC
Genomics</em> 11.1 (Dec. 2010), p. 683. ISSN: 1471-2164. DOI:
10.1186/1471-2164-11-683. <a href="https://doi.org/10.1186/1471-2164-11-683" class="uri">https://doi.org/10.1186/1471-2164-11-683</a>.</p>
<p>[96] L. S. Mofatto, F. d. A. Carneiro, N. G. Vieira, et al.
“Identification of candidate genes for drought tolerance in coffee by
high-throughput sequencing in the shoot apex of different Coffea
arabica cultivars”. In: <em>BMC Plant Biology</em> 16.1 (Apr. 2016), p. 94.
ISSN: 1471-2229. DOI: 10.1186/s12870-016-0777-5.
<a href="https://doi.org/10.1186/s12870-016-0777-5" class="uri">https://doi.org/10.1186/s12870-016-0777-5</a>.</p>
<p>[97] M. C. Munoz-Torres, J. T. Reese, C. P. Childers, et al.
“Hymenoptera Genome Database: integrated community resources for insect
species of the order Hymenoptera”. In: <em>Nucleic Acids Research</em>
39.suppl 1 (2011), pp. D658-D662. DOI: 10.1093/nar/gkq1145.</p>
<p>[98] Z. Niu, J. Pan, S. Zhu, et al. “Comparative Analysis of the
Complete Plastomes of Apostasia wallichii and Neuwiedia singapureana
(Apostasioideae) Reveals Different Evolutionary Dynamics of IR/SSC
Boundary among Photosynthetic Orchids”. In: <em>Frontiers in Plant
Science</em> 8 (2017), p. 1713. ISSN: 1664-462X. DOI:
10.3389/fpls.2017.01713.
<a href="https://www.frontiersin.org/article/10.3389/fpls.2017.01713" class="uri">https://www.frontiersin.org/article/10.3389/fpls.2017.01713</a>.</p>
<p>[99] M. H. O’Leary. “Carbon Isotopes in Photosynthesis”. In:
<em>BioScience</em> 38.5 (1988), pp. 328-336. ISSN: 00063568, 15253244.
<a href="http://www.jstor.org/stable/1310735" class="uri">http://www.jstor.org/stable/1310735</a>.</p>
<p>[100] H. Pagès, P. Aboyoun, R. Gentleman, et al. <em>Biostrings: Efficient
manipulation of biological strings</em>. R package version 2.54.0. 2019.</p>
<p>[101] E. Paradis. <em>Analysis of Phylogenetics and Evolution with R</em>.
Springer, 2012.</p>
<p>[102] E. Paradis, J. Claude, and K. Strimmer. “APE: analyses of
phylogenetics and evolution in R language”. In: <em>Bioinformatics</em> 20
(2004), pp. 289-290.</p>
<p>[103] E. Paradis and K. Schliep. “ape 5.0: an environment for modern
phylogenetics and evolutionary analyses in R”. In: <em>Bioinformatics</em> 35
(2019), pp. 526-528.</p>
<p>[104] J. Pellicer, M. F. Fay, and I. J. Leitch. “The largest eukaryotic
genome of them all?” In: <em>Botanical Journal of the Linnean Society</em>
164.1 (2010), pp. 10-15. DOI: 10.1111/j.1095-8339.2010.01072.x. eprint:
/oup/backfile/content_public/journal/botlinnean/164/1/10.1111_j.1095-8339.2010.01072.x/2/j.1095-8339.2010.01072.x.pdf.
&lt;+%20http://dx.doi.org/10.1111/j.1095-8339.2010.01072.x&gt;.</p>
<p>[105] R. Poplin, P. Chang, D. Alexander, et al. “A universal SNP and
small-indel variant caller using deep neural networks”. In: <em>Nature
Biotechnology</em> 36.10 (2018), pp. 983-987. ISSN: 1546-1696. DOI:
10.1038/nbt.4235. <a href="https://doi.org/10.1038/nbt.4235" class="uri">https://doi.org/10.1038/nbt.4235</a>.</p>
<p>[106] C. A. Quinto, R. Tinoco, and R. S. Hellberg. “DNA barcoding
reveals mislabeling of game meat species on the US commercial market”.
In: <em>Food Control</em> 59 (2016), pp. 386-392.</p>
<p>[107] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2016. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
<p>[108] A. Raj, M. Stephens, and J. K. Pritchard. “fastSTRUCTURE:
Variational Inference of Population Structure in Large SNP Data Sets”.
In: <em>Genetics</em> 197.2 (2014), pp. 573-589. ISSN: 0016-6731. DOI:
10.1534/genetics.114.164350. eprint:
<a href="http://www.genetics.org/content/197/2/573.full.pdf" class="uri">http://www.genetics.org/content/197/2/573.full.pdf</a>.
<a href="http://www.genetics.org/content/197/2/573" class="uri">http://www.genetics.org/content/197/2/573</a>.</p>
<p>[109] M. G. Reese and R. Guigó. “EGASP: Introduction”. In: <em>Genome
Biology</em> 7.1 (Aug. 2006), p. S1. ISSN: 1474-760X. DOI:
10.1186/gb-2006-7-s1-s1. <a href="https://doi.org/10.1186/gb-2006-7-s1-s1" class="uri">https://doi.org/10.1186/gb-2006-7-s1-s1</a>.</p>
<p>[110] J. A. Reuter, D. V. Spacek, and M. P. Snyder. “High-Throughput
Sequencing Technologies”. In: <em>Molecular Cell</em> 58.4 (2015), pp. 586 -
597. ISSN: 1097-2765. DOI:
<a href="https://doi.org/10.1016/j.molcel.2015.05.004" class="uri">https://doi.org/10.1016/j.molcel.2015.05.004</a>.
<a href="http://www.sciencedirect.com/science/article/pii/S1097276515003408" class="uri">http://www.sciencedirect.com/science/article/pii/S1097276515003408</a>.</p>
<p>[111] L. J. Revell. “phytools: An R package for phylogenetic
comparative biology (and other things).” In: <em>Methods in Ecology and
Evolution</em> 3 (2012), pp. 217-223.</p>
<p>[112] A. Rhoads and K. F. Au. “PacBio Sequencing and Its Applications”.
In: <em>Genomics, Proteomics and Bioinformatics</em> 13.5 (2015), pp. 278 -
289.</p>
<p>[113] D. W. Rice, A. J. Alverson, A. O. Richardson, et al. “Horizontal
Transfer of Entire Genomes via Mitochondrial Fusion in the Angiosperm
Amborella”. In: <em>Science</em> 342.6165 (2013), pp. 1468-1473. ISSN:
0036-8075. DOI: 10.1126/science.1246275. eprint:
<a href="https://science.sciencemag.org/content/342/6165/1468.full.pdf" class="uri">https://science.sciencemag.org/content/342/6165/1468.full.pdf</a>.
<a href="https://science.sciencemag.org/content/342/6165/1468" class="uri">https://science.sciencemag.org/content/342/6165/1468</a>.</p>
<p>[114] B. A. Richardson, S. G. Kitchen, R. L. Pendleton, et al.
“Adaptive responses reveal contemporary and future ecotypes in a desert
shrub”. In: <em>Ecological Applications</em> 24.2 (2014), pp. 413-427. ISSN:
1939-5582. DOI: 10.1890/13-0587.1.
<a href="http://dx.doi.org/10.1890/13-0587.1" class="uri">http://dx.doi.org/10.1890/13-0587.1</a>.</p>
<p>[115] G. Robertson, J. Schein, R. Chiu, et al. “De novo assembly and
analysis of RNA-seq data”. In: <em>Nature Methods</em> 7 (Oct. 2010), pp. 909
EP -. <a href="http://dx.doi.org/10.1038/nmeth.1517" class="uri">http://dx.doi.org/10.1038/nmeth.1517</a>.</p>
<p>[116] J. D. Robin, A. T. Ludlow, R. LaRanger, et al. “Comparison of DNA
Quantification Methods for Next Generation Sequencing”. In: <em>Scientific
Reports</em> 6 (Apr. 2016). Article, pp. 24067 EP -.
<a href="http://dx.doi.org/10.1038/srep24067" class="uri">http://dx.doi.org/10.1038/srep24067</a>.</p>
<p>[117] K. Schliep. “phangorn: phylogenetic analysis in R”. In:
<em>Bioinformatics</em> 27.4 (2011), pp. 592-593.</p>
<p>[118] L. Sen, M. A. Fares, B. Liang, et al. “Molecular evolution of
rbcL in three gymnosperm families: identifying adaptive and
coevolutionary patterns”. In: <em>Biology Direct</em> 6.1 (Jun. 2011), p. 29.
ISSN: 1745-6150. DOI: 10.1186/1745-6150-6-29.
<a href="https://doi.org/10.1186/1745-6150-6-29" class="uri">https://doi.org/10.1186/1745-6150-6-29</a>.</p>
<p>[119] K. Shinozaki and K. Yamaguchi-Shinozaki. “Gene networks involved
in drought stress response and tolerance”. In: <em>Journal of Experimental
Botany</em> 58.2 (2007), pp. 221-227. DOI: 10.1093/jxb/erl164. eprint:
/oup/backfile/content_public/journal/jxb/58/2/10.1093/jxb/erl164/2/erl164.pdf.
&lt;+%20http://dx.doi.org/10.1093/jxb/erl164&gt;.</p>
<p>[120] J. T. Simpson. “Exploring genome characteristics and sequence
quality without a reference”. In: <em>Bioinformatics</em> 30.9 (2014), pp.
1228-1235. DOI: 10.1093/bioinformatics/btu023. eprint:
/oup/backfile/content_public/journal/bioinformatics/30/9/10.1093_bioinformatics_btu023/3/btu023.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btu023&gt;.</p>
<p>[121] J. T. Simpson and R. Durbin. “Efficient de novo assembly of large
genomes using compressed data structures”. In: <em>Genome Research</em> 22.3
(2012), pp. 549-556. DOI: 10.1101/gr.126953.111. eprint:
<a href="http://genome.cshlp.org/content/22/3/549.full.pdf+html" class="uri">http://genome.cshlp.org/content/22/3/549.full.pdf+html</a>.
<a href="http://genome.cshlp.org/content/22/3/549.abstract" class="uri">http://genome.cshlp.org/content/22/3/549.abstract</a>.</p>
<p>[122] J. T. Simpson and M. Pop. “The Theory and Practice of Genome
Sequence Assembly”. In: <em>Annual Review of Genomics and Human Genetics</em>
16.1 (2015). PMID: 25939056, pp. 153-172. DOI:
10.1146/annurev-genom-090314-050032. eprint:
<a href="https://doi.org/10.1146/annurev-genom-090314-050032" class="uri">https://doi.org/10.1146/annurev-genom-090314-050032</a></p>
<p>. <a href="https://doi.org/10.1146/annurev-genom-090314-050032" class="uri">https://doi.org/10.1146/annurev-genom-090314-050032</a>.</p>
<p>[123] A. Soorni, D. Haak, D. Zaitlin, et al. “Organelle_PBA, a
pipeline for assembling chloroplast and mitochondrial genomes from
PacBio DNA sequencing data”. In: <em>BMC Genomics</em> 18.1 (Jan. 2017), p.
49. ISSN: 1471-2164. DOI: 10.1186/s12864-016-3412-9.
<a href="https://doi.org/10.1186/s12864-016-3412-9" class="uri">https://doi.org/10.1186/s12864-016-3412-9</a>.</p>
<p>[124] M. A. Soto Arenas. “Conservation of the genetic resources of
Vanilla”. PhD thesis. UNAM University, 1999.</p>
<p>[125] M. A. Soto Arenas. <em>Filogeografia y recursos geneticos de las
vainillas de Mexico</em>. Informe final SNIB-CONABIO proyecto No. J101.
Instituto Chinoin AC, 1999.</p>
<p>[126] M. A. Soto Arenas and P. Cribb. “A NEW INFRAGENERIC
CLASSIFICATION AND SYNOPSIS OF THE GENUS VANILLA PLUM. EX MILL.
(ORCHIDACEAE: VANILLINAE)”. In: <em>Lankesteriana International Journal on
Orchidology</em> 9 (2009), pp. 355-398.</p>
<p>[127] A. Srivathsan, A. Ang, A. P. Vogler, et al. “Fecal metagenomics
for the simultaneous assessment of diet, parasites, and population
genetics of an understudied primate”. In: <em>Frontiers in Zoology</em> 13.1
(Apr. 2016), p. 17. ISSN: 1742-9994. DOI: 10.1186/s12983-016-0150-4.
<a href="https://doi.org/10.1186/s12983-016-0150-4" class="uri">https://doi.org/10.1186/s12983-016-0150-4</a>.</p>
<p>[128] A. Stamatakis. “RAxML version 8: a tool for phylogenetic analysis
and post-analysis of large phylogenies”. Eng. In: <em>Bioinformatics
(Oxford, England)</em> 30.9 (May. 2014). 24451623[pmid], pp. 1312-1313.
ISSN: 1367-4811. DOI: 10.1093/bioinformatics/btu033.
<a href="https://pubmed.ncbi.nlm.nih.gov/24451623" class="uri">https://pubmed.ncbi.nlm.nih.gov/24451623</a>.</p>
<p>[129] A. C. Telfer, M. R. Young, J. Quinn, et al. “Biodiversity
inventories in high gear: DNA barcoding facilitates a rapid biotic
survey of a temperate nature reserve”. In: <em>Biodiversity data journal</em>
(2015).</p>
<p>[130] J. N. Timmis, M. A. Ayliffe, C. Y. Huang, et al. “Endosymbiotic
gene transfer: organelle genomes forge eukaryotic chromosomes”. In:
<em>Nature Reviews Genetics</em> 5 (Feb. 2004). Review Article, pp. 123 EP -.
<a href="http://dx.doi.org/10.1038/nrg1271" class="uri">http://dx.doi.org/10.1038/nrg1271</a>.</p>
<p>[131] A. D. Twyford and R. W. Ness. “Strategies for complete plastid
genome sequencing”. In: <em>Molecular Ecology Resources</em> 17.5 (2017), pp.
858-868. ISSN: 1755-0998. DOI: 10.1111/1755-0998.12626.
<a href="http://dx.doi.org/10.1111/1755-0998.12626" class="uri">http://dx.doi.org/10.1111/1755-0998.12626</a>.</p>
<p>[132] G. W. Vurture, F. J. Sedlazeck, M. Nattestad, et al.
“GenomeScope: fast reference-free genome profiling from short reads”.
In: <em>Bioinformatics</em> 33.14 (2017), pp. 2202-2204. DOI:
10.1093/bioinformatics/btx153. eprint:
/oup/backfile/content_public/journal/bioinformatics/33/14/10.1093_bioinformatics_btx153/2/btx153.pdf.
&lt;+%20http://dx.doi.org/10.1093/bioinformatics/btx153&gt;.</p>
<p>[133] B. J. Walker, T. Abeel, T. Shea, et al. “Pilon: An Integrated
Tool for Comprehensive Microbial Variant Detection and Genome Assembly
Improvement”. In: <em>PLoS One</em> 9.11 (Nov. 2014). Ed. by J. Wang.
PONE-D-14-38252[PII], p. e112963. ISSN: 1932-6203. DOI:
10.1371/journal.pone.0112963.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4237348/" class="uri">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4237348/</a>.</p>
<p>[134] Q. D. Wheeler. “Why the phylogenetic species
concept?—Elementary”. In: <em>Journal of nematology</em> 31.2 (1999), p. 134.</p>
<p>[135] D. J. Winter. “rentrez: an R package for the NCBI eUtils API”.
In: <em>The R Journal</em> 9 (2 2017), pp. 520-526.</p>
<p>[136] M. Yandell and D. Ence. “A beginner&amp;#39;s guide to eukaryotic
genome annotation”. In: <em>Nature Reviews Genetics</em> 13 (Apr. 2012).
Review Article, pp. 329 EP -. <a href="http://dx.doi.org/10.1038/nrg3174" class="uri">http://dx.doi.org/10.1038/nrg3174</a>.</p>
<p>[137] X. Yang, J. C. Cushman, A. M. Borland, et al. “A roadmap for
research on crassulacean acid metabolism (CAM) to enhance sustainable
food and bioenergy production in a hotter, drier world”. In: <em>New
Phytologist</em> 207.3 (2015). 2015-19119, pp. 491-504. ISSN: 1469-8137.
DOI: 10.1111/nph.13393. <a href="http://dx.doi.org/10.1111/nph.13393" class="uri">http://dx.doi.org/10.1111/nph.13393</a>.</p>
<p>[138] Y. Yin, P. Zhong, G. Zhang, et al. “Morphological, genome-size
and molecular analyses of Apostasia fogangica (Apostasioideae,
Orchidaceae), a new species from China”. English. In: <em>PHYTOTAXA</em> 277.1
(), pp. 59-67. ISSN: 1179-3155. DOI: 10.11646/phytotaxa.277.1.5.</p>
<p>[139] D. R. Zerbino and E. Birney. “Velvet: Algorithms for de novo
short read assembly using de Bruijn graphs”. In: <em>Genome Res</em> 18.5
(May. 2008). 18349386[pmid], pp. 821-829. ISSN: 1088-9051. DOI:
10.1101/gr.074492.107.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336801/" class="uri">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336801/</a>.</p>
<p>[140] G. Zhang, K. Liu, Z. Li, et al. “The Apostasia genome and the
evolution of orchids”. In: <em>Nature</em> 549 (Sep. 2017), pp. 379 EP -.
<a href="http://dx.doi.org/10.1038/nature23897" class="uri">http://dx.doi.org/10.1038/nature23897</a>.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
