<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2021-01-11" />

<title>References</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Mini_reports.html">Mini-reports</a>
</li>
<li>
  <a href="Tutorials.html">Tutorials</a>
</li>
<li>
  <a href="References.html">References</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">References</h1>
<h4 class="date">2021-01-11</h4>

</div>


<p>References used in the BIOL 497/597 - Genomics &amp; Bioinformatics course are cited here. The citation of references supports transparency in science as well as provides opportunities for students to further their knowledge by accessing articles and books used to develop this course. The <code>BibTex</code> document is accessible <a href="References.bib">here</a>. Please contact the instructor (<a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a>) to add missing references to this document.</p>
<p>[1] S. Asaf, A. L. Khan, A. R. Khan, et al. “Complete Chloroplast
Genome of Nicotiana otophora and its Comparison with Related Species.”
In: <em>Frontiers in Plant Science</em> 7 (2016), p. 843. ISSN: 1664-462X.
DOI: 10.3389/fpls.2016.00843. &lt;URL:
<a href="https://www.frontiersin.org/article/10.3389/fpls.2016.00843" class="uri">https://www.frontiersin.org/article/10.3389/fpls.2016.00843</a>&gt;.</p>
<p>[2] A. Bankevich, S. Nurk, D. Antipov, et al. “SPAdes: A New Genome
Assembly Algorithm and Its Applications to Single-Cell Sequencing.” In:
<em>J Comput Biol</em> 19.5 (May. 2012). 22506599[pmid], pp. 455-477. ISSN:
1066-5277. DOI: 10.1089/cmb.2012.0021[PII]. &lt;URL:
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3342519/" class="uri">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3342519/</a>&gt;.</p>
<p>[3] A. Beckerman, D. Childs, and O. Petchey. <em>Getting Started with R:
An Introduction for Biologists</em>. Oxford University Press, 2017.</p>
<p>[4] D. A. Benson, I. Karsch-Mizrachi, D. J. Lipman, et al. “GenBank.”
In: <em>Nucleic Acids Research</em> 33 (2005), pp. D34-D38.</p>
<p>[5] C. Bleidorn. <em>Phylogenomics</em>. Springer International Publishing,
2017.</p>
<p>[6] U. Bodenhofer, E. Bonatesta, C. Horejš-Kainrath, et al. “msa: an R
package for multiple sequence alignment.” In: <em>Bioinformatics</em> 31.24
(2015), pp. 3997-3999. DOI: 10.1093/bioinformatics/btv494. eprint:
/oup/backfile/content_public/journal/bioinformatics/31/24/10.1093/bioinformatics/btv494/2/btv494.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/bioinformatics/btv494" class="uri">http://dx.doi.org/10.1093/bioinformatics/btv494</a>&gt;.</p>
<p>[7] R. E. Bone, J. A. C. Smith, N. Arrigo, et al. “A macro-ecological
perspective on crassulacean acid metabolism (CAM) photosynthesis
evolution in Afro-Madagascan drylands: Eulophiinae orchids as a case
study.” In: <em>New Phytologist</em> 208.2 (2015). 2015-19192, pp. 469-481.
ISSN: 1469-8137. DOI: 10.1111/nph.13572. &lt;URL:
<a href="http://dx.doi.org/10.1111/nph.13572" class="uri">http://dx.doi.org/10.1111/nph.13572</a>&gt;.</p>
<p>[8] S. Bory, O. Catrice, S. Brown, et al. “Natural polyploidy in
Vanilla planifolia (Orchidaceae).” In: <em>Genome</em> 51.10 (2008). PMID:
18923533, pp. 816-826. DOI: 10.1139/G08-068. eprint:
<a href="https://doi.org/10.1139/G08-068" class="uri">https://doi.org/10.1139/G08-068</a></p>
<p>. &lt;URL: <a href="https://doi.org/10.1139/G08-068" class="uri">https://doi.org/10.1139/G08-068</a>&gt;.</p>
<p>[9] S. Bory, M. Grisoni, M. Duval, et al. “Biodiversity and
preservation of vanilla: present state of knowledge.” In: <em>Genetic
Resources and Crop Evolution</em> 55.4 (Jun. 2008), pp. 551-571. ISSN:
1573-5109. DOI: 10.1007/s10722-007-9260-3. &lt;URL:
<a href="https://doi.org/10.1007/s10722-007-9260-3" class="uri">https://doi.org/10.1007/s10722-007-9260-3</a>&gt;.</p>
<p>[10] K. R. Bradnam, J. N. Fass, A. Alexandrov, et al. “Assemblathon 2:
evaluating de novo methods of genome assembly in three vertebrate
species.” In: <em>GigaScience</em> 2.1 (2013), pp. 1-31. DOI:
10.1186/2047-217X-2-10. eprint:
/oup/backfile/content_public/journal/gigascience/2/1/10.1186_2047-217x-2-10/7/13742_2013_article_29.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1186/2047-217X-2-10" class="uri">http://dx.doi.org/10.1186/2047-217X-2-10</a>&gt;.</p>
<p>[11] T. A. Brown. <em>Genomes 4</em>. Garland Science, 2017.</p>
<p>[12] S. Buerki, F. Forest, N. Alvarez, et al. “An evaluation of new
parsimony-based versus parametric inference methods in biogeography: a
case study using the globally distributed plant family Sapindaceae.”
In: <em>Journal of Biogeography</em> 38.3 (2011), pp. 531-550. ISSN:
1365-2699. DOI: 10.1111/j.1365-2699.2010.02432.x. &lt;URL:
<a href="http://dx.doi.org/10.1111/j.1365-2699.2010.02432.x" class="uri">http://dx.doi.org/10.1111/j.1365-2699.2010.02432.x</a>&gt;.</p>
<p>[13] J. Burnier, S. Buerki, N. Arrigo, et al. “Genetic structure and
evolution of Alpine polyploid complexes: Ranunculus kuepferi
(Ranunculaceae) as a case study.” In: <em>Molecular Ecology</em> 18.17 (2009),
pp. 3730-3744. ISSN: 1365-294X. DOI: 10.1111/j.1365-294X.2009.04281.x.
&lt;URL: <a href="http://dx.doi.org/10.1111/j.1365-294X.2009.04281.x" class="uri">http://dx.doi.org/10.1111/j.1365-294X.2009.04281.x</a>&gt;.</p>
<p>[14] M. A. Butler and A. A. King. “Phylogenetic comparative analysis: a
modeling approach for adaptive evolution.” In: <em>American Naturalist</em>
164 (2004), pp. 683-695. &lt;URL:
<a href="http://www.press.uchicago.edu/ucp/journals/journal/an.html" class="uri">http://www.press.uchicago.edu/ucp/journals/journal/an.html</a>&gt;.</p>
<p>[15] J. Cai, X. Liu, K. Vanneste, et al. “The genome sequence of the
orchid Phalaenopsis equestris.” In: <em>Nature Genetics</em> 47.1 (2015), pp.
65-+. ISSN: 1061-4036. DOI: 10.1038/ng.3149. &lt;URL: <Go to
ISI>://WOS:000346990400013&gt;.</p>
<p>[16] S. Chamberlain. <em>rgbif: Interface to the Global ‘Biodiversity’
Information Facility ‘API’</em>. 2016. &lt;URL:
<a href="https://CRAN.R-project.org/package=rgbif" class="uri">https://CRAN.R-project.org/package=rgbif</a>&gt;.</p>
<p>[17] D. Charif and J. Lobry. “SeqinR 1.0-2: a contributed package to
the R project for statistical computing devoted to biological sequences
retrieval and analysis.” In: <em>Structural approaches to sequence
evolution: Molecules, networks, populations</em>. Ed. by U. Bastolla, M.
Porto, H. Roman and M. Vendruscolo. Biological and Medical Physics,
Biomedical Engineering. ISBN : 978-3-540-35305-8. New York: Springer
Verlag, 2007, pp. 207-232.</p>
<p>[18] J. Claverie and C. Notredame. <em>Bioinformatics For Dummies, 2nd
Edition</em>. Wiley, 2006.</p>
<p>[19] A. Coghlan, T. J. Fiedler, S. J. McKay, et al. “nGASP - the
nematode genome annotation assessment project.” In: <em>BMC
Bioinformatics</em> 9.1 (Dec. 2008), p. 549. ISSN: 1471-2105. DOI:
10.1186/1471-2105-9-549. &lt;URL:
<a href="https://doi.org/10.1186/1471-2105-9-549" class="uri">https://doi.org/10.1186/1471-2105-9-549</a>&gt;.</p>
<p>[20] M. J. Crawley. <em>The R Book, 2nd Edition</em>. Wiley, 2012.</p>
<p>[21] A. Dai. “Drought under global warming: a review.” In: <em>Wiley
Interdisciplinary Reviews: Climate Change</em> 2.1 (2011), pp. 45-65. ISSN:
1757-7799. DOI: 10.1002/wcc.81. &lt;URL:
<a href="http://dx.doi.org/10.1002/wcc.81" class="uri">http://dx.doi.org/10.1002/wcc.81</a>&gt;.</p>
<p>[22] J. W. Dale, M. von Schantz, and N. Plant. <em>From Genes to Genomes:
Concepts and Applications of DNA Technology, 3rd Edition</em>. Wiley, 2012.</p>
<p>[23] B. E. Davidson, S. J. Novak, and M. D. Serpe. “Consequences of
inoculation with native arbuscular mycorrhizal fungi for root
colonization and survival of Artemisia tridentata ssp. wyomingensis
seedlings after transplanting.” In: <em>Mycorrhiza</em> 26.6 (Aug. 2016), pp.
595-608. ISSN: 1432-1890. DOI: 10.1007/s00572-016-0696-1. &lt;URL:
<a href="https://doi.org/10.1007/s00572-016-0696-1" class="uri">https://doi.org/10.1007/s00572-016-0696-1</a>&gt;.</p>
<p>[24] R. deSalle and J. Rosenfeld. <em>Phylogenomics: A Primer 1st
Edition</em>. Garland Science, 2013.</p>
<p>[25] S. Dodsworth, M. W. Chase, L. J. Kelly, et al. “Genomic Repeat
Abundances Contain Phylogenetic Signal.” In: <em>Systematic Biology</em> 64.1
(2015), pp. 112-126. DOI: 10.1093/sysbio/syu080. eprint:
/oup/backfile/content_public/journal/sysbio/64/1/10.1093_sysbio_syu080/1/syu080.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/sysbio/syu080" class="uri">http://dx.doi.org/10.1093/sysbio/syu080</a>&gt;.</p>
<p>[26] S. Dray and A. Dufour. “The ade4 package: implementing the duality
diagram for ecologists.” In: <em>Journal of Statistical Software</em> 22.4
(2007), pp. 1-20.</p>
<p>[27] R. Ekblom and J. B. W. Wolf. “A field guide to whole-genome
sequencing, assembly and annotation.” In: <em>Evolutionary Applications</em>
7.9 (2014), pp. 1026-1042. ISSN: 1752-4571. DOI: 10.1111/eva.12178.
&lt;URL: <a href="http://dx.doi.org/10.1111/eva.12178" class="uri">http://dx.doi.org/10.1111/eva.12178</a>&gt;.</p>
<p>[28] J. Flexas, J. M. Escalona, S. Evain, et al. “Steady-state
chlorophyll fluorescence (Fs) measurements as a tool to follow
variations of net CO2 assimilation and stomatal conductance during
water-stress in C3 plants.” In: <em>Physiologia Plantarum</em> 114.2 (2002),
pp. 231-240. ISSN: 1399-3054. DOI: 10.1034/j.1399-3054.2002.1140209.x.
&lt;URL: <a href="http://dx.doi.org/10.1034/j.1399-3054.2002.1140209.x" class="uri">http://dx.doi.org/10.1034/j.1399-3054.2002.1140209.x</a>&gt;.</p>
<p>[29] H. Gehrig, K. Faist, and M. Kluge. “Identification of
phosphoenolpyruvate carboxylase isoforms in leaf, stem and roots of the
obligate CAM plant Vanilla planifolia Salib. (Orchidaceae): a
physiological and molecular approach.” In: <em>Plant Molecular Biology</em>
38.6 (Dec. 1998), pp. 1215-1223. ISSN: 1573-5028. DOI:
10.1023/A:1006006331011. &lt;URL:
<a href="https://doi.org/10.1023/A:1006006331011" class="uri">https://doi.org/10.1023/A:1006006331011</a>&gt;.</p>
<p>[30] S. Gnerre, I. MacCallum, D. Przybylski, et al. “High-quality draft
assemblies of mammalian genomes from massively parallel sequence data.”
In: <em>Proceedings of the National Academy of Sciences</em> 108.4 (2011), pp.
1513-1518. DOI: 10.1073/pnas.1017351108. eprint:
<a href="http://www.pnas.org/content/108/4/1513.full.pdf" class="uri">http://www.pnas.org/content/108/4/1513.full.pdf</a>. &lt;URL:
<a href="http://www.pnas.org/content/108/4/1513.abstract" class="uri">http://www.pnas.org/content/108/4/1513.abstract</a>&gt;.</p>
<p>[31] M. G. Grabherr, B. J. Haas, M. Yassour, et al. “Full-length
transcriptome assembly from RNA-Seq data without a reference genome.”
In: <em>Nature Biotechnology</em> 29 (May. 2011), pp. 644 EP -. &lt;URL:
<a href="http://dx.doi.org/10.1038/nbt.1883" class="uri">http://dx.doi.org/10.1038/nbt.1883</a>&gt;.</p>
<p>[32] C. Hahn, L. Bachmann, and B. Chevreux. “Reconstructing
mitochondrial genomes directly from genomic next-generation sequencing
reads—a baiting and iterative mapping approach.” In: <em>Nucleic Acids
Research</em> 41.13 (2013), p. e129. DOI: 10.1093/nar/gkt371. eprint:
/oup/backfile/content_public/journal/nar/41/13/10.1093_nar_gkt371/3/gkt371.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/nar/gkt371" class="uri">http://dx.doi.org/10.1093/nar/gkt371</a>&gt;.</p>
<p>[33] D. L. Hartl. “Fly meets shotgun: shotgun wins.” In: <em>Nature
Genetics</em> 24 (Apr. 2000), p. 327 EP. &lt;URL:
<a href="http://dx.doi.org/10.1038/74125" class="uri">http://dx.doi.org/10.1038/74125</a>&gt;.</p>
<p>[34] R. J. Hijmans, S. E. Cameron, J. L. Parra, et al. “Very high
resolution interpolated climate surfaces for global land areas.” In:
<em>International Journal of Climatology</em> 25.15 (2005), pp. 1965-1978.
ISSN: 1097-0088. DOI: 10.1002/joc.1276. &lt;URL:
<a href="http://dx.doi.org/10.1002/joc.1276" class="uri">http://dx.doi.org/10.1002/joc.1276</a>&gt;.</p>
<p>[35] C. Holt and M. Yandell. “MAKER2: an annotation pipeline and
genome-database management tool for second-generation genome projects.”
In: <em>BMC Bioinformatics</em> 12.1 (Dec. 2011), p. 491. ISSN: 1471-2105.
DOI: 10.1186/1471-2105-12-491. &lt;URL:
<a href="https://doi.org/10.1186/1471-2105-12-491" class="uri">https://doi.org/10.1186/1471-2105-12-491</a>&gt;.</p>
<p>[36] S. D. Jackman, B. P. Vandervalk, H. Mohamadi, et al. “ABySS 2.0:
resource-efficient assembly of large genomes using a Bloom filter.” In:
<em>Genome Research</em> 27.5 (2017), pp. 768-777. DOI: 10.1101/gr.214346.116.
eprint: <a href="http://genome.cshlp.org/content/27/5/768.full.pdf+html" class="uri">http://genome.cshlp.org/content/27/5/768.full.pdf+html</a>. &lt;URL:
<a href="http://genome.cshlp.org/content/27/5/768.abstract" class="uri">http://genome.cshlp.org/content/27/5/768.abstract</a>&gt;.</p>
<p>[37] J. Jersáková, P. Trávníček, B. Kubátová, et al. “Genome size
variation in Orchidaceae subfamily Apostasioideae: filling the
phylogenetic gap.” In: <em>Botanical Journal of the Linnean Society</em> 172.1
(2013), pp. 95-105. ISSN: 1095-8339. DOI: 10.1111/boj.12027. &lt;URL:
<a href="http://dx.doi.org/10.1111/boj.12027" class="uri">http://dx.doi.org/10.1111/boj.12027</a>&gt;.</p>
<p>[38] M. Kearse, R. Moir, A. Wilson, et al. “Geneious Basic: An
integrated and extendable desktop software platform for the
organization and analysis of sequence data.” In: <em>Bioinformatics</em> 28.12
(2012), pp. 1647-1649. DOI: 10.1093/bioinformatics/bts199. eprint:
/oup/backfile/content_public/journal/bioinformatics/28/12/10.1093/bioinformatics/bts199/2/bts199.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/bioinformatics/bts199" class="uri">http://dx.doi.org/10.1093/bioinformatics/bts199</a>&gt;.</p>
<p>[39] J. M. Keith. <em>Bioinformatics, Volume I: Data, Sequence Analysis,
and Evolution</em>. Springer Science+Business Media New York, 2017.</p>
<p>[40] D. R. Kelley, M. C. Schatz, and S. L. Salzberg. “Quake:
quality-aware detection and correction of sequencing errors.” In:
<em>Genome Biology</em> 11.11 (2010), p. R116. ISSN: 1474-760X. DOI:
10.1186/gb-2010-11-11-r116. &lt;URL:
<a href="https://doi.org/10.1186/gb-2010-11-11-r116" class="uri">https://doi.org/10.1186/gb-2010-11-11-r116</a>&gt;.</p>
<p>[41] S. Kembel, P. Cowan, M. Helmus, et al. “Picante: R tools for
integrating phylogenies and ecology.” In: <em>Bioinformatics</em> 26 (2010),
pp. 1463-1464.</p>
<p>[42] A. A. Kolesnikov and E. S. Gerasimov. “Diversity of mitochondrial
genome organization.” English. In: <em>BIOCHEMISTRY-MOSCOW</em> 77.13 (), pp.
1424-1435. ISSN: 0006-2979. DOI: 10.1134/S0006297912130020.</p>
<p>[43] S. Koren, B. P. Walenz, K. Berlin, et al. “Canu: scalable and
accurate long-read assembly via adaptive k-mer weighting and repeat
separation.” In: <em>Genome Research</em> 27.5 (2017), pp. 722-736. DOI:
10.1101/gr.215087.116. eprint:
<a href="http://genome.cshlp.org/content/27/5/722.full.pdf+html" class="uri">http://genome.cshlp.org/content/27/5/722.full.pdf+html</a>. &lt;URL:
<a href="http://genome.cshlp.org/content/27/5/722.abstract" class="uri">http://genome.cshlp.org/content/27/5/722.abstract</a>&gt;.</p>
<p>[44] P. Lemey, M. Salemi, and A. Vandamme. <em>The Phylogenetic Handbook:
A Practical Approach to Phylogenetic Analysis and Hypothesis Testing,
Edition 2</em>. Cambridge University Press, 2009.</p>
<p>[45] A. Lesk. <em>Introduction to Genomics, Third edition</em>. Oxford
University Press, 2017.</p>
<p>[46] R. Li, W. Fan, G. Tian, et al. “The sequence and de novo assembly
of the giant panda genome.” In: <em>Nature</em> 463 (Dec. 2009). Article, pp.
311 EP -. &lt;URL: <a href="http://dx.doi.org/10.1038/nature08696" class="uri">http://dx.doi.org/10.1038/nature08696</a>&gt;.</p>
<p>[47] C. Lin, J. J. W. Chen, Y. Huang, et al. “The location and
translocation of ndh genes of chloroplast origin in the Orchidaceae
family.” In: <em>Scientific Reports</em> 5 (Mar. 2015). Article, pp. 9040 EP
-. &lt;URL: <a href="http://dx.doi.org/10.1038/srep09040" class="uri">http://dx.doi.org/10.1038/srep09040</a>&gt;.</p>
<p>[48] J. Males and H. Griffiths. “Stomatal biology of CAM plants.” In:
<em>Plant Physiology</em> (2017). ISSN: 0032-0889. DOI: 10.1104/pp.17.00114.
eprint:
<a href="http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114.full.pdf" class="uri">http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114.full.pdf</a>.
&lt;URL:
<a href="http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114" class="uri">http://www.plantphysiol.org/content/early/2017/02/27/pp.17.00114</a>&gt;.</p>
<p>[49] D. Mapleson, G. Garcia Accinelli, G. Kettleborough, et al. “KAT: a
K-mer analysis toolkit to quality control NGS datasets and genome
assemblies.” In: <em>Bioinformatics</em> 33.4 (2017), pp. 574-576. DOI:
10.1093/bioinformatics/btw663. eprint:
/oup/backfile/content_public/journal/bioinformatics/33/4/10.1093_bioinformatics_btw663/4/btw663.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/bioinformatics/btw663" class="uri">http://dx.doi.org/10.1093/bioinformatics/btw663</a>&gt;.</p>
<p>[50] J. A. Martin and Z. Wang. “Next-generation transcriptome
assembly.” In: <em>Nature Reviews Genetics</em> 12 (Sep. 2011). Review
Article, pp. 671 EP -. &lt;URL: <a href="http://dx.doi.org/10.1038/nrg3068" class="uri">http://dx.doi.org/10.1038/nrg3068</a>&gt;.</p>
<p>[51] N. J. Matzke. <em>BioGeoBEARS: BioGeography with Bayesian (and
Likelihood) Evolutionary Analysis in R Scripts</em>. The
rexpokit/cladoRcpp/BioGeoBEARS packages were first published in:
Matzke, Nicholas J. (2012). Founder-event speciation in BioGeoBEARS
package dramatically improves likelihoods and alters parameter
inference in Dispersal-Extinction-Cladogenesis (DEC) analyses.
Frontiers of Biogeography, 4(suppl. 1): 210, December 2012. Link to
abstract and PDF of poster:
<a href="http://phylo.wikidot.com/matzke-2013-international-biogeography-society-poster" class="uri">http://phylo.wikidot.com/matzke-2013-international-biogeography-society-poster</a>
. (Poster abstract published in the Conference Program and Abstracts of
the International Biogeography Society 6th Biannual Meeting, Miami,
Florida. Poster Session P10: Historical and Paleo-Biogeography. Poster
129B. January 11, 2013.). University of California, Berkeley. Berkeley,
CA, 2013. &lt;URL: <a href="http://CRAN.R-project.org/package=BioGeoBEARS" class="uri">http://CRAN.R-project.org/package=BioGeoBEARS</a>&gt;.</p>
<p>[52] K. Maxwell and G. N. Johnson. “Chlorophyll fluorescence—a
practical guide.” In: <em>Journal of Experimental Botany</em> 51.345 (2000),
pp. 659-668. DOI: 10.1093/jexbot/51.345.659. eprint:
/oup/backfile/content_public/journal/jxb/51/345/10.1093_jexbot_51.345.659/1/510659.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/jexbot/51.345.659" class="uri">http://dx.doi.org/10.1093/jexbot/51.345.659</a>&gt;.</p>
<p>[53] R. Ming, R. VanBuren, C. M. Wai, et al. “The pineapple genome and
the evolution of CAM photosynthesis.” In: <em>Nature Genetics</em> 47.12 (Dec.
2015), p. 1435. ISSN: 1061-4036. DOI: 10.1038/ng.3435.</p>
<p>[54] H. Mizuno, Y. Kawahara, H. Sakai, et al. “Massive parallel
sequencing of mRNA in identification of unannotated salinity
stress-inducible transcripts in rice (Oryza sativa L.).” In: <em>BMC
Genomics</em> 11.1 (Dec. 2010), p. 683. ISSN: 1471-2164. DOI:
10.1186/1471-2164-11-683. &lt;URL:
<a href="https://doi.org/10.1186/1471-2164-11-683" class="uri">https://doi.org/10.1186/1471-2164-11-683</a>&gt;.</p>
<p>[55] L. S. Mofatto, F. d. A. Carneiro, N. G. Vieira, et al.
“Identification of candidate genes for drought tolerance in coffee by
high-throughput sequencing in the shoot apex of different Coffea
arabica cultivars.” In: <em>BMC Plant Biology</em> 16.1 (Apr. 2016), p. 94.
ISSN: 1471-2229. DOI: 10.1186/s12870-016-0777-5. &lt;URL:
<a href="https://doi.org/10.1186/s12870-016-0777-5" class="uri">https://doi.org/10.1186/s12870-016-0777-5</a>&gt;.</p>
<p>[56] M. C. Munoz-Torres, J. T. Reese, C. P. Childers, et al.
“Hymenoptera Genome Database: integrated community resources for insect
species of the order Hymenoptera.” In: <em>Nucleic Acids Research</em>
39.suppl 1 (2011), pp. D658-D662. DOI: 10.1093/nar/gkq1145.</p>
<p>[57] Z. Niu, J. Pan, S. Zhu, et al. “Comparative Analysis of the
Complete Plastomes of Apostasia wallichii and Neuwiedia singapureana
(Apostasioideae) Reveals Different Evolutionary Dynamics of IR/SSC
Boundary among Photosynthetic Orchids.” In: <em>Frontiers in Plant
Science</em> 8 (2017), p. 1713. ISSN: 1664-462X. DOI:
10.3389/fpls.2017.01713. &lt;URL:
<a href="https://www.frontiersin.org/article/10.3389/fpls.2017.01713" class="uri">https://www.frontiersin.org/article/10.3389/fpls.2017.01713</a>&gt;.</p>
<p>[58] M. H. O’Leary. “Carbon Isotopes in Photosynthesis.” In:
<em>BioScience</em> 38.5 (1988), pp. 328-336. ISSN: 00063568, 15253244. &lt;URL:
<a href="http://www.jstor.org/stable/1310735" class="uri">http://www.jstor.org/stable/1310735</a>&gt;.</p>
<p>[59] E. Paradis. <em>Analysis of Phylogenetics and Evolution with R</em>.
Springer, 2012.</p>
<p>[60] E. Paradis, J. Claude, and K. Strimmer. “APE: analyses of
phylogenetics and evolution in R language.” In: <em>Bioinformatics</em> 20
(2004), pp. 289-290.</p>
<p>[61] J. Pellicer, M. F. Fay, and I. J. Leitch. “The largest eukaryotic
genome of them all?” In: <em>Botanical Journal of the Linnean Society</em>
164.1 (2010), pp. 10-15. DOI: 10.1111/j.1095-8339.2010.01072.x. eprint:
/oup/backfile/content_public/journal/botlinnean/164/1/10.1111_j.1095-8339.2010.01072.x/2/j.1095-8339.2010.01072.x.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1111/j.1095-8339.2010.01072.x" class="uri">http://dx.doi.org/10.1111/j.1095-8339.2010.01072.x</a>&gt;.</p>
<p>[62] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2016. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[63] A. Raj, M. Stephens, and J. K. Pritchard. “fastSTRUCTURE:
Variational Inference of Population Structure in Large SNP Data Sets.”
In: <em>Genetics</em> 197.2 (2014), pp. 573-589. ISSN: 0016-6731. DOI:
10.1534/genetics.114.164350. eprint:
<a href="http://www.genetics.org/content/197/2/573.full.pdf" class="uri">http://www.genetics.org/content/197/2/573.full.pdf</a>. &lt;URL:
<a href="http://www.genetics.org/content/197/2/573" class="uri">http://www.genetics.org/content/197/2/573</a>&gt;.</p>
<p>[64] M. G. Reese and R. Guigó. “EGASP: Introduction.” In: <em>Genome
Biology</em> 7.1 (Aug. 2006), p. S1. ISSN: 1474-760X. DOI:
10.1186/gb-2006-7-s1-s1. &lt;URL:
<a href="https://doi.org/10.1186/gb-2006-7-s1-s1" class="uri">https://doi.org/10.1186/gb-2006-7-s1-s1</a>&gt;.</p>
<p>[65] J. Reuter, D. V. Spacek, and M. Snyder. “High-Throughput
Sequencing Technologies.” In: <em>Molecular Cell</em> 58.4 (2015), pp. 586 -
597. ISSN: 1097-2765. DOI:
<a href="https://doi.org/10.1016/j.molcel.2015.05.004" class="uri">https://doi.org/10.1016/j.molcel.2015.05.004</a>. &lt;URL:
<a href="http://www.sciencedirect.com/science/article/pii/S1097276515003408" class="uri">http://www.sciencedirect.com/science/article/pii/S1097276515003408</a>&gt;.</p>
<p>[66] L. J. Revell. “phytools: An R package for phylogenetic comparative
biology (and other things).” In: <em>Methods in Ecology and Evolution</em> 3
(2012), pp. 217-223.</p>
<p>[67] A. Rhoads and K. F. Au. “PacBio Sequencing and Its Applications.”
In: <em>Genomics, Proteomics and Bioinformatics</em> 13.5 (2015), pp. 278 -
289.</p>
<p>[68] D. W. Rice, A. J. Alverson, A. O. Richardson, et al. “Horizontal
Transfer of Entire Genomes via Mitochondrial Fusion in the Angiosperm
Amborella.” In: <em>Science</em> 342.6165 (2013), pp. 1468-1473. ISSN:
0036-8075. DOI: 10.1126/science.1246275. eprint:
<a href="https://science.sciencemag.org/content/342/6165/1468.full.pdf" class="uri">https://science.sciencemag.org/content/342/6165/1468.full.pdf</a>. &lt;URL:
<a href="https://science.sciencemag.org/content/342/6165/1468" class="uri">https://science.sciencemag.org/content/342/6165/1468</a>&gt;.</p>
<p>[69] B. A. Richardson, S. G. Kitchen, R. L. Pendleton, et al. “Adaptive
responses reveal contemporary and future ecotypes in a desert shrub.”
In: <em>Ecological Applications</em> 24.2 (2014), pp. 413-427. ISSN:
1939-5582. DOI: 10.1890/13-0587.1. &lt;URL:
<a href="http://dx.doi.org/10.1890/13-0587.1" class="uri">http://dx.doi.org/10.1890/13-0587.1</a>&gt;.</p>
<p>[70] G. Robertson, J. Schein, R. Chiu, et al. “De novo assembly and
analysis of RNA-seq data.” In: <em>Nature Methods</em> 7 (Oct. 2010), pp. 909
EP -. &lt;URL: <a href="http://dx.doi.org/10.1038/nmeth.1517" class="uri">http://dx.doi.org/10.1038/nmeth.1517</a>&gt;.</p>
<p>[71] J. D. Robin, A. T. Ludlow, R. LaRanger, et al. “Comparison of DNA
Quantification Methods for Next Generation Sequencing.” In: <em>Scientific
Reports</em> 6 (Apr. 2016). Article, pp. 24067 EP -. &lt;URL:
<a href="http://dx.doi.org/10.1038/srep24067" class="uri">http://dx.doi.org/10.1038/srep24067</a>&gt;.</p>
<p>[72] K. Schliep. “phangorn: phylogenetic analysis in R.” In:
<em>Bioinformatics</em> 27.4 (2011), pp. 592-593.</p>
<p>[73] L. Sen, M. A. Fares, B. Liang, et al. “Molecular evolution of rbcL
in three gymnosperm families: identifying adaptive and coevolutionary
patterns.” In: <em>Biology Direct</em> 6.1 (Jun. 2011), p. 29. ISSN:
1745-6150. DOI: 10.1186/1745-6150-6-29. &lt;URL:
<a href="https://doi.org/10.1186/1745-6150-6-29" class="uri">https://doi.org/10.1186/1745-6150-6-29</a>&gt;.</p>
<p>[74] K. Shinozaki and K. Yamaguchi-Shinozaki. “Gene networks involved
in drought stress response and tolerance.” In: <em>Journal of Experimental
Botany</em> 58.2 (2007), pp. 221-227. DOI: 10.1093/jxb/erl164. eprint:
/oup/backfile/content_public/journal/jxb/58/2/10.1093/jxb/erl164/2/erl164.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/jxb/erl164" class="uri">http://dx.doi.org/10.1093/jxb/erl164</a>&gt;.</p>
<p>[75] J. T. Simpson. “Exploring genome characteristics and sequence
quality without a reference.” In: <em>Bioinformatics</em> 30.9 (2014), pp.
1228-1235. DOI: 10.1093/bioinformatics/btu023. eprint:
/oup/backfile/content_public/journal/bioinformatics/30/9/10.1093_bioinformatics_btu023/3/btu023.pdf.
&lt;URL: + <a href="http://dx.doi.org/10.1093/bioinformatics/btu023" class="uri">http://dx.doi.org/10.1093/bioinformatics/btu023</a>&gt;.</p>
<p>[76] J. T. Simpson and R. Durbin. “Efficient de novo assembly of large
genomes using compressed data structures.” In: <em>Genome Research</em> 22.3
(2012), pp. 549-556. DOI: 10.1101/gr.126953.111. eprint:
<a href="http://genome.cshlp.org/content/22/3/549.full.pdf+html" class="uri">http://genome.cshlp.org/content/22/3/549.full.pdf+html</a>. &lt;URL:
<a href="http://genome.cshlp.org/content/22/3/549.abstract" class="uri">http://genome.cshlp.org/content/22/3/549.abstract</a>&gt;.</p>
<p>[77] J. T. Simpson and M. Pop. “The Theory and Practice of Genome
Sequence Assembly.” In: <em>Annual Review of Genomics and Human Genetics</em>
16.1 (2015). PMID: 25939056, pp. 153-172. DOI:
10.1146/annurev-genom-090314-050032. eprint:
<a href="https://doi.org/10.1146/annurev-genom-090314-050032" class="uri">https://doi.org/10.1146/annurev-genom-090314-050032</a></p>
<p>. &lt;URL: <a href="https://doi.org/10.1146/annurev-genom-090314-050032" class="uri">https://doi.org/10.1146/annurev-genom-090314-050032</a>&gt;.</p>
<p>[78] A. Soorni, D. Haak, D. Zaitlin, et al. “Organelle_PBA, a pipeline
for assembling chloroplast and mitochondrial genomes from PacBio DNA
sequencing data.” In: <em>BMC Genomics</em> 18.1 (Jan. 2017), p. 49. ISSN:
1471-2164. DOI: 10.1186/s12864-016-3412-9. &lt;URL:
<a href="https://doi.org/10.1186/s12864-016-3412-9" class="uri">https://doi.org/10.1186/s12864-016-3412-9</a>&gt;.</p>
<p>[79] M. A. Soto Arenas. “Conservation of the genetic resources of
Vanilla.” PhD thesis. UNAM University, 1999.</p>
<p>[80] M. A. Soto Arenas. <em>Filogeografia y recursos geneticos de las
vainillas de Mexico</em>. Informe final SNIB-CONABIO proyecto No. J101.
Instituto Chinoin AC, 1999.</p>
<p>[81] M. A. Soto Arenas and P. Cribb. “A NEW INFRAGENERIC CLASSIFICATION
AND SYNOPSIS OF THE GENUS VANILLA PLUM. EX MILL. (ORCHIDACEAE:
VANILLINAE).” In: <em>Lankesteriana International Journal on Orchidology</em>
9 (2009), pp. 355-398.</p>
<p>[82] A. Srivathsan, A. Ang, A. P. Vogler, et al. “Fecal metagenomics
for the simultaneous assessment of diet, parasites, and population
genetics of an understudied primate.” In: <em>Frontiers in Zoology</em> 13.1
(Apr. 2016), p. 17. ISSN: 1742-9994. DOI: 10.1186/s12983-016-0150-4.
&lt;URL: <a href="https://doi.org/10.1186/s12983-016-0150-4" class="uri">https://doi.org/10.1186/s12983-016-0150-4</a>&gt;.</p>
<p>[83] J. N. Timmis, M. A. Ayliffe, C. Y. Huang, et al. “Endosymbiotic
gene transfer: organelle genomes forge eukaryotic chromosomes.” In:
<em>Nature Reviews Genetics</em> 5 (Feb. 2004). Review Article, pp. 123 EP -.
&lt;URL: <a href="http://dx.doi.org/10.1038/nrg1271" class="uri">http://dx.doi.org/10.1038/nrg1271</a>&gt;.</p>
<p>[84] A. D. Twyford and R. W. Ness. “Strategies for complete plastid
genome sequencing.” In: <em>Molecular Ecology Resources</em> 17.5 (2017), pp.
858-868. ISSN: 1755-0998. DOI: 10.1111/1755-0998.12626. &lt;URL:
<a href="http://dx.doi.org/10.1111/1755-0998.12626" class="uri">http://dx.doi.org/10.1111/1755-0998.12626</a>&gt;.</p>
<p>[85] B. J. Walker, T. Abeel, T. Shea, et al. “Pilon: An Integrated Tool
for Comprehensive Microbial Variant Detection and Genome Assembly
Improvement.” In: <em>PLoS One</em> 9.11 (Nov. 2014). Ed. by J. Wang.
PONE-D-14-38252[PII], p. e112963. ISSN: 1932-6203. DOI:
10.1371/journal.pone.0112963. &lt;URL:
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4237348/" class="uri">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4237348/</a>&gt;.</p>
<p>[86] M. Yandell and D. Ence. “A beginner&amp;#39;s guide to eukaryotic
genome annotation.” In: <em>Nature Reviews Genetics</em> 13 (Apr. 2012).
Review Article, pp. 329 EP -. &lt;URL: <a href="http://dx.doi.org/10.1038/nrg3174" class="uri">http://dx.doi.org/10.1038/nrg3174</a>&gt;.</p>
<p>[87] X. Yang, J. C. Cushman, A. M. Borland, et al. “A roadmap for
research on crassulacean acid metabolism (CAM) to enhance sustainable
food and bioenergy production in a hotter, drier world.” In: <em>New
Phytologist</em> 207.3 (2015). 2015-19119, pp. 491-504. ISSN: 1469-8137.
DOI: 10.1111/nph.13393. &lt;URL: <a href="http://dx.doi.org/10.1111/nph.13393" class="uri">http://dx.doi.org/10.1111/nph.13393</a>&gt;.</p>
<p>[88] Y. Yin, P. Zhong, G. Zhang, et al. “Morphological, genome-size and
molecular analyses of Apostasia fogangica (Apostasioideae,
Orchidaceae), a new species from China.” English. In: <em>PHYTOTAXA</em> 277.1
(), pp. 59-67. ISSN: 1179-3155. DOI: 10.11646/phytotaxa.277.1.5.</p>
<p>[89] D. R. Zerbino and E. Birney. “Velvet: Algorithms for de novo short
read assembly using de Bruijn graphs.” In: <em>Genome Res</em> 18.5 (May.
2008). 18349386[pmid], pp. 821-829. ISSN: 1088-9051. DOI:
10.1101/gr.074492.107. &lt;URL:
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336801/" class="uri">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336801/</a>&gt;.</p>
<p>[90] G. Zhang, K. Liu, Z. Li, et al. “The Apostasia genome and the
evolution of orchids.” In: <em>Nature</em> 549 (Sep. 2017), pp. 379 EP -.
&lt;URL: <a href="http://dx.doi.org/10.1038/nature23897" class="uri">http://dx.doi.org/10.1038/nature23897</a>&gt;.</p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
